{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/frazer01/projects/CARDIPS/analysis/cardiac_eqtls\")\n",
    "\n",
    "source(\"script/packages.R\"  )\n",
    "source(\"script/input_data.R\")\n",
    "source(\"script/functions.R\" )\n",
    "source(\"script/colors.R\"    )\n",
    "\n",
    "suppressPackageStartupMessages(library(coloc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "outfolder = \"pipeline/6.1.coloc_gwas/intersect_rbps\"\n",
    "\n",
    "dir.create(outfolder, showWarnings = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "geneinfo = fread(\"pipeline/1.2.expression/isoform_info.txt\"                   , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "coloc    = fread(\"pipeline/6.1.coloc_gwas/coloc.eqtls.isoform.txt\"            , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "fm       = fread(\"pipeline/3.2.eqtls/eqtls_fine_map/cardiac_eqtls.isoform.txt\", sep = \"\\t\", header = TRUE , data.table = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "rbps        = data.frame(rbp = c(\"FMR1\", \"FUS\", \"FXR2\", \"HNRNPA1\", \"HNRNPL\", \"RBM5\", \"SRSF7\"), folder = \"\")\n",
    "rbps$folder = paste(\"/frazer01/home/tarthur/projects/CARDIPS/fine_mapping/analysis/2020-10-08-fgwas\", tolower(rbps$rbp), \"peakfiles\", sep = \"/\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Write BED file with all fine mapped eVariants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "fm2iso       = merge(fm, geneinfo[,c(\"transcript_id\", \"start\", \"end\", \"strand\")])\n",
    "fm2iso       = fm2iso[fm2iso$pos >= fm2iso$start & fm2iso$pos <= fm2iso$end,]\n",
    "fm2iso$tr2id = paste(fm2iso$transcript_id, fm2iso$type, fm2iso$id, sep = \":\")\n",
    "fm2iso$chrom = paste(\"chr\", fm2iso$chrom, sep = \"\")\n",
    "fm2iso       = fm2iso[order(fm2iso$chrom, fm2iso$pos),]\n",
    "fm2iso$score = 0\n",
    "fm_bed       = paste(outfolder, \"fm.bed\", sep = \"/\")\n",
    "\n",
    "fwrite(unique(fm2iso[,c(\"chrom\", \"pos\", \"pos\", \"tr2id\", \"score\", \"strand\")]), fm_bed, sep = \"\\t\", col.names = FALSE, row.names = FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Intersect:\n",
    "- fine mapped eVariants with RBP loci (eCLIP)\n",
    "- coloc with GWAS (by eIsoform or by PPA from coloc)\n",
    "- eVariants that affect RBP motifs (FIMO): all eVariants that coloc?\n",
    "- find examples (hopefully for some RBPs that are enriched for GWAS traits in FGWAS)\n",
    "\n",
    "## intersect fine mapped eVariants with RBP loci (eCLIP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersect_rbp = function(rbp, rbps, fm)\n",
    "{\n",
    "    #bedfiles = list.files(rbps[rbps$rbp == rbp, \"folder\"], full.names = TRUE)\n",
    "    command  = paste(\"bedtools\", \"intersect\", \"-wa\", \"-a\", fm_bed, \"-b\", paste(rbps[rbps$rbp == rbp, \"folder\"], \"*\", sep = \"/\"), \"|\", \"cut\", \"-f\", 4)\n",
    "    overlap  = system(command, intern = TRUE)\n",
    "    \n",
    "    return(overlap)\n",
    "}\n",
    "\n",
    "overlap_list        = lapply(rbps$rbp, function(rbp){intersect_rbp(rbp, rbps, fm)})\n",
    "names(overlap_list) = rbps$rbp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "intdata       = fm2iso\n",
    "intdata$score = NULL\n",
    "\n",
    "for(rbp in rbps$rbp)\n",
    "{\n",
    "    intdata[, paste(\"eclip\", rbp, sep = \".\")] = FALSE\n",
    "    intdata[ intdata$tr2id %in% overlap_list[[rbp]], paste(\"eclip\", rbp, sep = \".\")] = TRUE\n",
    "}\n",
    "\n",
    "intdata$eclip = FALSE\n",
    "intdata[intdata$tr2id %in% unique(unlist(overlap_list)), \"eclip\"] = TRUE\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## intersect with coloc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "intdata$tr2type         = paste(intdata$transcript_id, intdata$type)\n",
    "int2coloc               = intdata[intdata$eclip == TRUE, c(\"transcript_id\", \"type\", \"id\", \"pp\")]\n",
    "int2coloc$pp_fm         = int2coloc$pp\n",
    "int2coloc$pp            = NULL\n",
    "int2coloc               = merge(int2coloc, coloc[, c(\"transcript_id\", \"type\", \"trait\", \"PP.H4.abf\")], by = c(\"transcript_id\", \"type\"))\n",
    "int2coloc$pp_coloc_gene = int2coloc$PP.H4.abf\n",
    "int2coloc$PP.H4.abf     = NULL\n",
    "#int2coloc               = int2coloc[ int2coloc$pp_coloc_gene >= 0.5,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "int2coloc_var = as.data.frame(rbindlist(lapply(sort(unique(int2coloc$transcript_id)), function(x)\n",
    "{\n",
    "    indata  = fread(paste(\"pipeline/6.1.coloc_gwas/coloc.isoform\", paste(\"snp\", x, \"txt\", sep = \".\"), sep = \"/\"), sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "    outdata = merge(int2coloc, indata[,c(\"transcript_id\", \"type\", \"id\", \"trait\", \"pp_snp\")], by = c(\"transcript_id\", \"type\", \"id\", \"trait\"))\n",
    "    return(outdata)\n",
    "})), stringsAsFactors = FALSE)\n",
    "\n",
    "fwrite(int2coloc_var, paste(outfolder, \"int2coloc_var.txt\", sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "intdata  $coloc_gene = FALSE\n",
    "int2coloc$tr2id      = paste(int2coloc$transcript_id, int2coloc$type, int2coloc$id, sep = \":\")\n",
    "\n",
    "for(trait in sort(unique(int2coloc$trait)))\n",
    "{\n",
    "    intdata[, paste(\"coloc_gene\", trait, sep = \".\")] = FALSE\n",
    "    intdata[ intdata$tr2id %in% int2coloc[int2coloc$trait == trait & int2coloc$pp_coloc_gene >= 0.5, \"tr2id\"], paste(\"coloc_gene\", trait, sep = \".\")] = TRUE\n",
    "}\n",
    "\n",
    "intdata[rowSums(intdata[, paste(\"coloc_gene\", sort(unique(int2coloc$trait)), sep = \".\")]) > 0, \"coloc_gene\"] = TRUE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "intdata      $coloc_snp = FALSE\n",
    "int2coloc_var$tr2id     = paste(int2coloc_var$transcript_id, int2coloc_var$type, int2coloc_var$id, sep = \":\")\n",
    "\n",
    "for(trait in sort(unique(int2coloc_var$trait)))\n",
    "{\n",
    "    intdata[, paste(\"coloc_snp\", trait, sep = \".\")] = FALSE\n",
    "    intdata[ intdata$tr2id %in% int2coloc_var[int2coloc_var$trait == trait & int2coloc_var$pp_snp >= 0.01, \"tr2id\"], paste(\"coloc_snp\", trait, sep = \".\")] = TRUE\n",
    "}\n",
    "\n",
    "intdata[rowSums(intdata[, paste(\"coloc_snp\", sort(unique(int2coloc$trait)), sep = \".\")]) > 0, \"coloc_snp\"] = TRUE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 1548 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>transcript_id</th><th scope=col>type</th><th scope=col>gene_name</th><th scope=col>id</th><th scope=col>pp</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>58451</th><td>ENST00000472194.6_1</td><td>0</td><td>ATAD3B  </td><td>VAR_1_1415063_T_G </td><td>0.43642791</td></tr>\n",
       "\t<tr><th scope=row>58452</th><td>ENST00000472194.6_1</td><td>0</td><td>ATAD3B  </td><td>VAR_1_1415476_G_C </td><td>0.44240107</td></tr>\n",
       "\t<tr><th scope=row>62109</th><td>ENST00000480991.1_2</td><td>0</td><td>SLC35E2B</td><td>VAR_1_1596500_C_T </td><td>0.83906331</td></tr>\n",
       "\t<tr><th scope=row>102009</th><td>ENST00000611123.1_2</td><td>0</td><td>SLC35E2B</td><td>VAR_1_1596500_C_T </td><td>0.47855651</td></tr>\n",
       "\t<tr><th scope=row>102013</th><td>ENST00000611123.1_2</td><td>1</td><td>SLC35E2B</td><td>VAR_1_1596500_C_T </td><td>0.01638872</td></tr>\n",
       "\t<tr><th scope=row>103095</th><td>ENST00000617444.5_3</td><td>0</td><td>SLC35E2B</td><td>VAR_1_1596500_C_T </td><td>0.01864093</td></tr>\n",
       "\t<tr><th scope=row>102015</th><td>ENST00000611123.1_2</td><td>2</td><td>SLC35E2B</td><td>VAR_1_1602587_C_T </td><td>0.47960877</td></tr>\n",
       "\t<tr><th scope=row>69003</th><td>ENST00000496676.5_1</td><td>0</td><td>RNF207  </td><td>VAR_1_6278414_A_G </td><td>0.33245518</td></tr>\n",
       "\t<tr><th scope=row>30465</th><td>ENST00000377705.6_3</td><td>0</td><td>NOL9    </td><td>VAR_1_6583270_C_G </td><td>0.02425345</td></tr>\n",
       "\t<tr><th scope=row>30470</th><td>ENST00000377705.6_3</td><td>0</td><td>NOL9    </td><td>VAR_1_6589641_C_T </td><td>0.01234203</td></tr>\n",
       "\t<tr><th scope=row>30475</th><td>ENST00000377705.6_3</td><td>0</td><td>NOL9    </td><td>VAR_1_6614535_G_A </td><td>0.01374464</td></tr>\n",
       "\t<tr><th scope=row>242</th><td>ENST00000054650.8_1</td><td>0</td><td>THAP3   </td><td>VAR_1_6689319_G_A </td><td>0.01087085</td></tr>\n",
       "\t<tr><th scope=row>59003</th><td>ENST00000473578.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6853160_T_C </td><td>0.03433626</td></tr>\n",
       "\t<tr><th scope=row>59973</th><td>ENST00000476163.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6853160_T_C </td><td>0.10605258</td></tr>\n",
       "\t<tr><th scope=row>59005</th><td>ENST00000473578.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6855091_A_G </td><td>0.01647571</td></tr>\n",
       "\t<tr><th scope=row>59975</th><td>ENST00000476163.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6855091_A_G </td><td>0.03510509</td></tr>\n",
       "\t<tr><th scope=row>59008</th><td>ENST00000473578.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6863342_G_A </td><td>0.02809342</td></tr>\n",
       "\t<tr><th scope=row>59978</th><td>ENST00000476163.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6863342_G_A </td><td>0.17743252</td></tr>\n",
       "\t<tr><th scope=row>59980</th><td>ENST00000476163.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6893053_G_A </td><td>0.01377665</td></tr>\n",
       "\t<tr><th scope=row>59014</th><td>ENST00000473578.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6914046_C_T </td><td>0.03998138</td></tr>\n",
       "\t<tr><th scope=row>59983</th><td>ENST00000476163.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6914046_C_T </td><td>0.09403662</td></tr>\n",
       "\t<tr><th scope=row>59984</th><td>ENST00000476163.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6939528_G_A </td><td>0.02416656</td></tr>\n",
       "\t<tr><th scope=row>59986</th><td>ENST00000476163.5_1</td><td>0</td><td>CAMTA1  </td><td>VAR_1_6945530_G_A </td><td>0.01414979</td></tr>\n",
       "\t<tr><th scope=row>2647</th><td>ENST00000240185.8_2</td><td>0</td><td>TARDBP  </td><td>VAR_1_11079413_T_A</td><td>0.03171186</td></tr>\n",
       "\t<tr><th scope=row>37516</th><td>ENST00000400892.3_2</td><td>0</td><td>CLCN6   </td><td>VAR_1_11898228_T_C</td><td>0.04601849</td></tr>\n",
       "\t<tr><th scope=row>37521</th><td>ENST00000400892.3_2</td><td>0</td><td>CLCN6   </td><td>VAR_1_11900802_T_C</td><td>0.04601849</td></tr>\n",
       "\t<tr><th scope=row>37522</th><td>ENST00000400892.3_2</td><td>0</td><td>CLCN6   </td><td>VAR_1_11900825_G_T</td><td>0.04601849</td></tr>\n",
       "\t<tr><th scope=row>37523</th><td>ENST00000400892.3_2</td><td>0</td><td>CLCN6   </td><td>VAR_1_11901592_T_C</td><td>0.04601849</td></tr>\n",
       "\t<tr><th scope=row>37527</th><td>ENST00000400892.3_2</td><td>0</td><td>CLCN6   </td><td>VAR_1_11905633_T_C</td><td>0.06511452</td></tr>\n",
       "\t<tr><th scope=row>44343</th><td>ENST00000428417.5_1</td><td>0</td><td>TMEM51  </td><td>VAR_1_15480612_A_T</td><td>0.02919739</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>75123</th><td>ENST00000518166.5_5 </td><td>0</td><td>NCALD     </td><td>VAR_8_102902865_A_G</td><td>0.06359243</td></tr>\n",
       "\t<tr><th scope=row>38913</th><td>ENST00000407775.7_2 </td><td>0</td><td>ZFPM2     </td><td>VAR_8_106456683_A_G</td><td>0.01046428</td></tr>\n",
       "\t<tr><th scope=row>38915</th><td>ENST00000407775.7_2 </td><td>0</td><td>ZFPM2     </td><td>VAR_8_106460871_C_T</td><td>0.01949268</td></tr>\n",
       "\t<tr><th scope=row>38919</th><td>ENST00000407775.7_2 </td><td>0</td><td>ZFPM2     </td><td>VAR_8_106463774_C_T</td><td>0.01369919</td></tr>\n",
       "\t<tr><th scope=row>75000</th><td>ENST00000517878.5_2 </td><td>0</td><td>SLC45A4   </td><td>VAR_8_142229891_A_G</td><td>0.01889503</td></tr>\n",
       "\t<tr><th scope=row>75593</th><td>ENST00000520105.5_1 </td><td>0</td><td>PTP4A3    </td><td>VAR_8_142440608_T_C</td><td>0.09070998</td></tr>\n",
       "\t<tr><th scope=row>76937</th><td>ENST00000524028.1_1 </td><td>0</td><td>PTP4A3    </td><td>VAR_8_142440608_T_C</td><td>0.12922554</td></tr>\n",
       "\t<tr><th scope=row>75594</th><td>ENST00000520105.5_1 </td><td>0</td><td>PTP4A3    </td><td>VAR_8_142440751_A_G</td><td>0.09070998</td></tr>\n",
       "\t<tr><th scope=row>76938</th><td>ENST00000524028.1_1 </td><td>0</td><td>PTP4A3    </td><td>VAR_8_142440751_A_G</td><td>0.12922554</td></tr>\n",
       "\t<tr><th scope=row>77601</th><td>ENST00000526007.5_2 </td><td>0</td><td>PARP10    </td><td>VAR_8_145057868_A_G</td><td>0.25435154</td></tr>\n",
       "\t<tr><th scope=row>78139</th><td>ENST00000527222.5_3 </td><td>0</td><td>ZNF252P   </td><td>VAR_8_146220011_G_T</td><td>0.01802411</td></tr>\n",
       "\t<tr><th scope=row>32208</th><td>ENST00000382286.2_1 </td><td>0</td><td>KANK1     </td><td>VAR_9_745289_C_T   </td><td>0.07556146</td></tr>\n",
       "\t<tr><th scope=row>31765</th><td>ENST00000380502.8_4 </td><td>0</td><td>HAUS6     </td><td>VAR_9_19056147_T_TA</td><td>0.15782824</td></tr>\n",
       "\t<tr><th scope=row>94477</th><td>ENST00000580900.5_2 </td><td>0</td><td>MTAP      </td><td>VAR_9_21813718_C_T </td><td>0.89192898</td></tr>\n",
       "\t<tr><th scope=row>23391</th><td>ENST00000360384.9_3 </td><td>0</td><td>HNRNPK    </td><td>VAR_9_86594798_C_T </td><td>0.01078080</td></tr>\n",
       "\t<tr><th scope=row>30013</th><td>ENST00000376263.8_4 </td><td>0</td><td>HNRNPK    </td><td>VAR_9_86594798_C_T </td><td>0.04192885</td></tr>\n",
       "\t<tr><th scope=row>23392</th><td>ENST00000360384.9_3 </td><td>0</td><td>HNRNPK    </td><td>VAR_9_86595070_C_T </td><td>0.01078080</td></tr>\n",
       "\t<tr><th scope=row>30014</th><td>ENST00000376263.8_4 </td><td>0</td><td>HNRNPK    </td><td>VAR_9_86595070_C_T </td><td>0.04192885</td></tr>\n",
       "\t<tr><th scope=row>111640</th><td>ENST00000670189.1_2 </td><td>1</td><td>LINC00476 </td><td>VAR_9_98637430_C_T </td><td>0.12664946</td></tr>\n",
       "\t<tr><th scope=row>66654</th><td>ENST00000490599.2_3 </td><td>0</td><td>SLC35D2   </td><td>VAR_9_99111702_C_G </td><td>0.03201607</td></tr>\n",
       "\t<tr><th scope=row>39677</th><td>ENST00000412285.6_2 </td><td>0</td><td>CDC14B    </td><td>VAR_9_99270577_C_T </td><td>0.02002162</td></tr>\n",
       "\t<tr><th scope=row>39681</th><td>ENST00000412285.6_2 </td><td>0</td><td>CDC14B    </td><td>VAR_9_99282269_G_C </td><td>0.01477226</td></tr>\n",
       "\t<tr><th scope=row>39008</th><td>ENST00000409155.8_3 </td><td>0</td><td>ALAD      </td><td>VAR_9_116152990_C_G</td><td>1.00000000</td></tr>\n",
       "\t<tr><th scope=row>56868</th><td>ENST00000468504.5_2 </td><td>0</td><td>ALAD      </td><td>VAR_9_116152990_C_G</td><td>1.00000000</td></tr>\n",
       "\t<tr><th scope=row>28636</th><td>ENST00000373903.1_2 </td><td>0</td><td>PSMD5     </td><td>VAR_9_123586737_T_G</td><td>0.12414100</td></tr>\n",
       "\t<tr><th scope=row>106245</th><td>ENST00000640327.1_1 </td><td>0</td><td>CUTALP    </td><td>VAR_9_123610288_C_T</td><td>0.01671827</td></tr>\n",
       "\t<tr><th scope=row>28426</th><td>ENST00000373146.5_2 </td><td>0</td><td>ST6GALNAC6</td><td>VAR_9_130657833_G_A</td><td>0.05914459</td></tr>\n",
       "\t<tr><th scope=row>23837</th><td>ENST00000361256.10_3</td><td>0</td><td>SPOUT1    </td><td>VAR_9_131583170_T_C</td><td>0.08328318</td></tr>\n",
       "\t<tr><th scope=row>61847</th><td>ENST00000480366.1_2 </td><td>0</td><td>SPOUT1    </td><td>VAR_9_131586883_C_T</td><td>0.01037822</td></tr>\n",
       "\t<tr><th scope=row>48927</th><td>ENST00000444897.2_3 </td><td>0</td><td>PMPCA     </td><td>VAR_9_139317869_T_C</td><td>0.04693706</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1548 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & transcript\\_id & type & gene\\_name & id & pp\\\\\n",
       "  & <chr> & <int> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t58451 & ENST00000472194.6\\_1 & 0 & ATAD3B   & VAR\\_1\\_1415063\\_T\\_G  & 0.43642791\\\\\n",
       "\t58452 & ENST00000472194.6\\_1 & 0 & ATAD3B   & VAR\\_1\\_1415476\\_G\\_C  & 0.44240107\\\\\n",
       "\t62109 & ENST00000480991.1\\_2 & 0 & SLC35E2B & VAR\\_1\\_1596500\\_C\\_T  & 0.83906331\\\\\n",
       "\t102009 & ENST00000611123.1\\_2 & 0 & SLC35E2B & VAR\\_1\\_1596500\\_C\\_T  & 0.47855651\\\\\n",
       "\t102013 & ENST00000611123.1\\_2 & 1 & SLC35E2B & VAR\\_1\\_1596500\\_C\\_T  & 0.01638872\\\\\n",
       "\t103095 & ENST00000617444.5\\_3 & 0 & SLC35E2B & VAR\\_1\\_1596500\\_C\\_T  & 0.01864093\\\\\n",
       "\t102015 & ENST00000611123.1\\_2 & 2 & SLC35E2B & VAR\\_1\\_1602587\\_C\\_T  & 0.47960877\\\\\n",
       "\t69003 & ENST00000496676.5\\_1 & 0 & RNF207   & VAR\\_1\\_6278414\\_A\\_G  & 0.33245518\\\\\n",
       "\t30465 & ENST00000377705.6\\_3 & 0 & NOL9     & VAR\\_1\\_6583270\\_C\\_G  & 0.02425345\\\\\n",
       "\t30470 & ENST00000377705.6\\_3 & 0 & NOL9     & VAR\\_1\\_6589641\\_C\\_T  & 0.01234203\\\\\n",
       "\t30475 & ENST00000377705.6\\_3 & 0 & NOL9     & VAR\\_1\\_6614535\\_G\\_A  & 0.01374464\\\\\n",
       "\t242 & ENST00000054650.8\\_1 & 0 & THAP3    & VAR\\_1\\_6689319\\_G\\_A  & 0.01087085\\\\\n",
       "\t59003 & ENST00000473578.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6853160\\_T\\_C  & 0.03433626\\\\\n",
       "\t59973 & ENST00000476163.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6853160\\_T\\_C  & 0.10605258\\\\\n",
       "\t59005 & ENST00000473578.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6855091\\_A\\_G  & 0.01647571\\\\\n",
       "\t59975 & ENST00000476163.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6855091\\_A\\_G  & 0.03510509\\\\\n",
       "\t59008 & ENST00000473578.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6863342\\_G\\_A  & 0.02809342\\\\\n",
       "\t59978 & ENST00000476163.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6863342\\_G\\_A  & 0.17743252\\\\\n",
       "\t59980 & ENST00000476163.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6893053\\_G\\_A  & 0.01377665\\\\\n",
       "\t59014 & ENST00000473578.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6914046\\_C\\_T  & 0.03998138\\\\\n",
       "\t59983 & ENST00000476163.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6914046\\_C\\_T  & 0.09403662\\\\\n",
       "\t59984 & ENST00000476163.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6939528\\_G\\_A  & 0.02416656\\\\\n",
       "\t59986 & ENST00000476163.5\\_1 & 0 & CAMTA1   & VAR\\_1\\_6945530\\_G\\_A  & 0.01414979\\\\\n",
       "\t2647 & ENST00000240185.8\\_2 & 0 & TARDBP   & VAR\\_1\\_11079413\\_T\\_A & 0.03171186\\\\\n",
       "\t37516 & ENST00000400892.3\\_2 & 0 & CLCN6    & VAR\\_1\\_11898228\\_T\\_C & 0.04601849\\\\\n",
       "\t37521 & ENST00000400892.3\\_2 & 0 & CLCN6    & VAR\\_1\\_11900802\\_T\\_C & 0.04601849\\\\\n",
       "\t37522 & ENST00000400892.3\\_2 & 0 & CLCN6    & VAR\\_1\\_11900825\\_G\\_T & 0.04601849\\\\\n",
       "\t37523 & ENST00000400892.3\\_2 & 0 & CLCN6    & VAR\\_1\\_11901592\\_T\\_C & 0.04601849\\\\\n",
       "\t37527 & ENST00000400892.3\\_2 & 0 & CLCN6    & VAR\\_1\\_11905633\\_T\\_C & 0.06511452\\\\\n",
       "\t44343 & ENST00000428417.5\\_1 & 0 & TMEM51   & VAR\\_1\\_15480612\\_A\\_T & 0.02919739\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t75123 & ENST00000518166.5\\_5  & 0 & NCALD      & VAR\\_8\\_102902865\\_A\\_G & 0.06359243\\\\\n",
       "\t38913 & ENST00000407775.7\\_2  & 0 & ZFPM2      & VAR\\_8\\_106456683\\_A\\_G & 0.01046428\\\\\n",
       "\t38915 & ENST00000407775.7\\_2  & 0 & ZFPM2      & VAR\\_8\\_106460871\\_C\\_T & 0.01949268\\\\\n",
       "\t38919 & ENST00000407775.7\\_2  & 0 & ZFPM2      & VAR\\_8\\_106463774\\_C\\_T & 0.01369919\\\\\n",
       "\t75000 & ENST00000517878.5\\_2  & 0 & SLC45A4    & VAR\\_8\\_142229891\\_A\\_G & 0.01889503\\\\\n",
       "\t75593 & ENST00000520105.5\\_1  & 0 & PTP4A3     & VAR\\_8\\_142440608\\_T\\_C & 0.09070998\\\\\n",
       "\t76937 & ENST00000524028.1\\_1  & 0 & PTP4A3     & VAR\\_8\\_142440608\\_T\\_C & 0.12922554\\\\\n",
       "\t75594 & ENST00000520105.5\\_1  & 0 & PTP4A3     & VAR\\_8\\_142440751\\_A\\_G & 0.09070998\\\\\n",
       "\t76938 & ENST00000524028.1\\_1  & 0 & PTP4A3     & VAR\\_8\\_142440751\\_A\\_G & 0.12922554\\\\\n",
       "\t77601 & ENST00000526007.5\\_2  & 0 & PARP10     & VAR\\_8\\_145057868\\_A\\_G & 0.25435154\\\\\n",
       "\t78139 & ENST00000527222.5\\_3  & 0 & ZNF252P    & VAR\\_8\\_146220011\\_G\\_T & 0.01802411\\\\\n",
       "\t32208 & ENST00000382286.2\\_1  & 0 & KANK1      & VAR\\_9\\_745289\\_C\\_T    & 0.07556146\\\\\n",
       "\t31765 & ENST00000380502.8\\_4  & 0 & HAUS6      & VAR\\_9\\_19056147\\_T\\_TA & 0.15782824\\\\\n",
       "\t94477 & ENST00000580900.5\\_2  & 0 & MTAP       & VAR\\_9\\_21813718\\_C\\_T  & 0.89192898\\\\\n",
       "\t23391 & ENST00000360384.9\\_3  & 0 & HNRNPK     & VAR\\_9\\_86594798\\_C\\_T  & 0.01078080\\\\\n",
       "\t30013 & ENST00000376263.8\\_4  & 0 & HNRNPK     & VAR\\_9\\_86594798\\_C\\_T  & 0.04192885\\\\\n",
       "\t23392 & ENST00000360384.9\\_3  & 0 & HNRNPK     & VAR\\_9\\_86595070\\_C\\_T  & 0.01078080\\\\\n",
       "\t30014 & ENST00000376263.8\\_4  & 0 & HNRNPK     & VAR\\_9\\_86595070\\_C\\_T  & 0.04192885\\\\\n",
       "\t111640 & ENST00000670189.1\\_2  & 1 & LINC00476  & VAR\\_9\\_98637430\\_C\\_T  & 0.12664946\\\\\n",
       "\t66654 & ENST00000490599.2\\_3  & 0 & SLC35D2    & VAR\\_9\\_99111702\\_C\\_G  & 0.03201607\\\\\n",
       "\t39677 & ENST00000412285.6\\_2  & 0 & CDC14B     & VAR\\_9\\_99270577\\_C\\_T  & 0.02002162\\\\\n",
       "\t39681 & ENST00000412285.6\\_2  & 0 & CDC14B     & VAR\\_9\\_99282269\\_G\\_C  & 0.01477226\\\\\n",
       "\t39008 & ENST00000409155.8\\_3  & 0 & ALAD       & VAR\\_9\\_116152990\\_C\\_G & 1.00000000\\\\\n",
       "\t56868 & ENST00000468504.5\\_2  & 0 & ALAD       & VAR\\_9\\_116152990\\_C\\_G & 1.00000000\\\\\n",
       "\t28636 & ENST00000373903.1\\_2  & 0 & PSMD5      & VAR\\_9\\_123586737\\_T\\_G & 0.12414100\\\\\n",
       "\t106245 & ENST00000640327.1\\_1  & 0 & CUTALP     & VAR\\_9\\_123610288\\_C\\_T & 0.01671827\\\\\n",
       "\t28426 & ENST00000373146.5\\_2  & 0 & ST6GALNAC6 & VAR\\_9\\_130657833\\_G\\_A & 0.05914459\\\\\n",
       "\t23837 & ENST00000361256.10\\_3 & 0 & SPOUT1     & VAR\\_9\\_131583170\\_T\\_C & 0.08328318\\\\\n",
       "\t61847 & ENST00000480366.1\\_2  & 0 & SPOUT1     & VAR\\_9\\_131586883\\_C\\_T & 0.01037822\\\\\n",
       "\t48927 & ENST00000444897.2\\_3  & 0 & PMPCA      & VAR\\_9\\_139317869\\_T\\_C & 0.04693706\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1548 × 5\n",
       "\n",
       "| <!--/--> | transcript_id &lt;chr&gt; | type &lt;int&gt; | gene_name &lt;chr&gt; | id &lt;chr&gt; | pp &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 58451 | ENST00000472194.6_1 | 0 | ATAD3B   | VAR_1_1415063_T_G  | 0.43642791 |\n",
       "| 58452 | ENST00000472194.6_1 | 0 | ATAD3B   | VAR_1_1415476_G_C  | 0.44240107 |\n",
       "| 62109 | ENST00000480991.1_2 | 0 | SLC35E2B | VAR_1_1596500_C_T  | 0.83906331 |\n",
       "| 102009 | ENST00000611123.1_2 | 0 | SLC35E2B | VAR_1_1596500_C_T  | 0.47855651 |\n",
       "| 102013 | ENST00000611123.1_2 | 1 | SLC35E2B | VAR_1_1596500_C_T  | 0.01638872 |\n",
       "| 103095 | ENST00000617444.5_3 | 0 | SLC35E2B | VAR_1_1596500_C_T  | 0.01864093 |\n",
       "| 102015 | ENST00000611123.1_2 | 2 | SLC35E2B | VAR_1_1602587_C_T  | 0.47960877 |\n",
       "| 69003 | ENST00000496676.5_1 | 0 | RNF207   | VAR_1_6278414_A_G  | 0.33245518 |\n",
       "| 30465 | ENST00000377705.6_3 | 0 | NOL9     | VAR_1_6583270_C_G  | 0.02425345 |\n",
       "| 30470 | ENST00000377705.6_3 | 0 | NOL9     | VAR_1_6589641_C_T  | 0.01234203 |\n",
       "| 30475 | ENST00000377705.6_3 | 0 | NOL9     | VAR_1_6614535_G_A  | 0.01374464 |\n",
       "| 242 | ENST00000054650.8_1 | 0 | THAP3    | VAR_1_6689319_G_A  | 0.01087085 |\n",
       "| 59003 | ENST00000473578.5_1 | 0 | CAMTA1   | VAR_1_6853160_T_C  | 0.03433626 |\n",
       "| 59973 | ENST00000476163.5_1 | 0 | CAMTA1   | VAR_1_6853160_T_C  | 0.10605258 |\n",
       "| 59005 | ENST00000473578.5_1 | 0 | CAMTA1   | VAR_1_6855091_A_G  | 0.01647571 |\n",
       "| 59975 | ENST00000476163.5_1 | 0 | CAMTA1   | VAR_1_6855091_A_G  | 0.03510509 |\n",
       "| 59008 | ENST00000473578.5_1 | 0 | CAMTA1   | VAR_1_6863342_G_A  | 0.02809342 |\n",
       "| 59978 | ENST00000476163.5_1 | 0 | CAMTA1   | VAR_1_6863342_G_A  | 0.17743252 |\n",
       "| 59980 | ENST00000476163.5_1 | 0 | CAMTA1   | VAR_1_6893053_G_A  | 0.01377665 |\n",
       "| 59014 | ENST00000473578.5_1 | 0 | CAMTA1   | VAR_1_6914046_C_T  | 0.03998138 |\n",
       "| 59983 | ENST00000476163.5_1 | 0 | CAMTA1   | VAR_1_6914046_C_T  | 0.09403662 |\n",
       "| 59984 | ENST00000476163.5_1 | 0 | CAMTA1   | VAR_1_6939528_G_A  | 0.02416656 |\n",
       "| 59986 | ENST00000476163.5_1 | 0 | CAMTA1   | VAR_1_6945530_G_A  | 0.01414979 |\n",
       "| 2647 | ENST00000240185.8_2 | 0 | TARDBP   | VAR_1_11079413_T_A | 0.03171186 |\n",
       "| 37516 | ENST00000400892.3_2 | 0 | CLCN6    | VAR_1_11898228_T_C | 0.04601849 |\n",
       "| 37521 | ENST00000400892.3_2 | 0 | CLCN6    | VAR_1_11900802_T_C | 0.04601849 |\n",
       "| 37522 | ENST00000400892.3_2 | 0 | CLCN6    | VAR_1_11900825_G_T | 0.04601849 |\n",
       "| 37523 | ENST00000400892.3_2 | 0 | CLCN6    | VAR_1_11901592_T_C | 0.04601849 |\n",
       "| 37527 | ENST00000400892.3_2 | 0 | CLCN6    | VAR_1_11905633_T_C | 0.06511452 |\n",
       "| 44343 | ENST00000428417.5_1 | 0 | TMEM51   | VAR_1_15480612_A_T | 0.02919739 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 75123 | ENST00000518166.5_5  | 0 | NCALD      | VAR_8_102902865_A_G | 0.06359243 |\n",
       "| 38913 | ENST00000407775.7_2  | 0 | ZFPM2      | VAR_8_106456683_A_G | 0.01046428 |\n",
       "| 38915 | ENST00000407775.7_2  | 0 | ZFPM2      | VAR_8_106460871_C_T | 0.01949268 |\n",
       "| 38919 | ENST00000407775.7_2  | 0 | ZFPM2      | VAR_8_106463774_C_T | 0.01369919 |\n",
       "| 75000 | ENST00000517878.5_2  | 0 | SLC45A4    | VAR_8_142229891_A_G | 0.01889503 |\n",
       "| 75593 | ENST00000520105.5_1  | 0 | PTP4A3     | VAR_8_142440608_T_C | 0.09070998 |\n",
       "| 76937 | ENST00000524028.1_1  | 0 | PTP4A3     | VAR_8_142440608_T_C | 0.12922554 |\n",
       "| 75594 | ENST00000520105.5_1  | 0 | PTP4A3     | VAR_8_142440751_A_G | 0.09070998 |\n",
       "| 76938 | ENST00000524028.1_1  | 0 | PTP4A3     | VAR_8_142440751_A_G | 0.12922554 |\n",
       "| 77601 | ENST00000526007.5_2  | 0 | PARP10     | VAR_8_145057868_A_G | 0.25435154 |\n",
       "| 78139 | ENST00000527222.5_3  | 0 | ZNF252P    | VAR_8_146220011_G_T | 0.01802411 |\n",
       "| 32208 | ENST00000382286.2_1  | 0 | KANK1      | VAR_9_745289_C_T    | 0.07556146 |\n",
       "| 31765 | ENST00000380502.8_4  | 0 | HAUS6      | VAR_9_19056147_T_TA | 0.15782824 |\n",
       "| 94477 | ENST00000580900.5_2  | 0 | MTAP       | VAR_9_21813718_C_T  | 0.89192898 |\n",
       "| 23391 | ENST00000360384.9_3  | 0 | HNRNPK     | VAR_9_86594798_C_T  | 0.01078080 |\n",
       "| 30013 | ENST00000376263.8_4  | 0 | HNRNPK     | VAR_9_86594798_C_T  | 0.04192885 |\n",
       "| 23392 | ENST00000360384.9_3  | 0 | HNRNPK     | VAR_9_86595070_C_T  | 0.01078080 |\n",
       "| 30014 | ENST00000376263.8_4  | 0 | HNRNPK     | VAR_9_86595070_C_T  | 0.04192885 |\n",
       "| 111640 | ENST00000670189.1_2  | 1 | LINC00476  | VAR_9_98637430_C_T  | 0.12664946 |\n",
       "| 66654 | ENST00000490599.2_3  | 0 | SLC35D2    | VAR_9_99111702_C_G  | 0.03201607 |\n",
       "| 39677 | ENST00000412285.6_2  | 0 | CDC14B     | VAR_9_99270577_C_T  | 0.02002162 |\n",
       "| 39681 | ENST00000412285.6_2  | 0 | CDC14B     | VAR_9_99282269_G_C  | 0.01477226 |\n",
       "| 39008 | ENST00000409155.8_3  | 0 | ALAD       | VAR_9_116152990_C_G | 1.00000000 |\n",
       "| 56868 | ENST00000468504.5_2  | 0 | ALAD       | VAR_9_116152990_C_G | 1.00000000 |\n",
       "| 28636 | ENST00000373903.1_2  | 0 | PSMD5      | VAR_9_123586737_T_G | 0.12414100 |\n",
       "| 106245 | ENST00000640327.1_1  | 0 | CUTALP     | VAR_9_123610288_C_T | 0.01671827 |\n",
       "| 28426 | ENST00000373146.5_2  | 0 | ST6GALNAC6 | VAR_9_130657833_G_A | 0.05914459 |\n",
       "| 23837 | ENST00000361256.10_3 | 0 | SPOUT1     | VAR_9_131583170_T_C | 0.08328318 |\n",
       "| 61847 | ENST00000480366.1_2  | 0 | SPOUT1     | VAR_9_131586883_C_T | 0.01037822 |\n",
       "| 48927 | ENST00000444897.2_3  | 0 | PMPCA      | VAR_9_139317869_T_C | 0.04693706 |\n",
       "\n"
      ],
      "text/plain": [
       "       transcript_id        type gene_name  id                  pp        \n",
       "58451  ENST00000472194.6_1  0    ATAD3B     VAR_1_1415063_T_G   0.43642791\n",
       "58452  ENST00000472194.6_1  0    ATAD3B     VAR_1_1415476_G_C   0.44240107\n",
       "62109  ENST00000480991.1_2  0    SLC35E2B   VAR_1_1596500_C_T   0.83906331\n",
       "102009 ENST00000611123.1_2  0    SLC35E2B   VAR_1_1596500_C_T   0.47855651\n",
       "102013 ENST00000611123.1_2  1    SLC35E2B   VAR_1_1596500_C_T   0.01638872\n",
       "103095 ENST00000617444.5_3  0    SLC35E2B   VAR_1_1596500_C_T   0.01864093\n",
       "102015 ENST00000611123.1_2  2    SLC35E2B   VAR_1_1602587_C_T   0.47960877\n",
       "69003  ENST00000496676.5_1  0    RNF207     VAR_1_6278414_A_G   0.33245518\n",
       "30465  ENST00000377705.6_3  0    NOL9       VAR_1_6583270_C_G   0.02425345\n",
       "30470  ENST00000377705.6_3  0    NOL9       VAR_1_6589641_C_T   0.01234203\n",
       "30475  ENST00000377705.6_3  0    NOL9       VAR_1_6614535_G_A   0.01374464\n",
       "242    ENST00000054650.8_1  0    THAP3      VAR_1_6689319_G_A   0.01087085\n",
       "59003  ENST00000473578.5_1  0    CAMTA1     VAR_1_6853160_T_C   0.03433626\n",
       "59973  ENST00000476163.5_1  0    CAMTA1     VAR_1_6853160_T_C   0.10605258\n",
       "59005  ENST00000473578.5_1  0    CAMTA1     VAR_1_6855091_A_G   0.01647571\n",
       "59975  ENST00000476163.5_1  0    CAMTA1     VAR_1_6855091_A_G   0.03510509\n",
       "59008  ENST00000473578.5_1  0    CAMTA1     VAR_1_6863342_G_A   0.02809342\n",
       "59978  ENST00000476163.5_1  0    CAMTA1     VAR_1_6863342_G_A   0.17743252\n",
       "59980  ENST00000476163.5_1  0    CAMTA1     VAR_1_6893053_G_A   0.01377665\n",
       "59014  ENST00000473578.5_1  0    CAMTA1     VAR_1_6914046_C_T   0.03998138\n",
       "59983  ENST00000476163.5_1  0    CAMTA1     VAR_1_6914046_C_T   0.09403662\n",
       "59984  ENST00000476163.5_1  0    CAMTA1     VAR_1_6939528_G_A   0.02416656\n",
       "59986  ENST00000476163.5_1  0    CAMTA1     VAR_1_6945530_G_A   0.01414979\n",
       "2647   ENST00000240185.8_2  0    TARDBP     VAR_1_11079413_T_A  0.03171186\n",
       "37516  ENST00000400892.3_2  0    CLCN6      VAR_1_11898228_T_C  0.04601849\n",
       "37521  ENST00000400892.3_2  0    CLCN6      VAR_1_11900802_T_C  0.04601849\n",
       "37522  ENST00000400892.3_2  0    CLCN6      VAR_1_11900825_G_T  0.04601849\n",
       "37523  ENST00000400892.3_2  0    CLCN6      VAR_1_11901592_T_C  0.04601849\n",
       "37527  ENST00000400892.3_2  0    CLCN6      VAR_1_11905633_T_C  0.06511452\n",
       "44343  ENST00000428417.5_1  0    TMEM51     VAR_1_15480612_A_T  0.02919739\n",
       "⋮      ⋮                    ⋮    ⋮          ⋮                   ⋮         \n",
       "75123  ENST00000518166.5_5  0    NCALD      VAR_8_102902865_A_G 0.06359243\n",
       "38913  ENST00000407775.7_2  0    ZFPM2      VAR_8_106456683_A_G 0.01046428\n",
       "38915  ENST00000407775.7_2  0    ZFPM2      VAR_8_106460871_C_T 0.01949268\n",
       "38919  ENST00000407775.7_2  0    ZFPM2      VAR_8_106463774_C_T 0.01369919\n",
       "75000  ENST00000517878.5_2  0    SLC45A4    VAR_8_142229891_A_G 0.01889503\n",
       "75593  ENST00000520105.5_1  0    PTP4A3     VAR_8_142440608_T_C 0.09070998\n",
       "76937  ENST00000524028.1_1  0    PTP4A3     VAR_8_142440608_T_C 0.12922554\n",
       "75594  ENST00000520105.5_1  0    PTP4A3     VAR_8_142440751_A_G 0.09070998\n",
       "76938  ENST00000524028.1_1  0    PTP4A3     VAR_8_142440751_A_G 0.12922554\n",
       "77601  ENST00000526007.5_2  0    PARP10     VAR_8_145057868_A_G 0.25435154\n",
       "78139  ENST00000527222.5_3  0    ZNF252P    VAR_8_146220011_G_T 0.01802411\n",
       "32208  ENST00000382286.2_1  0    KANK1      VAR_9_745289_C_T    0.07556146\n",
       "31765  ENST00000380502.8_4  0    HAUS6      VAR_9_19056147_T_TA 0.15782824\n",
       "94477  ENST00000580900.5_2  0    MTAP       VAR_9_21813718_C_T  0.89192898\n",
       "23391  ENST00000360384.9_3  0    HNRNPK     VAR_9_86594798_C_T  0.01078080\n",
       "30013  ENST00000376263.8_4  0    HNRNPK     VAR_9_86594798_C_T  0.04192885\n",
       "23392  ENST00000360384.9_3  0    HNRNPK     VAR_9_86595070_C_T  0.01078080\n",
       "30014  ENST00000376263.8_4  0    HNRNPK     VAR_9_86595070_C_T  0.04192885\n",
       "111640 ENST00000670189.1_2  1    LINC00476  VAR_9_98637430_C_T  0.12664946\n",
       "66654  ENST00000490599.2_3  0    SLC35D2    VAR_9_99111702_C_G  0.03201607\n",
       "39677  ENST00000412285.6_2  0    CDC14B     VAR_9_99270577_C_T  0.02002162\n",
       "39681  ENST00000412285.6_2  0    CDC14B     VAR_9_99282269_G_C  0.01477226\n",
       "39008  ENST00000409155.8_3  0    ALAD       VAR_9_116152990_C_G 1.00000000\n",
       "56868  ENST00000468504.5_2  0    ALAD       VAR_9_116152990_C_G 1.00000000\n",
       "28636  ENST00000373903.1_2  0    PSMD5      VAR_9_123586737_T_G 0.12414100\n",
       "106245 ENST00000640327.1_1  0    CUTALP     VAR_9_123610288_C_T 0.01671827\n",
       "28426  ENST00000373146.5_2  0    ST6GALNAC6 VAR_9_130657833_G_A 0.05914459\n",
       "23837  ENST00000361256.10_3 0    SPOUT1     VAR_9_131583170_T_C 0.08328318\n",
       "61847  ENST00000480366.1_2  0    SPOUT1     VAR_9_131586883_C_T 0.01037822\n",
       "48927  ENST00000444897.2_3  0    PMPCA      VAR_9_139317869_T_C 0.04693706"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tops = intdata[ intdata$eclip == TRUE & intdata$coloc_gene == TRUE & intdata$coloc_snp == TRUE, ]\n",
    "\n",
    "fwrite(int2coloc, paste(outfolder, \"int2coloc.txt\", sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n",
    "fwrite(intdata  , paste(outfolder, \"intdata.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n",
    "fwrite(tops[,c(\"transcript_id\", \"type\", \"gene_name\", \"id\", \"pp\")], paste(outfolder, \"tops.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n",
    "\n",
    "tops[,c(\"transcript_id\", \"type\", \"gene_name\", \"id\", \"pp\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "intdata_pp       = fm2iso\n",
    "intdata_pp$score = NULL\n",
    "\n",
    "for(rbp in rbps$rbp)\n",
    "{\n",
    "    intdata_pp[, paste(\"eclip\", rbp, sep = \".\")] = FALSE\n",
    "    intdata_pp[ intdata_pp$tr2id %in% overlap_list[[rbp]], paste(\"eclip\", rbp, sep = \".\")] = TRUE\n",
    "}\n",
    "\n",
    "intdata_pp$eclip = FALSE\n",
    "intdata_pp[intdata_pp$tr2id %in% unique(unlist(overlap_list)), \"eclip\"] = TRUE\n",
    "\n",
    "mycols = colnames(intdata_pp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "intdata_pp$coloc_gene = FALSE\n",
    "int2coloc $tr2id      = paste(int2coloc$transcript_id, int2coloc$type, int2coloc$id, sep = \":\")\n",
    "\n",
    "for(trait in sort(unique(int2coloc$trait)))\n",
    "{\n",
    "    tomerge           = int2coloc[int2coloc$trait == trait, c(\"tr2id\", \"pp_coloc_gene\")]\n",
    "    colnames(tomerge) = c(\"tr2id\", paste(\"coloc_gene\", trait, sep = \".\"))\n",
    "    intdata_pp        = merge(intdata_pp, tomerge, all.x = TRUE)\n",
    "}\n",
    "\n",
    "intdata_pp$coloc_gene = unlist(lapply(1:nrow(intdata_pp), function(ii)\n",
    "{\n",
    "    out    = 0\n",
    "    mydata = intdata_pp[ii, grepl(\"coloc_gene\\\\.\", colnames(intdata_pp)) ==TRUE]\n",
    "    if(length(mydata[is.na(mydata) == FALSE]) > 0){out = max(mydata, na.rm = TRUE)}\n",
    "    \n",
    "    return(out)\n",
    "}))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "biomarkers-30680-both_sexes-irnt 32682\n",
      "\n",
      "biomarkers-30690-both_sexes-irnt 32682\n",
      "\n",
      "biomarkers-30760-both_sexes-irnt 32682\n",
      "\n",
      "biomarkers-30780-both_sexes-irnt 32682\n",
      "\n",
      "biomarkers-30790-both_sexes-irnt 32682\n",
      "\n",
      "biomarkers-30870-both_sexes-irnt 32682\n",
      "\n",
      "categorical-20002-both_sexes-1074 32682\n",
      "\n",
      "categorical-20002-both_sexes-1075 32682\n",
      "\n",
      "categorical-20002-both_sexes-1081 32682\n",
      "\n",
      "categorical-6150-both_sexes-100 32682\n",
      "\n",
      "categorical-6150-both_sexes-2 32682\n",
      "\n",
      "categorical-6150-both_sexes-3 32682\n",
      "\n",
      "categorical-6150-both_sexes-4 32682\n",
      "\n",
      "categorical-6153-both_sexes-1 32682\n",
      "\n",
      "categorical-6153-both_sexes-2 32682\n",
      "\n",
      "categorical-6177-both_sexes-1 32682\n",
      "\n",
      "categorical-6177-both_sexes-100 32682\n",
      "\n",
      "categorical-6177-both_sexes-2 32682\n",
      "\n",
      "continuous-102-both_sexes-irnt 32682\n",
      "\n",
      "continuous-12336-both_sexes-irnt 32682\n",
      "\n",
      "continuous-12340-both_sexes-irnt 32682\n",
      "\n",
      "continuous-21001-both_sexes-irnt 32682\n",
      "\n",
      "continuous-3894-both_sexes-irnt 32682\n",
      "\n",
      "continuous-4079-both_sexes-irnt 32682\n",
      "\n",
      "continuous-4080-both_sexes-irnt 32682\n",
      "\n",
      "continuous-5983-both_sexes-irnt 32682\n",
      "\n",
      "continuous-6033-both_sexes-irnt 32682\n",
      "\n",
      "continuous-93-both_sexes-irnt 32682\n",
      "\n",
      "continuous-94-both_sexes-irnt 32682\n",
      "\n",
      "continuous-DBP-both_sexes-auto_medadj_irnt 32682\n",
      "\n",
      "continuous-DBP-both_sexes-auto_medadj_raw 32682\n",
      "\n",
      "continuous-DBP-both_sexes-combined_irnt 32682\n",
      "\n",
      "continuous-DBP-both_sexes-combined_medadj_irnt 32682\n",
      "\n",
      "continuous-DBP-both_sexes-combined_medadj_raw 32682\n",
      "\n",
      "continuous-DBP-both_sexes-combined_raw 32682\n",
      "\n",
      "continuous-DBP-both_sexes-manual_medadj_irnt 32682\n",
      "\n",
      "continuous-DBP-both_sexes-manual_medadj_raw 32682\n",
      "\n",
      "continuous-LDLC-both_sexes-medadj_irnt 32682\n",
      "\n",
      "continuous-LDLC-both_sexes-medadj_raw 32682\n",
      "\n",
      "continuous-MAP-both_sexes-auto_irnt 32682\n",
      "\n",
      "continuous-MAP-both_sexes-auto_medadj_irnt 32682\n",
      "\n",
      "continuous-MAP-both_sexes-auto_medadj_raw 32682\n",
      "\n",
      "continuous-MAP-both_sexes-auto_raw 32682\n",
      "\n",
      "continuous-MAP-both_sexes-combined_irnt 32682\n",
      "\n",
      "continuous-MAP-both_sexes-combined_medadj_irnt 32682\n",
      "\n",
      "continuous-MAP-both_sexes-combined_medadj_raw 32682\n",
      "\n",
      "continuous-MAP-both_sexes-combined_raw 32682\n",
      "\n",
      "continuous-MAP-both_sexes-manual_irnt 32682\n",
      "\n",
      "continuous-MAP-both_sexes-manual_medadj_irnt 32682\n",
      "\n",
      "continuous-MAP-both_sexes-manual_medadj_raw 32682\n",
      "\n",
      "continuous-MAP-both_sexes-manual_raw 32682\n",
      "\n",
      "continuous-PP-both_sexes-auto_irnt 32682\n",
      "\n",
      "continuous-PP-both_sexes-auto_medadj_irnt 32682\n",
      "\n",
      "continuous-PP-both_sexes-auto_medadj_raw 32682\n",
      "\n",
      "continuous-PP-both_sexes-auto_raw 32682\n",
      "\n",
      "continuous-PP-both_sexes-combined_irnt 32682\n",
      "\n",
      "continuous-PP-both_sexes-combined_medadj_irnt 32682\n",
      "\n",
      "continuous-PP-both_sexes-combined_medadj_raw 32682\n",
      "\n",
      "continuous-PP-both_sexes-combined_raw 32682\n",
      "\n",
      "continuous-PP-both_sexes-manual_irnt 32682\n",
      "\n",
      "continuous-PP-both_sexes-manual_medadj_irnt 32682\n",
      "\n",
      "continuous-PP-both_sexes-manual_medadj_raw 32682\n",
      "\n",
      "continuous-PP-both_sexes-manual_raw 32682\n",
      "\n",
      "continuous-SBP-both_sexes-auto_medadj_irnt 32682\n",
      "\n",
      "continuous-SBP-both_sexes-auto_medadj_raw 32682\n",
      "\n",
      "continuous-SBP-both_sexes-combined_irnt 32682\n",
      "\n",
      "continuous-SBP-both_sexes-combined_medadj_irnt 32682\n",
      "\n",
      "continuous-SBP-both_sexes-combined_medadj_raw 32682\n",
      "\n",
      "continuous-SBP-both_sexes-combined_raw 32682\n",
      "\n",
      "continuous-SBP-both_sexes-manual_medadj_irnt 32682\n",
      "\n",
      "continuous-SBP-both_sexes-manual_medadj_raw 32682\n",
      "\n",
      "icd10-I20-both_sexes 32682\n",
      "\n",
      "icd10-I21-both_sexes 32682\n",
      "\n",
      "icd10-I25-both_sexes 32682\n",
      "\n",
      "icd10-I26-both_sexes 32682\n",
      "\n",
      "icd10-I44-both_sexes 32682\n",
      "\n",
      "icd10-I48-both_sexes 32682\n",
      "\n",
      "icd10-I50-both_sexes 32682\n",
      "\n",
      "icd10-I51-both_sexes 32682\n",
      "\n",
      "phecode-278-both_sexes 32682\n",
      "\n",
      "phecode-278.1-both_sexes 32682\n",
      "\n",
      "phecode-401-both_sexes 32682\n",
      "\n",
      "phecode-401.1-both_sexes 32682\n",
      "\n",
      "phecode-411-both_sexes 32682\n",
      "\n",
      "phecode-411.1-both_sexes 32682\n",
      "\n",
      "phecode-411.2-both_sexes 32682\n",
      "\n",
      "phecode-411.3-both_sexes 32682\n",
      "\n",
      "phecode-411.4-both_sexes 32682\n",
      "\n",
      "phecode-411.8-both_sexes 32682\n",
      "\n",
      "phecode-415-both_sexes 32682\n",
      "\n",
      "phecode-415.1-both_sexes 32682\n",
      "\n",
      "phecode-415.11-both_sexes 32682\n",
      "\n",
      "phecode-418.1-both_sexes 32682\n",
      "\n",
      "phecode-426-both_sexes 32682\n",
      "\n",
      "phecode-426.3-both_sexes 32682\n",
      "\n",
      "phecode-427.2-both_sexes 32682\n",
      "\n",
      "phecode-427.9-both_sexes 32682\n",
      "\n",
      "phecode-428-both_sexes 32682\n",
      "\n",
      "phecode-428.2-both_sexes 32682\n",
      "\n",
      "phecode-443-both_sexes 32682\n",
      "\n",
      "phecode-451-both_sexes 32682\n",
      "\n",
      "phecode-454-both_sexes 32682\n",
      "\n",
      "phecode-454.1-both_sexes 32682\n",
      "\n",
      "prescriptions-amlodipine-both_sexes 32682\n",
      "\n",
      "prescriptions-aspirin-both_sexes 32682\n",
      "\n",
      "prescriptions-atenolol-both_sexes 32682\n",
      "\n",
      "prescriptions-bisoprolol-both_sexes 32682\n",
      "\n",
      "prescriptions-candesartan-both_sexes 32682\n",
      "\n",
      "prescriptions-clopidogrel-both_sexes 32682\n",
      "\n",
      "prescriptions-lisinopril-both_sexes 32682\n",
      "\n",
      "prescriptions-losartan-both_sexes 32682\n",
      "\n",
      "prescriptions-ramipril-both_sexes 32682\n",
      "\n"
     ]
    }
   ],
   "source": [
    "intdata_pp   $coloc_snp = FALSE\n",
    "int2coloc_var$tr2id     = paste(int2coloc_var$transcript_id, int2coloc_var$type, int2coloc_var$id, sep = \":\")\n",
    "\n",
    "for(trait in sort(unique(int2coloc_var$trait)))\n",
    "{\n",
    "    tomerge           = int2coloc_var[int2coloc_var$trait == trait, c(\"tr2id\", \"pp_snp\")]\n",
    "    tomerge           = aggregate(pp_snp ~ tr2id, data = tomerge, FUN = max)\n",
    "    colnames(tomerge) = c(\"tr2id\", paste(\"coloc_snp\", trait, sep = \".\"))\n",
    "    intdata_pp        = merge(intdata_pp, tomerge, all.x = TRUE)\n",
    "}\n",
    "\n",
    "intdata_pp$coloc_snp = unlist(lapply(1:nrow(intdata_pp), function(ii)\n",
    "{\n",
    "    out    = 0\n",
    "    mydata = intdata_pp[ii, grepl(\"coloc_snp\\\\.\", colnames(intdata_pp)) ==TRUE]\n",
    "    if(length(mydata[is.na(mydata) == FALSE]) > 0){out = max(mydata, na.rm = TRUE)}\n",
    "    \n",
    "    return(out)\n",
    "}))\n",
    "\n",
    "fwrite(intdata_pp  , paste(outfolder, \"intdata_pp.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n",
    "\n",
    "fwrite(intdata_pp[,c(\"transcript_id\", \"type\", \"gene_name\", \"id\", \"eclip\", \"pp\", \"coloc_gene\", \"coloc_snp\")], paste(outfolder, \"tops_n.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "intdata_pp$score = (1 + intdata_pp$pp) * (1 + intdata_pp$coloc_gene) * (1 + intdata_pp$coloc_snp)\n",
    "\n",
    "fwrite(intdata_pp[,c(\"tr2id\", \"transcript_id\", \"type\", \"gene_name\", \"id\", \"eclip\", \"pp\", \"coloc_gene\", \"coloc_snp\", \"score\")], paste(outfolder, \"tops_n.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get FIMO data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "myvars       = merge(unique(fm[fm$id %in% intdata_pp$id, c(\"transcript_id\", \"chrom\", \"pos\", \"ref\", \"alt\", \"id\")]), geneinfo[,c(\"transcript_id\", \"strand\")])\n",
    "myvars       = unique(myvars[,c(\"transcript_id\", \"chrom\", \"pos\", \"ref\", \"alt\", \"id\", \"strand\")])\n",
    "myvars$chrom = paste(\"chr\", myvars$chrom, sep = \"\")\n",
    "myvars       = myvars[order(myvars$chrom, myvars$pos),]\n",
    "myvars       = myvars[myvars$ref %in% c(\"A\", \"C\", \"G\", \"T\") & myvars$alt %in% c(\"A\", \"C\", \"G\", \"T\"),]\n",
    "myvars$from  = myvars$pos - 11\n",
    "myvars$to    = myvars$pos + 10\n",
    "myvars$score = 0\n",
    "myvars$coord = paste(myvars$chrom, \":\", myvars$from, \"-\", myvars$to, \"(\", myvars$strand, \")\", sep = \"\")\n",
    "mybed        = paste(outfolder, \"variants2fasta.bed\", sep = \"/\")\n",
    "myfasta      = paste(outfolder, \"variants2fasta.txt\", sep = \"/\")\n",
    "\n",
    "fwrite(unique(myvars[,c(\"chrom\", \"from\", \"to\", \"id\", \"score\", \"strand\")]), mybed, sep = \"\\t\", col.names = FALSE, row.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [],
   "source": [
    "command = paste(\"bedtools\", \"getfasta\", \"-tab\", \"-s\", \"-fi\", \"/frazer01/reference/public/hg19/hg19.fa\", \"-bed\", mybed, \">\", myfasta)\n",
    "\n",
    "system(command)\n",
    "\n",
    "seqdata            = fread(myfasta, sep = \"\\t\", header = FALSE, data.table = FALSE)\n",
    "colnames(seqdata)  = c(\"coord\", \"sequence\")\n",
    "seqdata            = merge(myvars, seqdata)\n",
    "seqdata$sequence   = toupper(seqdata$sequence)\n",
    "seqdata$ref_strand = seqdata$ref\n",
    "seqdata$alt_strand = seqdata$alt\n",
    "seqdata[seqdata$strand == \"-\", \"ref_strand\"] = mapvalues(seqdata[seqdata$strand == \"-\", \"ref\"], from = c(\"A\", \"T\", \"C\", \"G\"), to = c(\"T\", \"A\", \"G\", \"C\"), warn_missing = FALSE)\n",
    "seqdata[seqdata$strand == \"-\", \"alt_strand\"] = mapvalues(seqdata[seqdata$strand == \"-\", \"alt\"], from = c(\"A\", \"T\", \"C\", \"G\"), to = c(\"T\", \"A\", \"G\", \"C\"), warn_missing = FALSE)\n",
    "seqdata$sequence_alt = unlist(lapply(1:nrow(seqdata), function(ii)\n",
    "{\n",
    "    paste(substr(seqdata[ii, \"sequence\"], start = 1, stop = 10), \n",
    "          seqdata[ii, \"alt_strand\"], \n",
    "          substr(seqdata[ii, \"sequence\"], start = 12, stop = 21), \n",
    "          sep = \"\"\n",
    "         )\n",
    "}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [],
   "source": [
    "tofasta = unique(seqdata[,c(\"coord\", \"sequence\", \"sequence_alt\")])\n",
    "tofasta = unlist(lapply(1:nrow(tofasta), function(ii)\n",
    "{\n",
    "    out1 = paste(\">\", tofasta[ii, \"coord\"], \".ref\\n\", tofasta[ii, \"sequence\"    ], sep = \"\")\n",
    "    out2 = paste(\">\", tofasta[ii, \"coord\"], \".alt\\n\", tofasta[ii, \"sequence_alt\"], sep = \"\")\n",
    "    \n",
    "    return(paste(out1, out2, sep = \"\\n\"))\n",
    "}))\n",
    "\n",
    "fasta2fimo      = paste(outfolder, \"fasta2fimo.fa\"        , sep = \"/\")\n",
    "fimo_motifs     = paste(outfolder, \"motifs.all.homer.meme\", sep = \"/\")\n",
    "writeLines(tofasta, fasta2fimo, sep = \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/frazer01/projects/CARDIPS/analysis/cardiac_eqtls'"
      ],
      "text/latex": [
       "'/frazer01/projects/CARDIPS/analysis/cardiac\\_eqtls'"
      ],
      "text/markdown": [
       "'/frazer01/projects/CARDIPS/analysis/cardiac_eqtls'"
      ],
      "text/plain": [
       "[1] \"/frazer01/projects/CARDIPS/analysis/cardiac_eqtls\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fimo_out = paste(outfolder, \"fimo_results.txt\", sep = \"/\")\n",
    "command  = paste(\"/frazer01/software/meme_4.12.0/bin/fimo\", \"--norc\", \"--text\", \"--o\", fimo_out_folder, \"--thresh\", 1, fimo_motifs, fasta2fimo, \">\", fimo_out)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [],
   "source": [
    "fimo           = fread(fimo_out, sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "fimo           = fimo[fimo$start <= 11 & fimo$stop >= 11,]\n",
    "colnames(fimo) = gsub(\"-\", \"\", gsub(\"# \", \"\", colnames(fimo)))\n",
    "fimo$coord     = unlist(lapply(fimo$sequence_name, function(x){unlist(strsplit(x, \"\\\\.\"))[[1]]}))\n",
    "fimo$is_ref    = FALSE\n",
    "\n",
    "fimo[grepl(\"ref$\", fimo$sequence_name) == TRUE, \"is_ref\"] = TRUE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [],
   "source": [
    "fimo2delta = merge(fimo[fimo$is_ref == TRUE , c(\"motif_id\", \"motif_alt_id\", \"coord\", \"start\", \"stop\", \"score\", \"matched_sequence\")],\n",
    "                   fimo[fimo$is_ref == FALSE, c(\"motif_id\", \"motif_alt_id\", \"coord\", \"start\", \"stop\", \"score\", \"matched_sequence\")],\n",
    "                   by       = c(\"motif_id\", \"motif_alt_id\", \"coord\", \"start\", \"stop\"),\n",
    "                   suffixes = c(\"_ref\", \"_alt\")\n",
    "                  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "fimo2delta$max = unlist(apply(fimo2delta[,c(\"score_ref\", \"score_alt\")], 1, max))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1000\n",
      "\n",
      "2000\n",
      "\n",
      "3000\n",
      "\n",
      "4000\n",
      "\n",
      "5000\n",
      "\n",
      "6000\n",
      "\n",
      "7000\n",
      "\n",
      "8000\n",
      "\n",
      "9000\n",
      "\n",
      "10000\n",
      "\n",
      "11000\n",
      "\n",
      "12000\n",
      "\n",
      "13000\n",
      "\n",
      "14000\n",
      "\n",
      "15000\n",
      "\n",
      "16000\n",
      "\n",
      "17000\n",
      "\n",
      "18000\n",
      "\n",
      "19000\n",
      "\n",
      "20000\n",
      "\n",
      "21000\n",
      "\n",
      "22000\n",
      "\n",
      "23000\n",
      "\n",
      "24000\n",
      "\n",
      "25000\n",
      "\n",
      "26000\n",
      "\n",
      "27000\n",
      "\n",
      "28000\n",
      "\n",
      "29000\n",
      "\n",
      "30000\n",
      "\n",
      "31000\n",
      "\n",
      "32000\n",
      "\n",
      "33000\n",
      "\n",
      "34000\n",
      "\n",
      "35000\n",
      "\n",
      "36000\n",
      "\n",
      "37000\n",
      "\n",
      "38000\n",
      "\n",
      "39000\n",
      "\n",
      "40000\n",
      "\n",
      "41000\n",
      "\n",
      "42000\n",
      "\n",
      "43000\n",
      "\n",
      "44000\n",
      "\n",
      "45000\n",
      "\n",
      "46000\n",
      "\n",
      "47000\n",
      "\n",
      "48000\n",
      "\n",
      "49000\n",
      "\n",
      "50000\n",
      "\n",
      "51000\n",
      "\n",
      "52000\n",
      "\n",
      "53000\n",
      "\n",
      "54000\n",
      "\n",
      "55000\n",
      "\n",
      "56000\n",
      "\n",
      "57000\n",
      "\n",
      "58000\n",
      "\n",
      "59000\n",
      "\n",
      "60000\n",
      "\n",
      "61000\n",
      "\n",
      "62000\n",
      "\n",
      "63000\n",
      "\n",
      "64000\n",
      "\n",
      "65000\n",
      "\n",
      "66000\n",
      "\n",
      "67000\n",
      "\n",
      "68000\n",
      "\n",
      "69000\n",
      "\n",
      "70000\n",
      "\n",
      "71000\n",
      "\n",
      "72000\n",
      "\n",
      "73000\n",
      "\n",
      "74000\n",
      "\n",
      "75000\n",
      "\n",
      "76000\n",
      "\n",
      "77000\n",
      "\n",
      "78000\n",
      "\n",
      "79000\n",
      "\n",
      "80000\n",
      "\n",
      "81000\n",
      "\n",
      "82000\n",
      "\n",
      "83000\n",
      "\n",
      "84000\n",
      "\n",
      "85000\n",
      "\n",
      "86000\n",
      "\n",
      "87000\n",
      "\n",
      "88000\n",
      "\n",
      "89000\n",
      "\n",
      "90000\n",
      "\n",
      "91000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "to_reduce  = unique(fimo2delta[ fimo2delta$max > 0, c(\"coord\", \"motif_id\", \"motif_alt_id\")])\n",
    "fimo2motif = as.data.frame(rbindlist(lapply(1:nrow(to_reduce), function(ii)\n",
    "{\n",
    "    if(ii %in% ((1:100) * 1000)){message(ii)}\n",
    "    \n",
    "    coord        = to_reduce[ii, \"coord\"       ]\n",
    "    motif_id     = to_reduce[ii, \"motif_id\"    ]\n",
    "    motif_alt_id = to_reduce[ii, \"motif_alt_id\"]\n",
    "    \n",
    "    this = fimo2delta[ fimo2delta$coord == coord & fimo2delta$motif_id == motif_id & fimo2delta$motif_alt_id == motif_alt_id,]\n",
    "    return(this[ which.max(this$max),])\n",
    "    \n",
    "})), stringsAsFactors = FALSE)\n",
    "\n",
    "fimo2motif$delta = fimo2motif$score_alt - fimo2motif$score_ref\n",
    "fwrite(fimo2motif, paste(outfolder, \"fimo2motif.txt\", sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [],
   "source": [
    "intdata_pp2       = intdata_pp\n",
    "intdata_pp2$coord = paste(intdata_pp2$chrom, \":\", intdata_pp2$pos - 11, \"-\", intdata_pp2$pos + 10, \"(\", intdata_pp2$strand, \")\", sep = \"\")\n",
    "\n",
    "for(motif in sort(unique(fimo2motif$motif_alt_id)))\n",
    "{\n",
    "    tomerge           = fimo2motif[fimo2motif$motif_alt_id == motif, c(\"coord\", \"max\", \"delta\")]\n",
    "    #tomerge           = aggregate(. ~ coord, data = tomerge, FUN = max)\n",
    "    colnames(tomerge) = c(\"coord\", paste(c(\"score\", \"delta\"), motif, sep = \".\"))\n",
    "    intdata_pp2        = merge(intdata_pp2, tomerge, all.x = TRUE)\n",
    "}\n",
    "\n",
    "fwrite(intdata_pp2, paste(outfolder, \"intdata_pp_with_motifs.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n",
    "fwrite(intdata_pp2[,colnames(intdata_pp2) %in% c(\"transcript_id\", \"type\", \"gene_name\", \"id\", \"eclip\", \"pp\", \"coloc_gene\", \"coloc_snp\", \"score\") | (grepl(\"^eclip\\\\.\", colnames(intdata_pp2)) == TRUE | grepl(\"^score\\\\.\", colnames(intdata_pp2)) == TRUE | grepl(\"^delta\\\\.\", colnames(intdata_pp2)) == TRUE)], paste(outfolder, \"tops_n_with_motifs.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {},
   "outputs": [],
   "source": [
    "int_isoforms  = fread(\"pipeline/3.2.eqtls/eqtls/cardiac_eqtls.isoform.interactions.txt\", sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "int_isoforms  = int_isoforms[int_isoforms$cell == TRUE,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "metadata": {},
   "outputs": [],
   "source": [
    "intdata_pp2$tr2type = paste(intdata_pp2$transcript_id, intdata_pp2$type)\n",
    "\n",
    "for(cell in sort(unique(int_isoforms$interaction)))\n",
    "{\n",
    "    tomerge           = int_isoforms[int_isoforms$interaction == cell, c(\"transcript_id\", \"type\")]\n",
    "    tomerge           = paste(tomerge$transcript_id, tomerge$type)\n",
    "    intdata_pp2[,paste(\"cell\", cell, sep = \".\")] = FALSE\n",
    "    intdata_pp2[intdata_pp2$tr2type %in% tomerge,paste(\"cell\", cell, sep = \".\")] = TRUE\n",
    "    \n",
    "}\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "metadata": {},
   "outputs": [],
   "source": [
    "fwrite(intdata_pp2, paste(outfolder, \"intdata_pp_with_motifs.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n",
    "fwrite(intdata_pp2[,colnames(intdata_pp2) %in% c(\"transcript_id\", \"type\", \"gene_name\", \"id\", \"eclip\", \"pp\", \"coloc_gene\", \"coloc_snp\", \"score\") | (grepl(\"^eclip\\\\.\", colnames(intdata_pp2)) == TRUE | grepl(\"^score\\\\.\", colnames(intdata_pp2)) == TRUE | grepl(\"^delta\\\\.\", colnames(intdata_pp2)) == TRUE)], paste(outfolder, \"tops_n_with_motifs.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "metadata": {},
   "outputs": [],
   "source": [
    "fwrite(intdata_pp2[,colnames(intdata_pp2) %in% c(\"transcript_id\", \"type\", \"gene_name\", \"id\", \"eclip\", \"pp\", \"coloc_gene\", \"coloc_snp\", \"score\") | (grepl(\"^cell\\\\.\", colnames(intdata_pp2)) == TRUE | grepl(\"_HNRNPL_\", colnames(intdata_pp2)) == TRUE | grepl(\"_RBM5_\", colnames(intdata_pp2)) == TRUE)], paste(outfolder, \"tops_n_with_motifs.txt\"  , sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Try to filter\n",
    "- ENST00000565571.5_1:0:VAR_16_89627460_T_G: iPSC-CVPC, overlaps eCLIP, but does not alter motifs\n",
    "- ENST00000567815.5_3:0:VAR_16_89627460_T_G: adult, alters FMR1 (not strong)\n",
    "- ENST00000563270.5_1:0:VAR_16_89627460_T_G: adult, alters FMR1 (not strong)\n",
    "\n",
    "\n",
    "- ENST00000336783.8_1:0:VAR_16_28844284_A_G\n",
    "- ENST00000513239.5_1:0:VAR_4_526848_A_G\n",
    "- ENST00000513239.5_1:0:VAR_4_527176_G_C\n",
    "- ENST00000523267.1_2:0:VAR_5_179264731_T_C\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "4945"
      ],
      "text/latex": [
       "4945"
      ],
      "text/markdown": [
       "4945"
      ],
      "text/plain": [
       "[1] 4945"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1914"
      ],
      "text/latex": [
       "1914"
      ],
      "text/markdown": [
       "1914"
      ],
      "text/plain": [
       "[1] 1914"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "353"
      ],
      "text/latex": [
       "353"
      ],
      "text/markdown": [
       "353"
      ],
      "text/plain": [
       "[1] 353"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "433"
      ],
      "text/latex": [
       "433"
      ],
      "text/markdown": [
       "433"
      ],
      "text/plain": [
       "[1] 433"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "638"
      ],
      "text/latex": [
       "638"
      ],
      "text/markdown": [
       "638"
      ],
      "text/plain": [
       "[1] 638"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "241"
      ],
      "text/latex": [
       "241"
      ],
      "text/markdown": [
       "241"
      ],
      "text/plain": [
       "[1] 241"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "506"
      ],
      "text/latex": [
       "506"
      ],
      "text/markdown": [
       "506"
      ],
      "text/plain": [
       "[1] 506"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "106"
      ],
      "text/latex": [
       "106"
      ],
      "text/markdown": [
       "106"
      ],
      "text/plain": [
       "[1] 106"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "69"
      ],
      "text/latex": [
       "69"
      ],
      "text/markdown": [
       "69"
      ],
      "text/plain": [
       "[1] 69"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "intdata_pp3 = intdata_pp2\n",
    "\n",
    "pp.fm        = intdata_pp3[ intdata_pp3$pp         >= 0.1, \"tr2id\"]\n",
    "pp.gwas.gene = intdata_pp3[ intdata_pp3$coloc_gene >= 0.5, \"tr2id\"]\n",
    "pp.gwas.snp  = intdata_pp3[ intdata_pp3$coloc_snp  >= 0.1, \"tr2id\"]\n",
    "\n",
    "cell.ipsc_cvpc = intdata_pp3[ intdata_pp3$cell.ipsc_cvpc == TRUE, \"tr2id\"]\n",
    "cell.adult     = intdata_pp3[ intdata_pp3$cell.adult     == TRUE, \"tr2id\"]\n",
    "\n",
    "\n",
    "rbp.hnrnpl.score = intdata_pp3[ intdata_pp3$eclip.HNRNPL == TRUE & apply(intdata_pp3[,grepl(\"^score\\\\.\", colnames(intdata_pp2)) == TRUE & grepl(\"_HNRNPL_\", colnames(intdata_pp2)) == TRUE], 1, function(x){max(    c(0, x) , na.rm = TRUE)}) >= 4, \"tr2id\"]\n",
    "rbp.hnrnpl.delta = intdata_pp3[ intdata_pp3$eclip.HNRNPL == TRUE & apply(intdata_pp3[,grepl(\"^delta\\\\.\", colnames(intdata_pp2)) == TRUE & grepl(\"_HNRNPL_\", colnames(intdata_pp2)) == TRUE], 1, function(x){max(abs(c(0, x)), na.rm = TRUE)}) >= 4, \"tr2id\"]\n",
    "rbp.rbm5.score   = intdata_pp3[ intdata_pp3$eclip.RBM5   == TRUE & is.na(intdata_pp3$score.RBPmap_RBM5_garggwr_human_PSSM) == FALSE & intdata_pp3$score.RBPmap_RBM5_garggwr_human_PSSM >= 4, \"tr2id\"]\n",
    "rbp.rbm5.delta   = intdata_pp3[ intdata_pp3$eclip.RBM5   == TRUE & is.na(intdata_pp3$delta.RBPmap_RBM5_garggwr_human_PSSM) == FALSE & intdata_pp3$delta.RBPmap_RBM5_garggwr_human_PSSM >= 4, \"tr2id\"]\n",
    "\n",
    "length(pp.fm           )\n",
    "length(pp.gwas.gene    )\n",
    "length(pp.gwas.snp     )\n",
    "length(cell.ipsc_cvpc  )\n",
    "length(cell.adult      )\n",
    "length(rbp.hnrnpl.score)\n",
    "length(rbp.hnrnpl.delta)\n",
    "length(rbp.rbm5.score  )\n",
    "length(rbp.rbm5.delta  )\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 318,
   "metadata": {},
   "outputs": [],
   "source": [
    "toint = unique(intdata_pp3[,c(\"tr2id\", \"transcript_id\", \"type\", \"gene_name\", \"gene_id\", \"id\", \"pp\", \"coloc_gene\", \"coloc_snp\")])\n",
    "toint       $tr2type = paste(toint       $transcript_id, toint       $type, sep = \":\")\n",
    "int_isoforms$tr2type = paste(int_isoforms$transcript_id, int_isoforms$type, sep = \":\")\n",
    "\n",
    "toint$pp.fm            = FALSE\n",
    "toint$pp.gwas.gene     = FALSE\n",
    "toint$pp.gwas.snp      = FALSE\n",
    "#toint$cell.ipsc_cvpc   = FALSE\n",
    "#toint$cell.adult       = FALSE\n",
    "#toint$rbp.hnrnpl.score = FALSE\n",
    "#toint$rbp.hnrnpl.delta = FALSE\n",
    "#toint$rbp.rbm5.score   = FALSE\n",
    "#toint$rbp.rbm5.delta   = FALSE\n",
    "\n",
    "\n",
    "toint[ toint$tr2id %in% pp.fm           , \"pp.fm\"           ] = TRUE\n",
    "toint[ toint$tr2id %in% pp.gwas.gene    , \"pp.gwas.gene\"    ] = TRUE\n",
    "toint[ toint$tr2id %in% pp.gwas.snp     , \"pp.gwas.snp\"     ] = TRUE\n",
    "#toint[ toint$tr2id %in% cell.ipsc_cvpc  , \"cell.ipsc_cvpc\"  ] = TRUE\n",
    "#toint[ toint$tr2id %in% cell.adult      , \"cell.adult\"      ] = TRUE\n",
    "#toint[ toint$tr2id %in% rbp.hnrnpl.score, \"rbp.hnrnpl.score\"] = TRUE\n",
    "#toint[ toint$tr2id %in% rbp.hnrnpl.delta, \"rbp.hnrnpl.delta\"] = TRUE\n",
    "#toint[ toint$tr2id %in% rbp.rbm5.score  , \"rbp.rbm5.score\"  ] = TRUE\n",
    "#toint[ toint$tr2id %in% rbp.rbm5.delta  , \"rbp.rbm5.delta\"  ] = TRUE\n",
    "\n",
    "\n",
    "for(cell in sort(unique(int_isoforms$interaction)))\n",
    "{\n",
    "    this = unique(int_isoforms[int_isoforms$interaction == cell, \"tr2type\"])\n",
    "    \n",
    "    toint[,paste(\"cell\", sub(\"^cibersort.regular.\", \"\", cell), sep = \".\")] = FALSE\n",
    "    toint[toint$tr2type %in% this, paste(\"cell\", sub(\"^cibersort.regular.\", \"\", cell), sep = \".\")] = TRUE\n",
    "}\n",
    "\n",
    "\n",
    "for(rbp in rbps$rbp)\n",
    "{\n",
    "    mycols_score = colnames(intdata_pp3[,grepl(\"^score\\\\.\", colnames(intdata_pp2)) == TRUE & grepl(paste(\"_\", rbp, \"_\", sep = \"\"), colnames(intdata_pp2)) == TRUE])\n",
    "    mycols_delta = colnames(intdata_pp3[,grepl(\"^delta\\\\.\", colnames(intdata_pp2)) == TRUE & grepl(paste(\"_\", rbp, \"_\", sep = \"\"), colnames(intdata_pp2)) == TRUE])\n",
    "    \n",
    "    if(length(mycols_score) > 1)\n",
    "    {\n",
    "        totest = intdata_pp3[intdata_pp3[, paste(\"eclip\", rbp, sep = \".\")] == TRUE &\n",
    "                             apply(intdata_pp3[,mycols_score], 1, function(x){max(    c(0, x) , na.rm = TRUE)}) >= 4 &\n",
    "                             apply(intdata_pp3[,mycols_delta], 1, function(x){max(abs(c(0, x)), na.rm = TRUE)}) >= 3, ]\n",
    "    }else\n",
    "    {\n",
    "        totest = intdata_pp3[intdata_pp3[, paste(\"eclip\", rbp, sep = \".\")] == TRUE &\n",
    "                             is.na(intdata_pp3[, mycols_score]) == FALSE &     intdata_pp3[, mycols_score]  >= 4 & \n",
    "                             is.na(intdata_pp3[, mycols_delta]) == FALSE & abs(intdata_pp3[, mycols_delta]) >= 3, ]\n",
    "    }\n",
    "    \n",
    "    toint[                             , paste(\"rbp\", rbp, sep = \".\")] = FALSE\n",
    "    toint[toint$tr2id %in% totest$tr2id, paste(\"rbp\", rbp, sep = \".\")] = TRUE\n",
    "}\n",
    "\n",
    "toint$score   = rowSums(toint[,colnames(toint) %in% c(\"pp.fm\", \"pp.gwas.gene\", \"pp.gwas.snp\", paste(\"rbp\", rbps$rbp, sep = \".\")) | grepl(\"^cell\\\\.\", colnames(toint)) == TRUE])\n",
    "toint$is_cell = FALSE\n",
    "toint$is_rbp  = FALSE\n",
    "\n",
    "toint[rowSums(toint[,grepl(\"^cell\\\\.\", colnames(toint)) == TRUE]) > 0, \"is_cell\"] = TRUE\n",
    "toint[rowSums(toint[,paste(\"rbp\", rbps$rbp, sep = \".\")         ]) > 0, \"is_rbp\" ] = TRUE\n",
    "\n",
    "\n",
    "\n",
    "fwrite(toint, paste(outfolder, \"filtered_for_examples.txt\", sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t1446 obs. of  40 variables:\n",
      " $ transcript_id: chr  \"ENST00000190983.4_1\" \"ENST00000216416.9_4\" \"ENST00000220853.8_2\" \"ENST00000220931.11_3\" ...\n",
      " $ gene_id      : chr  \"ENSG00000064205.10_4\" \"ENSG00000100528.12_6\" \"ENSG00000104412.8_5\" \"ENSG00000104490.18_8\" ...\n",
      " $ gene_name    : chr  \"CCN5\" \"CNIH1\" \"EMC2\" \"NCALD\" ...\n",
      " $ gene_type    : chr  \"protein_coding\" \"protein_coding\" \"protein_coding\" \"protein_coding\" ...\n",
      " $ start        : int  43343886 54890279 109455853 102698770 43678846 43678846 37321791 196678409 110210679 7281376 ...\n",
      " $ end          : int  43356452 54908092 109501425 102803197 43769140 43769140 37358944 196695931 110217908 7313432 ...\n",
      " $ strand       : chr  \"+\" \"-\" \"+\" \"-\" ...\n",
      " $ chrom        : int  20 14 8 8 7 7 6 3 1 6 ...\n",
      " $ pos          : int  43348735 54890379 109587552 102900082 43642005 43690946 37348920 196674879 110212293 7280780 ...\n",
      " $ ref          : chr  \"C\" \"C\" \"G\" \"C\" ...\n",
      " $ alt          : chr  \"A\" \"A\" \"A\" \"T\" ...\n",
      " $ rsid         : chr  \"rs2296530\" \"rs3204303\" \"rs34516160\" \"rs486496\" ...\n",
      " $ id           : chr  \"VAR_20_43348735_C_A\" \"VAR_14_54890379_C_A\" \"VAR_8_109587552_G_A\" \"VAR_8_102900082_C_T\" ...\n",
      " $ af           : num  0.38 0.096 0.289 0.259 0.289 0.074 0.505 0.526 0.433 0.712 ...\n",
      " $ beta         : num  0.195 0.886 -0.642 -0.938 -0.321 ...\n",
      " $ se           : num  0.0456 0.0738 0.0658 0.09 0.0377 ...\n",
      " $ pval         : num  1.27e-05 1.37e-36 4.77e-24 1.71e-27 1.38e-18 ...\n",
      " $ bonferroni   : num  2.17e-02 2.51e-33 1.02e-20 4.72e-24 3.43e-15 ...\n",
      " $ fdr          : num  5.19e-03 6.01e-34 1.28e-21 1.19e-24 6.99e-16 ...\n",
      " $ tests        : int  409 439 269 697 506 506 679 756 478 792 ...\n",
      " $ type         : int  0 0 0 0 0 1 0 0 0 0 ...\n",
      " $ beta_int     : num  0.36 1.447 -0.488 -0.849 -0.273 ...\n",
      " $ se_int       : num  0.0796 0.117 0.1045 0.132 0.0601 ...\n",
      " $ pval_int     : num  5.31e-06 6.64e-33 3.34e-06 1.71e-10 5.10e-06 ...\n",
      " $ interaction  : chr  \"arteria\" \"arteria\" \"arteria\" \"arteria\" ...\n",
      " $ beta_0       : num  -0.0748 0.2366 -0.3192 -0.5131 -0.4066 ...\n",
      " $ se_0         : num  0.0996 0.1668 0.138 0.131 0.0866 ...\n",
      " $ pval_0       : num  4.53e-01 1.57e-01 2.11e-02 9.96e-05 3.27e-06 ...\n",
      " $ beta_1       : num  0.38 1.871 -0.78 -1.408 -0.483 ...\n",
      " $ se_1         : num  0.0887 0.1468 0.1279 0.101 0.0859 ...\n",
      " $ pval_1       : num  2.34e-05 7.84e-31 2.75e-09 1.77e-35 3.86e-08 ...\n",
      " $ distance     : int  4849 17713 131699 -96885 127135 78194 27129 21052 1614 32652 ...\n",
      " $ qval_int     : num  4.51e-02 5.63e-29 2.84e-02 1.45e-06 4.33e-02 ...\n",
      " $ qval_1       : num  5.72e-05 1.96e-29 1.26e-08 5.92e-34 1.47e-07 ...\n",
      " $ qval_0       : num  4.82e-01 1.80e-01 2.81e-02 2.12e-04 9.31e-06 ...\n",
      " $ int_signif   : logi  TRUE TRUE TRUE TRUE TRUE TRUE ...\n",
      " $ cell         : logi  TRUE TRUE TRUE TRUE TRUE TRUE ...\n",
      " $ specific     : logi  TRUE TRUE FALSE FALSE FALSE TRUE ...\n",
      " $ associated   : logi  FALSE FALSE TRUE TRUE TRUE FALSE ...\n",
      " $ tr2type      : chr  \"ENST00000190983.4_1:0\" \"ENST00000216416.9_4:0\" \"ENST00000220853.8_2:0\" \"ENST00000220931.11_3:0\" ...\n",
      "'data.frame':\t32682 obs. of  37 variables:\n",
      " $ tr2id                : chr  \"ENST00000496843.1_1:0:VAR_1_100149308_G_A\" \"ENST00000496843.1_1:1:VAR_1_100156658_T_C\" \"ENST00000496843.1_1:0:VAR_1_100156658_T_C\" \"ENST00000294724.8_2:0:VAR_1_100316589_A_G\" ...\n",
      " $ transcript_id        : chr  \"ENST00000496843.1_1\" \"ENST00000496843.1_1\" \"ENST00000496843.1_1\" \"ENST00000294724.8_2\" ...\n",
      " $ type                 : int  0 1 0 0 0 0 0 0 0 0 ...\n",
      " $ gene_name            : chr  \"PALMD\" \"PALMD\" \"PALMD\" \"AGL\" ...\n",
      " $ gene_id              : chr  \"ENSG00000099260.11_5\" \"ENSG00000099260.11_5\" \"ENSG00000099260.11_5\" \"ENSG00000162688.17_5\" ...\n",
      " $ id                   : chr  \"VAR_1_100149308_G_A\" \"VAR_1_100156658_T_C\" \"VAR_1_100156658_T_C\" \"VAR_1_100316589_A_G\" ...\n",
      " $ pp                   : num  0.01 0.0115 0.0321 0.0328 0.0302 ...\n",
      " $ coloc_gene           : num  0.974 0 0 0 0 ...\n",
      " $ coloc_snp            : num  5.69e-05 0.00 0.00 0.00 0.00 ...\n",
      " $ tr2type              : chr  \"ENST00000496843.1_1:0\" \"ENST00000496843.1_1:1\" \"ENST00000496843.1_1:0\" \"ENST00000294724.8_2:0\" ...\n",
      " $ pp.fm                : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ pp.gwas.gene         : logi  TRUE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ pp.gwas.snp          : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.adult           : logi  TRUE FALSE TRUE FALSE FALSE FALSE ...\n",
      " $ cell.arteria         : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.arteria_aorta   : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.arteria_coronary: logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.cardiac_muscle  : logi  TRUE FALSE TRUE FALSE FALSE FALSE ...\n",
      " $ cell.cardiac_neuron  : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.endocardial     : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.endothelial     : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.fibroblast      : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.immune          : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.myofibroblast   : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.smooth_muscle   : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.heart           : logi  TRUE FALSE TRUE FALSE FALSE FALSE ...\n",
      " $ cell.heart_atrium    : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.heart_ventricle : logi  TRUE FALSE TRUE FALSE FALSE FALSE ...\n",
      " $ cell.ipsc_cvpc       : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.FMR1             : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.FUS              : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.FXR2             : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.HNRNPA1          : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.HNRNPL           : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.RBM5             : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.SRSF7            : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ score                : num  5 0 4 0 0 0 0 0 0 0 ...\n"
     ]
    }
   ],
   "source": [
    "str(int_isoforms)\n",
    "str(toint)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.0.1",
   "language": "R",
   "name": "ir401"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
