{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/frazer01/projects/CARDIPS/analysis/cardiac_eqtls\")\n",
    "\n",
    "source(\"script/packages.R\"  )\n",
    "source(\"script/input_data.R\")\n",
    "source(\"script/functions.R\" )\n",
    "source(\"script/colors.R\"    )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "manifest = fread(\"pipeline/6.1.coloc_gwas/traits.manifest.txt\"                      , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "fimo     = fread(\"pipeline/6.1.coloc_gwas//intersect_rbps/fimo2motif.txt\"           , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "coloc    = fread(\"pipeline/6.1.coloc_gwas/coloc.eqtls.isoform.txt\"                  , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "indata   = fread(\"pipeline/6.1.coloc_gwas//intersect_rbps/filtered_for_examples.txt\", sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "exons    = fread(\"/frazer01/reference/private/Gencode.v34lift37/exon_info.txt\"      , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "toplot   = indata[ indata$pp.gwas.gene == TRUE & indata$is_cell == TRUE & indata$is_rbp == TRUE,]\n",
    "toplot   = toplot[ order(toplot$coloc_gene, decreasing = TRUE),]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "rbps        = data.frame(rbp = c(\"FMR1\", \"FUS\", \"FXR2\", \"HNRNPA1\", \"HNRNPL\", \"RBM5\", \"SRSF7\"), folder = \"\")\n",
    "rbps$folder = paste(\"/frazer01/home/tarthur/projects/CARDIPS/fine_mapping/analysis/2020-10-08-fgwas\", tolower(rbps$rbp), \"peakfiles\", sep = \"/\")\n",
    "rbps$color  = qualitative_hcl(n = nrow(rbps), palette = \"Set 2\")\n",
    "\n",
    "read_rbp = function(rbp, rbps, fm)\n",
    "{\n",
    "    bedfiles         = list.files(rbps[rbps$rbp == rbp, \"folder\"], full.names = TRUE)\n",
    "    indata           = as.data.frame(rbindlist(lapply(bedfiles, function(x){fread(x, sep = \"\\t\", header = FALSE, data.table = FALSE)[,1:3]})), stringsAsFactors = FALSE)\n",
    "    colnames(indata) = c(\"chrom\", \"from\", \"to\")\n",
    "    return(indata)\n",
    "}\n",
    "\n",
    "rbp_list        = lapply(rbps$rbp, function(rbp){read_rbp(rbp, rbps)})\n",
    "names(rbp_list) = rbps$rbp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 22 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>tr2id</th><th scope=col>gene_name</th><th scope=col>pp</th><th scope=col>coloc_gene</th><th scope=col>coloc_snp</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>18026</th><td>ENST00000356150.10_3:0:VAR_2_231741_C_T  </td><td>SH3YL1</td><td>0.01823861</td><td>0.9793186</td><td>0.023701347</td></tr>\n",
       "\t<tr><th scope=row>18027</th><td>ENST00000403712.6_2:0:VAR_2_231741_C_T   </td><td>SH3YL1</td><td>0.01297159</td><td>0.9670757</td><td>0.015370886</td></tr>\n",
       "\t<tr><th scope=row>25893</th><td>ENST00000523267.1_2:0:VAR_5_179264731_T_C</td><td>MRNIP </td><td>0.03639948</td><td>0.9645763</td><td>0.067428181</td></tr>\n",
       "\t<tr><th scope=row>23881</th><td>ENST00000512289.5_1:1:VAR_4_1323070_G_A  </td><td>MAEA  </td><td>0.01021599</td><td>0.9485075</td><td>0.054658936</td></tr>\n",
       "\t<tr><th scope=row>11660</th><td>ENST00000562879.5_1:0:VAR_16_89629736_C_T</td><td>RPL13 </td><td>0.11222078</td><td>0.9454696</td><td>0.122675867</td></tr>\n",
       "\t<tr><th scope=row>30775</th><td>ENST00000518166.5_5:0:VAR_8_102894466_C_G</td><td>NCALD </td><td>0.01343205</td><td>0.9441154</td><td>0.065894025</td></tr>\n",
       "\t<tr><th scope=row>10620</th><td>ENST00000336783.8_1:0:VAR_16_28844284_A_G</td><td>ATXN2L</td><td>0.01891547</td><td>0.9435859</td><td>0.021415372</td></tr>\n",
       "\t<tr><th scope=row>10265</th><td>ENST00000397514.8_2:0:VAR_16_1365250_G_A </td><td>UBE2I </td><td>0.01554395</td><td>0.9434889</td><td>0.017434607</td></tr>\n",
       "\t<tr><th scope=row>10277</th><td>ENST00000397514.8_2:0:VAR_16_1368517_C_T </td><td>UBE2I </td><td>0.03088314</td><td>0.9434889</td><td>0.027465977</td></tr>\n",
       "\t<tr><th scope=row>10308</th><td>ENST00000397514.8_2:0:VAR_16_1372163_C_G </td><td>UBE2I </td><td>0.01554395</td><td>0.9434889</td><td>0.021460073</td></tr>\n",
       "\t<tr><th scope=row>27698</th><td>ENST00000462256.5_2:0:VAR_6_31501737_C_T </td><td>DDX39B</td><td>0.06335733</td><td>0.9083003</td><td>0.187874411</td></tr>\n",
       "\t<tr><th scope=row>27703</th><td>ENST00000462256.5_2:0:VAR_6_31502767_C_T </td><td>DDX39B</td><td>0.01344197</td><td>0.9083003</td><td>0.020692279</td></tr>\n",
       "\t<tr><th scope=row>14826</th><td>ENST00000429941.6_1:0:VAR_19_16336641_G_C</td><td>AP1M1 </td><td>0.03323213</td><td>0.8483905</td><td>0.037668433</td></tr>\n",
       "\t<tr><th scope=row>30310</th><td>ENST00000265758.7_3:0:VAR_7_73098456_G_A </td><td>BUD23 </td><td>0.03922880</td><td>0.8462387</td><td>0.038616760</td></tr>\n",
       "\t<tr><th scope=row>30315</th><td>ENST00000265758.7_3:0:VAR_7_73103958_G_A </td><td>BUD23 </td><td>0.04123853</td><td>0.8462387</td><td>0.041398543</td></tr>\n",
       "\t<tr><th scope=row>24134</th><td>ENST00000448662.6_3:0:VAR_4_186692434_C_T</td><td>SORBS2</td><td>0.01456426</td><td>0.8385452</td><td>0.028207810</td></tr>\n",
       "\t<tr><th scope=row>30958</th><td>ENST00000520105.5_1:0:VAR_8_142440751_A_G</td><td>PTP4A3</td><td>0.09070998</td><td>0.7985123</td><td>0.090242190</td></tr>\n",
       "\t<tr><th scope=row>9996</th><td>ENST00000559126.5_2:0:VAR_15_85197253_T_C</td><td>WDR73 </td><td>0.49925609</td><td>0.5998369</td><td>0.499540255</td></tr>\n",
       "\t<tr><th scope=row>6427</th><td>ENST00000435753.6_1:0:VAR_12_31240334_C_T</td><td>DDX11 </td><td>0.01294198</td><td>0.5936584</td><td>0.020707343</td></tr>\n",
       "\t<tr><th scope=row>6454</th><td>ENST00000435753.6_1:0:VAR_12_31244924_G_A</td><td>DDX11 </td><td>0.01261124</td><td>0.5936584</td><td>0.017848965</td></tr>\n",
       "\t<tr><th scope=row>24642</th><td>ENST00000513239.5_1:0:VAR_4_526848_A_G   </td><td>PIGG  </td><td>0.01088551</td><td>0.5005399</td><td>0.009832962</td></tr>\n",
       "\t<tr><th scope=row>24643</th><td>ENST00000513239.5_1:0:VAR_4_527176_G_C   </td><td>PIGG  </td><td>0.04148680</td><td>0.5005399</td><td>0.047647058</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 22 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & tr2id & gene\\_name & pp & coloc\\_gene & coloc\\_snp\\\\\n",
       "  & <chr> & <chr> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t18026 & ENST00000356150.10\\_3:0:VAR\\_2\\_231741\\_C\\_T   & SH3YL1 & 0.01823861 & 0.9793186 & 0.023701347\\\\\n",
       "\t18027 & ENST00000403712.6\\_2:0:VAR\\_2\\_231741\\_C\\_T    & SH3YL1 & 0.01297159 & 0.9670757 & 0.015370886\\\\\n",
       "\t25893 & ENST00000523267.1\\_2:0:VAR\\_5\\_179264731\\_T\\_C & MRNIP  & 0.03639948 & 0.9645763 & 0.067428181\\\\\n",
       "\t23881 & ENST00000512289.5\\_1:1:VAR\\_4\\_1323070\\_G\\_A   & MAEA   & 0.01021599 & 0.9485075 & 0.054658936\\\\\n",
       "\t11660 & ENST00000562879.5\\_1:0:VAR\\_16\\_89629736\\_C\\_T & RPL13  & 0.11222078 & 0.9454696 & 0.122675867\\\\\n",
       "\t30775 & ENST00000518166.5\\_5:0:VAR\\_8\\_102894466\\_C\\_G & NCALD  & 0.01343205 & 0.9441154 & 0.065894025\\\\\n",
       "\t10620 & ENST00000336783.8\\_1:0:VAR\\_16\\_28844284\\_A\\_G & ATXN2L & 0.01891547 & 0.9435859 & 0.021415372\\\\\n",
       "\t10265 & ENST00000397514.8\\_2:0:VAR\\_16\\_1365250\\_G\\_A  & UBE2I  & 0.01554395 & 0.9434889 & 0.017434607\\\\\n",
       "\t10277 & ENST00000397514.8\\_2:0:VAR\\_16\\_1368517\\_C\\_T  & UBE2I  & 0.03088314 & 0.9434889 & 0.027465977\\\\\n",
       "\t10308 & ENST00000397514.8\\_2:0:VAR\\_16\\_1372163\\_C\\_G  & UBE2I  & 0.01554395 & 0.9434889 & 0.021460073\\\\\n",
       "\t27698 & ENST00000462256.5\\_2:0:VAR\\_6\\_31501737\\_C\\_T  & DDX39B & 0.06335733 & 0.9083003 & 0.187874411\\\\\n",
       "\t27703 & ENST00000462256.5\\_2:0:VAR\\_6\\_31502767\\_C\\_T  & DDX39B & 0.01344197 & 0.9083003 & 0.020692279\\\\\n",
       "\t14826 & ENST00000429941.6\\_1:0:VAR\\_19\\_16336641\\_G\\_C & AP1M1  & 0.03323213 & 0.8483905 & 0.037668433\\\\\n",
       "\t30310 & ENST00000265758.7\\_3:0:VAR\\_7\\_73098456\\_G\\_A  & BUD23  & 0.03922880 & 0.8462387 & 0.038616760\\\\\n",
       "\t30315 & ENST00000265758.7\\_3:0:VAR\\_7\\_73103958\\_G\\_A  & BUD23  & 0.04123853 & 0.8462387 & 0.041398543\\\\\n",
       "\t24134 & ENST00000448662.6\\_3:0:VAR\\_4\\_186692434\\_C\\_T & SORBS2 & 0.01456426 & 0.8385452 & 0.028207810\\\\\n",
       "\t30958 & ENST00000520105.5\\_1:0:VAR\\_8\\_142440751\\_A\\_G & PTP4A3 & 0.09070998 & 0.7985123 & 0.090242190\\\\\n",
       "\t9996 & ENST00000559126.5\\_2:0:VAR\\_15\\_85197253\\_T\\_C & WDR73  & 0.49925609 & 0.5998369 & 0.499540255\\\\\n",
       "\t6427 & ENST00000435753.6\\_1:0:VAR\\_12\\_31240334\\_C\\_T & DDX11  & 0.01294198 & 0.5936584 & 0.020707343\\\\\n",
       "\t6454 & ENST00000435753.6\\_1:0:VAR\\_12\\_31244924\\_G\\_A & DDX11  & 0.01261124 & 0.5936584 & 0.017848965\\\\\n",
       "\t24642 & ENST00000513239.5\\_1:0:VAR\\_4\\_526848\\_A\\_G    & PIGG   & 0.01088551 & 0.5005399 & 0.009832962\\\\\n",
       "\t24643 & ENST00000513239.5\\_1:0:VAR\\_4\\_527176\\_G\\_C    & PIGG   & 0.04148680 & 0.5005399 & 0.047647058\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 22 × 5\n",
       "\n",
       "| <!--/--> | tr2id &lt;chr&gt; | gene_name &lt;chr&gt; | pp &lt;dbl&gt; | coloc_gene &lt;dbl&gt; | coloc_snp &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 18026 | ENST00000356150.10_3:0:VAR_2_231741_C_T   | SH3YL1 | 0.01823861 | 0.9793186 | 0.023701347 |\n",
       "| 18027 | ENST00000403712.6_2:0:VAR_2_231741_C_T    | SH3YL1 | 0.01297159 | 0.9670757 | 0.015370886 |\n",
       "| 25893 | ENST00000523267.1_2:0:VAR_5_179264731_T_C | MRNIP  | 0.03639948 | 0.9645763 | 0.067428181 |\n",
       "| 23881 | ENST00000512289.5_1:1:VAR_4_1323070_G_A   | MAEA   | 0.01021599 | 0.9485075 | 0.054658936 |\n",
       "| 11660 | ENST00000562879.5_1:0:VAR_16_89629736_C_T | RPL13  | 0.11222078 | 0.9454696 | 0.122675867 |\n",
       "| 30775 | ENST00000518166.5_5:0:VAR_8_102894466_C_G | NCALD  | 0.01343205 | 0.9441154 | 0.065894025 |\n",
       "| 10620 | ENST00000336783.8_1:0:VAR_16_28844284_A_G | ATXN2L | 0.01891547 | 0.9435859 | 0.021415372 |\n",
       "| 10265 | ENST00000397514.8_2:0:VAR_16_1365250_G_A  | UBE2I  | 0.01554395 | 0.9434889 | 0.017434607 |\n",
       "| 10277 | ENST00000397514.8_2:0:VAR_16_1368517_C_T  | UBE2I  | 0.03088314 | 0.9434889 | 0.027465977 |\n",
       "| 10308 | ENST00000397514.8_2:0:VAR_16_1372163_C_G  | UBE2I  | 0.01554395 | 0.9434889 | 0.021460073 |\n",
       "| 27698 | ENST00000462256.5_2:0:VAR_6_31501737_C_T  | DDX39B | 0.06335733 | 0.9083003 | 0.187874411 |\n",
       "| 27703 | ENST00000462256.5_2:0:VAR_6_31502767_C_T  | DDX39B | 0.01344197 | 0.9083003 | 0.020692279 |\n",
       "| 14826 | ENST00000429941.6_1:0:VAR_19_16336641_G_C | AP1M1  | 0.03323213 | 0.8483905 | 0.037668433 |\n",
       "| 30310 | ENST00000265758.7_3:0:VAR_7_73098456_G_A  | BUD23  | 0.03922880 | 0.8462387 | 0.038616760 |\n",
       "| 30315 | ENST00000265758.7_3:0:VAR_7_73103958_G_A  | BUD23  | 0.04123853 | 0.8462387 | 0.041398543 |\n",
       "| 24134 | ENST00000448662.6_3:0:VAR_4_186692434_C_T | SORBS2 | 0.01456426 | 0.8385452 | 0.028207810 |\n",
       "| 30958 | ENST00000520105.5_1:0:VAR_8_142440751_A_G | PTP4A3 | 0.09070998 | 0.7985123 | 0.090242190 |\n",
       "| 9996 | ENST00000559126.5_2:0:VAR_15_85197253_T_C | WDR73  | 0.49925609 | 0.5998369 | 0.499540255 |\n",
       "| 6427 | ENST00000435753.6_1:0:VAR_12_31240334_C_T | DDX11  | 0.01294198 | 0.5936584 | 0.020707343 |\n",
       "| 6454 | ENST00000435753.6_1:0:VAR_12_31244924_G_A | DDX11  | 0.01261124 | 0.5936584 | 0.017848965 |\n",
       "| 24642 | ENST00000513239.5_1:0:VAR_4_526848_A_G    | PIGG   | 0.01088551 | 0.5005399 | 0.009832962 |\n",
       "| 24643 | ENST00000513239.5_1:0:VAR_4_527176_G_C    | PIGG   | 0.04148680 | 0.5005399 | 0.047647058 |\n",
       "\n"
      ],
      "text/plain": [
       "      tr2id                                     gene_name pp         coloc_gene\n",
       "18026 ENST00000356150.10_3:0:VAR_2_231741_C_T   SH3YL1    0.01823861 0.9793186 \n",
       "18027 ENST00000403712.6_2:0:VAR_2_231741_C_T    SH3YL1    0.01297159 0.9670757 \n",
       "25893 ENST00000523267.1_2:0:VAR_5_179264731_T_C MRNIP     0.03639948 0.9645763 \n",
       "23881 ENST00000512289.5_1:1:VAR_4_1323070_G_A   MAEA      0.01021599 0.9485075 \n",
       "11660 ENST00000562879.5_1:0:VAR_16_89629736_C_T RPL13     0.11222078 0.9454696 \n",
       "30775 ENST00000518166.5_5:0:VAR_8_102894466_C_G NCALD     0.01343205 0.9441154 \n",
       "10620 ENST00000336783.8_1:0:VAR_16_28844284_A_G ATXN2L    0.01891547 0.9435859 \n",
       "10265 ENST00000397514.8_2:0:VAR_16_1365250_G_A  UBE2I     0.01554395 0.9434889 \n",
       "10277 ENST00000397514.8_2:0:VAR_16_1368517_C_T  UBE2I     0.03088314 0.9434889 \n",
       "10308 ENST00000397514.8_2:0:VAR_16_1372163_C_G  UBE2I     0.01554395 0.9434889 \n",
       "27698 ENST00000462256.5_2:0:VAR_6_31501737_C_T  DDX39B    0.06335733 0.9083003 \n",
       "27703 ENST00000462256.5_2:0:VAR_6_31502767_C_T  DDX39B    0.01344197 0.9083003 \n",
       "14826 ENST00000429941.6_1:0:VAR_19_16336641_G_C AP1M1     0.03323213 0.8483905 \n",
       "30310 ENST00000265758.7_3:0:VAR_7_73098456_G_A  BUD23     0.03922880 0.8462387 \n",
       "30315 ENST00000265758.7_3:0:VAR_7_73103958_G_A  BUD23     0.04123853 0.8462387 \n",
       "24134 ENST00000448662.6_3:0:VAR_4_186692434_C_T SORBS2    0.01456426 0.8385452 \n",
       "30958 ENST00000520105.5_1:0:VAR_8_142440751_A_G PTP4A3    0.09070998 0.7985123 \n",
       "9996  ENST00000559126.5_2:0:VAR_15_85197253_T_C WDR73     0.49925609 0.5998369 \n",
       "6427  ENST00000435753.6_1:0:VAR_12_31240334_C_T DDX11     0.01294198 0.5936584 \n",
       "6454  ENST00000435753.6_1:0:VAR_12_31244924_G_A DDX11     0.01261124 0.5936584 \n",
       "24642 ENST00000513239.5_1:0:VAR_4_526848_A_G    PIGG      0.01088551 0.5005399 \n",
       "24643 ENST00000513239.5_1:0:VAR_4_527176_G_C    PIGG      0.04148680 0.5005399 \n",
       "      coloc_snp  \n",
       "18026 0.023701347\n",
       "18027 0.015370886\n",
       "25893 0.067428181\n",
       "23881 0.054658936\n",
       "11660 0.122675867\n",
       "30775 0.065894025\n",
       "10620 0.021415372\n",
       "10265 0.017434607\n",
       "10277 0.027465977\n",
       "10308 0.021460073\n",
       "27698 0.187874411\n",
       "27703 0.020692279\n",
       "14826 0.037668433\n",
       "30310 0.038616760\n",
       "30315 0.041398543\n",
       "24134 0.028207810\n",
       "30958 0.090242190\n",
       "9996  0.499540255\n",
       "6427  0.020707343\n",
       "6454  0.017848965\n",
       "24642 0.009832962\n",
       "24643 0.047647058"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t22 obs. of  39 variables:\n",
      " $ tr2id                : chr  \"ENST00000356150.10_3:0:VAR_2_231741_C_T\" \"ENST00000403712.6_2:0:VAR_2_231741_C_T\" \"ENST00000523267.1_2:0:VAR_5_179264731_T_C\" \"ENST00000512289.5_1:1:VAR_4_1323070_G_A\" ...\n",
      " $ transcript_id        : chr  \"ENST00000356150.10_3\" \"ENST00000403712.6_2\" \"ENST00000523267.1_2\" \"ENST00000512289.5_1\" ...\n",
      " $ type                 : int  0 0 0 1 0 0 0 0 0 0 ...\n",
      " $ gene_name            : chr  \"SH3YL1\" \"SH3YL1\" \"MRNIP\" \"MAEA\" ...\n",
      " $ gene_id              : chr  \"ENSG00000035115.22_8\" \"ENSG00000035115.22_8\" \"ENSG00000161010.15_8\" \"ENSG00000090316.16_5\" ...\n",
      " $ id                   : chr  \"VAR_2_231741_C_T\" \"VAR_2_231741_C_T\" \"VAR_5_179264731_T_C\" \"VAR_4_1323070_G_A\" ...\n",
      " $ pp                   : num  0.0182 0.013 0.0364 0.0102 0.1122 ...\n",
      " $ coloc_gene           : num  0.979 0.967 0.965 0.949 0.945 ...\n",
      " $ coloc_snp            : num  0.0237 0.0154 0.0674 0.0547 0.1227 ...\n",
      " $ tr2type              : chr  \"ENST00000356150.10_3:0\" \"ENST00000403712.6_2:0\" \"ENST00000523267.1_2:0\" \"ENST00000512289.5_1:1\" ...\n",
      " $ pp.fm                : logi  FALSE FALSE FALSE FALSE TRUE FALSE ...\n",
      " $ pp.gwas.gene         : logi  TRUE TRUE TRUE TRUE TRUE TRUE ...\n",
      " $ pp.gwas.snp          : logi  FALSE FALSE FALSE FALSE TRUE FALSE ...\n",
      " $ cell.adult           : logi  FALSE FALSE TRUE FALSE FALSE FALSE ...\n",
      " $ cell.arteria         : logi  TRUE TRUE FALSE TRUE TRUE TRUE ...\n",
      " $ cell.arteria_aorta   : logi  TRUE TRUE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.arteria_coronary: logi  TRUE FALSE FALSE FALSE TRUE FALSE ...\n",
      " $ cell.cardiac_muscle  : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.cardiac_neuron  : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.endocardial     : logi  TRUE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.endothelial     : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.fibroblast      : logi  TRUE FALSE FALSE FALSE TRUE FALSE ...\n",
      " $ cell.immune          : logi  TRUE TRUE FALSE FALSE TRUE FALSE ...\n",
      " $ cell.myofibroblast   : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.smooth_muscle   : logi  TRUE TRUE FALSE FALSE TRUE TRUE ...\n",
      " $ cell.heart           : logi  FALSE FALSE TRUE FALSE FALSE FALSE ...\n",
      " $ cell.heart_atrium    : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.heart_ventricle : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ cell.ipsc_cvpc       : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.FMR1             : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.FUS              : logi  FALSE FALSE TRUE TRUE FALSE FALSE ...\n",
      " $ rbp.FXR2             : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.HNRNPA1          : logi  TRUE TRUE FALSE FALSE TRUE FALSE ...\n",
      " $ rbp.HNRNPL           : logi  FALSE FALSE FALSE FALSE FALSE TRUE ...\n",
      " $ rbp.RBM5             : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ rbp.SRSF7            : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      " $ score                : int  9 6 4 3 9 4 3 5 5 5 ...\n",
      " $ is_cell              : logi  TRUE TRUE TRUE TRUE TRUE TRUE ...\n",
      " $ is_rbp               : logi  TRUE TRUE TRUE TRUE TRUE TRUE ...\n"
     ]
    }
   ],
   "source": [
    "toplot[,c(\"tr2id\", \"gene_name\", \"pp\", \"coloc_gene\", \"coloc_snp\")]\n",
    "str(toplot)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_signal = function(x, id, xlims, pos, mycol, colname = \"\", mytext = \"\")\n",
    "{\n",
    "    x$y   = x[, mycol]\n",
    "    x$cex = 1\n",
    "    x$col = \"#aaaaaa66\"\n",
    "    x[x$id == id, \"cex\"] = 3\n",
    "    x[x$id == id, \"col\"] = \"#ff0000ff\"\n",
    "    \n",
    "    par(mar = c(0,6,0,1))\n",
    "    plot(1,1, type = \"n\", xlim = xlims, ylim = c(0, max(x$y)), xlab = \"\", ylab = \"\", axes = FALSE)\n",
    "    abline(h = 0)\n",
    "    points(x$pos, x$y, cex = x$cex, col = x$col, pch = 16)\n",
    "    \n",
    "    axis(2)\n",
    "    \n",
    "    mtext(text = colname, side = 2, line = 2.5)\n",
    "    \n",
    "    text(x = xlims[[2]], y = max(x$y) * 0.5, labels = mytext, pos = 2, cex = 1.2)\n",
    "    \n",
    "    abline(v = pos, lty = \"dashed\", col = \"#ff0000\")\n",
    "}\n",
    "\n",
    "plot_transcript = function(transcript_id, xlims, pos, exons, rbps, rbp_list)\n",
    "{\n",
    "    chrom = unique(exons$chrom)\n",
    "    \n",
    "    par(mar = c(4,6,0,1))\n",
    "    plot(1,1, type = \"n\", xlim = xlims, ylim = c(1, -2), xlab = \"\", ylab = \"\", axes = FALSE)\n",
    "    axis(1, at = pretty(xlims, n = 5), labels = pretty(xlims, n = 5) * 1e-6)\n",
    "    \n",
    "    segments(x0    = min(exons$start), x1     = max(exons$end), y0      = 0.5, col = \"#0000ff\")\n",
    "    rect    (xleft =     exons$start , xright =     exons$end , ybottom = 0.1, col = \"#0000ff\", ytop = 0.9, border = \"#0000ff\")\n",
    "    \n",
    "    text(x = max(exons$end), y = 0.5, labels = transcript_id, col = \"#0000ff\", pos = 4, cex = 1.5, font = 3)\n",
    "    \n",
    "    mtext(text = paste(chrom, \"(Mb)\"), side = 1, line = 2.5)\n",
    "    \n",
    "    rbps$y = -(1:nrow(rbps)) / nrow(rbps) * 2\n",
    "    \n",
    "    mtext(text = \"Isoform\", side = 2, line = 0, las = 2, at = 0.5)\n",
    "    mtext(text = rbps$rbp , side = 2, line = 0, las = 2, at = rbps$y + (1/nrow(rbps)), cex= 1)\n",
    "    \n",
    "    for(ii in 1:nrow(rbps))\n",
    "    {\n",
    "        rbp   = rbps[ii, \"rbp\"  ]\n",
    "        y     = rbps[ii, \"y\"    ]\n",
    "        mycol = rbps[ii, \"color\"]\n",
    "        \n",
    "        this  = rbp_list[[rbp]]\n",
    "        this  = this[this$chrom == chrom & this$to >= xlims[[1]] - 1e6 & this$from <= xlims[[2]] + 1e6,]\n",
    "        \n",
    "        rect    (xleft = this$from , xright = this$to, ybottom = y + (1/nrow(rbps) * 2), ytop = y, col = mycol, border = mycol)\n",
    "    }\n",
    "    abline(v = pos, lty = \"dashed\", col = \"#ff0000\")\n",
    "}\n",
    "\n",
    "plot_example = function(tr2id, toplot, coloc, manifest, exons, rbps, rbp_list)\n",
    "{\n",
    "    transcript_id = toplot[toplot$tr2id == tr2id, \"transcript_id\"]\n",
    "    type          = toplot[toplot$tr2id == tr2id, \"type\"         ]\n",
    "    gene_id       = toplot[toplot$tr2id == tr2id, \"gene_id\"      ]\n",
    "    gene_name     = toplot[toplot$tr2id == tr2id, \"gene_name\"    ]\n",
    "    id            = toplot[toplot$tr2id == tr2id, \"id\"           ]\n",
    "    \n",
    "    exons         = exons[exons$transcript_id == transcript_id,] \n",
    "    \n",
    "    mycoloc       = coloc[coloc$transcript_id == transcript_id & coloc$type == type,]\n",
    "    mycoloc       = mycoloc[order(mycoloc$PP.H4.abf, decreasing = TRUE),]\n",
    "    mycoloc       = mycoloc[which.max(mycoloc$PP.H4.abf),]\n",
    "    trait         = mycoloc[1, \"trait\"]\n",
    "    trait_name    = manifest[manifest$id == trait, \"description\"]\n",
    "    \n",
    "    eqtl          = fread(paste(\"pipeline/3.2.eqtls/eqtls_by_gene/cardiac_eqtls.isoform\" , paste(\"qtl\", transcript_id,       \"txt\", sep = \".\"), sep = \"/\"), sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "    fm            = fread(paste(\"pipeline/3.2.eqtls/eqtls_fine_map/cardiac_eqtls.isoform\", paste(\"qtl\", transcript_id, type, \"txt\", sep = \".\"), sep = \"/\"), sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "    gwas          = fread(paste(\"pipeline/6.1.coloc_gwas/coloc.isoform\"                  , paste(\"snp\", transcript_id,       \"txt\", sep = \".\"), sep = \"/\"), sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "    eqtl          = eqtl[eqtl$type == type,]\n",
    "    eqtl$logp     = -log10(eqtl$pval)\n",
    "    fm            = merge(fm, eqtl[,c(\"id\", \"pos\")])\n",
    "    gwas          = gwas[gwas$type == type & gwas$trait == trait,]\n",
    "\n",
    "    xlims         = range(eqtl$pos)\n",
    "    pos           = eqtl[eqtl$id == id, \"pos\"]\n",
    "    \n",
    "    totext_cell   = unlist(lapply(colnames(toplot)[grepl(\"^cell\\\\.\", colnames(toplot)) == TRUE], function(cell)\n",
    "    {\n",
    "        out = \"\"\n",
    "        if(toplot[toplot$tr2id == tr2id, cell] == TRUE){out = sub(\"^cell\\\\.\", \"\", cell)}\n",
    "        \n",
    "        return(out)\n",
    "    }))\n",
    "    \n",
    "    totext_rbp   = unlist(lapply(colnames(toplot)[grepl(\"^rbp\\\\.\", colnames(toplot)) == TRUE], function(rbp)\n",
    "    {\n",
    "        out = \"\"\n",
    "        if(toplot[toplot$tr2id == tr2id, rbp] == TRUE){out = sub(\"^rbp\\\\.\", \"\", rbp)}\n",
    "        \n",
    "        return(out)\n",
    "    }))\n",
    " \n",
    "    png(filename = paste(\"pipeline/6.1.coloc_gwas/intersect_rbps/examples\", paste(gene_name, gsub(\":\", \".\", tr2id), \"png\", sep = \".\"), sep = \"/\"), width = 8.5, height = 8.5, units = \"in\", res = 300)\n",
    "    \n",
    "    mytext   = paste(c(gene_name, paste(totext_cell[totext_cell != \"\"], collapse = \"\\n\"), paste(totext_rbp[totext_rbp != \"\"], collapse = \"\\n\")), collapse = \"\\n\")\n",
    "    mylayout = c(1:4)\n",
    "    layout(cbind(mylayout, mylayout + 4))\n",
    "\n",
    "    \n",
    "    plot_signal    (eqtl, id     , xlims, pos, \"logp\"  , \"-log10 (p-value eQTL)\", mytext)\n",
    "    plot_signal    (fm  , id     , xlims, pos, \"pp\"    , \"PPA (fine-map)\"       , trait_name)\n",
    "    plot_signal    (gwas, id     , xlims, pos, \"pp_snp\", \"PPA (GWAS)\")\n",
    "    plot_transcript(transcript_id, xlims, pos, exons, rbps, rbp_list)\n",
    "    \n",
    "    plot_signal    (eqtl, id     , c(min(exons$start), max(exons$end)), pos, \"logp\"  , \"-log10 (p-value eQTL)\")\n",
    "    plot_signal    (fm  , id     , c(min(exons$start), max(exons$end)), pos, \"pp\"    , \"PPA (fine-map)\"       )\n",
    "    plot_signal    (gwas, id     , c(min(exons$start), max(exons$end)), pos, \"pp_snp\", \"PPA (GWAS)\"           )\n",
    "    plot_transcript(transcript_id, c(min(exons$start), max(exons$end)), pos, exons, rbps, rbp_list)\n",
    "    \n",
    "    dev.off()   \n",
    "}\n",
    "\n",
    "options(repr.plot.width = 8.5, repr.plot.height = 11)\n",
    "\n",
    "\n",
    "dir.create(\"pipeline/6.1.coloc_gwas/intersect_rbps/examples\", showWarnings = FALSE)\n",
    "invisible(lapply(toplot$tr2id, function(tr2id){try(plot_example(tr2id, toplot, coloc, manifest, exons, rbps, rbp_list))}))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'cell.adult'</li><li>'cell.arteria'</li><li>'cell.arteria_aorta'</li><li>'cell.arteria_coronary'</li><li>'cell.cardiac_muscle'</li><li>'cell.cardiac_neuron'</li><li>'cell.endocardial'</li><li>'cell.endothelial'</li><li>'cell.fibroblast'</li><li>'cell.immune'</li><li>'cell.myofibroblast'</li><li>'cell.smooth_muscle'</li><li>'cell.heart'</li><li>'cell.heart_atrium'</li><li>'cell.heart_ventricle'</li><li>'cell.ipsc_cvpc'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'cell.adult'\n",
       "\\item 'cell.arteria'\n",
       "\\item 'cell.arteria\\_aorta'\n",
       "\\item 'cell.arteria\\_coronary'\n",
       "\\item 'cell.cardiac\\_muscle'\n",
       "\\item 'cell.cardiac\\_neuron'\n",
       "\\item 'cell.endocardial'\n",
       "\\item 'cell.endothelial'\n",
       "\\item 'cell.fibroblast'\n",
       "\\item 'cell.immune'\n",
       "\\item 'cell.myofibroblast'\n",
       "\\item 'cell.smooth\\_muscle'\n",
       "\\item 'cell.heart'\n",
       "\\item 'cell.heart\\_atrium'\n",
       "\\item 'cell.heart\\_ventricle'\n",
       "\\item 'cell.ipsc\\_cvpc'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'cell.adult'\n",
       "2. 'cell.arteria'\n",
       "3. 'cell.arteria_aorta'\n",
       "4. 'cell.arteria_coronary'\n",
       "5. 'cell.cardiac_muscle'\n",
       "6. 'cell.cardiac_neuron'\n",
       "7. 'cell.endocardial'\n",
       "8. 'cell.endothelial'\n",
       "9. 'cell.fibroblast'\n",
       "10. 'cell.immune'\n",
       "11. 'cell.myofibroblast'\n",
       "12. 'cell.smooth_muscle'\n",
       "13. 'cell.heart'\n",
       "14. 'cell.heart_atrium'\n",
       "15. 'cell.heart_ventricle'\n",
       "16. 'cell.ipsc_cvpc'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"cell.adult\"            \"cell.arteria\"          \"cell.arteria_aorta\"   \n",
       " [4] \"cell.arteria_coronary\" \"cell.cardiac_muscle\"   \"cell.cardiac_neuron\"  \n",
       " [7] \"cell.endocardial\"      \"cell.endothelial\"      \"cell.fibroblast\"      \n",
       "[10] \"cell.immune\"           \"cell.myofibroblast\"    \"cell.smooth_muscle\"   \n",
       "[13] \"cell.heart\"            \"cell.heart_atrium\"     \"cell.heart_ventricle\" \n",
       "[16] \"cell.ipsc_cvpc\"       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fm[fm]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 2 × 39</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>tr2id</th><th scope=col>transcript_id</th><th scope=col>type</th><th scope=col>gene_name</th><th scope=col>gene_id</th><th scope=col>id</th><th scope=col>pp</th><th scope=col>coloc_gene</th><th scope=col>coloc_snp</th><th scope=col>tr2type</th><th scope=col>⋯</th><th scope=col>rbp.FMR1</th><th scope=col>rbp.FUS</th><th scope=col>rbp.FXR2</th><th scope=col>rbp.HNRNPA1</th><th scope=col>rbp.HNRNPL</th><th scope=col>rbp.RBM5</th><th scope=col>rbp.SRSF7</th><th scope=col>score</th><th scope=col>is_cell</th><th scope=col>is_rbp</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>27698</th><td>ENST00000462256.5_2:0:VAR_6_31501737_C_T</td><td>ENST00000462256.5_2</td><td>0</td><td>DDX39B</td><td>ENSG00000198563.14_9</td><td>VAR_6_31501737_C_T</td><td>0.06335733</td><td>0.9083003</td><td>0.18787441</td><td>ENST00000462256.5_2:0</td><td>⋯</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td><td>FALSE</td><td>4</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>27703</th><td>ENST00000462256.5_2:0:VAR_6_31502767_C_T</td><td>ENST00000462256.5_2</td><td>0</td><td>DDX39B</td><td>ENSG00000198563.14_9</td><td>VAR_6_31502767_C_T</td><td>0.01344197</td><td>0.9083003</td><td>0.02069228</td><td>ENST00000462256.5_2:0</td><td>⋯</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>FALSE</td><td>TRUE</td><td>FALSE</td><td>FALSE</td><td>3</td><td>TRUE</td><td>TRUE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 39\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & tr2id & transcript\\_id & type & gene\\_name & gene\\_id & id & pp & coloc\\_gene & coloc\\_snp & tr2type & ⋯ & rbp.FMR1 & rbp.FUS & rbp.FXR2 & rbp.HNRNPA1 & rbp.HNRNPL & rbp.RBM5 & rbp.SRSF7 & score & is\\_cell & is\\_rbp\\\\\n",
       "  & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & ⋯ & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <lgl> & <int> & <lgl> & <lgl>\\\\\n",
       "\\hline\n",
       "\t27698 & ENST00000462256.5\\_2:0:VAR\\_6\\_31501737\\_C\\_T & ENST00000462256.5\\_2 & 0 & DDX39B & ENSG00000198563.14\\_9 & VAR\\_6\\_31501737\\_C\\_T & 0.06335733 & 0.9083003 & 0.18787441 & ENST00000462256.5\\_2:0 & ⋯ & FALSE & FALSE & FALSE & FALSE & TRUE & FALSE & FALSE & 4 & TRUE & TRUE\\\\\n",
       "\t27703 & ENST00000462256.5\\_2:0:VAR\\_6\\_31502767\\_C\\_T & ENST00000462256.5\\_2 & 0 & DDX39B & ENSG00000198563.14\\_9 & VAR\\_6\\_31502767\\_C\\_T & 0.01344197 & 0.9083003 & 0.02069228 & ENST00000462256.5\\_2:0 & ⋯ & FALSE & FALSE & FALSE & FALSE & TRUE & FALSE & FALSE & 3 & TRUE & TRUE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 39\n",
       "\n",
       "| <!--/--> | tr2id &lt;chr&gt; | transcript_id &lt;chr&gt; | type &lt;int&gt; | gene_name &lt;chr&gt; | gene_id &lt;chr&gt; | id &lt;chr&gt; | pp &lt;dbl&gt; | coloc_gene &lt;dbl&gt; | coloc_snp &lt;dbl&gt; | tr2type &lt;chr&gt; | ⋯ ⋯ | rbp.FMR1 &lt;lgl&gt; | rbp.FUS &lt;lgl&gt; | rbp.FXR2 &lt;lgl&gt; | rbp.HNRNPA1 &lt;lgl&gt; | rbp.HNRNPL &lt;lgl&gt; | rbp.RBM5 &lt;lgl&gt; | rbp.SRSF7 &lt;lgl&gt; | score &lt;int&gt; | is_cell &lt;lgl&gt; | is_rbp &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 27698 | ENST00000462256.5_2:0:VAR_6_31501737_C_T | ENST00000462256.5_2 | 0 | DDX39B | ENSG00000198563.14_9 | VAR_6_31501737_C_T | 0.06335733 | 0.9083003 | 0.18787441 | ENST00000462256.5_2:0 | ⋯ | FALSE | FALSE | FALSE | FALSE | TRUE | FALSE | FALSE | 4 | TRUE | TRUE |\n",
       "| 27703 | ENST00000462256.5_2:0:VAR_6_31502767_C_T | ENST00000462256.5_2 | 0 | DDX39B | ENSG00000198563.14_9 | VAR_6_31502767_C_T | 0.01344197 | 0.9083003 | 0.02069228 | ENST00000462256.5_2:0 | ⋯ | FALSE | FALSE | FALSE | FALSE | TRUE | FALSE | FALSE | 3 | TRUE | TRUE |\n",
       "\n"
      ],
      "text/plain": [
       "      tr2id                                    transcript_id       type\n",
       "27698 ENST00000462256.5_2:0:VAR_6_31501737_C_T ENST00000462256.5_2 0   \n",
       "27703 ENST00000462256.5_2:0:VAR_6_31502767_C_T ENST00000462256.5_2 0   \n",
       "      gene_name gene_id              id                 pp         coloc_gene\n",
       "27698 DDX39B    ENSG00000198563.14_9 VAR_6_31501737_C_T 0.06335733 0.9083003 \n",
       "27703 DDX39B    ENSG00000198563.14_9 VAR_6_31502767_C_T 0.01344197 0.9083003 \n",
       "      coloc_snp  tr2type               ⋯ rbp.FMR1 rbp.FUS rbp.FXR2 rbp.HNRNPA1\n",
       "27698 0.18787441 ENST00000462256.5_2:0 ⋯ FALSE    FALSE   FALSE    FALSE      \n",
       "27703 0.02069228 ENST00000462256.5_2:0 ⋯ FALSE    FALSE   FALSE    FALSE      \n",
       "      rbp.HNRNPL rbp.RBM5 rbp.SRSF7 score is_cell is_rbp\n",
       "27698 TRUE       FALSE    FALSE     4     TRUE    TRUE  \n",
       "27703 TRUE       FALSE    FALSE     3     TRUE    TRUE  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "toplot[toplot$gene_name == \"DDX39B\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.0.1",
   "language": "R",
   "name": "ir401"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
