{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/frazer01/projects/CARDIPS/analysis/cardiac_eqtls\")\n",
    "\n",
    "source(\"script/packages.R\"  )\n",
    "source(\"script/input_data.R\")\n",
    "source(\"script/functions.R\" )\n",
    "source(\"script/colors.R\"    )\n",
    "\n",
    "library(shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "geneinfo_gene       = fread(\"pipeline/1.2.expression/gene_info.txt\"                          , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "geneinfo_isoform    = fread(\"pipeline/1.2.expression/isoform_info.txt\"                       , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "diffexp             = fread(\"pipeline/4.1.differential_expression/diffexp.txt\"               , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "diffexp_cell        = fread(\"pipeline/4.1.differential_expression/diffexp_cell.txt\"          , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "diffexp_cell_tissue = fread(\"pipeline/4.1.differential_expression/diffexp_cell_tissue.txt\"   , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "coloc_gene          = fread(\"pipeline/6.1.coloc_gwas/coloc.eqtls.gene.txt\"                   , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "coloc_isoform       = fread(\"pipeline/6.1.coloc_gwas/coloc.eqtls.isoform.txt\"                , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "eqtl_genes          = fread(\"pipeline/3.2.eqtls/eqtls/cardiac_eqtls.gene.egenes.txt\"         , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "int_genes           = fread(\"pipeline/3.2.eqtls/eqtls/cardiac_eqtls.gene.interactions.txt\"   , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "eqtl_isoforms       = fread(\"pipeline/3.2.eqtls/eqtls/cardiac_eqtls.isoform.egenes.txt\"      , sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "int_isoforms        = fread(\"pipeline/3.2.eqtls/eqtls/cardiac_eqtls.isoform.interactions.txt\", sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "\n",
    "# external\n",
    "diffexp_quadrant = fread(\"/frazer01/home/jennifer/projects/cardiac_eqtls/diffexp/results/summary.cell_stage_associations.txt\", sep = \"\\t\", header = TRUE , data.table = FALSE)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#diffexp_quadrant                = unique(diffexp_quadrant[,c(\"cell_type\", \"tissue1\", \"tissue2\")])\n",
    "diffexp            $tissue      = paste(diffexp$tissue1, diffexp$tissue2, sep = \"_vs_\")\n",
    "diffexp_cell       $cell_type   = gsub(\"cibersort.regular\\\\.\", \"\", diffexp_cell       $cell_type)\n",
    "diffexp_cell_tissue$cell_type   = gsub(\"cibersort.regular\\\\.\", \"\", diffexp_cell_tissue$cell_type)\n",
    "diffexp_cell_tissue$tissue2cell = paste(diffexp_cell_tissue$cell_type, paste(diffexp_cell_tissue$tissue1, diffexp_cell_tissue$tissue2, sep = \"_vs_\"), sep = \".\")\n",
    "diffexp_quadrant   $tissue2cell = paste(diffexp_quadrant   $cell_type, paste(diffexp_quadrant   $tissue1, diffexp_quadrant   $tissue2, sep = \"_vs_\"), sep = \".\")\n",
    "diffexp_quadrant   $is_tissue1  = FALSE\n",
    "diffexp_quadrant   $is_tissue2  = FALSE\n",
    "\n",
    "diffexp_quadrant[diffexp_quadrant$is_stage_associated == TRUE & diffexp_quadrant$stage.tissue1_qval < 0.05 & diffexp_quadrant$stage.tissue1_beta > 0, \"is_tissue1\"] = TRUE\n",
    "diffexp_quadrant[diffexp_quadrant$is_stage_associated == TRUE & diffexp_quadrant$stage.tissue2_qval < 0.05 & diffexp_quadrant$stage.tissue2_beta > 0, \"is_tissue2\"] = TRUE\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>diffexp.heart_vs_arteria</dt><dd>13366</dd><dt>diffexp.heart_vs_arteria.heart</dt><dd>5517</dd><dt>diffexp.heart_vs_arteria.arteria</dt><dd>7849</dd><dt>diffexp.ipsc_cvpc_vs_arteria</dt><dd>15468</dd><dt>diffexp.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>8812</dd><dt>diffexp.ipsc_cvpc_vs_arteria.arteria</dt><dd>6656</dd><dt>diffexp.ipsc_cvpc_vs_heart</dt><dd>14724</dd><dt>diffexp.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>8836</dd><dt>diffexp.ipsc_cvpc_vs_heart.heart</dt><dd>5888</dd><dt>diffexp_cell.cardiac_muscle</dt><dd>4942</dd><dt>diffexp_cell.cardiac_neuron</dt><dd>3017</dd><dt>diffexp_cell.endocardial</dt><dd>3750</dd><dt>diffexp_cell.endothelial</dt><dd>3665</dd><dt>diffexp_cell.fibroblast</dt><dd>4379</dd><dt>diffexp_cell.immune</dt><dd>2890</dd><dt>diffexp_cell.myofibroblast</dt><dd>1858</dd><dt>diffexp_cell.smooth_muscle</dt><dd>2756</dd><dt>diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>421</dd><dt>diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.heart</dt><dd>2706</dd><dt>diffexp_quadrant.endocardial.heart_vs_arteria.heart</dt><dd>3139</dd><dt>diffexp_quadrant.endocardial.heart_vs_arteria.arteria</dt><dd>211</dd><dt>diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>376</dd><dt>diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.arteria</dt><dd>622</dd><dt>diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>217</dd><dt>diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.heart</dt><dd>3404</dd><dt>diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>588</dd><dt>diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.heart</dt><dd>2078</dd><dt>diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>132</dd><dt>diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.arteria</dt><dd>1983</dd><dt>diffexp_quadrant.none.heart_vs_arteria.heart</dt><dd>426</dd><dt>diffexp_quadrant.none.heart_vs_arteria.arteria</dt><dd>183</dd><dt>diffexp_quadrant.none.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>185</dd><dt>diffexp_quadrant.none.ipsc_cvpc_vs_arteria.arteria</dt><dd>205</dd><dt>diffexp_quadrant.none.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>296</dd><dt>diffexp_quadrant.none.ipsc_cvpc_vs_heart.heart</dt><dd>796</dd><dt>diffexp_quadrant.smooth_muscle.heart_vs_arteria.heart</dt><dd>420</dd><dt>diffexp_quadrant.smooth_muscle.heart_vs_arteria.arteria</dt><dd>1947</dd><dt>diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>464</dd><dt>diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.arteria</dt><dd>2096</dd><dt>diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>572</dd><dt>diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.heart</dt><dd>525</dd><dt>egene</dt><dd>11692</dd><dt>eqtl_interaction.stage.adult</dt><dd>561</dd><dt>eqtl_interaction.tissue.arteria</dt><dd>588</dd><dt>eqtl_interaction.tissue.arteria_aorta</dt><dd>409</dd><dt>eqtl_interaction.tissue.arteria_coronary</dt><dd>80</dd><dt>eqtl_interaction.cell.cardiac_muscle</dt><dd>427</dd><dt>eqtl_interaction.cell.cardiac_neuron</dt><dd>118</dd><dt>eqtl_interaction.cell.endocardial</dt><dd>79</dd><dt>eqtl_interaction.cell.endothelial</dt><dd>10</dd><dt>eqtl_interaction.cell.fibroblast</dt><dd>219</dd><dt>eqtl_interaction.cell.immune</dt><dd>303</dd><dt>eqtl_interaction.cell.myofibroblast</dt><dd>3</dd><dt>eqtl_interaction.cell.smooth_muscle</dt><dd>468</dd><dt>eqtl_interaction.tissue.heart</dt><dd>505</dd><dt>eqtl_interaction.tissue.heart_atrium</dt><dd>184</dd><dt>eqtl_interaction.tissue.heart_ventricle</dt><dd>287</dd><dt>eqtl_interaction.stage.ipsc_cvpc</dt><dd>181</dd><dt>gwas</dt><dd>3722</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[diffexp.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria] 13366\n",
       "\\item[diffexp.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.heart] 5517\n",
       "\\item[diffexp.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 7849\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria] 15468\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 8812\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 6656\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart] 14724\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 8836\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 5888\n",
       "\\item[diffexp\\textbackslash{}\\_cell.cardiac\\textbackslash{}\\_muscle] 4942\n",
       "\\item[diffexp\\textbackslash{}\\_cell.cardiac\\textbackslash{}\\_neuron] 3017\n",
       "\\item[diffexp\\textbackslash{}\\_cell.endocardial] 3750\n",
       "\\item[diffexp\\textbackslash{}\\_cell.endothelial] 3665\n",
       "\\item[diffexp\\textbackslash{}\\_cell.fibroblast] 4379\n",
       "\\item[diffexp\\textbackslash{}\\_cell.immune] 2890\n",
       "\\item[diffexp\\textbackslash{}\\_cell.myofibroblast] 1858\n",
       "\\item[diffexp\\textbackslash{}\\_cell.smooth\\textbackslash{}\\_muscle] 2756\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.cardiac\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 421\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.cardiac\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 2706\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.heart] 3139\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 211\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 376\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 622\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 217\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 3404\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endothelial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 588\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endothelial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 2078\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.immune.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 132\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.immune.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 1983\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.heart] 426\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 183\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 185\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 205\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 296\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 796\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.heart] 420\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 1947\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 464\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 2096\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 572\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 525\n",
       "\\item[egene] 11692\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.stage.adult] 561\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.arteria] 588\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.arteria\\textbackslash{}\\_aorta] 409\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.arteria\\textbackslash{}\\_coronary] 80\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.cardiac\\textbackslash{}\\_muscle] 427\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.cardiac\\textbackslash{}\\_neuron] 118\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.endocardial] 79\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.endothelial] 10\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.fibroblast] 219\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.immune] 303\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.myofibroblast] 3\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.smooth\\textbackslash{}\\_muscle] 468\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.heart] 505\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.heart\\textbackslash{}\\_atrium] 184\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.heart\\textbackslash{}\\_ventricle] 287\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.stage.ipsc\\textbackslash{}\\_cvpc] 181\n",
       "\\item[gwas] 3722\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "diffexp.heart_vs_arteria\n",
       ":   13366diffexp.heart_vs_arteria.heart\n",
       ":   5517diffexp.heart_vs_arteria.arteria\n",
       ":   7849diffexp.ipsc_cvpc_vs_arteria\n",
       ":   15468diffexp.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   8812diffexp.ipsc_cvpc_vs_arteria.arteria\n",
       ":   6656diffexp.ipsc_cvpc_vs_heart\n",
       ":   14724diffexp.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   8836diffexp.ipsc_cvpc_vs_heart.heart\n",
       ":   5888diffexp_cell.cardiac_muscle\n",
       ":   4942diffexp_cell.cardiac_neuron\n",
       ":   3017diffexp_cell.endocardial\n",
       ":   3750diffexp_cell.endothelial\n",
       ":   3665diffexp_cell.fibroblast\n",
       ":   4379diffexp_cell.immune\n",
       ":   2890diffexp_cell.myofibroblast\n",
       ":   1858diffexp_cell.smooth_muscle\n",
       ":   2756diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   421diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.heart\n",
       ":   2706diffexp_quadrant.endocardial.heart_vs_arteria.heart\n",
       ":   3139diffexp_quadrant.endocardial.heart_vs_arteria.arteria\n",
       ":   211diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   376diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.arteria\n",
       ":   622diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   217diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.heart\n",
       ":   3404diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   588diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.heart\n",
       ":   2078diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   132diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.arteria\n",
       ":   1983diffexp_quadrant.none.heart_vs_arteria.heart\n",
       ":   426diffexp_quadrant.none.heart_vs_arteria.arteria\n",
       ":   183diffexp_quadrant.none.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   185diffexp_quadrant.none.ipsc_cvpc_vs_arteria.arteria\n",
       ":   205diffexp_quadrant.none.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   296diffexp_quadrant.none.ipsc_cvpc_vs_heart.heart\n",
       ":   796diffexp_quadrant.smooth_muscle.heart_vs_arteria.heart\n",
       ":   420diffexp_quadrant.smooth_muscle.heart_vs_arteria.arteria\n",
       ":   1947diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   464diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.arteria\n",
       ":   2096diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   572diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.heart\n",
       ":   525egene\n",
       ":   11692eqtl_interaction.stage.adult\n",
       ":   561eqtl_interaction.tissue.arteria\n",
       ":   588eqtl_interaction.tissue.arteria_aorta\n",
       ":   409eqtl_interaction.tissue.arteria_coronary\n",
       ":   80eqtl_interaction.cell.cardiac_muscle\n",
       ":   427eqtl_interaction.cell.cardiac_neuron\n",
       ":   118eqtl_interaction.cell.endocardial\n",
       ":   79eqtl_interaction.cell.endothelial\n",
       ":   10eqtl_interaction.cell.fibroblast\n",
       ":   219eqtl_interaction.cell.immune\n",
       ":   303eqtl_interaction.cell.myofibroblast\n",
       ":   3eqtl_interaction.cell.smooth_muscle\n",
       ":   468eqtl_interaction.tissue.heart\n",
       ":   505eqtl_interaction.tissue.heart_atrium\n",
       ":   184eqtl_interaction.tissue.heart_ventricle\n",
       ":   287eqtl_interaction.stage.ipsc_cvpc\n",
       ":   181gwas\n",
       ":   3722\n",
       "\n"
      ],
      "text/plain": [
       "                                     diffexp.heart_vs_arteria \n",
       "                                                        13366 \n",
       "                               diffexp.heart_vs_arteria.heart \n",
       "                                                         5517 \n",
       "                             diffexp.heart_vs_arteria.arteria \n",
       "                                                         7849 \n",
       "                                 diffexp.ipsc_cvpc_vs_arteria \n",
       "                                                        15468 \n",
       "                       diffexp.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                         8812 \n",
       "                         diffexp.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                         6656 \n",
       "                                   diffexp.ipsc_cvpc_vs_heart \n",
       "                                                        14724 \n",
       "                         diffexp.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                         8836 \n",
       "                             diffexp.ipsc_cvpc_vs_heart.heart \n",
       "                                                         5888 \n",
       "                                  diffexp_cell.cardiac_muscle \n",
       "                                                         4942 \n",
       "                                  diffexp_cell.cardiac_neuron \n",
       "                                                         3017 \n",
       "                                     diffexp_cell.endocardial \n",
       "                                                         3750 \n",
       "                                     diffexp_cell.endothelial \n",
       "                                                         3665 \n",
       "                                      diffexp_cell.fibroblast \n",
       "                                                         4379 \n",
       "                                          diffexp_cell.immune \n",
       "                                                         2890 \n",
       "                                   diffexp_cell.myofibroblast \n",
       "                                                         1858 \n",
       "                                   diffexp_cell.smooth_muscle \n",
       "                                                         2756 \n",
       " diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                          421 \n",
       "     diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.heart \n",
       "                                                         2706 \n",
       "          diffexp_quadrant.endocardial.heart_vs_arteria.heart \n",
       "                                                         3139 \n",
       "        diffexp_quadrant.endocardial.heart_vs_arteria.arteria \n",
       "                                                          211 \n",
       "  diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                          376 \n",
       "    diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                          622 \n",
       "    diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                          217 \n",
       "        diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.heart \n",
       "                                                         3404 \n",
       "    diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                          588 \n",
       "        diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.heart \n",
       "                                                         2078 \n",
       "       diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                          132 \n",
       "         diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                         1983 \n",
       "                 diffexp_quadrant.none.heart_vs_arteria.heart \n",
       "                                                          426 \n",
       "               diffexp_quadrant.none.heart_vs_arteria.arteria \n",
       "                                                          183 \n",
       "         diffexp_quadrant.none.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                          185 \n",
       "           diffexp_quadrant.none.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                          205 \n",
       "           diffexp_quadrant.none.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                          296 \n",
       "               diffexp_quadrant.none.ipsc_cvpc_vs_heart.heart \n",
       "                                                          796 \n",
       "        diffexp_quadrant.smooth_muscle.heart_vs_arteria.heart \n",
       "                                                          420 \n",
       "      diffexp_quadrant.smooth_muscle.heart_vs_arteria.arteria \n",
       "                                                         1947 \n",
       "diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                          464 \n",
       "  diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                         2096 \n",
       "  diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                          572 \n",
       "      diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.heart \n",
       "                                                          525 \n",
       "                                                        egene \n",
       "                                                        11692 \n",
       "                                 eqtl_interaction.stage.adult \n",
       "                                                          561 \n",
       "                              eqtl_interaction.tissue.arteria \n",
       "                                                          588 \n",
       "                        eqtl_interaction.tissue.arteria_aorta \n",
       "                                                          409 \n",
       "                     eqtl_interaction.tissue.arteria_coronary \n",
       "                                                           80 \n",
       "                         eqtl_interaction.cell.cardiac_muscle \n",
       "                                                          427 \n",
       "                         eqtl_interaction.cell.cardiac_neuron \n",
       "                                                          118 \n",
       "                            eqtl_interaction.cell.endocardial \n",
       "                                                           79 \n",
       "                            eqtl_interaction.cell.endothelial \n",
       "                                                           10 \n",
       "                             eqtl_interaction.cell.fibroblast \n",
       "                                                          219 \n",
       "                                 eqtl_interaction.cell.immune \n",
       "                                                          303 \n",
       "                          eqtl_interaction.cell.myofibroblast \n",
       "                                                            3 \n",
       "                          eqtl_interaction.cell.smooth_muscle \n",
       "                                                          468 \n",
       "                                eqtl_interaction.tissue.heart \n",
       "                                                          505 \n",
       "                         eqtl_interaction.tissue.heart_atrium \n",
       "                                                          184 \n",
       "                      eqtl_interaction.tissue.heart_ventricle \n",
       "                                                          287 \n",
       "                             eqtl_interaction.stage.ipsc_cvpc \n",
       "                                                          181 \n",
       "                                                         gwas \n",
       "                                                         3722 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>diffexp.heart_vs_arteria</dt><dd>11596</dd><dt>diffexp.heart_vs_arteria.heart</dt><dd>5124</dd><dt>diffexp.heart_vs_arteria.arteria</dt><dd>6472</dd><dt>diffexp.ipsc_cvpc_vs_arteria</dt><dd>21021</dd><dt>diffexp.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>11091</dd><dt>diffexp.ipsc_cvpc_vs_arteria.arteria</dt><dd>9930</dd><dt>diffexp.ipsc_cvpc_vs_heart</dt><dd>19270</dd><dt>diffexp.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>11127</dd><dt>diffexp.ipsc_cvpc_vs_heart.heart</dt><dd>8143</dd><dt>diffexp_cell.cardiac_muscle</dt><dd>2993</dd><dt>diffexp_cell.cardiac_neuron</dt><dd>1008</dd><dt>diffexp_cell.endocardial</dt><dd>2022</dd><dt>diffexp_cell.endothelial</dt><dd>866</dd><dt>diffexp_cell.fibroblast</dt><dd>3691</dd><dt>diffexp_cell.immune</dt><dd>740</dd><dt>diffexp_cell.myofibroblast</dt><dd>285</dd><dt>diffexp_cell.smooth_muscle</dt><dd>667</dd><dt>diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>176</dd><dt>diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.heart</dt><dd>1656</dd><dt>diffexp_quadrant.endocardial.heart_vs_arteria.heart</dt><dd>1279</dd><dt>diffexp_quadrant.endocardial.heart_vs_arteria.arteria</dt><dd>18</dd><dt>diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>189</dd><dt>diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.arteria</dt><dd>24</dd><dt>diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>108</dd><dt>diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.heart</dt><dd>1204</dd><dt>diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>117</dd><dt>diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.heart</dt><dd>359</dd><dt>diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>17</dd><dt>diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.arteria</dt><dd>304</dd><dt>diffexp_quadrant.none.heart_vs_arteria.heart</dt><dd>577</dd><dt>diffexp_quadrant.none.heart_vs_arteria.arteria</dt><dd>50</dd><dt>diffexp_quadrant.none.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>99</dd><dt>diffexp_quadrant.none.ipsc_cvpc_vs_arteria.arteria</dt><dd>70</dd><dt>diffexp_quadrant.none.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>179</dd><dt>diffexp_quadrant.none.ipsc_cvpc_vs_heart.heart</dt><dd>1088</dd><dt>diffexp_quadrant.smooth_muscle.heart_vs_arteria.heart</dt><dd>140</dd><dt>diffexp_quadrant.smooth_muscle.heart_vs_arteria.arteria</dt><dd>264</dd><dt>diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.ipsc_cvpc</dt><dd>50</dd><dt>diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.arteria</dt><dd>266</dd><dt>diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc</dt><dd>46</dd><dt>diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.heart</dt><dd>145</dd><dt>egene</dt><dd>7165</dd><dt>eqtl_interaction.stage.adult</dt><dd>183</dd><dt>eqtl_interaction.tissue.arteria</dt><dd>243</dd><dt>eqtl_interaction.tissue.arteria_aorta</dt><dd>125</dd><dt>eqtl_interaction.tissue.arteria_coronary</dt><dd>29</dd><dt>eqtl_interaction.cell.cardiac_muscle</dt><dd>86</dd><dt>eqtl_interaction.cell.cardiac_neuron</dt><dd>32</dd><dt>eqtl_interaction.cell.endocardial</dt><dd>31</dd><dt>eqtl_interaction.cell.endothelial</dt><dd>3</dd><dt>eqtl_interaction.cell.fibroblast</dt><dd>68</dd><dt>eqtl_interaction.cell.immune</dt><dd>114</dd><dt>eqtl_interaction.cell.myofibroblast</dt><dd>2</dd><dt>eqtl_interaction.cell.smooth_muscle</dt><dd>195</dd><dt>eqtl_interaction.tissue.heart</dt><dd>83</dd><dt>eqtl_interaction.tissue.heart_atrium</dt><dd>32</dd><dt>eqtl_interaction.tissue.heart_ventricle</dt><dd>31</dd><dt>eqtl_interaction.stage.ipsc_cvpc</dt><dd>102</dd><dt>gwas</dt><dd>2364</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[diffexp.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria] 11596\n",
       "\\item[diffexp.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.heart] 5124\n",
       "\\item[diffexp.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 6472\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria] 21021\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 11091\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 9930\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart] 19270\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 11127\n",
       "\\item[diffexp.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 8143\n",
       "\\item[diffexp\\textbackslash{}\\_cell.cardiac\\textbackslash{}\\_muscle] 2993\n",
       "\\item[diffexp\\textbackslash{}\\_cell.cardiac\\textbackslash{}\\_neuron] 1008\n",
       "\\item[diffexp\\textbackslash{}\\_cell.endocardial] 2022\n",
       "\\item[diffexp\\textbackslash{}\\_cell.endothelial] 866\n",
       "\\item[diffexp\\textbackslash{}\\_cell.fibroblast] 3691\n",
       "\\item[diffexp\\textbackslash{}\\_cell.immune] 740\n",
       "\\item[diffexp\\textbackslash{}\\_cell.myofibroblast] 285\n",
       "\\item[diffexp\\textbackslash{}\\_cell.smooth\\textbackslash{}\\_muscle] 667\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.cardiac\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 176\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.cardiac\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 1656\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.heart] 1279\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 18\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 189\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 24\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 108\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endocardial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 1204\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endothelial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 117\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.endothelial.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 359\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.immune.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 17\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.immune.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 304\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.heart] 577\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 50\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 99\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 70\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 179\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.none.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 1088\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.heart] 140\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.heart\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 264\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.ipsc\\textbackslash{}\\_cvpc] 50\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_arteria.arteria] 266\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.ipsc\\textbackslash{}\\_cvpc] 46\n",
       "\\item[diffexp\\textbackslash{}\\_quadrant.smooth\\textbackslash{}\\_muscle.ipsc\\textbackslash{}\\_cvpc\\textbackslash{}\\_vs\\textbackslash{}\\_heart.heart] 145\n",
       "\\item[egene] 7165\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.stage.adult] 183\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.arteria] 243\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.arteria\\textbackslash{}\\_aorta] 125\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.arteria\\textbackslash{}\\_coronary] 29\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.cardiac\\textbackslash{}\\_muscle] 86\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.cardiac\\textbackslash{}\\_neuron] 32\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.endocardial] 31\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.endothelial] 3\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.fibroblast] 68\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.immune] 114\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.myofibroblast] 2\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.cell.smooth\\textbackslash{}\\_muscle] 195\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.heart] 83\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.heart\\textbackslash{}\\_atrium] 32\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.tissue.heart\\textbackslash{}\\_ventricle] 31\n",
       "\\item[eqtl\\textbackslash{}\\_interaction.stage.ipsc\\textbackslash{}\\_cvpc] 102\n",
       "\\item[gwas] 2364\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "diffexp.heart_vs_arteria\n",
       ":   11596diffexp.heart_vs_arteria.heart\n",
       ":   5124diffexp.heart_vs_arteria.arteria\n",
       ":   6472diffexp.ipsc_cvpc_vs_arteria\n",
       ":   21021diffexp.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   11091diffexp.ipsc_cvpc_vs_arteria.arteria\n",
       ":   9930diffexp.ipsc_cvpc_vs_heart\n",
       ":   19270diffexp.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   11127diffexp.ipsc_cvpc_vs_heart.heart\n",
       ":   8143diffexp_cell.cardiac_muscle\n",
       ":   2993diffexp_cell.cardiac_neuron\n",
       ":   1008diffexp_cell.endocardial\n",
       ":   2022diffexp_cell.endothelial\n",
       ":   866diffexp_cell.fibroblast\n",
       ":   3691diffexp_cell.immune\n",
       ":   740diffexp_cell.myofibroblast\n",
       ":   285diffexp_cell.smooth_muscle\n",
       ":   667diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   176diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.heart\n",
       ":   1656diffexp_quadrant.endocardial.heart_vs_arteria.heart\n",
       ":   1279diffexp_quadrant.endocardial.heart_vs_arteria.arteria\n",
       ":   18diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   189diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.arteria\n",
       ":   24diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   108diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.heart\n",
       ":   1204diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   117diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.heart\n",
       ":   359diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   17diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.arteria\n",
       ":   304diffexp_quadrant.none.heart_vs_arteria.heart\n",
       ":   577diffexp_quadrant.none.heart_vs_arteria.arteria\n",
       ":   50diffexp_quadrant.none.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   99diffexp_quadrant.none.ipsc_cvpc_vs_arteria.arteria\n",
       ":   70diffexp_quadrant.none.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   179diffexp_quadrant.none.ipsc_cvpc_vs_heart.heart\n",
       ":   1088diffexp_quadrant.smooth_muscle.heart_vs_arteria.heart\n",
       ":   140diffexp_quadrant.smooth_muscle.heart_vs_arteria.arteria\n",
       ":   264diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.ipsc_cvpc\n",
       ":   50diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.arteria\n",
       ":   266diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc\n",
       ":   46diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.heart\n",
       ":   145egene\n",
       ":   7165eqtl_interaction.stage.adult\n",
       ":   183eqtl_interaction.tissue.arteria\n",
       ":   243eqtl_interaction.tissue.arteria_aorta\n",
       ":   125eqtl_interaction.tissue.arteria_coronary\n",
       ":   29eqtl_interaction.cell.cardiac_muscle\n",
       ":   86eqtl_interaction.cell.cardiac_neuron\n",
       ":   32eqtl_interaction.cell.endocardial\n",
       ":   31eqtl_interaction.cell.endothelial\n",
       ":   3eqtl_interaction.cell.fibroblast\n",
       ":   68eqtl_interaction.cell.immune\n",
       ":   114eqtl_interaction.cell.myofibroblast\n",
       ":   2eqtl_interaction.cell.smooth_muscle\n",
       ":   195eqtl_interaction.tissue.heart\n",
       ":   83eqtl_interaction.tissue.heart_atrium\n",
       ":   32eqtl_interaction.tissue.heart_ventricle\n",
       ":   31eqtl_interaction.stage.ipsc_cvpc\n",
       ":   102gwas\n",
       ":   2364\n",
       "\n"
      ],
      "text/plain": [
       "                                     diffexp.heart_vs_arteria \n",
       "                                                        11596 \n",
       "                               diffexp.heart_vs_arteria.heart \n",
       "                                                         5124 \n",
       "                             diffexp.heart_vs_arteria.arteria \n",
       "                                                         6472 \n",
       "                                 diffexp.ipsc_cvpc_vs_arteria \n",
       "                                                        21021 \n",
       "                       diffexp.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                        11091 \n",
       "                         diffexp.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                         9930 \n",
       "                                   diffexp.ipsc_cvpc_vs_heart \n",
       "                                                        19270 \n",
       "                         diffexp.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                        11127 \n",
       "                             diffexp.ipsc_cvpc_vs_heart.heart \n",
       "                                                         8143 \n",
       "                                  diffexp_cell.cardiac_muscle \n",
       "                                                         2993 \n",
       "                                  diffexp_cell.cardiac_neuron \n",
       "                                                         1008 \n",
       "                                     diffexp_cell.endocardial \n",
       "                                                         2022 \n",
       "                                     diffexp_cell.endothelial \n",
       "                                                          866 \n",
       "                                      diffexp_cell.fibroblast \n",
       "                                                         3691 \n",
       "                                          diffexp_cell.immune \n",
       "                                                          740 \n",
       "                                   diffexp_cell.myofibroblast \n",
       "                                                          285 \n",
       "                                   diffexp_cell.smooth_muscle \n",
       "                                                          667 \n",
       " diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                          176 \n",
       "     diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.heart \n",
       "                                                         1656 \n",
       "          diffexp_quadrant.endocardial.heart_vs_arteria.heart \n",
       "                                                         1279 \n",
       "        diffexp_quadrant.endocardial.heart_vs_arteria.arteria \n",
       "                                                           18 \n",
       "  diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                          189 \n",
       "    diffexp_quadrant.endocardial.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                           24 \n",
       "    diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                          108 \n",
       "        diffexp_quadrant.endocardial.ipsc_cvpc_vs_heart.heart \n",
       "                                                         1204 \n",
       "    diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                          117 \n",
       "        diffexp_quadrant.endothelial.ipsc_cvpc_vs_heart.heart \n",
       "                                                          359 \n",
       "       diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                           17 \n",
       "         diffexp_quadrant.immune.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                          304 \n",
       "                 diffexp_quadrant.none.heart_vs_arteria.heart \n",
       "                                                          577 \n",
       "               diffexp_quadrant.none.heart_vs_arteria.arteria \n",
       "                                                           50 \n",
       "         diffexp_quadrant.none.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                           99 \n",
       "           diffexp_quadrant.none.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                           70 \n",
       "           diffexp_quadrant.none.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                          179 \n",
       "               diffexp_quadrant.none.ipsc_cvpc_vs_heart.heart \n",
       "                                                         1088 \n",
       "        diffexp_quadrant.smooth_muscle.heart_vs_arteria.heart \n",
       "                                                          140 \n",
       "      diffexp_quadrant.smooth_muscle.heart_vs_arteria.arteria \n",
       "                                                          264 \n",
       "diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.ipsc_cvpc \n",
       "                                                           50 \n",
       "  diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_arteria.arteria \n",
       "                                                          266 \n",
       "  diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc \n",
       "                                                           46 \n",
       "      diffexp_quadrant.smooth_muscle.ipsc_cvpc_vs_heart.heart \n",
       "                                                          145 \n",
       "                                                        egene \n",
       "                                                         7165 \n",
       "                                 eqtl_interaction.stage.adult \n",
       "                                                          183 \n",
       "                              eqtl_interaction.tissue.arteria \n",
       "                                                          243 \n",
       "                        eqtl_interaction.tissue.arteria_aorta \n",
       "                                                          125 \n",
       "                     eqtl_interaction.tissue.arteria_coronary \n",
       "                                                           29 \n",
       "                         eqtl_interaction.cell.cardiac_muscle \n",
       "                                                           86 \n",
       "                         eqtl_interaction.cell.cardiac_neuron \n",
       "                                                           32 \n",
       "                            eqtl_interaction.cell.endocardial \n",
       "                                                           31 \n",
       "                            eqtl_interaction.cell.endothelial \n",
       "                                                            3 \n",
       "                             eqtl_interaction.cell.fibroblast \n",
       "                                                           68 \n",
       "                                 eqtl_interaction.cell.immune \n",
       "                                                          114 \n",
       "                          eqtl_interaction.cell.myofibroblast \n",
       "                                                            2 \n",
       "                          eqtl_interaction.cell.smooth_muscle \n",
       "                                                          195 \n",
       "                                eqtl_interaction.tissue.heart \n",
       "                                                           83 \n",
       "                         eqtl_interaction.tissue.heart_atrium \n",
       "                                                           32 \n",
       "                      eqtl_interaction.tissue.heart_ventricle \n",
       "                                                           31 \n",
       "                             eqtl_interaction.stage.ipsc_cvpc \n",
       "                                                          102 \n",
       "                                                         gwas \n",
       "                                                         2364 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "build_tables = function(name, geneinfo, diffexp, diffexp_cell, diffexp_quadrant, eqtls, ints, coloc)\n",
    "{\n",
    "    out = geneinfo[,c(\"gene_id\", \"gene_name\", \"transcript_id\")]\n",
    "    \n",
    "    for(tissue in sort(unique(diffexp$tissue)))\n",
    "    {\n",
    "        mycol        = paste(\"diffexp\", tissue, sep = \".\")\n",
    "        out[, mycol] = FALSE\n",
    "        out[out$transcript_id %in% diffexp[diffexp$tissue == tissue & diffexp$diffexp == TRUE & diffexp$type == name, \"transcript_id\"], mycol] = TRUE\n",
    "        \n",
    "        this         = diffexp[diffexp$tissue == tissue & diffexp$diffexp == TRUE & diffexp$type == name,]\n",
    "        mycol        = paste(\"diffexp\", tissue, unique(this$tissue1), sep = \".\")\n",
    "        out[, mycol] = FALSE\n",
    "        out[out$transcript_id %in% this[this$beta > 0, \"transcript_id\"], mycol] = TRUE\n",
    "        \n",
    "        mycol        = paste(\"diffexp\", tissue, unique(this$tissue2), sep = \".\")\n",
    "        out[, mycol] = FALSE\n",
    "        out[out$transcript_id %in% this[this$beta < 0, \"transcript_id\"], mycol] = TRUE\n",
    "        \n",
    "    }\n",
    "    \n",
    "    for(cell in sort(unique(diffexp_cell$cell_type)))\n",
    "    {\n",
    "        mycol        = paste(\"diffexp_cell\", cell, sep = \".\")\n",
    "        out[, mycol] = FALSE\n",
    "        out[out$transcript_id %in% diffexp_cell[diffexp_cell$cell_type == cell & diffexp_cell$qval <= 0.05 & diffexp_cell$beta > 0 & diffexp_cell$type == name, \"transcript_id\"], mycol] = TRUE\n",
    "    }\n",
    "    \n",
    "    for(cell in sort(unique(diffexp_quadrant$tissue2cell)))\n",
    "    {\n",
    "        this = diffexp_quadrant[diffexp_quadrant$type == name & diffexp_quadrant$tissue2cell == cell,]\n",
    "        \n",
    "        mycol        = paste(\"diffexp_quadrant\", cell, unique(this$tissue1), sep = \".\")\n",
    "        out[, mycol] = FALSE\n",
    "        out[out$transcript_id %in% this[this$is_tissue1 == TRUE, \"transcript_id\"], mycol] = TRUE\n",
    "        mycol        = paste(\"diffexp_quadrant\", cell, unique(this$tissue2), sep = \".\")\n",
    "        out[, mycol] = FALSE\n",
    "        out[out$transcript_id %in% this[this$is_tissue2 == TRUE, \"transcript_id\"], mycol] = TRUE\n",
    "    }\n",
    "    \n",
    "    out$egene = FALSE\n",
    "    out[out$transcript_id %in% eqtls[eqtls$egene == TRUE, \"transcript_id\"], \"egene\"] = TRUE\n",
    "    \n",
    "    tissues = c(\"heart\", \"arteria\", \"heart_atrium\", \"heart_ventricle\", \"arteria_aorta\", \"arteria_coronary\")\n",
    "    for(cell in sort(unique(ints$interaction)))\n",
    "    {\n",
    "        tissue       = \"cell\"\n",
    "        if(cell %in% c(\"ipsc_cvpc\", \"adult\")){tissue = \"stage\" }\n",
    "        if(cell %in% tissues                ){tissue = \"tissue\"}\n",
    "        \n",
    "        mycol        = paste(\"eqtl_interaction\", tissue, gsub(\"cibersort.regular\\\\.\", \"\", cell), sep = \".\")\n",
    "        out[, mycol] = FALSE\n",
    "        out[out$transcript_id %in% ints[ints$interaction == cell & ints$cell == TRUE, \"transcript_id\"], mycol] = TRUE\n",
    "    }\n",
    "    \n",
    "    out$gwas = FALSE\n",
    "    out[out$transcript_id %in% coloc[coloc$PP.H4.abf > 0.5, \"transcript_id\"], \"gwas\"] = TRUE\n",
    "    \n",
    "    fwrite(out, paste(\"pipeline/X.1.combine\", paste(\"combined\", name, \"txt\", sep = \".\"), sep = \"/\"), sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n",
    "    \n",
    "    return(out)\n",
    "}\n",
    "\n",
    "tab_gene    = build_tables(\"gene_tpm\"   , geneinfo_gene   , diffexp, diffexp_cell, diffexp_quadrant, eqtl_genes   , int_genes   , coloc_gene   )\n",
    "tab_isoform = build_tables(\"isoform_use\", geneinfo_isoform, diffexp, diffexp_cell, diffexp_quadrant, eqtl_isoforms, int_isoforms, coloc_isoform)\n",
    "\n",
    "colSums(tab_gene   [,4:ncol(tab_gene   )])\n",
    "colSums(tab_isoform[,4:ncol(tab_isoform)])\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Find interesting candidates (genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "tab_isof = tab_isoform"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "iPSC-CVPC-specific eGenes that are also overexpressed in iPSC-CVPCs and colocalize with GWAS: LINC01405; ENPP4; TMEM71\n",
      "\n",
      "Adult-specific eGenes that are also overexpressed in adult heart and colocalize with GWAS: PFKFB2; SLC22A18; CARNS1; AC116903.1; PGPEP1L; CIAO3; GNAO1; LINC01535; PIGF; LAMA5-AS1; KREMEN1; MKRN2; PDZRN3; AC117489.1; PFN2; PLD1; FBN2; HDDC2; DMTN; PEBP4\n",
      "\n",
      "iPSC-CVPC-specific eIsoforms that are also overexpressed in iPSC-CVPCs and colocalize with GWAS: HES4; ADAM15; LGALS8; GPR137; CIB1; NOMO3; AC005670.3; AP3D1; HLA-C; NOD1\n",
      "\n",
      "Adult-specific eIsoforms that are also overexpressed in adult heart and colocalize with GWAS: PALMD; FMO2; MS4A7; XRRA1; TMEM255B; MLH3; ATXN3; XRCC3; XRCC3; HAGHL; EMP2; RPL13; NUP88; FAM106A; FAM106A; AC005670.3; PCAT19; ZNF814; ZSCAN18; USP34; AC073254.1; SERHL; FAM118A; PIGG; NPR3; ITGA1; ATG10; MRNIP; NOTCH4; HLA-DQA1; HLA-DPB1; HLA-DPB1; TMEM63B; PRKRIP1\n",
      "\n",
      "iPSC-CVPC-specific eIsoforms that colocalize with GWAS, but their associated eGene does not: HES4; ADAM15; TOR1AIP1; LGALS8; PRKG1; MXI1; DKK3; GPR137; UCHL3; CIB1; MLST8; NOMO3; ATXN2L; CTDNEP1; AC005670.3; AP3D1; KANK2; DMPK; DIDO1; THOC5; STAB1; LPP; RPL9; CAST; HLA-F-AS1; HLA-C; AC147651.1; NOD1; PTP4A3\n",
      "\n",
      "Adult-specific eIsoforms that colocalize with GWAS, but their associated eGene does not: PALMD; FMO2; B3GAT3; XRRA1; TMEM255B; MLH3; ATXN3; XRCC3; ICE2; HAGHL; EMP2; HP; RPL13; NUP88; GRAMD1A; PCAT19; AC073254.1; SERHL; FAM118A; TDGF1; PIGG; CRMP1; AC093752.1; NPR3; ITGA1; ATG10; MRNIP; HLA-C; NOTCH4; HLA-DQA1; HLA-DPB1; TMEM63B; AC147651.1; PRKRIP1; NMRK1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "message(paste(\"iPSC-CVPC-specific eGenes that are also overexpressed in iPSC-CVPCs and colocalize with GWAS\"   , paste(tab_gene[tab_gene$diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.ipsc_cvpc == TRUE & tab_gene$egene == TRUE & tab_gene$eqtl_interaction.stage.ipsc_cvpc == TRUE & tab_gene$gwas == TRUE, \"gene_name\"], collapse = \"; \"), sep = \": \"))\n",
    "message(paste(\"Adult-specific eGenes that are also overexpressed in adult heart and colocalize with GWAS\"      , paste(tab_gene[tab_gene$diffexp_quadrant.cardiac_muscle.ipsc_cvpc_vs_heart.heart     == TRUE & tab_gene$egene == TRUE & tab_gene$eqtl_interaction.stage.adult     == TRUE & tab_gene$gwas == TRUE, \"gene_name\"], collapse = \"; \"), sep = \": \"))\n",
    "message(paste(\"iPSC-CVPC-specific eIsoforms that are also overexpressed in iPSC-CVPCs and colocalize with GWAS\", paste(tab_isof[tab_isof$diffexp.ipsc_cvpc_vs_heart.ipsc_cvpc                         == TRUE & tab_isof$egene == TRUE & tab_isof$eqtl_interaction.stage.ipsc_cvpc == TRUE & tab_isof$gwas == TRUE, \"gene_name\"], collapse = \"; \"), sep = \": \"))\n",
    "message(paste(\"Adult-specific eIsoforms that are also overexpressed in adult heart and colocalize with GWAS\"   , paste(tab_isof[tab_isof$diffexp.ipsc_cvpc_vs_heart.heart                             == TRUE & tab_isof$egene == TRUE & tab_isof$eqtl_interaction.stage.adult     == TRUE & tab_isof$gwas == TRUE, \"gene_name\"], collapse = \"; \"), sep = \": \"))\n",
    "\n",
    "message(paste(\"iPSC-CVPC-specific eIsoforms that colocalize with GWAS, but their associated eGene does not\", paste(setdiff(tab_isof[tab_isof$gwas == TRUE & tab_isof$eqtl_interaction.stage.ipsc_cvpc == TRUE, \"gene_name\"], tab_gene[tab_gene$gwas == TRUE & tab_gene$eqtl_interaction.stage.ipsc_cvpc == TRUE, \"gene_name\"]), collapse = \"; \"), sep = \": \"))\n",
    "message(paste(\"Adult-specific eIsoforms that colocalize with GWAS, but their associated eGene does not\"    , paste(setdiff(tab_isof[tab_isof$gwas == TRUE & tab_isof$eqtl_interaction.stage.adult     == TRUE, \"gene_name\"], tab_gene[tab_gene$gwas == TRUE & tab_gene$eqtl_interaction.stage.adult     == TRUE, \"gene_name\"]), collapse = \"; \"), sep = \": \"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "manifest = fread(\"pipeline/6.1.coloc_gwas/traits.manifest.txt\", sep = \"\\t\", header = TRUE, data.table = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "coloc0 = fread(\"pipeline/3.2.eqtls/eqtl_overlap/cardiac_eqtls.coloc_by_gene.txt\"     , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "coloc  = coloc0\n",
    "coloc  = coloc[grepl(\"ENSG\", coloc$transcript_id2) == TRUE, c(\"transcript_id1\", \"transcript_id2\", \"type1\", \"type2\", \"PP.H3.abf\", \"PP.H4.abf\")]\n",
    "colnames(coloc) = c(\"transcript_id\", \"gene_id\", \"type_isoform\", \"type_gene\", \"pp3_iso2gene\", \"pp4_iso2gene\")\n",
    "\n",
    "coloc = aggregate(list(ppa = coloc$pp4_iso2gene), by = list(transcript_id = coloc$transcript_id, type = coloc$type_isoform), FUN = max)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 24 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>trait</th><th scope=col>id</th><th scope=col>chrom</th><th scope=col>pos</th><th scope=col>ref</th><th scope=col>alt</th><th scope=col>PP.H4.abf</th><th scope=col>pp_snp</th><th scope=col>description</th><th scope=col>category</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>7</th><td>continuous-DBP-both_sexes-combined_medadj_raw </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9956010</td><td>0.9677503</td><td>Diastolic blood pressure, combined automated + manual reading, adjusted by medication</td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>15</th><td>continuous-MAP-both_sexes-combined_medadj_raw </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9954974</td><td>0.9704706</td><td>Mean arterial pressure, combined automated + manual reading, adjusted by medication  </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>continuous-DBP-both_sexes-combined_medadj_irnt</td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9954903</td><td>0.9668499</td><td>Diastolic blood pressure, combined automated + manual reading, adjusted by medication</td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>14</th><td>continuous-MAP-both_sexes-combined_medadj_irnt</td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9953985</td><td>0.9695669</td><td>Mean arterial pressure, combined automated + manual reading, adjusted by medication  </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>continuous-DBP-both_sexes-auto_medadj_raw     </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9945712</td><td>0.9689628</td><td>Diastolic blood pressure, automated reading, adjusted by medication                  </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>continuous-DBP-both_sexes-auto_medadj_irnt    </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9944503</td><td>0.9681868</td><td>Diastolic blood pressure, automated reading, adjusted by medication                  </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>11</th><td>continuous-MAP-both_sexes-auto_medadj_raw     </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9914709</td><td>0.9721203</td><td>Mean arterial pressure, automated reading, adjusted by medication                    </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>10</th><td>continuous-MAP-both_sexes-auto_medadj_irnt    </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9904437</td><td>0.9713708</td><td>Mean arterial pressure, automated reading, adjusted by medication                    </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>continuous-DBP-both_sexes-combined_irnt       </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9869303</td><td>0.9647412</td><td>Diastolic blood pressure, combined automated + manual reading                        </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>8</th><td>continuous-DBP-both_sexes-combined_raw        </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9859885</td><td>0.9650935</td><td>Diastolic blood pressure, combined automated + manual reading                        </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>1</th><td><span style=white-space:pre-wrap>continuous-4079-both_sexes-irnt               </span></td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9841914</td><td>0.9657800</td><td><span style=white-space:pre-wrap>Diastolic blood pressure, automated reading                                          </span></td><td>UK Biobank Assessment Centre &gt; Physical measures &gt; Blood pressure</td></tr>\n",
       "\t<tr><th scope=row>23</th><td>phecode-454-both_sexes                        </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9840003</td><td>0.9650281</td><td>Varicose veins                                                                       </td><td>circulatory system                                               </td></tr>\n",
       "\t<tr><th scope=row>13</th><td>continuous-MAP-both_sexes-combined_irnt       </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9838585</td><td>0.9684947</td><td>Mean arterial pressure, combined automated + manual reading                          </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>16</th><td>continuous-MAP-both_sexes-combined_raw        </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9824609</td><td>0.9687345</td><td>Mean arterial pressure, combined automated + manual reading                          </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>9</th><td>continuous-MAP-both_sexes-auto_irnt           </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9705101</td><td>0.9700493</td><td>Mean arterial pressure, automated reading                                            </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>21</th><td>continuous-SBP-both_sexes-combined_medadj_raw </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9697528</td><td>0.9726712</td><td>Systolic blood pressure, combined automated + manual reading, adjusted by medication </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>12</th><td>continuous-MAP-both_sexes-auto_raw            </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9648485</td><td>0.9702371</td><td>Mean arterial pressure, automated reading                                            </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>24</th><td>phecode-454.1-both_sexes                      </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9607719</td><td>0.9629746</td><td>Varicose veins of lower extremity                                                    </td><td>circulatory system                                               </td></tr>\n",
       "\t<tr><th scope=row>20</th><td>continuous-SBP-both_sexes-combined_medadj_irnt</td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.9606794</td><td>0.9718908</td><td>Systolic blood pressure, combined automated + manual reading, adjusted by medication </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>19</th><td>continuous-SBP-both_sexes-combined_irnt       </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.8287958</td><td>0.9716750</td><td>Systolic blood pressure, combined automated + manual reading                         </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>18</th><td>continuous-SBP-both_sexes-auto_medadj_raw     </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.8087496</td><td>0.9740736</td><td>Systolic blood pressure, automated reading, adjusted by medication                   </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>17</th><td>continuous-SBP-both_sexes-auto_medadj_irnt    </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.7955508</td><td>0.9734419</td><td>Systolic blood pressure, automated reading, adjusted by medication                   </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>22</th><td>continuous-SBP-both_sexes-combined_raw        </td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.7822460</td><td>0.9718938</td><td>Systolic blood pressure, combined automated + manual reading                         </td><td>                                                                 </td></tr>\n",
       "\t<tr><th scope=row>2</th><td><span style=white-space:pre-wrap>continuous-4080-both_sexes-irnt               </span></td><td>VAR_3_188066953_T_G</td><td>3</td><td>188066953</td><td>T</td><td>G</td><td>0.5786907</td><td>0.9731255</td><td><span style=white-space:pre-wrap>Systolic blood pressure, automated reading                                           </span></td><td>UK Biobank Assessment Centre &gt; Physical measures &gt; Blood pressure</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 24 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & trait & id & chrom & pos & ref & alt & PP.H4.abf & pp\\_snp & description & category\\\\\n",
       "  & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t7 & continuous-DBP-both\\_sexes-combined\\_medadj\\_raw  & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9956010 & 0.9677503 & Diastolic blood pressure, combined automated + manual reading, adjusted by medication &                                                                  \\\\\n",
       "\t15 & continuous-MAP-both\\_sexes-combined\\_medadj\\_raw  & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9954974 & 0.9704706 & Mean arterial pressure, combined automated + manual reading, adjusted by medication   &                                                                  \\\\\n",
       "\t6 & continuous-DBP-both\\_sexes-combined\\_medadj\\_irnt & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9954903 & 0.9668499 & Diastolic blood pressure, combined automated + manual reading, adjusted by medication &                                                                  \\\\\n",
       "\t14 & continuous-MAP-both\\_sexes-combined\\_medadj\\_irnt & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9953985 & 0.9695669 & Mean arterial pressure, combined automated + manual reading, adjusted by medication   &                                                                  \\\\\n",
       "\t4 & continuous-DBP-both\\_sexes-auto\\_medadj\\_raw      & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9945712 & 0.9689628 & Diastolic blood pressure, automated reading, adjusted by medication                   &                                                                  \\\\\n",
       "\t3 & continuous-DBP-both\\_sexes-auto\\_medadj\\_irnt     & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9944503 & 0.9681868 & Diastolic blood pressure, automated reading, adjusted by medication                   &                                                                  \\\\\n",
       "\t11 & continuous-MAP-both\\_sexes-auto\\_medadj\\_raw      & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9914709 & 0.9721203 & Mean arterial pressure, automated reading, adjusted by medication                     &                                                                  \\\\\n",
       "\t10 & continuous-MAP-both\\_sexes-auto\\_medadj\\_irnt     & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9904437 & 0.9713708 & Mean arterial pressure, automated reading, adjusted by medication                     &                                                                  \\\\\n",
       "\t5 & continuous-DBP-both\\_sexes-combined\\_irnt        & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9869303 & 0.9647412 & Diastolic blood pressure, combined automated + manual reading                         &                                                                  \\\\\n",
       "\t8 & continuous-DBP-both\\_sexes-combined\\_raw         & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9859885 & 0.9650935 & Diastolic blood pressure, combined automated + manual reading                         &                                                                  \\\\\n",
       "\t1 & continuous-4079-both\\_sexes-irnt                & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9841914 & 0.9657800 & Diastolic blood pressure, automated reading                                           & UK Biobank Assessment Centre > Physical measures > Blood pressure\\\\\n",
       "\t23 & phecode-454-both\\_sexes                         & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9840003 & 0.9650281 & Varicose veins                                                                        & circulatory system                                               \\\\\n",
       "\t13 & continuous-MAP-both\\_sexes-combined\\_irnt        & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9838585 & 0.9684947 & Mean arterial pressure, combined automated + manual reading                           &                                                                  \\\\\n",
       "\t16 & continuous-MAP-both\\_sexes-combined\\_raw         & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9824609 & 0.9687345 & Mean arterial pressure, combined automated + manual reading                           &                                                                  \\\\\n",
       "\t9 & continuous-MAP-both\\_sexes-auto\\_irnt            & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9705101 & 0.9700493 & Mean arterial pressure, automated reading                                             &                                                                  \\\\\n",
       "\t21 & continuous-SBP-both\\_sexes-combined\\_medadj\\_raw  & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9697528 & 0.9726712 & Systolic blood pressure, combined automated + manual reading, adjusted by medication  &                                                                  \\\\\n",
       "\t12 & continuous-MAP-both\\_sexes-auto\\_raw             & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9648485 & 0.9702371 & Mean arterial pressure, automated reading                                             &                                                                  \\\\\n",
       "\t24 & phecode-454.1-both\\_sexes                       & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9607719 & 0.9629746 & Varicose veins of lower extremity                                                     & circulatory system                                               \\\\\n",
       "\t20 & continuous-SBP-both\\_sexes-combined\\_medadj\\_irnt & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.9606794 & 0.9718908 & Systolic blood pressure, combined automated + manual reading, adjusted by medication  &                                                                  \\\\\n",
       "\t19 & continuous-SBP-both\\_sexes-combined\\_irnt        & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.8287958 & 0.9716750 & Systolic blood pressure, combined automated + manual reading                          &                                                                  \\\\\n",
       "\t18 & continuous-SBP-both\\_sexes-auto\\_medadj\\_raw      & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.8087496 & 0.9740736 & Systolic blood pressure, automated reading, adjusted by medication                    &                                                                  \\\\\n",
       "\t17 & continuous-SBP-both\\_sexes-auto\\_medadj\\_irnt     & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.7955508 & 0.9734419 & Systolic blood pressure, automated reading, adjusted by medication                    &                                                                  \\\\\n",
       "\t22 & continuous-SBP-both\\_sexes-combined\\_raw         & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.7822460 & 0.9718938 & Systolic blood pressure, combined automated + manual reading                          &                                                                  \\\\\n",
       "\t2 & continuous-4080-both\\_sexes-irnt                & VAR\\_3\\_188066953\\_T\\_G & 3 & 188066953 & T & G & 0.5786907 & 0.9731255 & Systolic blood pressure, automated reading                                            & UK Biobank Assessment Centre > Physical measures > Blood pressure\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 24 × 10\n",
       "\n",
       "| <!--/--> | trait &lt;chr&gt; | id &lt;chr&gt; | chrom &lt;int&gt; | pos &lt;int&gt; | ref &lt;chr&gt; | alt &lt;chr&gt; | PP.H4.abf &lt;dbl&gt; | pp_snp &lt;dbl&gt; | description &lt;chr&gt; | category &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 7 | continuous-DBP-both_sexes-combined_medadj_raw  | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9956010 | 0.9677503 | Diastolic blood pressure, combined automated + manual reading, adjusted by medication | <!----> |\n",
       "| 15 | continuous-MAP-both_sexes-combined_medadj_raw  | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9954974 | 0.9704706 | Mean arterial pressure, combined automated + manual reading, adjusted by medication   | <!----> |\n",
       "| 6 | continuous-DBP-both_sexes-combined_medadj_irnt | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9954903 | 0.9668499 | Diastolic blood pressure, combined automated + manual reading, adjusted by medication | <!----> |\n",
       "| 14 | continuous-MAP-both_sexes-combined_medadj_irnt | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9953985 | 0.9695669 | Mean arterial pressure, combined automated + manual reading, adjusted by medication   | <!----> |\n",
       "| 4 | continuous-DBP-both_sexes-auto_medadj_raw      | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9945712 | 0.9689628 | Diastolic blood pressure, automated reading, adjusted by medication                   | <!----> |\n",
       "| 3 | continuous-DBP-both_sexes-auto_medadj_irnt     | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9944503 | 0.9681868 | Diastolic blood pressure, automated reading, adjusted by medication                   | <!----> |\n",
       "| 11 | continuous-MAP-both_sexes-auto_medadj_raw      | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9914709 | 0.9721203 | Mean arterial pressure, automated reading, adjusted by medication                     | <!----> |\n",
       "| 10 | continuous-MAP-both_sexes-auto_medadj_irnt     | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9904437 | 0.9713708 | Mean arterial pressure, automated reading, adjusted by medication                     | <!----> |\n",
       "| 5 | continuous-DBP-both_sexes-combined_irnt        | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9869303 | 0.9647412 | Diastolic blood pressure, combined automated + manual reading                         | <!----> |\n",
       "| 8 | continuous-DBP-both_sexes-combined_raw         | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9859885 | 0.9650935 | Diastolic blood pressure, combined automated + manual reading                         | <!----> |\n",
       "| 1 | continuous-4079-both_sexes-irnt                | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9841914 | 0.9657800 | Diastolic blood pressure, automated reading                                           | UK Biobank Assessment Centre &gt; Physical measures &gt; Blood pressure |\n",
       "| 23 | phecode-454-both_sexes                         | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9840003 | 0.9650281 | Varicose veins                                                                        | circulatory system                                                |\n",
       "| 13 | continuous-MAP-both_sexes-combined_irnt        | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9838585 | 0.9684947 | Mean arterial pressure, combined automated + manual reading                           | <!----> |\n",
       "| 16 | continuous-MAP-both_sexes-combined_raw         | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9824609 | 0.9687345 | Mean arterial pressure, combined automated + manual reading                           | <!----> |\n",
       "| 9 | continuous-MAP-both_sexes-auto_irnt            | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9705101 | 0.9700493 | Mean arterial pressure, automated reading                                             | <!----> |\n",
       "| 21 | continuous-SBP-both_sexes-combined_medadj_raw  | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9697528 | 0.9726712 | Systolic blood pressure, combined automated + manual reading, adjusted by medication  | <!----> |\n",
       "| 12 | continuous-MAP-both_sexes-auto_raw             | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9648485 | 0.9702371 | Mean arterial pressure, automated reading                                             | <!----> |\n",
       "| 24 | phecode-454.1-both_sexes                       | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9607719 | 0.9629746 | Varicose veins of lower extremity                                                     | circulatory system                                                |\n",
       "| 20 | continuous-SBP-both_sexes-combined_medadj_irnt | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.9606794 | 0.9718908 | Systolic blood pressure, combined automated + manual reading, adjusted by medication  | <!----> |\n",
       "| 19 | continuous-SBP-both_sexes-combined_irnt        | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.8287958 | 0.9716750 | Systolic blood pressure, combined automated + manual reading                          | <!----> |\n",
       "| 18 | continuous-SBP-both_sexes-auto_medadj_raw      | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.8087496 | 0.9740736 | Systolic blood pressure, automated reading, adjusted by medication                    | <!----> |\n",
       "| 17 | continuous-SBP-both_sexes-auto_medadj_irnt     | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.7955508 | 0.9734419 | Systolic blood pressure, automated reading, adjusted by medication                    | <!----> |\n",
       "| 22 | continuous-SBP-both_sexes-combined_raw         | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.7822460 | 0.9718938 | Systolic blood pressure, combined automated + manual reading                          | <!----> |\n",
       "| 2 | continuous-4080-both_sexes-irnt                | VAR_3_188066953_T_G | 3 | 188066953 | T | G | 0.5786907 | 0.9731255 | Systolic blood pressure, automated reading                                            | UK Biobank Assessment Centre &gt; Physical measures &gt; Blood pressure |\n",
       "\n"
      ],
      "text/plain": [
       "   trait                                          id                  chrom\n",
       "7  continuous-DBP-both_sexes-combined_medadj_raw  VAR_3_188066953_T_G 3    \n",
       "15 continuous-MAP-both_sexes-combined_medadj_raw  VAR_3_188066953_T_G 3    \n",
       "6  continuous-DBP-both_sexes-combined_medadj_irnt VAR_3_188066953_T_G 3    \n",
       "14 continuous-MAP-both_sexes-combined_medadj_irnt VAR_3_188066953_T_G 3    \n",
       "4  continuous-DBP-both_sexes-auto_medadj_raw      VAR_3_188066953_T_G 3    \n",
       "3  continuous-DBP-both_sexes-auto_medadj_irnt     VAR_3_188066953_T_G 3    \n",
       "11 continuous-MAP-both_sexes-auto_medadj_raw      VAR_3_188066953_T_G 3    \n",
       "10 continuous-MAP-both_sexes-auto_medadj_irnt     VAR_3_188066953_T_G 3    \n",
       "5  continuous-DBP-both_sexes-combined_irnt        VAR_3_188066953_T_G 3    \n",
       "8  continuous-DBP-both_sexes-combined_raw         VAR_3_188066953_T_G 3    \n",
       "1  continuous-4079-both_sexes-irnt                VAR_3_188066953_T_G 3    \n",
       "23 phecode-454-both_sexes                         VAR_3_188066953_T_G 3    \n",
       "13 continuous-MAP-both_sexes-combined_irnt        VAR_3_188066953_T_G 3    \n",
       "16 continuous-MAP-both_sexes-combined_raw         VAR_3_188066953_T_G 3    \n",
       "9  continuous-MAP-both_sexes-auto_irnt            VAR_3_188066953_T_G 3    \n",
       "21 continuous-SBP-both_sexes-combined_medadj_raw  VAR_3_188066953_T_G 3    \n",
       "12 continuous-MAP-both_sexes-auto_raw             VAR_3_188066953_T_G 3    \n",
       "24 phecode-454.1-both_sexes                       VAR_3_188066953_T_G 3    \n",
       "20 continuous-SBP-both_sexes-combined_medadj_irnt VAR_3_188066953_T_G 3    \n",
       "19 continuous-SBP-both_sexes-combined_irnt        VAR_3_188066953_T_G 3    \n",
       "18 continuous-SBP-both_sexes-auto_medadj_raw      VAR_3_188066953_T_G 3    \n",
       "17 continuous-SBP-both_sexes-auto_medadj_irnt     VAR_3_188066953_T_G 3    \n",
       "22 continuous-SBP-both_sexes-combined_raw         VAR_3_188066953_T_G 3    \n",
       "2  continuous-4080-both_sexes-irnt                VAR_3_188066953_T_G 3    \n",
       "   pos       ref alt PP.H4.abf pp_snp   \n",
       "7  188066953 T   G   0.9956010 0.9677503\n",
       "15 188066953 T   G   0.9954974 0.9704706\n",
       "6  188066953 T   G   0.9954903 0.9668499\n",
       "14 188066953 T   G   0.9953985 0.9695669\n",
       "4  188066953 T   G   0.9945712 0.9689628\n",
       "3  188066953 T   G   0.9944503 0.9681868\n",
       "11 188066953 T   G   0.9914709 0.9721203\n",
       "10 188066953 T   G   0.9904437 0.9713708\n",
       "5  188066953 T   G   0.9869303 0.9647412\n",
       "8  188066953 T   G   0.9859885 0.9650935\n",
       "1  188066953 T   G   0.9841914 0.9657800\n",
       "23 188066953 T   G   0.9840003 0.9650281\n",
       "13 188066953 T   G   0.9838585 0.9684947\n",
       "16 188066953 T   G   0.9824609 0.9687345\n",
       "9  188066953 T   G   0.9705101 0.9700493\n",
       "21 188066953 T   G   0.9697528 0.9726712\n",
       "12 188066953 T   G   0.9648485 0.9702371\n",
       "24 188066953 T   G   0.9607719 0.9629746\n",
       "20 188066953 T   G   0.9606794 0.9718908\n",
       "19 188066953 T   G   0.8287958 0.9716750\n",
       "18 188066953 T   G   0.8087496 0.9740736\n",
       "17 188066953 T   G   0.7955508 0.9734419\n",
       "22 188066953 T   G   0.7822460 0.9718938\n",
       "2  188066953 T   G   0.5786907 0.9731255\n",
       "   description                                                                          \n",
       "7  Diastolic blood pressure, combined automated + manual reading, adjusted by medication\n",
       "15 Mean arterial pressure, combined automated + manual reading, adjusted by medication  \n",
       "6  Diastolic blood pressure, combined automated + manual reading, adjusted by medication\n",
       "14 Mean arterial pressure, combined automated + manual reading, adjusted by medication  \n",
       "4  Diastolic blood pressure, automated reading, adjusted by medication                  \n",
       "3  Diastolic blood pressure, automated reading, adjusted by medication                  \n",
       "11 Mean arterial pressure, automated reading, adjusted by medication                    \n",
       "10 Mean arterial pressure, automated reading, adjusted by medication                    \n",
       "5  Diastolic blood pressure, combined automated + manual reading                        \n",
       "8  Diastolic blood pressure, combined automated + manual reading                        \n",
       "1  Diastolic blood pressure, automated reading                                          \n",
       "23 Varicose veins                                                                       \n",
       "13 Mean arterial pressure, combined automated + manual reading                          \n",
       "16 Mean arterial pressure, combined automated + manual reading                          \n",
       "9  Mean arterial pressure, automated reading                                            \n",
       "21 Systolic blood pressure, combined automated + manual reading, adjusted by medication \n",
       "12 Mean arterial pressure, automated reading                                            \n",
       "24 Varicose veins of lower extremity                                                    \n",
       "20 Systolic blood pressure, combined automated + manual reading, adjusted by medication \n",
       "19 Systolic blood pressure, combined automated + manual reading                         \n",
       "18 Systolic blood pressure, automated reading, adjusted by medication                   \n",
       "17 Systolic blood pressure, automated reading, adjusted by medication                   \n",
       "22 Systolic blood pressure, combined automated + manual reading                         \n",
       "2  Systolic blood pressure, automated reading                                           \n",
       "   category                                                         \n",
       "7                                                                   \n",
       "15                                                                  \n",
       "6                                                                   \n",
       "14                                                                  \n",
       "4                                                                   \n",
       "3                                                                   \n",
       "11                                                                  \n",
       "10                                                                  \n",
       "5                                                                   \n",
       "8                                                                   \n",
       "1  UK Biobank Assessment Centre > Physical measures > Blood pressure\n",
       "23 circulatory system                                               \n",
       "13                                                                  \n",
       "16                                                                  \n",
       "9                                                                   \n",
       "21                                                                  \n",
       "12                                                                  \n",
       "24 circulatory system                                               \n",
       "20                                                                  \n",
       "19                                                                  \n",
       "18                                                                  \n",
       "17                                                                  \n",
       "22                                                                  \n",
       "2  UK Biobank Assessment Centre > Physical measures > Blood pressure"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "a = merge(coloc_isoform[ coloc_isoform$transcript_id == \"ENST00000617246.4_3\" & coloc_isoform$PP.H4.abf > 0.5, c(\"id\", \"chrom\", \"pos\", \"ref\", \"alt\", \"PP.H4.abf\", \"pp_snp\", \"trait\")], manifest[,c(\"id\", \"description\", \"category\")], by.x = \"trait\", by.y = \"id\")\n",
    "a[order(a$PP.H4.abf, decreasing = TRUE),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "metadata": {},
   "outputs": [],
   "source": [
    "tab1                 = unique(tab_isof[tab_isof$egene == TRUE & tab_isof$eqtl_interaction.stage.ipsc_cvpc == TRUE & tab_isof$gwas == TRUE, c(\"gene_id\", \"transcript_id\", \"gene_name\")])\n",
    "tab1                 = merge(tab1, eqtl_isoforms[eqtl_isoforms$egene == TRUE & eqtl_isoforms$pos >= eqtl_isoforms$start & eqtl_isoforms$pos <= eqtl_isoforms$end, c(\"transcript_id\", \"type\", \"id\", \"chrom\", \"pos\", \"ref\", \"alt\", \"rsid\", \"af\", \"beta\", \"se\", \"pval\", \"fdr\", \"distance\")])\n",
    "tab1        $tr2type = paste(tab1        $transcript_id, tab1        $type)\n",
    "int_isoforms$tr2type = paste(int_isoforms$transcript_id, int_isoforms$type)\n",
    "tab1                 = tab1[tab1$tr2type %in% int_isoforms[ int_isoforms$cell == TRUE & int_isoforms$interaction == \"ipsc_cvpc\", \"tr2type\"],]\n",
    "tab1                 = merge(tab1, coloc, by = c(\"transcript_id\", \"type\"))\n",
    "tab1                 = tab1[tab1$ppa < 0.5,]\n",
    "tab1                 = as.data.frame(rbindlist(lapply(1:nrow(tab1), function(ii)\n",
    "{\n",
    "    this                   = tab1[ii,]\n",
    "    tr2type                = this[1, \"tr2type\"]\n",
    "    coloc_tab              = coloc_isoform[ coloc_isoform$tr2type == tr2type,]\n",
    "    coloc_tab              = coloc_tab[coloc_tab$PP.H4.abf > 0.5, ]\n",
    "    coloc_tab              = coloc_tab[order(coloc_tab$PP.H4.abf, decreasing = TRUE),]\n",
    "    coloc_tab$pval_gwas    = 1\n",
    "    \n",
    "    for(jj in 1:nrow(coloc_tab))\n",
    "    {\n",
    "        chrom           = coloc_tab[jj, \"chrom\"]\n",
    "        pos             = coloc_tab[jj, \"pos\"  ]\n",
    "        trait           = coloc_tab[jj, \"trait\"]\n",
    "        coord           = paste(chrom, \":\", pos - 1, \"-\", pos + 1, sep = \"\")\n",
    "        infile          = manifest[manifest$id == trait, \"filename\"]\n",
    "        command         = paste(\"tabix\", \"-h\", infile, coord)\n",
    "        gwas            = fread(cmd = command, sep = \"\\t\", header = FALSE, data.table = FALSE)\n",
    "        colnames(gwas)  = unlist(strsplit(system(paste(\"zcat\", infile, \"|\", \"head\", \"-n\", 1), intern = TRUE), \"\\t\"))\n",
    "        gwas            = gwas[gwas$pos == pos,]\n",
    "        gwas$id         = paste(\"VAR\", gwas$chr, gwas$pos, gwas$ref, gwas$alt, sep = \"_\")\n",
    "        gwas            = gwas[,c(\"id\", \"beta_meta\", \"se_meta\", \"pval_meta\")]\n",
    "        \n",
    "        coloc_tab[jj, \"pval_gwas\"] = gwas[1, \"pval_meta\"]\n",
    "    }\n",
    "    \n",
    "    out = merge(this, coloc_tab[, c(\"tr2type\", \"trait\", \"id\", \"pval_gwas\", \"PP.H4.abf\")], by = \"tr2type\", suffixes = c(\"_eqtl\", \"_coloc\"))\n",
    "    \n",
    "    return(out)\n",
    "})), stringsAsFactors = FALSE)\n",
    "\n",
    "tab1_all = tab1\n",
    "fwrite(tab1_all, \"pipeline/X.1.combine/eisoforms_no_coloc_with_egene_all.txt\", sep = \"\\t\", col.names = TRUE, row.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "metadata": {},
   "outputs": [],
   "source": [
    "tab1                 = unique(tab_isof[tab_isof$egene == TRUE & tab_isof$eqtl_interaction.stage.ipsc_cvpc == TRUE & tab_isof$gwas == TRUE, c(\"gene_id\", \"transcript_id\", \"gene_name\")])\n",
    "tab1                 = merge(tab1, eqtl_isoforms[eqtl_isoforms$egene == TRUE & eqtl_isoforms$pos >= eqtl_isoforms$start & eqtl_isoforms$pos <= eqtl_isoforms$end, c(\"transcript_id\", \"type\", \"id\", \"chrom\", \"pos\", \"ref\", \"alt\", \"rsid\", \"af\", \"beta\", \"se\", \"pval\", \"fdr\", \"distance\")])\n",
    "tab1        $tr2type = paste(tab1        $transcript_id, tab1        $type)\n",
    "int_isoforms$tr2type = paste(int_isoforms$transcript_id, int_isoforms$type)\n",
    "tab1                 = tab1[tab1$tr2type %in% int_isoforms[ int_isoforms$cell == TRUE & int_isoforms$interaction == \"ipsc_cvpc\", \"tr2type\"],]\n",
    "tab1                 = merge(tab1, coloc, by = c(\"transcript_id\", \"type\"))\n",
    "tab1                 = tab1[tab1$ppa < 0.5,]\n",
    "tab1                 = as.data.frame(rbindlist(lapply(1:nrow(tab1), function(ii)\n",
    "{\n",
    "    this                   = tab1[ii,]\n",
    "    tr2type                = this[1, \"tr2type\"]\n",
    "    coloc_tab              = coloc_isoform[ coloc_isoform$tr2type == tr2type,]\n",
    "    coloc_tab              = coloc_tab[coloc_tab$PP.H4.abf > 0.5, ]\n",
    "    coloc_tab              = coloc_tab[order(coloc_tab$PP.H4.abf, decreasing = TRUE),]\n",
    "    coloc_tab$pval_gwas    = 1\n",
    "    \n",
    "    for(jj in 1:nrow(coloc_tab))\n",
    "    {\n",
    "        chrom           = coloc_tab[jj, \"chrom\"]\n",
    "        pos             = coloc_tab[jj, \"pos\"  ]\n",
    "        trait           = coloc_tab[jj, \"trait\"]\n",
    "        coord           = paste(chrom, \":\", pos - 1, \"-\", pos + 1, sep = \"\")\n",
    "        infile          = manifest[manifest$id == trait, \"filename\"]\n",
    "        command         = paste(\"tabix\", \"-h\", infile, coord)\n",
    "        gwas            = fread(cmd = command, sep = \"\\t\", header = FALSE, data.table = FALSE)\n",
    "        colnames(gwas)  = unlist(strsplit(system(paste(\"zcat\", infile, \"|\", \"head\", \"-n\", 1), intern = TRUE), \"\\t\"))\n",
    "        gwas            = gwas[gwas$pos == pos,]\n",
    "        gwas$id         = paste(\"VAR\", gwas$chr, gwas$pos, gwas$ref, gwas$alt, sep = \"_\")\n",
    "        gwas            = gwas[,c(\"id\", \"beta_meta\", \"se_meta\", \"pval_meta\")]\n",
    "        \n",
    "        coloc_tab[jj, \"pval_gwas\"] = gwas[1, \"pval_meta\"]\n",
    "    }\n",
    "    \n",
    "    if(min(coloc_tab$pval_gwas) <= 1e-5)\n",
    "    {\n",
    "        trait = coloc_tab[which.min(coloc_tab$pval_gwas), \"trait\"    ]\n",
    "        pval  = coloc_tab[which.min(coloc_tab$pval_gwas), \"pval_gwas\"]\n",
    "        ppa   = coloc_tab[which.min(coloc_tab$pval_gwas), \"PP.H4.abf\"]\n",
    "    }else\n",
    "    {\n",
    "        trait = coloc_tab[1, \"trait\"    ]\n",
    "        pval  = coloc_tab[1, \"pval_gwas\"]\n",
    "        ppa   = coloc_tab[1, \"PP.H4.abf\"]\n",
    "        trait = coloc_tab[which.min(coloc_tab$pval_gwas), \"trait\"    ]\n",
    "        pval  = coloc_tab[which.min(coloc_tab$pval_gwas), \"pval_gwas\"]\n",
    "        ppa   = coloc_tab[which.min(coloc_tab$pval_gwas), \"PP.H4.abf\"]\n",
    "    }\n",
    "    \n",
    "    this$trait             = trait\n",
    "    this$trait_description = manifest[manifest$id == trait, \"description\"]\n",
    "    this$trait_category    = manifest[manifest$id == trait, \"category\"   ]\n",
    "    this$pval_gwas         = pval\n",
    "    this$ppa_gwas          = ppa\n",
    "    \n",
    "    return(this)\n",
    "})), stringsAsFactors = FALSE)\n",
    "\n",
    "tab1 = tab1[order(tab1$ppa_gwas, decreasing = TRUE),]\n",
    "\n",
    "fwrite(tab1, \"pipeline/X.1.combine/eisoforms_no_coloc_with_egene.txt\", sep = \"\\t\", col.names = TRUE, row.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [],
   "source": [
    "exons    = fread(\"/frazer01/reference/private/Gencode.v34lift37/exon_info.txt\"      , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "cds      = fread(\"/frazer01/reference/private/Gencode.v34lift37/cds_info.txt\"       , sep = \"\\t\", header = TRUE, data.table = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "LPP ENSG00000145012.13_7 ENST00000617246.4_3 0 rs6787621\n",
      "\n",
      "KANK2 ENSG00000197256.10_9 ENST00000588787.5_5 1 rs11668313\n",
      "\n",
      "PRKG1 ENSG00000185532.20_9 ENST00000643582.1_1 0 rs11000060\n",
      "\n",
      "NOD1 ENSG00000106100.11_7 ENST00000222823.9_3 0 rs1558067\n",
      "\n",
      "GPR137 ENSG00000173264.14_7 ENST00000536017.1_1 0 rs11231745\n",
      "\n",
      "MXI1 ENSG00000119950.21_7 ENST00000651516.1_1 0 rs2855469\n",
      "\n",
      "TOR1AIP1 ENSG00000143337.18_6 ENST00000435319.8_2 0 rs2245425\n",
      "\n",
      "LGALS8 ENSG00000116977.18_6 ENST00000528782.5_1 0 rs1969746\n",
      "\n",
      "UCHL3 ENSG00000118939.17_8 ENST00000377595.7_1 0 rs9530456\n",
      "\n",
      "DKK3 ENSG00000050165.18_5 ENST00000527132.1_2 0 rs142843448\n",
      "\n",
      "MLST8 ENSG00000167965.18_7 ENST00000397124.5_2 0 rs27425\n",
      "\n",
      "CIB1 ENSG00000185043.12_6 ENST00000328649.11_3 0 rs8041338\n",
      "\n"
     ]
    }
   ],
   "source": [
    "plot_sumstats = function(indata, myvar, myrange, ylim = 0, y_axis = \"\", rsid = FALSE, is_x_axis = TRUE, is_log = FALSE, is_legend = FALSE, ...)\n",
    "{\n",
    "    indata$myvar = indata[,myvar]\n",
    "    \n",
    "    if(is_log == TRUE){indata$myvar = -log10(indata$myvar)}\n",
    "    \n",
    "    if(ylim == 0){ylim = max(indata$myvar)}\n",
    "    \n",
    "    par(...)\n",
    "    plot(1,1, type = \"n\", xlim = myrange, ylim = c(min(indata$myvar), ylim), xlab = \"\", ylab = \"\", axes = FALSE)\n",
    "    axis(2)\n",
    "    if(is_x_axis == TRUE){axis(1)}\n",
    "    \n",
    "    mtext(text = y_axis, side = 2, line = 2.5)\n",
    "    \n",
    "    points(indata$pos, indata$myvar, pch = indata$pch, col = \"#000000\", bg = indata$ld_color, cex = indata$cex)\n",
    "    \n",
    "    abline(h = 0)\n",
    "    abline(v = indata[indata$pch == 23, \"pos\"], lty = \"dashed\", col = \"#ff0000\")\n",
    "    \n",
    "    mylead   = indata[which.max(indata$pp_snp_isoform), ]\n",
    "    mypos    =     3\n",
    "    x_legend =     0.6\n",
    "\n",
    "    if(mylead[1, \"pos\"] >= myrange[[1]] + (myrange[[2]] - myrange[[1]]) / 2)\n",
    "    {\n",
    "        x_legend = 0.01\n",
    "    }\n",
    "    \n",
    "    if(rsid == TRUE)\n",
    "    {\n",
    "        if(mylead[1, \"myvar\"] > 0.9){mypos = 1}\n",
    "        \n",
    "        text(x = mylead[1, \"pos\"], y = mylead[1, \"myvar\"], labels = mylead[1, \"rsid\"], pos = mypos, cex = 2)\n",
    "    }\n",
    "    \n",
    "    if(is_legend == TRUE)\n",
    "    {\n",
    "        tol = data.frame(r2 = (0:4) / 5, name = paste(\"(\", (0:4) / 5, \"-\", (1:5) / 5, \")\", sep = \"\"), y = (5:1 + 3)/10 * ylim, color = c(\"#000088\", \"#aaffff\", \"#00ff00\", \"#ffa500\", \"#ff0000\"))\n",
    "        \n",
    "        points(x = rep(myrange[[1]] + (myrange[[2]] - myrange[[1]]) * x_legend, nrow(tol)    ), y =   tol$y              , pch    = 22      , bg  = tol$color, cex = 3)\n",
    "        text  (x = rep(myrange[[1]] + (myrange[[2]] - myrange[[1]]) * x_legend, nrow(tol) + 1), y = c(tol$y, ylim * 9/10), labels = c(paste(\"\", tol$name, \"\"), \" Linkage disequilibrium \"), pos = 4, cex = 2)\n",
    "    }\n",
    "}\n",
    "\n",
    "plot_scatter = function(indata, x, y, is_log = FALSE)\n",
    "{\n",
    "    par(mar = c(4,4,1,1))\n",
    "    \n",
    "    if(is_log == TRUE)\n",
    "    {\n",
    "        indata$x = -log10(indata[,x])\n",
    "        indata$y = -log10(indata[,y])\n",
    "    }else\n",
    "    {\n",
    "        indata$x = indata[,x]\n",
    "        indata$y = indata[,y]\n",
    "    }\n",
    "    \n",
    "    mymax = max(c(indata$x, indata$y))\n",
    "    \n",
    "    #plot(indata$x, indata$y, pch = 16, col = \"#000000\", cex = 0.8, xlim = c(0, mymax), ylim = c(0, mymax), xlab = \"\", ylab = \"\")\n",
    "    plot(indata$x, indata$y, pch = 16, col = \"#000000\", cex = 0.8, xlab = \"\", ylab = \"\")\n",
    "    \n",
    "    mtext(x, side = 1, line = 2.5)\n",
    "    mtext(y, side = 2, line = 2.5)\n",
    "}\n",
    "\n",
    "plot_transcript = function(transcript_id, indata, xlims, exons, cds, ...)\n",
    "{\n",
    "    exons  = exons[exons$transcript_id == transcript_id,]\n",
    "    cds    = cds  [cds  $transcript_id == transcript_id,]\n",
    "    chrom  = unique(exons$chrom)\n",
    "    strand = unique(exons$strand)\n",
    "    \n",
    "    par(...)\n",
    "    plot(1,1, type = \"n\", xlim = xlims, ylim = c(0, 1), xlab = \"\", ylab = \"\", axes = FALSE)\n",
    "    axis(1, at = pretty(xlims, n = 5), labels = pretty(xlims, n = 5) * 1e-6)\n",
    "    \n",
    "    segments(x0 = min(exons$start), x1 = max(exons$end), y0 = 0.5, col = \"#0000ff\")\n",
    "    \n",
    "    if(nrow(exons) > 0){rect    (xleft =     exons$start , xright =     exons$end , ybottom = 0.3, col = \"#0000ff\", ytop = 0.7, border = \"#0000ff\")}\n",
    "    if(nrow(cds  ) > 0){rect    (xleft =     cds  $start , xright =     cds  $end , ybottom = 0.1, col = \"#0000ff\", ytop = 0.9, border = \"#0000ff\")}\n",
    "    \n",
    "    #text(x = max(exons$end), y = 0.5, labels = transcript_id, col = \"#0000ff\", pos = 4, cex = 1.5, font = 3)\n",
    "    \n",
    "    mtext(text = paste(chrom, \"(Mb)\"), side = 1, line = 2.5)\n",
    "    \n",
    "    abline(v = indata[indata$pch == 23, \"pos\"], lty = \"dashed\", col = \"#ff0000\")\n",
    "    \n",
    "    myangle = 0\n",
    "    if(strand == \"-\"){myangle = 180}\n",
    "    \n",
    "    toarrow = xlims[[1]] + (1:39) / 40 * (xlims[[2]] - xlims[[1]])\n",
    "    toarrow = toarrow[toarrow >= min(c(exons$start, cds$start)) & toarrow <= max(c(exons$end, cds$end))]\n",
    "    #toarrow = min(c(exons$start, cds$start)) + (max(c(exons$end, cds$end)) - min(c(exons$start, cds$start))) * (0:29) / 30\n",
    "    #Arrowhead(x0 = toarrow, y0 = 0.5, angle = myangle, arr.length = 0.1, arr.width = 0.1, arr.type = \"triangle\", arr.col = \"#0000ff\", arr.lwd = 0.01)\n",
    "    \n",
    "    if(strand == \"+\"){arrows(x0 = toarrow[1:(length(toarrow) - 1)], y0 = 0.5, x1 = toarrow[2:(length(toarrow)    )], length = 0.1, angle = 45, col = \"#0000ff\")}\n",
    "    if(strand == \"-\"){arrows(x0 = toarrow[2:(length(toarrow)    )], y0 = 0.5, x1 = toarrow[1:(length(toarrow) - 1)], length = 0.1, angle = 45, col = \"#0000ff\")}\n",
    "}\n",
    "\n",
    "\n",
    "plot_ex = function(transcript_id, type, trait, geneinfo, manifest, exons, cds)\n",
    "{\n",
    "    gene            = geneinfo[geneinfo$transcript_id == transcript_id,]\n",
    "    gene_id         = gene[1, \"gene_id\"]\n",
    "    coloc_gene_file = paste(\"pipeline/6.1.coloc_gwas/coloc.gene\", paste(\"snp\", gene_id      , \"txt\", sep = \".\"), sep = \"/\")\n",
    "    egene           =              fread(paste(\"pipeline/3.2.eqtls/eqtls_by_gene/cardiac_eqtls.gene\"   , paste(\"qtl\"    , gene_id      , \"txt\", sep = \".\"), sep = \"/\"), sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "    eisof           =              fread(paste(\"pipeline/3.2.eqtls/eqtls_by_gene/cardiac_eqtls.isoform\", paste(\"qtl\"    , transcript_id, \"txt\", sep = \".\"), sep = \"/\"), sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "    pp2snp_isof     =              fread(paste(\"pipeline/6.1.coloc_gwas/coloc.isoform\"                 , paste(\"snp\"    , transcript_id, \"txt\", sep = \".\"), sep = \"/\"), sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "    gtdata          = add_rownames(fread(paste(\"pipeline/1.3.genotype/use_isoform\"                     , paste(\"gt_data\", transcript_id, \"txt\", sep = \".\"), sep = \"/\"), sep = \"\\t\", header = TRUE, data.table = FALSE))\n",
    "    \n",
    "    if(file.exists(coloc_gene_file) == TRUE)\n",
    "    {\n",
    "        pp2snp_gene     = fread(coloc_gene_file, sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "        pp2snp_gene     = pp2snp_gene[pp2snp_gene$trait == trait & pp2snp_gene$type == 0,]\n",
    "    }\n",
    "    \n",
    "    pp2snp_isof     = pp2snp_isof[pp2snp_isof$trait == trait & pp2snp_isof$type == type,]\n",
    "    egene           = egene[egene$type == 0   ,]\n",
    "    eisof           = eisof[eisof$type == type,]\n",
    "    egene           = merge(eisof, egene[,c(\"id\", \"beta\", \"se\", \"pval\")], by = \"id\", suffixes = c(\"\", \"_gene\"))\n",
    "    infile          = manifest[manifest$id == trait, \"filename\"]\n",
    "    myrange         = range(egene$pos)\n",
    "    coord           = paste(sub(\"chr\", \"\", gene[1, \"chrom\"]), \":\", myrange[[1]], \"-\", myrange[[2]], sep = \"\")\n",
    "    command         = paste(\"tabix\", \"-h\", infile, coord)\n",
    "    gwas            = fread(cmd = command, sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "    colnames(gwas)  = unlist(strsplit(system(paste(\"zcat\", infile, \"|\", \"head\", \"-n\", 1), intern = TRUE), \"\\t\"))\n",
    "    gwas$id         = paste(\"VAR\", gwas$chr, gwas$pos, gwas$ref, gwas$alt, sep = \"_\")\n",
    "    gwas            = gwas[,c(\"id\", \"beta_meta\", \"se_meta\", \"pval_meta\")]\n",
    "    colnames(gwas)  = gsub(\"meta\", \"gwas\", colnames(gwas))\n",
    "    toplot          = merge(egene, gwas)\n",
    "    toplot          = merge(toplot, pp2snp_isof[,c(\"id\", \"pp_snp\")])\n",
    "    \n",
    "    if(file.exists(coloc_gene_file) == TRUE)\n",
    "    {\n",
    "        toplot          = merge(toplot, pp2snp_gene[,c(\"id\", \"pp_snp\")], by = \"id\", suffixes = c(\"_isoform\", \"_gene\"))\n",
    "    }else\n",
    "    {\n",
    "        toplot$pp_snp_isoform = toplot$pp_snp\n",
    "    }\n",
    "    \n",
    "    mylead    = toplot[which.max(toplot$pp_snp_isoform), \"id\"]\n",
    "    lddata    = data.frame(id = rownames(gtdata), r2 = 0, ld_color = \"#000088\")\n",
    "    mylead_gt = as.numeric(gtdata[mylead,])\n",
    "    lddata$r2 = unlist(lapply(1:nrow(gtdata), function(jj)\n",
    "    {\n",
    "        tocor = data.frame(x = mylead_gt, y = as.numeric(gtdata[jj, ]))\n",
    "        tocor = tocor[is.na(tocor$x) == FALSE & is.na(tocor$y) == FALSE, ]\n",
    "        out   = (cor(tocor$x, tocor$y))^2\n",
    "        \n",
    "        return(out)\n",
    "    }))\n",
    "    \n",
    "    if(nrow(lddata[lddata$r2 >= 0.2, ]) > 0){lddata[lddata$r2 >= 0.2, \"ld_color\"] = \"#aaffff\"}\n",
    "    if(nrow(lddata[lddata$r2 >= 0.4, ]) > 0){lddata[lddata$r2 >= 0.4, \"ld_color\"] = \"#00ff00\"}\n",
    "    if(nrow(lddata[lddata$r2 >= 0.6, ]) > 0){lddata[lddata$r2 >= 0.6, \"ld_color\"] = \"#ffa500\"}\n",
    "    if(nrow(lddata[lddata$r2 >= 0.8, ]) > 0){lddata[lddata$r2 >= 0.8, \"ld_color\"] = \"#ff0000\"}\n",
    "    \n",
    "    toplot            = merge(toplot, lddata)\n",
    "    ylim              = -log10(min(c(toplot$pval, toplot$pval_gene)))\n",
    "    #myrange           = range(c(gene$start, gene$end, egene[which.min(egene$pval), \"pos\"] - 10000, egene[which.min(egene$pval), \"pos\"] + 10000))\n",
    "    myrange           = range(c(gene$start, gene$end, toplot[toplot$id == mylead, \"pos\"] - 10000, toplot[toplot$id == mylead, \"pos\"] + 10000))\n",
    "    \n",
    "    #return(manifest[manifest$id == trait, c(\"id\", \"description\", \"category\")])\n",
    "    \n",
    "    if(manifest[manifest$id == trait, \"description\"] == \"\")\n",
    "    {\n",
    "        trait_description = sub(\"prescriptions-\", \"\", sub(\"-both_sexes\", \"\", manifest[manifest$id == trait, \"id\"]))\n",
    "    }else\n",
    "    {\n",
    "        trait_description = unlist(strsplit(manifest[manifest$id == trait, \"description\"], \",\"))[[1]]\n",
    "    }\n",
    "    \n",
    "    toplot$cex =  2\n",
    "    toplot$pch = 21\n",
    "    toplot     = toplot[order(toplot$r2),]\n",
    "    \n",
    "    toplot[toplot$id == mylead, \"cex\"] =  3\n",
    "    toplot[toplot$id == mylead, \"pch\"] = 23\n",
    "    \n",
    "    png(filename = paste(\"pipeline/X.1.combine\", paste(\"example\", gene[1, \"gene_name\"], \"png\", sep = \".\"), sep = \"/\"), width = 6.5, height = 11, units = \"in\", res = 300)\n",
    "    options(repr.plot.width = 6.5, repr.plot.height = 11)\n",
    "    #layout(cbind(c(1:5), c(1:4,6), c(1:4,7)))\n",
    "    layout(cbind(sort(c(rep(1:4, 2), 5))))\n",
    "    \n",
    "    plot_sumstats(toplot, \"pval\"          , myrange, ylim * 1.1, \"-log(p) eIsoform\"  , FALSE, FALSE, TRUE , FALSE, mar = c(0,4,0.5,1))\n",
    "    plot_sumstats(toplot, \"pval_gene\"     , myrange, ylim * 1.1, \"-log(p) eGene\"     , FALSE, FALSE, TRUE , FALSE, mar = c(0,4,0.5,1))\n",
    "    plot_sumstats(toplot, \"pval_gwas\"     , myrange, 0         , trait_description   , FALSE, FALSE, TRUE , FALSE, mar = c(0,4,0.5,1))\n",
    "    plot_sumstats(toplot, \"pp_snp_isoform\", myrange, 1         , \"PPA colocalization\", TRUE , FALSE, FALSE, TRUE , mar = c(0,4,0.5,1))\n",
    "    \n",
    "    plot_transcript(transcript_id, toplot, myrange, exons, cds, mar = c(4,4,0.5,1))\n",
    "    \n",
    "    #plot_scatter(toplot, \"pval\"     , \"pval_gene\", TRUE)\n",
    "    #plot_scatter(toplot, \"pval\"     , \"pval_gwas\", TRUE)\n",
    "    #plot_scatter(toplot, \"pval_gene\", \"pval_gwas\", TRUE)\n",
    "    \n",
    "    dev.off()\n",
    "    \n",
    "    message(paste(gene[1, \"gene_name\"], gene_id, transcript_id, type, toplot[which.max(toplot$pp_snp_isoform), \"rsid\"]))\n",
    "    \n",
    "}\n",
    "\n",
    "ii = 9\n",
    "#plot_ex(tab1[ii, \"transcript_id\"], tab1[ii, \"type\"], tab1[ii, \"trait\"], geneinfo_isoform, manifest, exons, cds)\n",
    "\n",
    "invisible(lapply(1:nrow(tab1), function(ii){plot_ex(tab1[ii, \"transcript_id\"], tab1[ii, \"type\"], tab1[ii, \"trait\"], geneinfo_isoform, manifest, exons, cds)}))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "render_html(\"TOR1AIP1\")\n",
      "render_html(\"LGALS8\")\n",
      "render_html(\"PRKG1\")\n",
      "render_html(\"MXI1\")\n",
      "render_html(\"DKK3\")\n",
      "render_html(\"GPR137\")\n",
      "render_html(\"UCHL3\")\n",
      "render_html(\"CIB1\")\n",
      "render_html(\"MLST8\")\n",
      "render_html(\"KANK2\")\n",
      "render_html(\"LPP\")\n",
      "render_html(\"NOD1\")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "message(paste('render_html(\"', geneinfo_isoform[geneinfo_isoform$transcript_id %in% tab1$transcript_id, \"gene_name\"], '\")', sep = \"\", collapse = \"\\n\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 12 × 23</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>transcript_id</th><th scope=col>type</th><th scope=col>gene_id</th><th scope=col>gene_name</th><th scope=col>id</th><th scope=col>chrom</th><th scope=col>pos</th><th scope=col>ref</th><th scope=col>alt</th><th scope=col>rsid</th><th scope=col>⋯</th><th scope=col>pval</th><th scope=col>fdr</th><th scope=col>distance</th><th scope=col>tr2type</th><th scope=col>ppa</th><th scope=col>trait</th><th scope=col>trait_description</th><th scope=col>trait_category</th><th scope=col>pval_gwas</th><th scope=col>ppa_gwas</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>10</th><td>ENST00000617246.4_3 </td><td>0</td><td>ENSG00000145012.13_7</td><td>LPP     </td><td>VAR_3_188066953_T_G      </td><td> 3</td><td>188066953</td><td>T      </td><td>G</td><td>rs6787621  </td><td>⋯</td><td>1.033506e-47</td><td>9.435914e-45</td><td>123760</td><td>ENST00000617246.4_3 0 </td><td>0.0511866064</td><td>continuous-MAP-both_sexes-combined_medadj_raw</td><td>Mean arterial pressure, combined automated + manual reading, adjusted by medication</td><td>                                                                                            </td><td>1.360e-07</td><td>0.9954974</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>ENST00000588787.5_5 </td><td>1</td><td>ENSG00000197256.10_9</td><td><span style=white-space:pre-wrap>KANK2   </span></td><td><span style=white-space:pre-wrap>VAR_19_11275811_CTG_C    </span></td><td>19</td><td> 11275811</td><td><span style=white-space:pre-wrap>CTG    </span></td><td>C</td><td><span style=white-space:pre-wrap>rs16423    </span></td><td>⋯</td><td>5.991958e-10</td><td>2.816220e-07</td><td> 10812</td><td>ENST00000588787.5_5 1 </td><td>0.0003362076</td><td><span style=white-space:pre-wrap>continuous-102-both_sexes-irnt               </span></td><td><span style=white-space:pre-wrap>Pulse rate, automated reading                                                      </span></td><td><span style=white-space:pre-wrap>UK Biobank Assessment Centre &gt; Physical measures &gt; Blood pressure                           </span></td><td>6.846e-14</td><td>0.9933029</td></tr>\n",
       "\t<tr><th scope=row>11</th><td>ENST00000643582.1_1 </td><td>0</td><td>ENSG00000185532.20_9</td><td>PRKG1   </td><td>VAR_10_53668890_T_C      </td><td>10</td><td> 53668890</td><td>T      </td><td>C</td><td>rs11000060 </td><td>⋯</td><td>5.553895e-14</td><td>4.037681e-11</td><td>834620</td><td>ENST00000643582.1_1 0 </td><td>0.0244994442</td><td>continuous-PP-both_sexes-combined_medadj_irnt</td><td>Pulse pressure, combined automated + manual reading, adjusted by medication        </td><td>                                                                                            </td><td>4.720e-09</td><td>0.9682591</td></tr>\n",
       "\t<tr><th scope=row>1</th><td>ENST00000222823.9_3 </td><td>0</td><td>ENSG00000106100.11_7</td><td>NOD1    </td><td>VAR_7_30503280_T_C       </td><td> 7</td><td> 30503280</td><td>T      </td><td>C</td><td>rs1558067  </td><td>⋯</td><td>9.393610e-26</td><td>3.860774e-23</td><td> 15120</td><td>ENST00000222823.9_3 0 </td><td>0.0261553526</td><td>continuous-LDLC-both_sexes-medadj_irnt       </td><td>LDL direct, adjusted by medication                                                 </td><td>                                                                                            </td><td>6.431e-06</td><td>0.9546053</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>ENST00000536017.1_1 </td><td>0</td><td>ENSG00000173264.14_7</td><td><span style=white-space:pre-wrap>GPR137  </span></td><td><span style=white-space:pre-wrap>VAR_11_64053848_GA_G     </span></td><td>11</td><td> 64053848</td><td><span style=white-space:pre-wrap>GA     </span></td><td>G</td><td><span style=white-space:pre-wrap>rs3831423  </span></td><td>⋯</td><td>2.632378e-28</td><td>7.897134e-26</td><td><span style=white-space:pre-wrap>   266</span></td><td>ENST00000536017.1_1 0 </td><td>0.0325164850</td><td><span style=white-space:pre-wrap>categorical-6150-both_sexes-100              </span></td><td><span style=white-space:pre-wrap>Vascular/heart problems diagnosed by doctor                                        </span></td><td>UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions</td><td>3.508e-05</td><td>0.9151178</td></tr>\n",
       "\t<tr><th scope=row>12</th><td>ENST00000651516.1_1 </td><td>0</td><td>ENSG00000119950.21_7</td><td><span style=white-space:pre-wrap>MXI1    </span></td><td><span style=white-space:pre-wrap>VAR_10_112041579_T_C     </span></td><td>10</td><td>112041579</td><td><span style=white-space:pre-wrap>T      </span></td><td>C</td><td><span style=white-space:pre-wrap>rs2855469  </span></td><td>⋯</td><td>6.310338e-33</td><td>2.871204e-30</td><td> 54966</td><td>ENST00000651516.1_1 0 </td><td>0.0262055309</td><td><span style=white-space:pre-wrap>categorical-6150-both_sexes-100              </span></td><td><span style=white-space:pre-wrap>Vascular/heart problems diagnosed by doctor                                        </span></td><td>UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions</td><td>2.222e-05</td><td>0.8427219</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ENST00000435319.8_2 </td><td>0</td><td>ENSG00000143337.18_6</td><td>TOR1AIP1</td><td>VAR_1_179858444_G_A      </td><td> 1</td><td>179858444</td><td>G      </td><td>A</td><td>rs2245425  </td><td>⋯</td><td>2.203381e-55</td><td>9.915216e-53</td><td>  6613</td><td>ENST00000435319.8_2 0 </td><td>0.0146587812</td><td>prescriptions-ramipril-both_sexes            </td><td>                                                                                   </td><td>ACE inhibitor,anti-hypertensive                                                             </td><td>6.660e-05</td><td>0.8262158</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>ENST00000528782.5_1 </td><td>0</td><td>ENSG00000116977.18_6</td><td>LGALS8  </td><td>VAR_1_236695882_C_T      </td><td> 1</td><td>236695882</td><td>C      </td><td>T</td><td>rs2794787  </td><td>⋯</td><td>9.024229e-06</td><td>6.208670e-03</td><td>  8597</td><td>ENST00000528782.5_1 0 </td><td>0.1176807258</td><td>continuous-PP-both_sexes-combined_irnt       </td><td>Pulse pressure, combined automated + manual reading                                </td><td>                                                                                            </td><td>1.687e-06</td><td>0.7636977</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>ENST00000377595.7_1 </td><td>0</td><td>ENSG00000118939.17_8</td><td>UCHL3   </td><td>VAR_13_76124151_C_T      </td><td>13</td><td> 76124151</td><td>C      </td><td>T</td><td>rs8192738  </td><td>⋯</td><td>1.268516e-07</td><td>6.012765e-05</td><td>   224</td><td>ENST00000377595.7_1 0 </td><td>0.0223341706</td><td>continuous-DBP-both_sexes-auto_medadj_raw    </td><td>Diastolic blood pressure, automated reading, adjusted by medication                </td><td>                                                                                            </td><td>3.578e-05</td><td>0.7393361</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ENST00000527132.1_2 </td><td>0</td><td>ENSG00000050165.18_5</td><td><span style=white-space:pre-wrap>DKK3    </span></td><td>VAR_11_11985133_GTTTACT_G</td><td>11</td><td> 11985133</td><td>GTTTACT</td><td>G</td><td>rs142843448</td><td>⋯</td><td>8.239718e-22</td><td>7.844211e-19</td><td> 45044</td><td>ENST00000527132.1_2 0 </td><td>0.0108097150</td><td><span style=white-space:pre-wrap>continuous-6033-both_sexes-irnt              </span></td><td><span style=white-space:pre-wrap>Maximum heart rate during fitness test                                             </span></td><td><span style=white-space:pre-wrap>UK Biobank Assessment Centre &gt; Physical measures &gt; ECG during exercise                      </span></td><td>5.205e-04</td><td>0.5672378</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>ENST00000397124.5_2 </td><td>0</td><td>ENSG00000167965.18_7</td><td><span style=white-space:pre-wrap>MLST8   </span></td><td><span style=white-space:pre-wrap>VAR_16_2256047_AC_A      </span></td><td>16</td><td><span style=white-space:pre-wrap>  2256047</span></td><td><span style=white-space:pre-wrap>AC     </span></td><td>A</td><td><span style=white-space:pre-wrap>rs3217423  </span></td><td>⋯</td><td>7.210944e-26</td><td>3.648738e-23</td><td><span style=white-space:pre-wrap>   305</span></td><td>ENST00000397124.5_2 0 </td><td>0.0277618574</td><td><span style=white-space:pre-wrap>continuous-21001-both_sexes-irnt             </span></td><td><span style=white-space:pre-wrap>Body mass index (BMI)                                                              </span></td><td><span style=white-space:pre-wrap>UK Biobank Assessment Centre &gt; Physical measures &gt; Anthropometry &gt; Body size measures       </span></td><td>1.163e-06</td><td>0.5656842</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>ENST00000328649.11_3</td><td>0</td><td>ENSG00000185043.12_6</td><td><span style=white-space:pre-wrap>CIB1    </span></td><td><span style=white-space:pre-wrap>VAR_15_90775913_C_T      </span></td><td>15</td><td> 90775913</td><td><span style=white-space:pre-wrap>C      </span></td><td>T</td><td><span style=white-space:pre-wrap>rs8041338  </span></td><td>⋯</td><td>1.343713e-06</td><td>7.242613e-04</td><td><span style=white-space:pre-wrap>  1260</span></td><td>ENST00000328649.11_3 0</td><td>0.2734090404</td><td><span style=white-space:pre-wrap>categorical-6150-both_sexes-4                </span></td><td><span style=white-space:pre-wrap>Vascular/heart problems diagnosed by doctor                                        </span></td><td>UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions</td><td>3.255e-04</td><td>0.5258842</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 12 × 23\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & transcript\\_id & type & gene\\_id & gene\\_name & id & chrom & pos & ref & alt & rsid & ⋯ & pval & fdr & distance & tr2type & ppa & trait & trait\\_description & trait\\_category & pval\\_gwas & ppa\\_gwas\\\\\n",
       "  & <chr> & <int> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & ⋯ & <dbl> & <dbl> & <int> & <chr> & <dbl> & <chr> & <chr> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t10 & ENST00000617246.4\\_3  & 0 & ENSG00000145012.13\\_7 & LPP      & VAR\\_3\\_188066953\\_T\\_G       &  3 & 188066953 & T       & G & rs6787621   & ⋯ & 1.033506e-47 & 9.435914e-45 & 123760 & ENST00000617246.4\\_3 0  & 0.0511866064 & continuous-MAP-both\\_sexes-combined\\_medadj\\_raw & Mean arterial pressure, combined automated + manual reading, adjusted by medication &                                                                                              & 1.360e-07 & 0.9954974\\\\\n",
       "\t9 & ENST00000588787.5\\_5  & 1 & ENSG00000197256.10\\_9 & KANK2    & VAR\\_19\\_11275811\\_CTG\\_C     & 19 &  11275811 & CTG     & C & rs16423     & ⋯ & 5.991958e-10 & 2.816220e-07 &  10812 & ENST00000588787.5\\_5 1  & 0.0003362076 & continuous-102-both\\_sexes-irnt                & Pulse rate, automated reading                                                       & UK Biobank Assessment Centre > Physical measures > Blood pressure                            & 6.846e-14 & 0.9933029\\\\\n",
       "\t11 & ENST00000643582.1\\_1  & 0 & ENSG00000185532.20\\_9 & PRKG1    & VAR\\_10\\_53668890\\_T\\_C       & 10 &  53668890 & T       & C & rs11000060  & ⋯ & 5.553895e-14 & 4.037681e-11 & 834620 & ENST00000643582.1\\_1 0  & 0.0244994442 & continuous-PP-both\\_sexes-combined\\_medadj\\_irnt & Pulse pressure, combined automated + manual reading, adjusted by medication         &                                                                                              & 4.720e-09 & 0.9682591\\\\\n",
       "\t1 & ENST00000222823.9\\_3  & 0 & ENSG00000106100.11\\_7 & NOD1     & VAR\\_7\\_30503280\\_T\\_C        &  7 &  30503280 & T       & C & rs1558067   & ⋯ & 9.393610e-26 & 3.860774e-23 &  15120 & ENST00000222823.9\\_3 0  & 0.0261553526 & continuous-LDLC-both\\_sexes-medadj\\_irnt        & LDL direct, adjusted by medication                                                  &                                                                                              & 6.431e-06 & 0.9546053\\\\\n",
       "\t8 & ENST00000536017.1\\_1  & 0 & ENSG00000173264.14\\_7 & GPR137   & VAR\\_11\\_64053848\\_GA\\_G      & 11 &  64053848 & GA      & G & rs3831423   & ⋯ & 2.632378e-28 & 7.897134e-26 &    266 & ENST00000536017.1\\_1 0  & 0.0325164850 & categorical-6150-both\\_sexes-100               & Vascular/heart problems diagnosed by doctor                                         & UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions & 3.508e-05 & 0.9151178\\\\\n",
       "\t12 & ENST00000651516.1\\_1  & 0 & ENSG00000119950.21\\_7 & MXI1     & VAR\\_10\\_112041579\\_T\\_C      & 10 & 112041579 & T       & C & rs2855469   & ⋯ & 6.310338e-33 & 2.871204e-30 &  54966 & ENST00000651516.1\\_1 0  & 0.0262055309 & categorical-6150-both\\_sexes-100               & Vascular/heart problems diagnosed by doctor                                         & UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions & 2.222e-05 & 0.8427219\\\\\n",
       "\t5 & ENST00000435319.8\\_2  & 0 & ENSG00000143337.18\\_6 & TOR1AIP1 & VAR\\_1\\_179858444\\_G\\_A       &  1 & 179858444 & G       & A & rs2245425   & ⋯ & 2.203381e-55 & 9.915216e-53 &   6613 & ENST00000435319.8\\_2 0  & 0.0146587812 & prescriptions-ramipril-both\\_sexes             &                                                                                     & ACE inhibitor,anti-hypertensive                                                              & 6.660e-05 & 0.8262158\\\\\n",
       "\t7 & ENST00000528782.5\\_1  & 0 & ENSG00000116977.18\\_6 & LGALS8   & VAR\\_1\\_236695882\\_C\\_T       &  1 & 236695882 & C       & T & rs2794787   & ⋯ & 9.024229e-06 & 6.208670e-03 &   8597 & ENST00000528782.5\\_1 0  & 0.1176807258 & continuous-PP-both\\_sexes-combined\\_irnt        & Pulse pressure, combined automated + manual reading                                 &                                                                                              & 1.687e-06 & 0.7636977\\\\\n",
       "\t3 & ENST00000377595.7\\_1  & 0 & ENSG00000118939.17\\_8 & UCHL3    & VAR\\_13\\_76124151\\_C\\_T       & 13 &  76124151 & C       & T & rs8192738   & ⋯ & 1.268516e-07 & 6.012765e-05 &    224 & ENST00000377595.7\\_1 0  & 0.0223341706 & continuous-DBP-both\\_sexes-auto\\_medadj\\_raw     & Diastolic blood pressure, automated reading, adjusted by medication                 &                                                                                              & 3.578e-05 & 0.7393361\\\\\n",
       "\t6 & ENST00000527132.1\\_2  & 0 & ENSG00000050165.18\\_5 & DKK3     & VAR\\_11\\_11985133\\_GTTTACT\\_G & 11 &  11985133 & GTTTACT & G & rs142843448 & ⋯ & 8.239718e-22 & 7.844211e-19 &  45044 & ENST00000527132.1\\_2 0  & 0.0108097150 & continuous-6033-both\\_sexes-irnt               & Maximum heart rate during fitness test                                              & UK Biobank Assessment Centre > Physical measures > ECG during exercise                       & 5.205e-04 & 0.5672378\\\\\n",
       "\t4 & ENST00000397124.5\\_2  & 0 & ENSG00000167965.18\\_7 & MLST8    & VAR\\_16\\_2256047\\_AC\\_A       & 16 &   2256047 & AC      & A & rs3217423   & ⋯ & 7.210944e-26 & 3.648738e-23 &    305 & ENST00000397124.5\\_2 0  & 0.0277618574 & continuous-21001-both\\_sexes-irnt              & Body mass index (BMI)                                                               & UK Biobank Assessment Centre > Physical measures > Anthropometry > Body size measures        & 1.163e-06 & 0.5656842\\\\\n",
       "\t2 & ENST00000328649.11\\_3 & 0 & ENSG00000185043.12\\_6 & CIB1     & VAR\\_15\\_90775913\\_C\\_T       & 15 &  90775913 & C       & T & rs8041338   & ⋯ & 1.343713e-06 & 7.242613e-04 &   1260 & ENST00000328649.11\\_3 0 & 0.2734090404 & categorical-6150-both\\_sexes-4                 & Vascular/heart problems diagnosed by doctor                                         & UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions & 3.255e-04 & 0.5258842\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 12 × 23\n",
       "\n",
       "| <!--/--> | transcript_id &lt;chr&gt; | type &lt;int&gt; | gene_id &lt;chr&gt; | gene_name &lt;chr&gt; | id &lt;chr&gt; | chrom &lt;int&gt; | pos &lt;int&gt; | ref &lt;chr&gt; | alt &lt;chr&gt; | rsid &lt;chr&gt; | ⋯ ⋯ | pval &lt;dbl&gt; | fdr &lt;dbl&gt; | distance &lt;int&gt; | tr2type &lt;chr&gt; | ppa &lt;dbl&gt; | trait &lt;chr&gt; | trait_description &lt;chr&gt; | trait_category &lt;chr&gt; | pval_gwas &lt;dbl&gt; | ppa_gwas &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 10 | ENST00000617246.4_3  | 0 | ENSG00000145012.13_7 | LPP      | VAR_3_188066953_T_G       |  3 | 188066953 | T       | G | rs6787621   | ⋯ | 1.033506e-47 | 9.435914e-45 | 123760 | ENST00000617246.4_3 0  | 0.0511866064 | continuous-MAP-both_sexes-combined_medadj_raw | Mean arterial pressure, combined automated + manual reading, adjusted by medication | <!----> | 1.360e-07 | 0.9954974 |\n",
       "| 9 | ENST00000588787.5_5  | 1 | ENSG00000197256.10_9 | KANK2    | VAR_19_11275811_CTG_C     | 19 |  11275811 | CTG     | C | rs16423     | ⋯ | 5.991958e-10 | 2.816220e-07 |  10812 | ENST00000588787.5_5 1  | 0.0003362076 | continuous-102-both_sexes-irnt                | Pulse rate, automated reading                                                       | UK Biobank Assessment Centre &gt; Physical measures &gt; Blood pressure                            | 6.846e-14 | 0.9933029 |\n",
       "| 11 | ENST00000643582.1_1  | 0 | ENSG00000185532.20_9 | PRKG1    | VAR_10_53668890_T_C       | 10 |  53668890 | T       | C | rs11000060  | ⋯ | 5.553895e-14 | 4.037681e-11 | 834620 | ENST00000643582.1_1 0  | 0.0244994442 | continuous-PP-both_sexes-combined_medadj_irnt | Pulse pressure, combined automated + manual reading, adjusted by medication         | <!----> | 4.720e-09 | 0.9682591 |\n",
       "| 1 | ENST00000222823.9_3  | 0 | ENSG00000106100.11_7 | NOD1     | VAR_7_30503280_T_C        |  7 |  30503280 | T       | C | rs1558067   | ⋯ | 9.393610e-26 | 3.860774e-23 |  15120 | ENST00000222823.9_3 0  | 0.0261553526 | continuous-LDLC-both_sexes-medadj_irnt        | LDL direct, adjusted by medication                                                  | <!----> | 6.431e-06 | 0.9546053 |\n",
       "| 8 | ENST00000536017.1_1  | 0 | ENSG00000173264.14_7 | GPR137   | VAR_11_64053848_GA_G      | 11 |  64053848 | GA      | G | rs3831423   | ⋯ | 2.632378e-28 | 7.897134e-26 |    266 | ENST00000536017.1_1 0  | 0.0325164850 | categorical-6150-both_sexes-100               | Vascular/heart problems diagnosed by doctor                                         | UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions | 3.508e-05 | 0.9151178 |\n",
       "| 12 | ENST00000651516.1_1  | 0 | ENSG00000119950.21_7 | MXI1     | VAR_10_112041579_T_C      | 10 | 112041579 | T       | C | rs2855469   | ⋯ | 6.310338e-33 | 2.871204e-30 |  54966 | ENST00000651516.1_1 0  | 0.0262055309 | categorical-6150-both_sexes-100               | Vascular/heart problems diagnosed by doctor                                         | UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions | 2.222e-05 | 0.8427219 |\n",
       "| 5 | ENST00000435319.8_2  | 0 | ENSG00000143337.18_6 | TOR1AIP1 | VAR_1_179858444_G_A       |  1 | 179858444 | G       | A | rs2245425   | ⋯ | 2.203381e-55 | 9.915216e-53 |   6613 | ENST00000435319.8_2 0  | 0.0146587812 | prescriptions-ramipril-both_sexes             | <!----> | ACE inhibitor,anti-hypertensive                                                              | 6.660e-05 | 0.8262158 |\n",
       "| 7 | ENST00000528782.5_1  | 0 | ENSG00000116977.18_6 | LGALS8   | VAR_1_236695882_C_T       |  1 | 236695882 | C       | T | rs2794787   | ⋯ | 9.024229e-06 | 6.208670e-03 |   8597 | ENST00000528782.5_1 0  | 0.1176807258 | continuous-PP-both_sexes-combined_irnt        | Pulse pressure, combined automated + manual reading                                 | <!----> | 1.687e-06 | 0.7636977 |\n",
       "| 3 | ENST00000377595.7_1  | 0 | ENSG00000118939.17_8 | UCHL3    | VAR_13_76124151_C_T       | 13 |  76124151 | C       | T | rs8192738   | ⋯ | 1.268516e-07 | 6.012765e-05 |    224 | ENST00000377595.7_1 0  | 0.0223341706 | continuous-DBP-both_sexes-auto_medadj_raw     | Diastolic blood pressure, automated reading, adjusted by medication                 | <!----> | 3.578e-05 | 0.7393361 |\n",
       "| 6 | ENST00000527132.1_2  | 0 | ENSG00000050165.18_5 | DKK3     | VAR_11_11985133_GTTTACT_G | 11 |  11985133 | GTTTACT | G | rs142843448 | ⋯ | 8.239718e-22 | 7.844211e-19 |  45044 | ENST00000527132.1_2 0  | 0.0108097150 | continuous-6033-both_sexes-irnt               | Maximum heart rate during fitness test                                              | UK Biobank Assessment Centre &gt; Physical measures &gt; ECG during exercise                       | 5.205e-04 | 0.5672378 |\n",
       "| 4 | ENST00000397124.5_2  | 0 | ENSG00000167965.18_7 | MLST8    | VAR_16_2256047_AC_A       | 16 |   2256047 | AC      | A | rs3217423   | ⋯ | 7.210944e-26 | 3.648738e-23 |    305 | ENST00000397124.5_2 0  | 0.0277618574 | continuous-21001-both_sexes-irnt              | Body mass index (BMI)                                                               | UK Biobank Assessment Centre &gt; Physical measures &gt; Anthropometry &gt; Body size measures        | 1.163e-06 | 0.5656842 |\n",
       "| 2 | ENST00000328649.11_3 | 0 | ENSG00000185043.12_6 | CIB1     | VAR_15_90775913_C_T       | 15 |  90775913 | C       | T | rs8041338   | ⋯ | 1.343713e-06 | 7.242613e-04 |   1260 | ENST00000328649.11_3 0 | 0.2734090404 | categorical-6150-both_sexes-4                 | Vascular/heart problems diagnosed by doctor                                         | UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions | 3.255e-04 | 0.5258842 |\n",
       "\n"
      ],
      "text/plain": [
       "   transcript_id        type gene_id              gene_name\n",
       "10 ENST00000617246.4_3  0    ENSG00000145012.13_7 LPP      \n",
       "9  ENST00000588787.5_5  1    ENSG00000197256.10_9 KANK2    \n",
       "11 ENST00000643582.1_1  0    ENSG00000185532.20_9 PRKG1    \n",
       "1  ENST00000222823.9_3  0    ENSG00000106100.11_7 NOD1     \n",
       "8  ENST00000536017.1_1  0    ENSG00000173264.14_7 GPR137   \n",
       "12 ENST00000651516.1_1  0    ENSG00000119950.21_7 MXI1     \n",
       "5  ENST00000435319.8_2  0    ENSG00000143337.18_6 TOR1AIP1 \n",
       "7  ENST00000528782.5_1  0    ENSG00000116977.18_6 LGALS8   \n",
       "3  ENST00000377595.7_1  0    ENSG00000118939.17_8 UCHL3    \n",
       "6  ENST00000527132.1_2  0    ENSG00000050165.18_5 DKK3     \n",
       "4  ENST00000397124.5_2  0    ENSG00000167965.18_7 MLST8    \n",
       "2  ENST00000328649.11_3 0    ENSG00000185043.12_6 CIB1     \n",
       "   id                        chrom pos       ref     alt rsid        ⋯\n",
       "10 VAR_3_188066953_T_G        3    188066953 T       G   rs6787621   ⋯\n",
       "9  VAR_19_11275811_CTG_C     19     11275811 CTG     C   rs16423     ⋯\n",
       "11 VAR_10_53668890_T_C       10     53668890 T       C   rs11000060  ⋯\n",
       "1  VAR_7_30503280_T_C         7     30503280 T       C   rs1558067   ⋯\n",
       "8  VAR_11_64053848_GA_G      11     64053848 GA      G   rs3831423   ⋯\n",
       "12 VAR_10_112041579_T_C      10    112041579 T       C   rs2855469   ⋯\n",
       "5  VAR_1_179858444_G_A        1    179858444 G       A   rs2245425   ⋯\n",
       "7  VAR_1_236695882_C_T        1    236695882 C       T   rs2794787   ⋯\n",
       "3  VAR_13_76124151_C_T       13     76124151 C       T   rs8192738   ⋯\n",
       "6  VAR_11_11985133_GTTTACT_G 11     11985133 GTTTACT G   rs142843448 ⋯\n",
       "4  VAR_16_2256047_AC_A       16      2256047 AC      A   rs3217423   ⋯\n",
       "2  VAR_15_90775913_C_T       15     90775913 C       T   rs8041338   ⋯\n",
       "   pval         fdr          distance tr2type                ppa         \n",
       "10 1.033506e-47 9.435914e-45 123760   ENST00000617246.4_3 0  0.0511866064\n",
       "9  5.991958e-10 2.816220e-07  10812   ENST00000588787.5_5 1  0.0003362076\n",
       "11 5.553895e-14 4.037681e-11 834620   ENST00000643582.1_1 0  0.0244994442\n",
       "1  9.393610e-26 3.860774e-23  15120   ENST00000222823.9_3 0  0.0261553526\n",
       "8  2.632378e-28 7.897134e-26    266   ENST00000536017.1_1 0  0.0325164850\n",
       "12 6.310338e-33 2.871204e-30  54966   ENST00000651516.1_1 0  0.0262055309\n",
       "5  2.203381e-55 9.915216e-53   6613   ENST00000435319.8_2 0  0.0146587812\n",
       "7  9.024229e-06 6.208670e-03   8597   ENST00000528782.5_1 0  0.1176807258\n",
       "3  1.268516e-07 6.012765e-05    224   ENST00000377595.7_1 0  0.0223341706\n",
       "6  8.239718e-22 7.844211e-19  45044   ENST00000527132.1_2 0  0.0108097150\n",
       "4  7.210944e-26 3.648738e-23    305   ENST00000397124.5_2 0  0.0277618574\n",
       "2  1.343713e-06 7.242613e-04   1260   ENST00000328649.11_3 0 0.2734090404\n",
       "   trait                                        \n",
       "10 continuous-MAP-both_sexes-combined_medadj_raw\n",
       "9  continuous-102-both_sexes-irnt               \n",
       "11 continuous-PP-both_sexes-combined_medadj_irnt\n",
       "1  continuous-LDLC-both_sexes-medadj_irnt       \n",
       "8  categorical-6150-both_sexes-100              \n",
       "12 categorical-6150-both_sexes-100              \n",
       "5  prescriptions-ramipril-both_sexes            \n",
       "7  continuous-PP-both_sexes-combined_irnt       \n",
       "3  continuous-DBP-both_sexes-auto_medadj_raw    \n",
       "6  continuous-6033-both_sexes-irnt              \n",
       "4  continuous-21001-both_sexes-irnt             \n",
       "2  categorical-6150-both_sexes-4                \n",
       "   trait_description                                                                  \n",
       "10 Mean arterial pressure, combined automated + manual reading, adjusted by medication\n",
       "9  Pulse rate, automated reading                                                      \n",
       "11 Pulse pressure, combined automated + manual reading, adjusted by medication        \n",
       "1  LDL direct, adjusted by medication                                                 \n",
       "8  Vascular/heart problems diagnosed by doctor                                        \n",
       "12 Vascular/heart problems diagnosed by doctor                                        \n",
       "5                                                                                     \n",
       "7  Pulse pressure, combined automated + manual reading                                \n",
       "3  Diastolic blood pressure, automated reading, adjusted by medication                \n",
       "6  Maximum heart rate during fitness test                                             \n",
       "4  Body mass index (BMI)                                                              \n",
       "2  Vascular/heart problems diagnosed by doctor                                        \n",
       "   trait_category                                                                              \n",
       "10                                                                                             \n",
       "9  UK Biobank Assessment Centre > Physical measures > Blood pressure                           \n",
       "11                                                                                             \n",
       "1                                                                                              \n",
       "8  UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions\n",
       "12 UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions\n",
       "5  ACE inhibitor,anti-hypertensive                                                             \n",
       "7                                                                                              \n",
       "3                                                                                              \n",
       "6  UK Biobank Assessment Centre > Physical measures > ECG during exercise                      \n",
       "4  UK Biobank Assessment Centre > Physical measures > Anthropometry > Body size measures       \n",
       "2  UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions\n",
       "   pval_gwas ppa_gwas \n",
       "10 1.360e-07 0.9954974\n",
       "9  6.846e-14 0.9933029\n",
       "11 4.720e-09 0.9682591\n",
       "1  6.431e-06 0.9546053\n",
       "8  3.508e-05 0.9151178\n",
       "12 2.222e-05 0.8427219\n",
       "5  6.660e-05 0.8262158\n",
       "7  1.687e-06 0.7636977\n",
       "3  3.578e-05 0.7393361\n",
       "6  5.205e-04 0.5672378\n",
       "4  1.163e-06 0.5656842\n",
       "2  3.255e-04 0.5258842"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "tab1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Comments\n",
    "\n",
    "## iPSC-CVPC-specific eGenes that are also overexpressed in iPSC-CVPCs and colocalize with GWAS: LINC01405; TBX2-AS1; ENPP4; TMEM71\n",
    "- ***TBX2***: TF involved in cardiac development. The TBX2/TBX2-AS1 locus is known to be associated with blood pressure. Maybe TBX2 and TBX2-AS1 are associated with the same eQTLs? \n",
    "- ***ENPP4***: involved in arterial septal defects (GeneCards)\n",
    "- *TMEM71* and *LINC01405* don't have a very well-defined function \n",
    "\n",
    "## Adult-specific eGenes that are also overexpressed in adult heart and colocalize with GWAS: PFKFB2; SLC22A18; CARNS1; AC116903.1; PGPEP1L; CIAO3; GNAO1; ROCK1P1; LINC01535; PIGF; GPR17; LAMA5-AS1; KREMEN1; MKRN2; PDZRN3; AC117489.1; PFN2; PLD1; FBN2; HDDC2; DMTN; PEBP4; AC084024.3\n",
    "- ***PFKFB2***: synthesis and degradation of fructose-2,6-bisphosphate, a regulatory molecule that controls glycolysis. important regulator of glycolytic flux in cardiac cells ([27802586](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5193105/)). Involved in blood pressure\n",
    "- ***CARNS1***: Carnosine syntase. Associated with LDL levels and pulse pressure. In animal studies, carnosine has been shown to suppress many biochemical processes that accompany aging and age related chronic diseases such as obesity, type 2 diabetes and diabetes complications, cardiovascular diseases, cancer and dementia (Introduction in [26439389](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4595442/))\n",
    "- ***ROCK1P1*** (LITTLE ROCK): pseudogene for ***ROCK1*** (involved in blood pressure by promoting smooth muscle contraction: [20398283](https://bmcgenet.biomedcentral.com/articles/10.1186/1471-2156-11-22)): has 5 exons that share 99% seuqence identity with ROCK1, is expressed (especially in smooth muscle) but seems to be NMD. This seems to be a case similar to the *PTEN*/*PTENP1* regulation by paralogous pseudogenes that are expressed only to regulate the expression of their associated protein coding gene. Maybe there is something more interesting here? Our eQTLs are associated with hypertension.\n",
    "- ***GPR17***: regulation of heart-resident mesenchymal cells and blood-borne cellular species recruitment following myocardial infarction, orchestrated by GPR17 ([24909956](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4196654/)). GPR17 is a novel negative regulator of Shh signalling in a wide range of cellular contexts ([31444216](https://dev.biologists.org/content/146/17/dev176784)). Its eQTL is associated with cholesterol\n",
    "- ***LAMA5-AS1***: AS of *LAMA5* (Laminins, a family of extracellular matrix glycoproteins, are the major noncollagenous constituent of basement membranes: GeneCards). A *LAMA5* variant (rs659822) is associated with low cholesterol in elderly individuals ([20951195](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2998567/)). *LAMA5-AS1* eQTLs are associated with pulse pressure, but *LAMA5* is not associated with GWAS.\n",
    "- ***KREMEN1***: Receptor for Dickkopf proteins. Cooperates with DKK1/2 to inhibit Wnt/beta-catenin signaling by promoting the endocytosis of Wnt receptors LRP5 and LRP6. In the absence of DKK1, potentiates Wnt-beta-catenin signaling by maintaining LRP5 or LRP6 at the cell membrane. Can trigger apoptosis in a Wnt-independent manner and this apoptotic activity is inhibited upon binding of the ligand DKK1 (GeneCards). *KREMEN1* eQTLs are associated with heart failure. Something interesting with the return to fetal?\n",
    "- ***PDZRN3***: cardiac Pdzrn3 deficiency protected against heart failure while over expression of Pdzrn3 in mouse cardiomyocytes during the first weeks of life, impaired postnatal cardiomyocyte maturation leading to premature death ([https://doi.org/10.1101/2020.07.29.226597](https://www.biorxiv.org/content/10.1101/2020.07.29.226597v1.full)). Is also required for cardiac differentiation ([17118964](https://jcs.biologists.org/content/119/24/5106)). *PDZRN3* eQTLs are associated with QRS duration.\n",
    "- ***PFN2***: regulates actin polymerization (GeneCards). *PFN2* eQTLs are associated with pulse pressure.\n",
    "- ***PLD1***: associated with the cardiac sarcolemmal (SL) membrane hydrolyses phosphatidyl-choline to produce phosphatidic acid, an important phospholipid signaling molecule known to influence cardiac func-tion ([15601581](https://onlinelibrary.wiley.com/doi/epdf/10.1111/j.1582-4934.2004.tb00477.x)). *PLD1* eQTLs are associated with pulse pressure.\n",
    "- ***FBN2***. Involved in Congenital contractural arachnodactyly (CCA), which is often associated with cardiac defects ([16740166](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1524931/)). *FBN2* eQTls are associated with heart failure. \n",
    "- ***DMTN***: actin binding protein. Involved in erythrocyte membrane stability: its absence results in anemia ([27073223](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4937361/)). Regulates Calcium Mobilization and Signaling in Platelets ([23060452](https://www.jbc.org/content/287/49/41218.full)). *DMTN* eQTLs are associated with MI and atherosclerosis.\n",
    "- ***PEBP4***: silencing inhibits TGF-β1-induced epithelial-mesenchymal transition of osteosarcoma cells by suppressing the PI3K/Akt pathway ([X](http://www.ijcep.com/files/ijcep0020912.pdf)). eQTLs are associated with pulse rate.\n",
    "\n",
    "## iPSC-CVPC-specific eIsoforms that are also overexpressed in iPSC-CVPCs and colocalize with GWAS: HES4; ADAM15; LGALS8; GPR137; CIB1; NOMO3; AC005670.3; AP3D1; HLA-C; NOD1\n",
    "- ***ADAM15***: ADAM (a Disintegrin and Metalloproteinase) 15 Deficiency Exacerbates Ang II (Angiotensin II)–Induced Aortic Remodeling Leading to Abdominal Aortic Aneurysm ([32522006](https://www.ahajournals.org/doi/10.1161/ATVBAHA.120.314600)). A Disintegrin and Metalloproteinase 15 Contributes to Atherosclerosis by Mediating Endothelial Barrier Dysfunction via Src Family Kinase Activity ([22904271](https://www.ahajournals.org/doi/full/10.1161/atvbaha.112.252205)). Is also associated with MI ([19563617](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2697557/)). eQTLs are associated with blood pressure and varicose veins.\n",
    "- ***LGALS8*** (galectin-8): ([review of functions](https://jcs.biologists.org/content/joces/131/9/jcs208884.full.pdf)). ieQTLs are associated with pulse pressure but not gene eQTLs.\n",
    "- ***GPR137***: G protein coupled receptor. Has different isoforms with known functions during development ([25514843](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4337595/)). Its alternative splicing is associated with ESRP1 ([28975893](https://elifesciences.org/articles/28366)) ieQTLs only are associated with heart failure. \n",
    "- ***AP3D1***: associated with albinism and a lot of developmental issues. ieQTLs only are associated with blood pressure.\n",
    "- ***NOD1***: NOD1 Activation Induces Cardiac Dysfunction and Modulates Cardiac Fibrosis and Cardiomyocyte Apoptosis ([23028889](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3445482/)). Deficiency of NOD1 Improves the β-Adrenergic Modulation of Ca2+ Handling in a Mouse Model of Heart Failure ([29962957](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6010671/)). ieQTLs only are associated with LDL levels, while gene eQTLs are associated with triglycerides (low PP)\n",
    "\n",
    "## Adult-specific eIsoforms that are also overexpressed in adult heart and colocalize with GWAS: PALMD; FMO2; MS4A7; XRRA1; TMEM255B; MLH3; ATXN3; XRCC3; HAGHL; EMP2; RPL13; NUP88; FAM106A; AC005670.3; PCAT19; ZNF814; ZSCAN18; USP34; AC073254.1; SERHL; FAM118A; PIGG; NPR3; ITGA1; ATG10; MRNIP; NOTCH4; HLA-DQA1; HLA-DPB1; HLA-DPB1; TMEM63B; PRKRIP1\n",
    "- ***PALMD*** (palmdelphin): susceptibility gene for calcific aortic valve stenosis ([29511167](https://www.nature.com/articles/s41467-018-03260-6)). ieQTL is associated with pulse rate, gene eQTL is associated with blood pressure\n",
    "- ***FMO2***: NADPH-dependent enzyme that catalyzes the N-oxidation of some primary alkylamines through an N-hydroxylamine intermediate. However, some human populations contain an allele (FMO2\\*2A) with a premature stop codon, resulting in a protein that is C-terminally-truncated, has no catalytic activity, and is likely degraded rapidly (GeneCards and [15864117](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1351039/)). eQTLs are associated with ventricular rate\n",
    "- ***XRCC3***: DNA repair. Associated with increased carotid artery thickness upon exposure to ionizing radiations ([X](https://academic.oup.com/eurheartj/article/34/suppl_1/P3922/2862076)). XRCC3 polymorphism is associated with hypertension-induced left ventricular hypertrophy ([29626209](https://www.nature.com/articles/s41440-018-0038-0?platform=hootsuite)). Xrcc3 Induces Cisplatin Resistance by Stimulation of Rad51-Related Recombinational Repair, S-Phase Checkpoint Activation, and Reduced Apoptosis ([15843498](https://jpet.aspetjournals.org/content/314/2/495.abstract)). iEQTLs are associated with pulse rate, BMI, atrial fibrillation, heart failure and hypertension (gene eQTLs only with LDL levels).\n",
    "- ***ATXN3***: . Splice Isoforms of the Polyglutamine Disease Protein Ataxin-3 Exhibit Similar Enzymatic yet Different Aggregation Properties ([21060878](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2965175/)). A knockin mouse model of spinocerebellar ataxia type 3 exhibits prominent aggregate pathology and aberrant splicing of the disease gene transcript ([25320121](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4321438/)). ieQTLs are associated with HDL levels and pulse rate (no associations for gene-level).\n",
    "- ***RPL13***: ribosomal protein. candidate for congenital heart disease ([31625562](https://academic.oup.com/hmg/article/28/23/3954/5589181?casa_token=qwbNCOmtIvQAAAAA:oIHuTEm5rI_a5uKnV2mm_cyOrh-O3z316JvNWNi3uuLMd4Xum37s-RMDq8HmKW4wzLlkLvMJgVcZ)). ieQTLs and gene eQTLs are associated with blood pressure and QRS duration.\n",
    "- ***NUP88***: nucleoporin complex. Other NUPs are associated with heart disease ([PMC5027676](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027676/)). ieQTLs only are associated with LDL levels.\n",
    "- ***PCAT19***: intereseting because this is a lncRNA with oncogenic properties. eQTLs are associated with cardiac conduction disorders.\n",
    "- ***USP34***: Acts as an activator of the Wnt signaling pathway downstream of the beta-catenin destruction complex by deubiquitinating and stabilizing AXIN1 and AXIN2, leading to promote nuclear accumulation of AXIN1 and AXIN2 and positively regulate beta-catenin (CTNBB1)-mediated transcription (GeneCards). Associated with CHD ([29555671](https://www.ahajournals.org/doi/full/10.1161/circgen.117.001978)). eQTLs are associated with blood pressure.\n",
    "- ***NPR3***: protects cardiomyocytes from apoptosis ([PMC5026813](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5026813/)). ieQTLs are associated with triglycerides levels; gene eQTLs with blood pressure.\n",
    "- ***ITGA1***: Cardiac Fibroblasts Regulate Myocardial Proliferation through β1 Integrin Signaling ([19217425](https://www.sciencedirect.com/science/article/pii/S1534580708005170)). ieQTLs only are associated with unstable angina.\n",
    "- ***ATG10***: involved in autophagy ([PMC2838272](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2838272/)). ieQTLs only are associated with HDL, triglycerides levels and palpitations.\n",
    "- ***NOTCH4***: a lot of associations with heart functions. eQTLs are associated with blood pressure\n",
    "- ***TMEM63B***: OSCA/TMEM63 are an evolutionarily conserved family of mechanically activated ion channels ([30382938](https://elifesciences.org/articles/41844)). ieQTLs only are associated with Acute pulmonary heart disease; Pulmonary embolism and infarction, acute\n",
    "\n",
    "## iPSC-CVPC-specific eIsoforms that colocalize with GWAS, but their associated eGene does not: HES4; ADAM15; TOR1AIP1; LGALS8; PRKG1; MXI1; DKK3; GPR137; UCHL3; CIB1; MLST8; NOMO3; ATXN2L; CTDNEP1; AC005670.3; AP3D1; KANK2; DMPK; DIDO1; THOC5; STAB1; LPP; RPL9; CAST; HLA-F-AS1; HLA-C; AC147651.1; NOD1; PTP4A3\n",
    "- ***ADAM15***: ADAM (a Disintegrin and Metalloproteinase) 15 Deficiency Exacerbates Ang II (Angiotensin II)–Induced Aortic Remodeling Leading to Abdominal Aortic Aneurysm ([32522006](https://www.ahajournals.org/doi/10.1161/ATVBAHA.120.314600)). A Disintegrin and Metalloproteinase 15 Contributes to Atherosclerosis by Mediating Endothelial Barrier Dysfunction via Src Family Kinase Activity ([22904271](https://www.ahajournals.org/doi/full/10.1161/atvbaha.112.252205)). Is also associated with MI ([19563617]\n",
    "- ***LGALS8*** (galectin-8): ([review of functions](https://jcs.biologists.org/content/joces/131/9/jcs208884.full.pdf)). ieQTLs are associated with pulse pressure but not gene eQTLs.\n",
    "- ***PRKG1***: key mediator of the nitric oxide (NO)/cGMP signaling pathway (GeneCards). Recurrent Gain-of-Function Mutation in PRKG1 Causes Thoracic Aortic Aneurysms and Acute Aortic Dissections ([PMC3738837](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3738837/)). genetic variability in PRKG1 might affect left ventricular (LV) function and structure ([24060898](https://www.ahajournals.org/doi/pdf/10.1161/HYPERTENSIONAHA.113.01630)).ieQTLs are associated with blood pressure.\n",
    "- ***MXI1***: transcriptional repressor thought to negatively regulate MYC function (GeneCards). MXI1-0, an Alternatively Transcribed Mxi1 Isoform, Is Overexpressed in Glioblastomas\n",
    " ([PMC1531670](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1531670/)). ieQTLs are associated with ischaemic heart disease.\n",
    "- ***DKK3***: involved in embryonic development through its interactions with the Wnt signaling pathway (GeneCards). Its function is associated with KREMEN, which is adult-specific (see above). Dickkopf-3 protects against cardiac dysfunction and ventricular remodelling following myocardial infarction ([25840773](https://pubmed.ncbi.nlm.nih.gov/25840773/)). ieQTLs are associated with heart rate\n",
    "- ***GPR137***: G protein coupled receptor. Has different isoforms with known functions during development ([25514843](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4337595/)). Its alternative splicing is associated with ESRP1 ([28975893](https://elifesciences.org/articles/28366)) ieQTLs only are associated with heart failure. \n",
    "- ***ATXN2L***: Mouse embryonal fibroblast cells revealed more multinucleated giant cells upon ATXN2L deficiency. In addition, in human neural cells, transcript levels of ATXN2L were induced upon starvation and glucose and amino acids exposure, but this induction was partially prevented by serum or low cholesterol administration ([32698485](https://www.mdpi.com/1422-0067/21/14/5124/pdf)). ieQTLs are associated with blood pressure, HDL levels, BMI.\n",
    "- ***CTDNEP1***: controls fatty acid metabolism (GeneCards, [PMC3283218](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3283218/)). ieQTLs are associated with blood pressure medication.\n",
    "- ***AP3D1***: associated with albinism and a lot of developmental issues. ieQTLs only are associated with blood pressure.\n",
    "- ***KANK2***: Involved in actin stress fibers formation(GeneCards). ieQTLs are associated with HDL cholesterol; Pulse rate, automated reading\n",
    "- ***DMPK***: critical to the modulation of cardiac contractility and to the maintenance of proper cardiac conduction activity probably through the regulation of cellular calcium homeostasis. Phosphorylates PLN, a regulator of calcium pumps and may regulate sarcoplasmic reticulum calcium uptake in myocytes (GeneCards). Dmpk gene deletion or antisense knockdown does not compromise cardiac or skeletal muscle function in mice ([27522499](https://pubmed.ncbi.nlm.nih.gov/27522499/)). ieQTLs are associated with calcium levels, atherosclerosis and blood pressure (also eQTLs)\n",
    "- ***LPP***: mechanosensitive protein regulated by nitric oxide in the heart ([PMC3642136](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3642136/)). ieQTLs are associated with blood pressure\n",
    "- ***CAST*** (calpastatin): involved in cardiac hypertrophy ([PMC3151485](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3151485/)). ieQTLs are associated with Medication for cholesterol, blood pressure or diabetes; high blood pressure\n",
    "- ***NOD1***: NOD1 Activation Induces Cardiac Dysfunction and Modulates Cardiac Fibrosis and Cardiomyocyte Apoptosis ([23028889](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3445482/)). Deficiency of NOD1 Improves the β-Adrenergic Modulation of Ca2+ Handling in a Mouse Model of Heart Failure ([29962957](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6010671/)). ieQTLs only are associated with LDL levels, while gene eQTLs are associated with triglycerides (low PP)\n",
    "\n",
    "\n",
    "## Adult-specific eIsoforms that colocalize with GWAS, but their associated eGene does not: PALMD; FMO2; B3GAT3; XRRA1; TMEM255B; MLH3; ATXN3; XRCC3; ICE2; HAGHL; EMP2; HP; RPL13; NUP88; GRAMD1A; PCAT19; AC073254.1; SERHL; FAM118A; TDGF1; PIGG; CRMP1; AC093752.1; NPR3; ITGA1; ATG10; MRNIP; HLA-C; NOTCH4; HLA-DQA1; HLA-DPB1; TMEM63B; AC147651.1; PRKRIP1; NMRK1\n",
    "- ***PALMD*** (palmdelphin): susceptibility gene for calcific aortic valve stenosis ([29511167](https://www.nature.com/articles/s41467-018-03260-6)). ieQTL is associated with pulse rate, gene eQTL is associated with blood pressure\n",
    "- ***FMO2***: NADPH-dependent enzyme that catalyzes the N-oxidation of some primary alkylamines through an N-hydroxylamine intermediate. However, some human populations contain an allele (FMO2\\*2A) with a premature stop codon, resulting in a protein that is C-terminally-truncated, has no catalytic activity, and is likely degraded rapidly (GeneCards and [15864117](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1351039/)). eQTLs are associated with ventricular rate\n",
    "- ***XRCC3***: DNA repair. Associated with increased carotid artery thickness upon exposure to ionizing radiations ([X](https://academic.oup.com/eurheartj/article/34/suppl_1/P3922/2862076)). XRCC3 polymorphism is associated with hypertension-induced left ventricular hypertrophy ([29626209](https://www.nature.com/articles/s41440-018-0038-0?platform=hootsuite)). Xrcc3 Induces Cisplatin Resistance by Stimulation of Rad51-Related Recombinational Repair, S-Phase Checkpoint Activation, and Reduced Apoptosis ([15843498](https://jpet.aspetjournals.org/content/314/2/495.abstract)). iEQTLs are associated with pulse rate, BMI, atrial fibrillation, heart failure and hypertension (gene eQTLs only with LDL levels).\n",
    "- ***ATXN3***: . Splice Isoforms of the Polyglutamine Disease Protein Ataxin-3 Exhibit Similar Enzymatic yet Different Aggregation Properties ([21060878](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2965175/)). A knockin mouse model of spinocerebellar ataxia type 3 exhibits prominent aggregate pathology and aberrant splicing of the disease gene transcript ([25320121](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4321438/)). ieQTLs are associated with HDL levels and pulse rate (no associations for gene-level).\n",
    "- ***RPL13***: ribosomal protein. candidate for congenital heart disease ([31625562](https://academic.oup.com/hmg/article/28/23/3954/5589181?casa_token=qwbNCOmtIvQAAAAA:oIHuTEm5rI_a5uKnV2mm_cyOrh-O3z316JvNWNi3uuLMd4Xum37s-RMDq8HmKW4wzLlkLvMJgVcZ)). ieQTLs and gene eQTLs are associated with blood pressure and QRS duration.\n",
    "- ***NUP88***: nucleoporin complex. Other NUPs are associated with heart disease ([PMC5027676](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027676/)). ieQTLs only are associated with LDL levels.\n",
    "- ***PCAT19***: intereseting because this is a lncRNA with oncogenic properties. eQTLs are associated with cardiac conduction disorders.\n",
    "- ***USP34***: Acts as an activator of the Wnt signaling pathway downstream of the beta-catenin destruction complex by deubiquitinating and stabilizing AXIN1 and AXIN2, leading to promote nuclear accumulation of AXIN1 and AXIN2 and positively regulate beta-catenin (CTNBB1)-mediated transcription (GeneCards). Associated with CHD ([29555671](https://www.ahajournals.org/doi/full/10.1161/circgen.117.001978)). eQTLs are associated with blood pressure.\n",
    "- ***NPR3***: protects cardiomyocytes from apoptosis ([PMC5026813](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5026813/)). ieQTLs are associated with triglycerides levels; gene eQTLs with blood pressure.\n",
    "- ***ITGA1***: Cardiac Fibroblasts Regulate Myocardial Proliferation through β1 Integrin Signaling ([19217425](https://www.sciencedirect.com/science/article/pii/S1534580708005170)). ieQTLs only are associated with unstable angina.\n",
    "- ***ATG10***: involved in autophagy ([PMC2838272](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2838272/)). ieQTLs only are associated with HDL, triglycerides levels and palpitations.\n",
    "- ***NOTCH4***: a lot of associations with heart functions. eQTLs are associated with blood pressure\n",
    "- ***TMEM63B***: OSCA/TMEM63 are an evolutionarily conserved family of mechanically activated ion channels ([30382938](https://elifesciences.org/articles/41844)). ieQTLs only are associated with Acute pulmonary heart disease; Pulmonary embolism and infarction, acute\n",
    "- ***B3GAT3***: catalyzes the formation of the glycosaminoglycan-protein linkage. Novel Splicing Mutation in B3GAT3 Associated with Short Stature, GH Deficiency, Hypoglycaemia, Developmental Delay, and Multiple Congenital Anomalies ([29318063](https://www.hindawi.com/journals/crig/2017/3941483/)). ieQTLs are associated with blood pressure\n",
    "- ***HP*** (Haptoglobin): associated with MI risk ([19657769](https://pubmed.ncbi.nlm.nih.gov/19657769/)) and cardiac disease in diabetes ([PMC5019011](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5019011/)). ieQTLs are associated with blood pressure, atherosclerosis, hypertension, pulse rate\n",
    "- ***GRAMD1A***: cholesterol transporter (GeneCards). ieQTLs are associated with HDL levels\n",
    "- ***TDGF1***: epdermal growth factor associated with NKX2-5 ([PMC3779917](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3779917/)) and MEF2C ([26811383](https://dev.biologists.org/content/143/5/774)) and involved in early cardiac development. ieQTLs are associated with atenolol prescription(beta blocker used to treat high blood pressure\n",
    "- ***NMRK1***: Nicotinamide adenine dinucleotide (NAD+) synthesis (GeneCards). ieQTLs are associated with MI\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "GENE = \n",
      "\n",
      "ISOFORM = I21 Acute myocardial infarction\n",
      "\n"
     ]
    }
   ],
   "source": [
    "gene = \"NMRK1\"\n",
    "message(paste(\"GENE\"   , paste(sort(unique(merge(coloc_gene   [ coloc_gene   $PP.H4.abf > 0.5 & coloc_gene   $transcript_id ==   geneinfo_gene   [geneinfo_gene   $gene_name == gene, \"transcript_id\"],], manifest[,c(\"id\", \"description\", \"coding_description\")], by.x = \"trait\", by.y = \"id\")[, \"description\"])), collapse = \"; \"), sep = \" = \"))\n",
    "message(paste(\"ISOFORM\", paste(sort(unique(merge(coloc_isoform[ coloc_isoform$PP.H4.abf > 0.5 & coloc_isoform$transcript_id %in% geneinfo_isoform[geneinfo_isoform$gene_name == gene, \"transcript_id\"],], manifest[,c(\"id\", \"description\", \"coding_description\")], by.x = \"trait\", by.y = \"id\")[, \"description\"])), collapse = \"; \"), sep = \" = \"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 4 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>id</th><th scope=col>trait_type</th><th scope=col>phenocode</th><th scope=col>description</th><th scope=col>description_more</th><th scope=col>coding_description</th><th scope=col>category</th><th scope=col>n_cases_full_cohort_both_sexes</th><th scope=col>n_controls_total</th><th scope=col>saige_heritability_EUR</th><th scope=col>filename</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>17</th><td><span style=white-space:pre-wrap>categorical-6150-both_sexes-3  </span></td><td>categorical</td><td>6150</td><td>Vascular/heart problems diagnosed by doctor</td><td><span style=white-space:pre-wrap>ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit.</span></td><td><span style=white-space:pre-wrap>Stroke             </span></td><td>UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions</td><td><span style=white-space:pre-wrap>  7332</span></td><td>428034</td><td>0.051107</td><td><span style=white-space:pre-wrap>/frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-3.txt.gz  </span></td></tr>\n",
       "\t<tr><th scope=row>18</th><td>categorical-6150-both_sexes-100</td><td>categorical</td><td>6150</td><td>Vascular/heart problems diagnosed by doctor</td><td><span style=white-space:pre-wrap>ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit.</span></td><td><span style=white-space:pre-wrap>None of the above  </span></td><td>UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions</td><td>338784</td><td>131602</td><td>0.076389</td><td>/frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-100.txt.gz</td></tr>\n",
       "\t<tr><th scope=row>19</th><td><span style=white-space:pre-wrap>categorical-6150-both_sexes-4  </span></td><td>categorical</td><td>6150</td><td>Vascular/heart problems diagnosed by doctor</td><td><span style=white-space:pre-wrap>ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit.</span></td><td>High blood pressure</td><td>UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions</td><td>130711</td><td>320160</td><td>0.079343</td><td><span style=white-space:pre-wrap>/frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-4.txt.gz  </span></td></tr>\n",
       "\t<tr><th scope=row>50</th><td><span style=white-space:pre-wrap>categorical-6150-both_sexes-2  </span></td><td>categorical</td><td>6150</td><td>Vascular/heart problems diagnosed by doctor</td><td><span style=white-space:pre-wrap>ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit.</span></td><td><span style=white-space:pre-wrap>Angina             </span></td><td>UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions</td><td> 15585</td><td>422159</td><td>0.098137</td><td><span style=white-space:pre-wrap>/frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-2.txt.gz  </span></td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & id & trait\\_type & phenocode & description & description\\_more & coding\\_description & category & n\\_cases\\_full\\_cohort\\_both\\_sexes & n\\_controls\\_total & saige\\_heritability\\_EUR & filename\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t17 & categorical-6150-both\\_sexes-3   & categorical & 6150 & Vascular/heart problems diagnosed by doctor & ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit. & Stroke              & UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions &   7332 & 428034 & 0.051107 & /frazer01/projects/CARDIPS/analysis/cardiac\\_eqtls/input/gwas/pan\\_ukbb/summary\\_statistics/categorical-6150-both\\_sexes-3.txt.gz  \\\\\n",
       "\t18 & categorical-6150-both\\_sexes-100 & categorical & 6150 & Vascular/heart problems diagnosed by doctor & ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit. & None of the above   & UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions & 338784 & 131602 & 0.076389 & /frazer01/projects/CARDIPS/analysis/cardiac\\_eqtls/input/gwas/pan\\_ukbb/summary\\_statistics/categorical-6150-both\\_sexes-100.txt.gz\\\\\n",
       "\t19 & categorical-6150-both\\_sexes-4   & categorical & 6150 & Vascular/heart problems diagnosed by doctor & ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit. & High blood pressure & UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions & 130711 & 320160 & 0.079343 & /frazer01/projects/CARDIPS/analysis/cardiac\\_eqtls/input/gwas/pan\\_ukbb/summary\\_statistics/categorical-6150-both\\_sexes-4.txt.gz  \\\\\n",
       "\t50 & categorical-6150-both\\_sexes-2   & categorical & 6150 & Vascular/heart problems diagnosed by doctor & ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit. & Angina              & UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions &  15585 & 422159 & 0.098137 & /frazer01/projects/CARDIPS/analysis/cardiac\\_eqtls/input/gwas/pan\\_ukbb/summary\\_statistics/categorical-6150-both\\_sexes-2.txt.gz  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 11\n",
       "\n",
       "| <!--/--> | id &lt;chr&gt; | trait_type &lt;chr&gt; | phenocode &lt;chr&gt; | description &lt;chr&gt; | description_more &lt;chr&gt; | coding_description &lt;chr&gt; | category &lt;chr&gt; | n_cases_full_cohort_both_sexes &lt;int&gt; | n_controls_total &lt;int&gt; | saige_heritability_EUR &lt;dbl&gt; | filename &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 17 | categorical-6150-both_sexes-3   | categorical | 6150 | Vascular/heart problems diagnosed by doctor | ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit. | Stroke              | UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions |   7332 | 428034 | 0.051107 | /frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-3.txt.gz   |\n",
       "| 18 | categorical-6150-both_sexes-100 | categorical | 6150 | Vascular/heart problems diagnosed by doctor | ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit. | None of the above   | UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions | 338784 | 131602 | 0.076389 | /frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-100.txt.gz |\n",
       "| 19 | categorical-6150-both_sexes-4   | categorical | 6150 | Vascular/heart problems diagnosed by doctor | ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit. | High blood pressure | UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions | 130711 | 320160 | 0.079343 | /frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-4.txt.gz   |\n",
       "| 50 | categorical-6150-both_sexes-2   | categorical | 6150 | Vascular/heart problems diagnosed by doctor | ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit. | Angina              | UK Biobank Assessment Centre &gt; Touchscreen &gt; Health and medical history &gt; Medical conditions |  15585 | 422159 | 0.098137 | /frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-2.txt.gz   |\n",
       "\n"
      ],
      "text/plain": [
       "   id                              trait_type  phenocode\n",
       "17 categorical-6150-both_sexes-3   categorical 6150     \n",
       "18 categorical-6150-both_sexes-100 categorical 6150     \n",
       "19 categorical-6150-both_sexes-4   categorical 6150     \n",
       "50 categorical-6150-both_sexes-2   categorical 6150     \n",
       "   description                                \n",
       "17 Vascular/heart problems diagnosed by doctor\n",
       "18 Vascular/heart problems diagnosed by doctor\n",
       "19 Vascular/heart problems diagnosed by doctor\n",
       "50 Vascular/heart problems diagnosed by doctor\n",
       "   description_more                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "17 ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit.\n",
       "18 ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit.\n",
       "19 ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit.\n",
       "50 ACE touchscreen question \"\"Has a doctor ever told you that you have had any of the following conditions? (You can select more than one answer)\"\" The following checks were performed:  If code -7 was selected, then no additional choices were allowed.  If code -3 was selected, then no additional choices were allowed.   If the participant activated the Help button they were shown the message:     If you do not know if you have had any of the listed conditions, enter None of the above. You can check this with an interviewer later in the visit.\n",
       "   coding_description \n",
       "17 Stroke             \n",
       "18 None of the above  \n",
       "19 High blood pressure\n",
       "50 Angina             \n",
       "   category                                                                                    \n",
       "17 UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions\n",
       "18 UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions\n",
       "19 UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions\n",
       "50 UK Biobank Assessment Centre > Touchscreen > Health and medical history > Medical conditions\n",
       "   n_cases_full_cohort_both_sexes n_controls_total saige_heritability_EUR\n",
       "17   7332                         428034           0.051107              \n",
       "18 338784                         131602           0.076389              \n",
       "19 130711                         320160           0.079343              \n",
       "50  15585                         422159           0.098137              \n",
       "   filename                                                                                                                       \n",
       "17 /frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-3.txt.gz  \n",
       "18 /frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-100.txt.gz\n",
       "19 /frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-4.txt.gz  \n",
       "50 /frazer01/projects/CARDIPS/analysis/cardiac_eqtls/input/gwas/pan_ukbb/summary_statistics/categorical-6150-both_sexes-2.txt.gz  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "manifest[grepl(\"categorical-6150-both_sexes\", manifest$id) == TRUE,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'data.frame':\t2011740 obs. of  17 variables:\n",
      " $ nsnps        : int  2817 2817 2817 2817 2815 2816 2563 2814 2817 2813 ...\n",
      " $ PP.H0.abf    : num  0.321 0.302 0.247 0.329 0.301 ...\n",
      " $ PP.H1.abf    : num  0.622 0.585 0.478 0.638 0.584 ...\n",
      " $ PP.H2.abf    : num  0.0167 0.0346 0.0834 0.0092 0.0336 ...\n",
      " $ PP.H3.abf    : num  0.0324 0.067 0.1616 0.0178 0.0651 ...\n",
      " $ PP.H4.abf    : num  0.00717 0.01221 0.03008 0.00592 0.0161 ...\n",
      " $ transcript_id: chr  \"ENSG00000000457.14_7\" \"ENSG00000000457.14_7\" \"ENSG00000000457.14_7\" \"ENSG00000000457.14_7\" ...\n",
      " $ type         : int  0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ id           : chr  \"VAR_1_169860528_C_A\" \"VAR_1_169860528_C_A\" \"VAR_1_169690313_ACT_A\" \"VAR_1_169852613_A_G\" ...\n",
      " $ chrom        : int  1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ pos          : int  169860528 169860528 169690313 169852613 169860528 169860528 169860528 169860528 169747630 169860528 ...\n",
      " $ ref          : chr  \"C\" \"C\" \"ACT\" \"A\" ...\n",
      " $ alt          : chr  \"A\" \"A\" \"A\" \"G\" ...\n",
      " $ af           : num  0.13 0.13 0.147 0.13 0.13 0.13 0.13 0.13 0.115 0.13 ...\n",
      " $ pp_snp       : num  0.0813 0.061 0.0341 0.0812 0.0813 ...\n",
      " $ trait        : chr  \"prescriptions-lisinopril-both_sexes\" \"prescriptions-candesartan-both_sexes\" \"phecode-401.1-both_sexes\" \"phecode-278-both_sexes\" ...\n",
      " $ tr2type      : chr  \"ENSG00000000457.14_7 0\" \"ENSG00000000457.14_7 0\" \"ENSG00000000457.14_7 0\" \"ENSG00000000457.14_7 0\" ...\n"
     ]
    }
   ],
   "source": [
    "str(coloc_gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.0.1",
   "language": "R",
   "name": "ir401"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
