{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Meta-analysis of GWAS traits\n",
    "- for each signal at each locus, perform meta-analysis to get one single GWAS signal (which is equal to the combined signal of all colocalizing traits)\n",
    "- colocalize this meta-signal with eQTLs (only for eGenes that already colocalize with one of the individual GWAS signals, PPH4 > 0.8)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/frazer01/projects/CARDIPS/analysis/cardiac_gwas_coloc\")\n",
    "\n",
    "source(\"script/functions.R\"  )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "dir.create(\"pipeline/5.5.meta_analysis\"           , showWarnings = FALSE)\n",
    "dir.create(\"pipeline/5.5.meta_analysis/coloc_data\", showWarnings = FALSE)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get GWAS data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "suppressPackageStartupMessages(library(kohonen))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "manifest   = add_rownames(fread(\"pipeline/1.1.sumstats/manifest.txt\"                     , sep = \"\\t\", header = TRUE, data.table = FALSE))\n",
    "loci       = add_rownames(fread(\"pipeline/1.2.genomewide_significant_loci/loci.txt\"      , sep = \"\\t\", header = TRUE, data.table = FALSE))\n",
    "loci2study = add_rownames(fread(\"pipeline/1.2.genomewide_significant_loci/loci2study.txt\", sep = \"\\t\", header = TRUE, data.table = FALSE))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "populations = c('meta','AFR','AMR','CSA','EAS','EUR','MID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "moloc_df       = fread  (\"pipeline/2.2.moloc/moloc.txt\", sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "moloc_list     = readRDS(\"pipeline/2.2.moloc/moloc_list.rds\")\n",
    "moloc_map_list = readRDS(\"pipeline/2.2.moloc/moloc_map_list.rds\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "moloc_loci_groups = moloc_map_list[[\"som\"]][[\"input\"]]\n",
    "moloc_loci2class  = moloc_map_list[[\"loci2class\"]]\n",
    "moloc_trait2class = moloc_map_list[[\"trait2class\"]]\n",
    "\n",
    "moloc_loci2class$class = paste0(\"V\", moloc_loci2class$som)\n",
    "\n",
    "rownames(moloc_loci2class) = moloc_loci2class$id"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get eQTL coloc data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "qtl_list = readRDS(\"/frazer01/projects/CARDIPS/analysis/cardiac_qtls_combined/input/qtl/qtls.RDS\"   )\n",
    "exp_list = readRDS(\"/frazer01/projects/CARDIPS/analysis/cardiac_qtls_combined/input/expdata_qtl.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "phenotypes  = qtl_list$phenotypes\n",
    "qtls        = qtl_list$qtl\n",
    "coordinates = exp_list$coordinates\n",
    "rownames(coordinates) = coordinates$transcript_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "coloc_qtls = fread(\"pipeline/4.1.coloc_qtls/coloc_eqtls.txt\", sep = \"\\t\", header = TRUE, data.table = FALSE)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data for meta analysis\n",
    "- https://bookdown.org/MathiasHarrer/Doing_Meta_Analysis_in_R/fixed.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "suppressPackageStartupMessages(library(meta))\n",
    "suppressPackageStartupMessages(library(coloc))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "coloc_list        = readRDS(\"pipeline/5.4.analyze_coloc_qtl_som_maps/coloc_list.rds\")\n",
    "locus2coloc       = fread  (\"pipeline/5.4.analyze_coloc_qtl_som_maps//locus2coloc.txt\", sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "intersected       = fread  (\"pipeline/4.1.coloc_qtls/intersected_qtls_loci.txt\"       , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "qtls2test         = fread  (\"pipeline/4.1.coloc_qtls/qtls2test.txt\"                   , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "intersected       = intersected[ intersected$phenotype %in% c(\"rna\", \"isoform\"),]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create qsub for each locus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "run_qsub = function(id)\n",
    "{\n",
    "    qsub = paste(\"qsub\", paste(getwd(), \"script\", \"5.5.coloc_qtl_meta_analysis.sh\", sep = \"/\"), id)\n",
    "    \n",
    "    system(qsub)\n",
    "    \n",
    "    #message(qsub)\n",
    "}\n",
    "\n",
    "invisible(lapply(locus2coloc$id, run_qsub))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check what went wrong"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 0 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>som</th><th scope=col>id</th><th scope=col>locus</th><th scope=col>cluster</th><th scope=col>class</th><th scope=col>traits</th><th scope=col>traits_n</th><th scope=col>eqtls</th><th scope=col>eqtls_n</th><th scope=col>file</th><th scope=col>exists</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " som & id & locus & cluster & class & traits & traits\\_n & eqtls & eqtls\\_n & file & exists\\\\\n",
       " <int> & <chr> & <chr> & <int> & <chr> & <chr> & <int> & <chr> & <int> & <chr> & <lgl>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 11\n",
       "\n",
       "| som &lt;int&gt; | id &lt;chr&gt; | locus &lt;chr&gt; | cluster &lt;int&gt; | class &lt;chr&gt; | traits &lt;chr&gt; | traits_n &lt;int&gt; | eqtls &lt;chr&gt; | eqtls_n &lt;int&gt; | file &lt;chr&gt; | exists &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     som id locus cluster class traits traits_n eqtls eqtls_n file exists"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "locus2file        = locus2coloc\n",
    "locus2file$file   = paste0(\"pipeline/5.5.meta_analysis/coloc_data/\", locus2file$id, \".rds\")\n",
    "locus2file$exists = file.exists(locus2file$file)\n",
    "\n",
    "locus2file[ locus2file$exists == FALSE, ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SCRATCH - Meta-analysis of GWAS traits\n",
    "- https://bookdown.org/MathiasHarrer/Doing_Meta_Analysis_in_R/fixed.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "suppressPackageStartupMessages(library(meta))\n",
    "suppressPackageStartupMessages(library(coloc))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "coloc_list        = readRDS(\"pipeline/5.4.analyze_coloc_qtl_som_maps/coloc_list.rds\")\n",
    "locus2coloc       = fread  (\"pipeline/5.4.analyze_coloc_qtl_som_maps//locus2coloc.txt\", sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "intersected       = fread  (\"pipeline/4.1.coloc_qtls/intersected_qtls_loci.txt\"       , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "qtls2test         = fread  (\"pipeline/4.1.coloc_qtls/qtls2test.txt\"                   , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "intersected       = intersected[ intersected$phenotype %in% c(\"rna\", \"isoform\"),]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_gwas_data = function(coord, gwas_file)\n",
    "{\n",
    "    my_head              = colnames(fread(cmd = paste(\"zcat\", gwas_file, \"|\", \"head -n 2\"), sep = \"\\t\", header = TRUE, data.table = FALSE))\n",
    "    gwas_data            = suppressWarnings(tabix.read.table(gwas_file, coord, col.names = TRUE, stringsAsFactors = FALSE))\n",
    "    colnames(gwas_data)  = my_head\n",
    "    gwas_data$id         = paste(\"VAR\", gwas_data$chr, gwas_data$pos, gwas_data$ref, gwas_data$alt, sep = \"_\")\n",
    "    rownames(gwas_data)  = gwas_data$id\n",
    "    \n",
    "    if(\"af_meta\"          %in% colnames(gwas_data)){gwas_data$af = gwas_data$af_meta         }\n",
    "    if(\"af_controls_meta\" %in% colnames(gwas_data)){gwas_data$af = gwas_data$af_controls_meta}\n",
    "    \n",
    "    return(gwas_data)\n",
    "}\n",
    "\n",
    "get_qtl_data = function(qtl_file)\n",
    "{\n",
    "    indata = fread(qtl_file, sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "    return(indata)\n",
    "}\n",
    "\n",
    "calculate_meta = function(id, traits_cluster, tometa_te, tometa_se, tometa_pval, tometa_af)\n",
    "{\n",
    "    totest = data.frame(trait = traits_cluster, \n",
    "                        TE    = as.numeric(tometa_te  [id, traits_cluster]),\n",
    "                        seTE  = as.numeric(tometa_se  [id, traits_cluster]),\n",
    "                        pval  = as.numeric(tometa_pval[id, traits_cluster])\n",
    "                       )\n",
    "    totest = totest[is.na(totest$TE) == FALSE,]\n",
    "    \n",
    "    if(nrow(totest) > 1)\n",
    "    {\n",
    "        m = metagen(TE          = TE,\n",
    "                    seTE        = seTE,\n",
    "                    data        = totest,\n",
    "                    studlab     = paste(trait),\n",
    "                    comb.fixed  = TRUE,\n",
    "                    comb.random = FALSE,\n",
    "                    prediction  = TRUE,\n",
    "                    sm=\"SMD\")\n",
    "        \n",
    "        out = data.frame(id       = id,\n",
    "                         beta     = m$TE.fixed  ,  \n",
    "                         se       = m$seTE.fixed,  \n",
    "                         pval     = m$pval.fixed,\n",
    "                         af       = mean(as.numeric(tometa_af[id,]), na.rm = TRUE),\n",
    "                         pval_het = m$pval.Q\n",
    "                        )\n",
    "    }else\n",
    "    {\n",
    "        out = data.frame(id       = id,\n",
    "                         beta     = totest$TE  ,  \n",
    "                         se       = totest$seTE,  \n",
    "                         pval     = totest$pval,\n",
    "                         af       = mean(as.numeric(tometa_af[id,]), na.rm = TRUE),\n",
    "                         pval_het = 1\n",
    "                        )\n",
    "    }\n",
    "    \n",
    "    if(nrow(out[is.na(out$pval_het) == TRUE, ]) > 0){out[is.na(out$pval_het) == TRUE, \"pval_het\"] = 1}\n",
    "    \n",
    "    return(out)\n",
    "}\n",
    "\n",
    "create_dataset = function(study, trait_type, totest, variants, populations, manifest)\n",
    "{\n",
    "    if(trait_type %in% c(\"categorical\", \"icd10\", \"phecode\"))\n",
    "    {\n",
    "        n = sum(manifest[study, paste(\"n_cases\", populations, sep = \"_\")]) + sum(manifest[study, paste(\"n_controls\", populations, sep = \"_\")])\n",
    "    }\n",
    "    if(trait_type %in% c(\"biomarkers\", \"continuous\"))\n",
    "    {\n",
    "        n = sum(manifest[study, paste(\"n_cases\", populations, sep = \"_\")])\n",
    "    }\n",
    "    if(trait_type == \"qtl\")\n",
    "    {\n",
    "        n = 966\n",
    "    }\n",
    "\n",
    "    rownames(totest) = totest$id\n",
    "    dataset = list(snp     = variants, \n",
    "                   pvalues = totest     [variants, \"pval\"], \n",
    "                   N       = n, \n",
    "                   MAF     = totest     [variants, \"af\"], \n",
    "                   type    = \"quant\")\n",
    "    \n",
    "    return(dataset)\n",
    "}\n",
    "\n",
    "finemap_gwas = function(dataset)\n",
    "{\n",
    "    if(length(dataset$pvalues) > 10000)\n",
    "    {\n",
    "        new_dataset = data.frame(pval = dataset$pvalues, maf = dataset$MAF, snp = dataset$snp)\n",
    "        new_dataset = new_dataset[order(new_dataset$pval),]\n",
    "        new_dataset = new_dataset[1:10000,]\n",
    "        dataset     = list(pvalues = new_dataset$pval, MAF = new_dataset$maf, snp = new_dataset$snp, N = dataset$N, sdY = 1, type = dataset$type)\n",
    "    }\n",
    "    \n",
    "    finemap           = finemap.abf(dataset = dataset)\n",
    "    finemap           = finemap[,c(\"snp\", \"pvalues.\", \"SNP.PP\")]\n",
    "    colnames(finemap) = c(\"id\", \"pval\", \"pp_snp\")\n",
    "    myres             = finemap\n",
    "    myres             = myres[order(myres$pp_snp, decreasing = TRUE), ]\n",
    "    myres$cum         = cumsum(myres$pp_snp)\n",
    "    myres$cs          = FALSE\n",
    "    to_cs             = myres[myres$cum >= 0.99, ][1, \"id\"]\n",
    "    \n",
    "    myres[myres$cum < 0.99 | myres$id == to_cs, \"cs\"] = TRUE\n",
    "\n",
    "    credible_set = myres[ myres$cs == TRUE, \"id\"]\n",
    "    \n",
    "    return(list(pp = myres, credible_set = credible_set))\n",
    "}\n",
    "\n",
    "quiet = function(x) \n",
    "{ \n",
    "    sink(tempfile()) \n",
    "    on.exit(sink()) \n",
    "    invisible(force(x)) \n",
    "} \n",
    "\n",
    "run_coloc = function(id, study1, study2, populations, totest1, totest2, qtls2test, manifest)\n",
    "{\n",
    "    trait_type1 = manifest[study1, \"trait_type\"]\n",
    "    trait_type2 = \"qtl\"\n",
    "    variants1   = totest1[is.na(totest1$pval) == FALSE & is.na(totest1$af) == FALSE, \"id\"]\n",
    "    types       = sort  (qtls2test[qtls2test$transcript_id == study2, \"type\"     ])\n",
    "    phenotype   = unique(qtls2test[qtls2test$transcript_id == study2, \"phenotype\"])\n",
    "    \n",
    "    out = lapply(types, function(type)\n",
    "    {\n",
    "        totest_type = totest2[totest2$type == type,]\n",
    "        variants2   = totest_type$id\n",
    "        variants    = intersect(variants1, variants2)\n",
    "\n",
    "        if(length(variants) > 100)\n",
    "        {\n",
    "            dataset1        = create_dataset(study1, trait_type1, totest1    , variants, populations, manifest)\n",
    "            dataset2        = create_dataset(study2, trait_type2, totest_type, variants, c()        , manifest)\n",
    "            coloc_mapped    = coloc.abf(dataset1 = dataset1, dataset2 = dataset2) \n",
    "            probs           = as.data.frame(t(coloc_mapped$summary))\n",
    "            myres           = coloc_mapped$results\n",
    "            myres           = myres[, c(which(colnames(myres) == \"snp\"), ncol(myres))]\n",
    "            colnames(myres) = c(\"id\", \"pp_snp\")\n",
    "            myres           = myres[order(myres$pp_snp, decreasing = TRUE), ]\n",
    "            out             = cbind(probs, myres[1, ])\n",
    "            \n",
    "            out = list(by_snp = myres, pp = out)\n",
    "        }else\n",
    "        {\n",
    "            out = list(by_snp = data.frame(id = \"\", pp_snp = 0),\n",
    "                       pp     = data.frame(nsnps = 0, PP.H0.abf = 1, PP.H1.abf = 0, PP.H2.abf = 0, PP.H3.abf = 0, PP.H4.abf = 0, id = \"\", pp_snp = 0)\n",
    "                      )\n",
    "        }\n",
    "\n",
    "        return(out)\n",
    "    })\n",
    "    \n",
    "    names(out) = paste(\"type\", types)\n",
    "    return(out)\n",
    "}\n",
    "\n",
    "\n",
    "meta_analysis = function(id, moloc_loci_groups, coloc_list, loci, manifest, intersected, qtls2test, to_return = FALSE)\n",
    "{\n",
    "    som            = coloc_list[[id]][[\"som\"           ]]\n",
    "    eqtls          = coloc_list[[id]][[\"eqtls\"         ]]\n",
    "    traits         = coloc_list[[id]][[\"traits\"        ]]\n",
    "    coloc          = coloc_list[[id]][[\"coloc\"         ]]   \n",
    "    traits_cluster = data.frame(trait = colnames(moloc_loci_groups), x = as.numeric(moloc_loci_groups[id,]))\n",
    "    traits_cluster = traits_cluster[ traits_cluster$x == 1, \"trait\"]\n",
    "    \n",
    "    locus         = som[1, \"locus\"]\n",
    "    coloc         = coloc[ coloc$PP.H4.abf >= 0.8,]\n",
    "    coloc         = coloc[order(coloc$PP.H4.abf, decreasing = TRUE),]\n",
    "    coloc$tr2type = paste(coloc$transcript_id, coloc$type)\n",
    "    coord         = loci[locus,]\n",
    "    coord         = paste0(coord$chrom, \":\", coord$from, \"-\", coord$to)\n",
    "    \n",
    "    # get GWAS data\n",
    "    totest_gwas_list        = lapply(traits_cluster, function(gwas){get_gwas_data(coord, manifest[gwas, \"sumstat_file\"])})\n",
    "    names(totest_gwas_list) = traits_cluster\n",
    "    \n",
    "    if(length(totest_gwas_list) > 1)\n",
    "    {\n",
    "        gwas = totest_gwas_list[[1]]\n",
    "        gwas = data.frame(id       = gwas$id,\n",
    "                          beta     = gwas$beta_meta,\n",
    "                          se       = gwas$se_meta  ,\n",
    "                          pval     = gwas$pval_meta,\n",
    "                          af       = gwas$af,\n",
    "                          pval_het = 1\n",
    "                         )\n",
    "\n",
    "        var_ids     = sort(unique(unlist(lapply(totest_gwas_list, function(x){x$id}))))\n",
    "        tometa      = as.data.frame(matrix(0, nrow = length(var_ids), ncol = length(totest_gwas_list), dimnames = list(var_ids, traits_cluster)))\n",
    "        tometa_te   = tometa\n",
    "        tometa_se   = tometa\n",
    "        tometa_pval = tometa\n",
    "        tometa_af   = tometa\n",
    "        \n",
    "        for(trait in traits_cluster)\n",
    "        {\n",
    "            x = totest_gwas_list[[trait]]\n",
    "            tometa_te  [ x$id, trait] = x$beta_meta\n",
    "            tometa_se  [ x$id, trait] = x$se_meta\n",
    "            tometa_pval[ x$id, trait] = x$pval_meta\n",
    "            tometa_af  [ x$id, trait] = x$af\n",
    "        }\n",
    "        \n",
    "        gwas = suppressWarnings(as.data.frame(rbindlist(lapply(var_ids, function(id){calculate_meta(id, traits_cluster, tometa_te, tometa_se, tometa_pval, tometa_af)}))))\n",
    "    }else\n",
    "    {\n",
    "        gwas = totest_gwas_list[[1]]\n",
    "        gwas = data.frame(id       = gwas$id,\n",
    "                          beta     = gwas$beta_meta,\n",
    "                          se       = gwas$se_meta  ,\n",
    "                          pval     = gwas$pval_meta,\n",
    "                          af       = gwas$af,\n",
    "                          pval_het = 1\n",
    "                         )\n",
    "    }\n",
    "    \n",
    "    rownames(gwas)    = gwas$id\n",
    "    trait             = traits_cluster[[1]]\n",
    "    tocoloc           = intersected[ intersected$locus == locus, ]\n",
    "    trait_type1       = manifest[trait, \"trait_type\"]\n",
    "    trait_type2       = \"qtl\"\n",
    "    populations       = unlist(strsplit(manifest[trait, \"pops\"], \",\"))\n",
    "    totest_qtl        = lapply(tocoloc$qtl_file, get_qtl_data)\n",
    "    names(totest_qtl) = tocoloc$transcript_id\n",
    "    qtls2test         = qtls2test[ qtls2test$transcript_id %in% tocoloc$transcript_id,]\n",
    "    dataset           = create_dataset(trait, \n",
    "                                       manifest[trait, \"trait_type\"], \n",
    "                                       gwas, \n",
    "                                       gwas[is.na(gwas$pval) == FALSE & is.na(gwas$af) == FALSE, \"id\"], \n",
    "                                       populations, \n",
    "                                       manifest)\n",
    "    \n",
    "    finemapped = finemap_gwas(dataset)\n",
    "\n",
    "    #transcript_id = names(totest_qtl)[[1]]\n",
    "    #out           = run_coloc(id, trait, transcript_id, populations, gwas, totest_qtl[[transcript_id]], qtls2test, manifest)\n",
    "    \n",
    "    coloc_list = quiet(lapply(names(totest_qtl), function(transcript_id)\n",
    "    {\n",
    "        run_coloc(id, trait, transcript_id, populations, gwas, totest_qtl[[transcript_id]], qtls2test, manifest)\n",
    "    }))\n",
    "    \n",
    "    names(coloc_list) = names(totest_qtl)\n",
    "    \n",
    "    out = list(som                 = som,\n",
    "               traits_cluster      = traits_cluster   ,\n",
    "               eqtls               = qtls2test        ,\n",
    "               eqtl_transcript_ids = names(totest_qtl),\n",
    "               eqtl_data           = totest_qtl       ,\n",
    "               gwas_data           = totest_gwas_list ,\n",
    "               meta_analysis_gwas  = gwas             ,\n",
    "               finemapped_gwas     = finemapped       ,\n",
    "               coloc               = coloc_list\n",
    "              )\n",
    "    \n",
    "    saveRDS(out, paste0(\"pipeline/5.5.meta_analysis/coloc_data/\", id, \".rds\"))\n",
    "    \n",
    "    if(to_return == TRUE){return(out)}\n",
    "}\n",
    "\n",
    "id = \"5_136864391_137972681.2\" # atrial fibrillation + pulse rate\n",
    "id = \"7_19692053_19892053.1\" \n",
    "\n",
    "x = meta_analysis(id, moloc_loci_groups, coloc_list, loci, manifest, intersected, qtls2test, TRUE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "List of 9\n",
      " $ som                :'data.frame':\t1 obs. of  5 variables:\n",
      "  ..$ som    : num 48\n",
      "  ..$ id     : chr \"7_19692053_19892053.1\"\n",
      "  ..$ locus  : chr \"7_19692053_19892053\"\n",
      "  ..$ cluster: int 14\n",
      "  ..$ class  : chr \"V48\"\n",
      " $ traits_cluster     : chr \"categorical-20003-both_sexes-1140866738\"\n",
      " $ eqtls              :'data.frame':\t1 obs. of  8 variables:\n",
      "  ..$ gene_id      : chr \"ENSG00000105849.6_5\"\n",
      "  ..$ gene_name    : chr \"TWISTNB\"\n",
      "  ..$ transcript_id: chr \"ENSG00000105849.6_5\"\n",
      "  ..$ type         : int 0\n",
      "  ..$ chrom        : int 7\n",
      "  ..$ start        : int 19735084\n",
      "  ..$ end          : int 19748660\n",
      "  ..$ phenotype    : chr \"rna\"\n",
      " $ eqtl_transcript_ids: chr \"ENSG00000105849.6_5\"\n",
      " $ eqtl_data          :List of 1\n",
      "  ..$ ENSG00000105849.6_5:'data.frame':\t5216 obs. of  13 variables:\n",
      "  .. ..$ chrom     : int [1:5216] 7 7 7 7 7 7 7 7 7 7 ...\n",
      "  .. ..$ pos       : int [1:5216] 19240546 19241659 19241735 19241953 19241980 19247180 19247377 19248160 19249169 19249778 ...\n",
      "  .. ..$ ref       : chr [1:5216] \"T\" \"C\" \"A\" \"G\" ...\n",
      "  .. ..$ alt       : chr [1:5216] \"G\" \"T\" \"C\" \"A\" ...\n",
      "  .. ..$ rsid      : chr [1:5216] \"rs724810\" \"rs7779453\" \"rs7779362\" \"rs1544373\" ...\n",
      "  .. ..$ id        : chr [1:5216] \"VAR_7_19240546_T_G\" \"VAR_7_19241659_C_T\" \"VAR_7_19241735_A_C\" \"VAR_7_19241953_G_A\" ...\n",
      "  .. ..$ af        : num [1:5216] 0.079 0.413 0.707 0.07 0.074 0.074 0.074 0.07 0.072 0.637 ...\n",
      "  .. ..$ beta      : num [1:5216] -0.0886 0.0598 0.0551 0.0406 -0.0885 ...\n",
      "  .. ..$ se        : num [1:5216] 0.078 0.0342 0.0349 0.0571 0.0782 ...\n",
      "  .. ..$ pval      : num [1:5216] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ gene_id   : chr [1:5216] \"ENSG00000105849.6_5\" \"ENSG00000105849.6_5\" \"ENSG00000105849.6_5\" \"ENSG00000105849.6_5\" ...\n",
      "  .. ..$ bonferroni: num [1:5216] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type      : int [1:5216] 0 0 0 0 0 0 0 0 0 0 ...\n",
      " $ gwas_data          :List of 1\n",
      "  ..$ categorical-20003-both_sexes-1140866738:'data.frame':\t908 obs. of  42 variables:\n",
      "  .. ..$ chr               : int [1:908] 7 7 7 7 7 7 7 7 7 7 ...\n",
      "  .. ..$ pos               : int [1:908] 19692124 19692174 19692670 19692863 19693464 19693604 19693626 19693755 19693996 19694131 ...\n",
      "  .. ..$ ref               : chr [1:908] \"C\" \"C\" \"C\" \"C\" ...\n",
      "  .. ..$ alt               : chr [1:908] \"T\" \"T\" \"G\" \"G\" ...\n",
      "  .. ..$ af_cases_meta     : num [1:908] 0.3049 0.3054 0.0176 0.305 0.0153 ...\n",
      "  .. ..$ af_controls_meta  : num [1:908] 0.3062 0.3063 0.0177 0.3063 0.015 ...\n",
      "  .. ..$ beta_meta         : num [1:908] -0.00628 -0.00477 0.00188 -0.00711 0.02807 ...\n",
      "  .. ..$ se_meta           : num [1:908] 0.0124 0.0124 0.0452 0.0124 0.0475 ...\n",
      "  .. ..$ pval_meta         : num [1:908] 0.614 0.702 0.967 0.568 0.554 ...\n",
      "  .. ..$ pval_heterogeneity: num [1:908] 0.7581 0.742 0.0034 0.7672 0.7557 ...\n",
      "  .. ..$ af_cases_AFR      : num [1:908] 6.97e-01 6.96e-01 8.64e-06 6.96e-01 5.50e-02 ...\n",
      "  .. ..$ af_cases_CSA      : num [1:908] 0.3739 0.3724 0.0104 0.3735 0.041 ...\n",
      "  .. ..$ af_cases_EAS      : num [1:908] 0.69 0.69 NA 0.69 0.069 ...\n",
      "  .. ..$ af_cases_EUR      : num [1:908] 0.2945 0.2951 0.0178 0.2946 0.0138 ...\n",
      "  .. ..$ af_cases_MID      : num [1:908] 0.39 0.39 NA 0.388 0 ...\n",
      "  .. ..$ af_controls_AFR   : num [1:908] 0.7124 0.7117 0.00142 0.7116 0.05294 ...\n",
      "  .. ..$ af_controls_CSA   : num [1:908] 0.3702 0.3704 0.00436 0.3682 0.0342 ...\n",
      "  .. ..$ af_controls_EAS   : num [1:908] 0.7052 0.7054 NA 0.6975 0.0522 ...\n",
      "  .. ..$ af_controls_EUR   : num [1:908] 0.2955 0.2956 0.018 0.2958 0.0138 ...\n",
      "  .. ..$ af_controls_MID   : num [1:908] 0.4074 0.4082 NA 0.406 0.00944 ...\n",
      "  .. ..$ beta_AFR          : num [1:908] -0.1454 -0.148 -1.219 -0.1492 0.0657 ...\n",
      "  .. ..$ beta_CSA          : num [1:908] 0.00402 -0.00264 2.361 0.00737 0.1499 ...\n",
      "  .. ..$ beta_EAS          : num [1:908] -0.1174 -0.1155 NA -0.0868 0.4129 ...\n",
      "  .. ..$ beta_EUR          : num [1:908] -0.00432 -0.00255 -0.00554 -0.00531 0.0144 ...\n",
      "  .. ..$ beta_MID          : num [1:908] -0.00301 -0.0035 NA -0.0032 -1.291 ...\n",
      "  .. ..$ se_AFR            : num [1:908] 0.111 0.111 1.552 0.111 0.217 ...\n",
      "  .. ..$ se_CSA            : num [1:908] 0.0824 0.0823 0.8067 0.0828 0.216 ...\n",
      "  .. ..$ se_EAS            : num [1:908] 0.214 0.214 NA 0.212 0.439 ...\n",
      "  .. ..$ se_EUR            : num [1:908] 0.0127 0.0127 0.0452 0.0127 0.0503 ...\n",
      "  .. ..$ se_MID            : num [1:908] 0.197 0.197 NA 0.197 0.995 ...\n",
      "  .. ..$ pval_AFR          : num [1:908] 0.19 0.182 0.432 0.179 0.762 ...\n",
      "  .. ..$ pval_CSA          : num [1:908] 0.9611 0.9744 0.00342 0.929 0.4877 ...\n",
      "  .. ..$ pval_EAS          : num [1:908] 0.584 0.589 NA 0.683 0.347 ...\n",
      "  .. ..$ pval_EUR          : num [1:908] 0.734 0.841 0.902 0.676 0.775 ...\n",
      "  .. ..$ pval_MID          : num [1:908] 0.988 0.986 NA 0.987 0.195 ...\n",
      "  .. ..$ low_confidence_AFR: logi [1:908] FALSE FALSE TRUE FALSE FALSE FALSE ...\n",
      "  .. ..$ low_confidence_CSA: logi [1:908] FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      "  .. ..$ low_confidence_EAS: logi [1:908] FALSE FALSE NA FALSE FALSE FALSE ...\n",
      "  .. ..$ low_confidence_EUR: logi [1:908] FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      "  .. ..$ low_confidence_MID: chr [1:908] \"FALSE\" \"FALSE\" \"7\" \"FALSE\" ...\n",
      "  .. ..$ id                : chr [1:908] \"VAR_7_19692124_C_T\" \"VAR_7_19692174_C_T\" \"VAR_7_19692670_C_G\" \"VAR_7_19692863_C_G\" ...\n",
      "  .. ..$ af                : num [1:908] 0.3062 0.3063 0.0177 0.3063 0.015 ...\n",
      " $ meta_analysis_gwas :'data.frame':\t908 obs. of  6 variables:\n",
      "  ..$ id      : chr [1:908] \"VAR_7_19692124_C_T\" \"VAR_7_19692174_C_T\" \"VAR_7_19692670_C_G\" \"VAR_7_19692863_C_G\" ...\n",
      "  ..$ beta    : num [1:908] -0.00628 -0.00477 0.00188 -0.00711 0.02807 ...\n",
      "  ..$ se      : num [1:908] 0.0124 0.0124 0.0452 0.0124 0.0475 ...\n",
      "  ..$ pval    : num [1:908] 0.614 0.702 0.967 0.568 0.554 ...\n",
      "  ..$ af      : num [1:908] 0.3062 0.3063 0.0177 0.3063 0.015 ...\n",
      "  ..$ pval_het: num [1:908] 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ finemapped_gwas    :List of 2\n",
      "  ..$ pp          :'data.frame':\t909 obs. of  5 variables:\n",
      "  .. ..$ id    : chr [1:909] \"VAR_7_19792053_A_T\" \"VAR_7_19778888_C_G\" \"VAR_7_19808196_A_G\" \"VAR_7_19808497_T_C\" ...\n",
      "  .. ..$ pval  : num [1:909] 3.34e-08 1.26e-07 4.64e-07 4.99e-07 NA ...\n",
      "  .. ..$ pp_snp: num [1:909] 0.4887 0.1171 0.0325 0.0303 0.025 ...\n",
      "  .. ..$ cum   : num [1:909] 0.489 0.606 0.638 0.669 0.694 ...\n",
      "  .. ..$ cs    : logi [1:909] TRUE TRUE TRUE TRUE TRUE TRUE ...\n",
      "  ..$ credible_set: chr [1:47] \"VAR_7_19792053_A_T\" \"VAR_7_19778888_C_G\" \"VAR_7_19808196_A_G\" \"VAR_7_19808497_T_C\" ...\n",
      " $ coloc              :List of 1\n",
      "  ..$ ENSG00000105849.6_5:List of 1\n",
      "  .. ..$ type 0:List of 2\n",
      "  .. .. ..$ by_snp:'data.frame':\t475 obs. of  2 variables:\n",
      "  .. .. .. ..$ id    : chr [1:475] \"VAR_7_19801016_T_C\" \"VAR_7_19750541_A_G\" \"VAR_7_19757519_G_T\" \"VAR_7_19753807_G_A\" ...\n",
      "  .. .. .. ..$ pp_snp: num [1:475] 0.969501 0.018428 0.00255 0.000679 0.00042 ...\n",
      "  .. .. ..$ pp    :'data.frame':\t1 obs. of  8 variables:\n",
      "  .. .. .. ..$ nsnps    : num 475\n",
      "  .. .. .. ..$ PP.H0.abf: num 0.655\n",
      "  .. .. .. ..$ PP.H1.abf: num 0.205\n",
      "  .. .. .. ..$ PP.H2.abf: num 0.0979\n",
      "  .. .. .. ..$ PP.H3.abf: num 0.0307\n",
      "  .. .. .. ..$ PP.H4.abf: num 0.0108\n",
      "  .. .. .. ..$ id       : chr \"VAR_7_19801016_T_C\"\n",
      "  .. .. .. ..$ pp_snp   : num 0.97\n"
     ]
    }
   ],
   "source": [
    "str(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 909 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>id</th><th scope=col>pval</th><th scope=col>pp_snp</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>VAR_7_19692124_C_T</td><td>0.6139000</td><td>4.811418e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19692174_C_T</td><td>0.7018000</td><td>4.558298e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19692670_C_G</td><td>0.9668000</td><td>1.481438e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19692863_C_G</td><td>0.5683000</td><td>4.984796e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19693464_T_C</td><td>0.5545000</td><td>1.909496e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19693604_A_G</td><td>0.4721000</td><td>2.106482e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19693626_C_T</td><td>0.0007141</td><td>3.779624e-05</td></tr>\n",
       "\t<tr><td>VAR_7_19693755_C_T</td><td>0.0009416</td><td>2.936939e-05</td></tr>\n",
       "\t<tr><td>VAR_7_19693996_G_A</td><td>0.0006517</td><td>4.135361e-05</td></tr>\n",
       "\t<tr><td>VAR_7_19694131_T_G</td><td>0.0716300</td><td>3.663994e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19694196_G_T</td><td>0.0006526</td><td>4.130068e-05</td></tr>\n",
       "\t<tr><td>VAR_7_19694243_T_C</td><td>0.4625000</td><td>2.131246e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19694376_C_T</td><td>0.0961600</td><td>2.893296e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19694384_C_T</td><td>0.0006522</td><td>4.132419e-05</td></tr>\n",
       "\t<tr><td>VAR_7_19694393_G_T</td><td>0.4491000</td><td>2.168130e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19694632_G_A</td><td>0.2062000</td><td>1.493414e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19694720_C_A</td><td>0.2081000</td><td>1.483512e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19694853_G_C</td><td>0.6086000</td><td>4.885853e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19694917_C_A</td><td>0.6236000</td><td>4.833438e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19695194_T_C</td><td>0.5374000</td><td>5.084969e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19695335_C_A</td><td>0.4492000</td><td>2.167857e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19695613_C_G</td><td>0.4494000</td><td>2.166581e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19695617_A_C</td><td>0.4653000</td><td>5.566101e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19695772_C_T</td><td>0.4453000</td><td>5.706559e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19695799_C_A</td><td>0.4696000</td><td>2.114779e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19695837_G_T</td><td>0.4695000</td><td>2.115027e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19695848_G_A</td><td>0.2135000</td><td>8.696362e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19695935_C_T</td><td>0.5971000</td><td>4.935843e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19696224_C_T</td><td>0.0976100</td><td>2.859063e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19696256_G_A</td><td>0.5995000</td><td>4.927361e-08</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>VAR_7_19883135_A_T</td><td>0.85880</td><td>4.437409e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19883487_T_A</td><td>0.91600</td><td>1.430593e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19884073_T_A</td><td>0.92330</td><td>1.428905e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19884215_G_C</td><td>0.84800</td><td>5.638465e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19884274_C_A</td><td>0.50260</td><td>1.747456e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19884500_T_A</td><td>0.85750</td><td>5.625807e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19884564_G_A</td><td>0.95640</td><td>5.841276e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19884578_A_C</td><td>0.44850</td><td>1.696315e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19885393_C_T</td><td>0.52670</td><td>7.505636e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19885637_A_G</td><td>0.89550</td><td>1.429872e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19885864_T_C</td><td>0.94890</td><td>5.846479e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19886762_G_C</td><td>0.98500</td><td>5.831656e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19887079_T_C</td><td>0.77920</td><td>5.718867e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19887236_A_T</td><td>0.66420</td><td>6.029903e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19887579_T_A</td><td>0.85390</td><td>4.448424e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19887617_A_T</td><td>0.59750</td><td>6.511431e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19888054_G_A</td><td>0.53730</td><td>6.828427e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19888161_C_T</td><td>0.85730</td><td>4.439920e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19888629_C_T</td><td>0.56730</td><td>7.193810e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19889258_A_G</td><td>0.72770</td><td>8.853237e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19889268_G_A</td><td>0.86480</td><td>4.433505e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19889820_G_A</td><td>0.87130</td><td>4.427427e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19890181_A_G</td><td>0.97150</td><td>5.840151e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19890410_A_G</td><td>0.60720</td><td>6.968404e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19890805_G_A</td><td>0.92670</td><td>1.347766e-07</td></tr>\n",
       "\t<tr><td>VAR_7_19891018_G_C</td><td>0.03135</td><td>1.647415e-06</td></tr>\n",
       "\t<tr><td>VAR_7_19891221_C_G</td><td>0.62700</td><td>6.832826e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19891587_T_C</td><td>0.99640</td><td>5.809604e-08</td></tr>\n",
       "\t<tr><td>VAR_7_19891604_G_T</td><td>0.68260</td><td>4.797553e-08</td></tr>\n",
       "\t<tr><td>null              </td><td>     NA</td><td>2.499197e-02</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 909 × 3\n",
       "\\begin{tabular}{lll}\n",
       " id & pval & pp\\_snp\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t VAR\\_7\\_19692124\\_C\\_T & 0.6139000 & 4.811418e-08\\\\\n",
       "\t VAR\\_7\\_19692174\\_C\\_T & 0.7018000 & 4.558298e-08\\\\\n",
       "\t VAR\\_7\\_19692670\\_C\\_G & 0.9668000 & 1.481438e-07\\\\\n",
       "\t VAR\\_7\\_19692863\\_C\\_G & 0.5683000 & 4.984796e-08\\\\\n",
       "\t VAR\\_7\\_19693464\\_T\\_C & 0.5545000 & 1.909496e-07\\\\\n",
       "\t VAR\\_7\\_19693604\\_A\\_G & 0.4721000 & 2.106482e-07\\\\\n",
       "\t VAR\\_7\\_19693626\\_C\\_T & 0.0007141 & 3.779624e-05\\\\\n",
       "\t VAR\\_7\\_19693755\\_C\\_T & 0.0009416 & 2.936939e-05\\\\\n",
       "\t VAR\\_7\\_19693996\\_G\\_A & 0.0006517 & 4.135361e-05\\\\\n",
       "\t VAR\\_7\\_19694131\\_T\\_G & 0.0716300 & 3.663994e-07\\\\\n",
       "\t VAR\\_7\\_19694196\\_G\\_T & 0.0006526 & 4.130068e-05\\\\\n",
       "\t VAR\\_7\\_19694243\\_T\\_C & 0.4625000 & 2.131246e-07\\\\\n",
       "\t VAR\\_7\\_19694376\\_C\\_T & 0.0961600 & 2.893296e-07\\\\\n",
       "\t VAR\\_7\\_19694384\\_C\\_T & 0.0006522 & 4.132419e-05\\\\\n",
       "\t VAR\\_7\\_19694393\\_G\\_T & 0.4491000 & 2.168130e-07\\\\\n",
       "\t VAR\\_7\\_19694632\\_G\\_A & 0.2062000 & 1.493414e-07\\\\\n",
       "\t VAR\\_7\\_19694720\\_C\\_A & 0.2081000 & 1.483512e-07\\\\\n",
       "\t VAR\\_7\\_19694853\\_G\\_C & 0.6086000 & 4.885853e-08\\\\\n",
       "\t VAR\\_7\\_19694917\\_C\\_A & 0.6236000 & 4.833438e-08\\\\\n",
       "\t VAR\\_7\\_19695194\\_T\\_C & 0.5374000 & 5.084969e-08\\\\\n",
       "\t VAR\\_7\\_19695335\\_C\\_A & 0.4492000 & 2.167857e-07\\\\\n",
       "\t VAR\\_7\\_19695613\\_C\\_G & 0.4494000 & 2.166581e-07\\\\\n",
       "\t VAR\\_7\\_19695617\\_A\\_C & 0.4653000 & 5.566101e-08\\\\\n",
       "\t VAR\\_7\\_19695772\\_C\\_T & 0.4453000 & 5.706559e-08\\\\\n",
       "\t VAR\\_7\\_19695799\\_C\\_A & 0.4696000 & 2.114779e-07\\\\\n",
       "\t VAR\\_7\\_19695837\\_G\\_T & 0.4695000 & 2.115027e-07\\\\\n",
       "\t VAR\\_7\\_19695848\\_G\\_A & 0.2135000 & 8.696362e-08\\\\\n",
       "\t VAR\\_7\\_19695935\\_C\\_T & 0.5971000 & 4.935843e-08\\\\\n",
       "\t VAR\\_7\\_19696224\\_C\\_T & 0.0976100 & 2.859063e-07\\\\\n",
       "\t VAR\\_7\\_19696256\\_G\\_A & 0.5995000 & 4.927361e-08\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t VAR\\_7\\_19883135\\_A\\_T & 0.85880 & 4.437409e-08\\\\\n",
       "\t VAR\\_7\\_19883487\\_T\\_A & 0.91600 & 1.430593e-07\\\\\n",
       "\t VAR\\_7\\_19884073\\_T\\_A & 0.92330 & 1.428905e-07\\\\\n",
       "\t VAR\\_7\\_19884215\\_G\\_C & 0.84800 & 5.638465e-08\\\\\n",
       "\t VAR\\_7\\_19884274\\_C\\_A & 0.50260 & 1.747456e-07\\\\\n",
       "\t VAR\\_7\\_19884500\\_T\\_A & 0.85750 & 5.625807e-08\\\\\n",
       "\t VAR\\_7\\_19884564\\_G\\_A & 0.95640 & 5.841276e-08\\\\\n",
       "\t VAR\\_7\\_19884578\\_A\\_C & 0.44850 & 1.696315e-07\\\\\n",
       "\t VAR\\_7\\_19885393\\_C\\_T & 0.52670 & 7.505636e-08\\\\\n",
       "\t VAR\\_7\\_19885637\\_A\\_G & 0.89550 & 1.429872e-07\\\\\n",
       "\t VAR\\_7\\_19885864\\_T\\_C & 0.94890 & 5.846479e-08\\\\\n",
       "\t VAR\\_7\\_19886762\\_G\\_C & 0.98500 & 5.831656e-08\\\\\n",
       "\t VAR\\_7\\_19887079\\_T\\_C & 0.77920 & 5.718867e-08\\\\\n",
       "\t VAR\\_7\\_19887236\\_A\\_T & 0.66420 & 6.029903e-08\\\\\n",
       "\t VAR\\_7\\_19887579\\_T\\_A & 0.85390 & 4.448424e-08\\\\\n",
       "\t VAR\\_7\\_19887617\\_A\\_T & 0.59750 & 6.511431e-08\\\\\n",
       "\t VAR\\_7\\_19888054\\_G\\_A & 0.53730 & 6.828427e-08\\\\\n",
       "\t VAR\\_7\\_19888161\\_C\\_T & 0.85730 & 4.439920e-08\\\\\n",
       "\t VAR\\_7\\_19888629\\_C\\_T & 0.56730 & 7.193810e-08\\\\\n",
       "\t VAR\\_7\\_19889258\\_A\\_G & 0.72770 & 8.853237e-08\\\\\n",
       "\t VAR\\_7\\_19889268\\_G\\_A & 0.86480 & 4.433505e-08\\\\\n",
       "\t VAR\\_7\\_19889820\\_G\\_A & 0.87130 & 4.427427e-08\\\\\n",
       "\t VAR\\_7\\_19890181\\_A\\_G & 0.97150 & 5.840151e-08\\\\\n",
       "\t VAR\\_7\\_19890410\\_A\\_G & 0.60720 & 6.968404e-08\\\\\n",
       "\t VAR\\_7\\_19890805\\_G\\_A & 0.92670 & 1.347766e-07\\\\\n",
       "\t VAR\\_7\\_19891018\\_G\\_C & 0.03135 & 1.647415e-06\\\\\n",
       "\t VAR\\_7\\_19891221\\_C\\_G & 0.62700 & 6.832826e-08\\\\\n",
       "\t VAR\\_7\\_19891587\\_T\\_C & 0.99640 & 5.809604e-08\\\\\n",
       "\t VAR\\_7\\_19891604\\_G\\_T & 0.68260 & 4.797553e-08\\\\\n",
       "\t null               &      NA & 2.499197e-02\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 909 × 3\n",
       "\n",
       "| id &lt;chr&gt; | pval &lt;dbl&gt; | pp_snp &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| VAR_7_19692124_C_T | 0.6139000 | 4.811418e-08 |\n",
       "| VAR_7_19692174_C_T | 0.7018000 | 4.558298e-08 |\n",
       "| VAR_7_19692670_C_G | 0.9668000 | 1.481438e-07 |\n",
       "| VAR_7_19692863_C_G | 0.5683000 | 4.984796e-08 |\n",
       "| VAR_7_19693464_T_C | 0.5545000 | 1.909496e-07 |\n",
       "| VAR_7_19693604_A_G | 0.4721000 | 2.106482e-07 |\n",
       "| VAR_7_19693626_C_T | 0.0007141 | 3.779624e-05 |\n",
       "| VAR_7_19693755_C_T | 0.0009416 | 2.936939e-05 |\n",
       "| VAR_7_19693996_G_A | 0.0006517 | 4.135361e-05 |\n",
       "| VAR_7_19694131_T_G | 0.0716300 | 3.663994e-07 |\n",
       "| VAR_7_19694196_G_T | 0.0006526 | 4.130068e-05 |\n",
       "| VAR_7_19694243_T_C | 0.4625000 | 2.131246e-07 |\n",
       "| VAR_7_19694376_C_T | 0.0961600 | 2.893296e-07 |\n",
       "| VAR_7_19694384_C_T | 0.0006522 | 4.132419e-05 |\n",
       "| VAR_7_19694393_G_T | 0.4491000 | 2.168130e-07 |\n",
       "| VAR_7_19694632_G_A | 0.2062000 | 1.493414e-07 |\n",
       "| VAR_7_19694720_C_A | 0.2081000 | 1.483512e-07 |\n",
       "| VAR_7_19694853_G_C | 0.6086000 | 4.885853e-08 |\n",
       "| VAR_7_19694917_C_A | 0.6236000 | 4.833438e-08 |\n",
       "| VAR_7_19695194_T_C | 0.5374000 | 5.084969e-08 |\n",
       "| VAR_7_19695335_C_A | 0.4492000 | 2.167857e-07 |\n",
       "| VAR_7_19695613_C_G | 0.4494000 | 2.166581e-07 |\n",
       "| VAR_7_19695617_A_C | 0.4653000 | 5.566101e-08 |\n",
       "| VAR_7_19695772_C_T | 0.4453000 | 5.706559e-08 |\n",
       "| VAR_7_19695799_C_A | 0.4696000 | 2.114779e-07 |\n",
       "| VAR_7_19695837_G_T | 0.4695000 | 2.115027e-07 |\n",
       "| VAR_7_19695848_G_A | 0.2135000 | 8.696362e-08 |\n",
       "| VAR_7_19695935_C_T | 0.5971000 | 4.935843e-08 |\n",
       "| VAR_7_19696224_C_T | 0.0976100 | 2.859063e-07 |\n",
       "| VAR_7_19696256_G_A | 0.5995000 | 4.927361e-08 |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| VAR_7_19883135_A_T | 0.85880 | 4.437409e-08 |\n",
       "| VAR_7_19883487_T_A | 0.91600 | 1.430593e-07 |\n",
       "| VAR_7_19884073_T_A | 0.92330 | 1.428905e-07 |\n",
       "| VAR_7_19884215_G_C | 0.84800 | 5.638465e-08 |\n",
       "| VAR_7_19884274_C_A | 0.50260 | 1.747456e-07 |\n",
       "| VAR_7_19884500_T_A | 0.85750 | 5.625807e-08 |\n",
       "| VAR_7_19884564_G_A | 0.95640 | 5.841276e-08 |\n",
       "| VAR_7_19884578_A_C | 0.44850 | 1.696315e-07 |\n",
       "| VAR_7_19885393_C_T | 0.52670 | 7.505636e-08 |\n",
       "| VAR_7_19885637_A_G | 0.89550 | 1.429872e-07 |\n",
       "| VAR_7_19885864_T_C | 0.94890 | 5.846479e-08 |\n",
       "| VAR_7_19886762_G_C | 0.98500 | 5.831656e-08 |\n",
       "| VAR_7_19887079_T_C | 0.77920 | 5.718867e-08 |\n",
       "| VAR_7_19887236_A_T | 0.66420 | 6.029903e-08 |\n",
       "| VAR_7_19887579_T_A | 0.85390 | 4.448424e-08 |\n",
       "| VAR_7_19887617_A_T | 0.59750 | 6.511431e-08 |\n",
       "| VAR_7_19888054_G_A | 0.53730 | 6.828427e-08 |\n",
       "| VAR_7_19888161_C_T | 0.85730 | 4.439920e-08 |\n",
       "| VAR_7_19888629_C_T | 0.56730 | 7.193810e-08 |\n",
       "| VAR_7_19889258_A_G | 0.72770 | 8.853237e-08 |\n",
       "| VAR_7_19889268_G_A | 0.86480 | 4.433505e-08 |\n",
       "| VAR_7_19889820_G_A | 0.87130 | 4.427427e-08 |\n",
       "| VAR_7_19890181_A_G | 0.97150 | 5.840151e-08 |\n",
       "| VAR_7_19890410_A_G | 0.60720 | 6.968404e-08 |\n",
       "| VAR_7_19890805_G_A | 0.92670 | 1.347766e-07 |\n",
       "| VAR_7_19891018_G_C | 0.03135 | 1.647415e-06 |\n",
       "| VAR_7_19891221_C_G | 0.62700 | 6.832826e-08 |\n",
       "| VAR_7_19891587_T_C | 0.99640 | 5.809604e-08 |\n",
       "| VAR_7_19891604_G_T | 0.68260 | 4.797553e-08 |\n",
       "| null               |      NA | 2.499197e-02 |\n",
       "\n"
      ],
      "text/plain": [
       "    id                 pval      pp_snp      \n",
       "1   VAR_7_19692124_C_T 0.6139000 4.811418e-08\n",
       "2   VAR_7_19692174_C_T 0.7018000 4.558298e-08\n",
       "3   VAR_7_19692670_C_G 0.9668000 1.481438e-07\n",
       "4   VAR_7_19692863_C_G 0.5683000 4.984796e-08\n",
       "5   VAR_7_19693464_T_C 0.5545000 1.909496e-07\n",
       "6   VAR_7_19693604_A_G 0.4721000 2.106482e-07\n",
       "7   VAR_7_19693626_C_T 0.0007141 3.779624e-05\n",
       "8   VAR_7_19693755_C_T 0.0009416 2.936939e-05\n",
       "9   VAR_7_19693996_G_A 0.0006517 4.135361e-05\n",
       "10  VAR_7_19694131_T_G 0.0716300 3.663994e-07\n",
       "11  VAR_7_19694196_G_T 0.0006526 4.130068e-05\n",
       "12  VAR_7_19694243_T_C 0.4625000 2.131246e-07\n",
       "13  VAR_7_19694376_C_T 0.0961600 2.893296e-07\n",
       "14  VAR_7_19694384_C_T 0.0006522 4.132419e-05\n",
       "15  VAR_7_19694393_G_T 0.4491000 2.168130e-07\n",
       "16  VAR_7_19694632_G_A 0.2062000 1.493414e-07\n",
       "17  VAR_7_19694720_C_A 0.2081000 1.483512e-07\n",
       "18  VAR_7_19694853_G_C 0.6086000 4.885853e-08\n",
       "19  VAR_7_19694917_C_A 0.6236000 4.833438e-08\n",
       "20  VAR_7_19695194_T_C 0.5374000 5.084969e-08\n",
       "21  VAR_7_19695335_C_A 0.4492000 2.167857e-07\n",
       "22  VAR_7_19695613_C_G 0.4494000 2.166581e-07\n",
       "23  VAR_7_19695617_A_C 0.4653000 5.566101e-08\n",
       "24  VAR_7_19695772_C_T 0.4453000 5.706559e-08\n",
       "25  VAR_7_19695799_C_A 0.4696000 2.114779e-07\n",
       "26  VAR_7_19695837_G_T 0.4695000 2.115027e-07\n",
       "27  VAR_7_19695848_G_A 0.2135000 8.696362e-08\n",
       "28  VAR_7_19695935_C_T 0.5971000 4.935843e-08\n",
       "29  VAR_7_19696224_C_T 0.0976100 2.859063e-07\n",
       "30  VAR_7_19696256_G_A 0.5995000 4.927361e-08\n",
       "⋮   ⋮                  ⋮         ⋮           \n",
       "880 VAR_7_19883135_A_T 0.85880   4.437409e-08\n",
       "881 VAR_7_19883487_T_A 0.91600   1.430593e-07\n",
       "882 VAR_7_19884073_T_A 0.92330   1.428905e-07\n",
       "883 VAR_7_19884215_G_C 0.84800   5.638465e-08\n",
       "884 VAR_7_19884274_C_A 0.50260   1.747456e-07\n",
       "885 VAR_7_19884500_T_A 0.85750   5.625807e-08\n",
       "886 VAR_7_19884564_G_A 0.95640   5.841276e-08\n",
       "887 VAR_7_19884578_A_C 0.44850   1.696315e-07\n",
       "888 VAR_7_19885393_C_T 0.52670   7.505636e-08\n",
       "889 VAR_7_19885637_A_G 0.89550   1.429872e-07\n",
       "890 VAR_7_19885864_T_C 0.94890   5.846479e-08\n",
       "891 VAR_7_19886762_G_C 0.98500   5.831656e-08\n",
       "892 VAR_7_19887079_T_C 0.77920   5.718867e-08\n",
       "893 VAR_7_19887236_A_T 0.66420   6.029903e-08\n",
       "894 VAR_7_19887579_T_A 0.85390   4.448424e-08\n",
       "895 VAR_7_19887617_A_T 0.59750   6.511431e-08\n",
       "896 VAR_7_19888054_G_A 0.53730   6.828427e-08\n",
       "897 VAR_7_19888161_C_T 0.85730   4.439920e-08\n",
       "898 VAR_7_19888629_C_T 0.56730   7.193810e-08\n",
       "899 VAR_7_19889258_A_G 0.72770   8.853237e-08\n",
       "900 VAR_7_19889268_G_A 0.86480   4.433505e-08\n",
       "901 VAR_7_19889820_G_A 0.87130   4.427427e-08\n",
       "902 VAR_7_19890181_A_G 0.97150   5.840151e-08\n",
       "903 VAR_7_19890410_A_G 0.60720   6.968404e-08\n",
       "904 VAR_7_19890805_G_A 0.92670   1.347766e-07\n",
       "905 VAR_7_19891018_G_C 0.03135   1.647415e-06\n",
       "906 VAR_7_19891221_C_G 0.62700   6.832826e-08\n",
       "907 VAR_7_19891587_T_C 0.99640   5.809604e-08\n",
       "908 VAR_7_19891604_G_T 0.68260   4.797553e-08\n",
       "909 null                    NA   2.499197e-02"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "x$finemapped_gwas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A data.frame: 10001 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>id</th><th scope=col>pval</th><th scope=col>pp_snp</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>VAR_19_45242740_A_G</td><td>1.057829e-264</td><td> 9.987847e-01</td></tr>\n",
       "\t<tr><td>VAR_19_45418790_T_C</td><td>9.430279e-262</td><td> 1.215255e-03</td></tr>\n",
       "\t<tr><td>VAR_19_45416178_G_T</td><td>6.215120e-246</td><td> 1.916881e-19</td></tr>\n",
       "\t<tr><td>VAR_19_45416741_C_T</td><td>7.463390e-244</td><td> 1.604392e-21</td></tr>\n",
       "\t<tr><td>VAR_19_45220896_T_C</td><td>4.514966e-220</td><td> 3.379888e-45</td></tr>\n",
       "\t<tr><td>VAR_19_45384338_G_A</td><td>4.761959e-219</td><td> 2.395260e-46</td></tr>\n",
       "\t<tr><td>VAR_19_45687093_C_T</td><td>3.200399e-206</td><td> 2.782476e-59</td></tr>\n",
       "\t<tr><td>VAR_19_45384105_C_T</td><td>5.042943e-206</td><td> 2.347705e-59</td></tr>\n",
       "\t<tr><td>VAR_19_45372959_C_T</td><td>2.187813e-202</td><td> 6.078236e-63</td></tr>\n",
       "\t<tr><td>VAR_19_45304061_G_A</td><td>2.825650e-202</td><td> 5.970324e-63</td></tr>\n",
       "\t<tr><td>VAR_19_45372184_A_C</td><td>1.773714e-200</td><td> 7.542997e-65</td></tr>\n",
       "\t<tr><td>VAR_19_45395266_G_A</td><td>1.177904e-196</td><td> 1.053249e-68</td></tr>\n",
       "\t<tr><td>VAR_19_45372354_T_G</td><td>3.842682e-188</td><td> 3.566300e-77</td></tr>\n",
       "\t<tr><td>VAR_19_45524593_A_G</td><td>3.867867e-186</td><td> 1.828176e-79</td></tr>\n",
       "\t<tr><td>VAR_19_45372707_G_A</td><td>6.645271e-184</td><td> 2.087437e-81</td></tr>\n",
       "\t<tr><td>VAR_19_45395330_A_G</td><td>5.710477e-183</td><td> 2.276577e-82</td></tr>\n",
       "\t<tr><td>VAR_19_45372129_A_T</td><td>2.179755e-181</td><td> 6.420226e-84</td></tr>\n",
       "\t<tr><td>VAR_19_45186873_G_T</td><td>6.898697e-180</td><td> 2.789779e-85</td></tr>\n",
       "\t<tr><td>VAR_19_45421100_G_A</td><td>2.281116e-173</td><td> 5.823871e-92</td></tr>\n",
       "\t<tr><td>VAR_19_45913250_C_T</td><td>9.821882e-172</td><td> 1.557319e-93</td></tr>\n",
       "\t<tr><td>VAR_19_45410444_G_A</td><td>1.581927e-171</td><td> 8.507279e-94</td></tr>\n",
       "\t<tr><td>VAR_19_45239185_A_G</td><td>9.988133e-171</td><td> 1.396435e-94</td></tr>\n",
       "\t<tr><td>VAR_19_45416291_C_T</td><td>3.572415e-169</td><td> 3.796783e-96</td></tr>\n",
       "\t<tr><td>VAR_19_45160086_C_A</td><td>7.714231e-168</td><td> 1.341096e-97</td></tr>\n",
       "\t<tr><td>VAR_19_45170097_A_G</td><td>3.514722e-167</td><td> 6.221276e-98</td></tr>\n",
       "\t<tr><td>VAR_19_45237812_T_C</td><td>3.646089e-166</td><td> 3.893403e-99</td></tr>\n",
       "\t<tr><td>VAR_19_45380970_G_T</td><td>3.875917e-166</td><td> 3.479225e-99</td></tr>\n",
       "\t<tr><td>VAR_19_45385759_G_C</td><td>4.248585e-163</td><td>3.268907e-102</td></tr>\n",
       "\t<tr><td>VAR_19_45381917_C_T</td><td>9.058701e-163</td><td>1.783299e-102</td></tr>\n",
       "\t<tr><td>VAR_19_45402477_C_T</td><td>8.993050e-162</td><td>1.550301e-103</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>VAR_19_45103698_T_C</td><td>0.7733042</td><td>5.690196e-263</td></tr>\n",
       "\t<tr><td>VAR_19_44850314_C_T</td><td>0.7734388</td><td>2.568215e-262</td></tr>\n",
       "\t<tr><td>VAR_19_44441694_G_A</td><td>0.7734785</td><td>5.551229e-263</td></tr>\n",
       "\t<tr><td>VAR_19_46986829_G_C</td><td>0.7750160</td><td>7.494146e-263</td></tr>\n",
       "\t<tr><td>VAR_19_46950437_G_C</td><td>0.7751292</td><td>1.415613e-262</td></tr>\n",
       "\t<tr><td>VAR_19_46815373_C_T</td><td>0.7755157</td><td>2.113053e-262</td></tr>\n",
       "\t<tr><td>VAR_19_44111890_C_T</td><td>0.7757396</td><td>1.880876e-262</td></tr>\n",
       "\t<tr><td>VAR_19_46615656_A_G</td><td>0.7759433</td><td>1.969228e-262</td></tr>\n",
       "\t<tr><td>VAR_19_44899111_C_T</td><td>0.7760955</td><td>7.347045e-263</td></tr>\n",
       "\t<tr><td>VAR_19_46327245_G_T</td><td>0.7761084</td><td>1.465946e-262</td></tr>\n",
       "\t<tr><td>VAR_19_46950906_A_G</td><td>0.7764213</td><td>1.414934e-262</td></tr>\n",
       "\t<tr><td>VAR_19_46718040_C_G</td><td>0.7768282</td><td>1.718177e-262</td></tr>\n",
       "\t<tr><td>VAR_19_44899142_G_A</td><td>0.7770098</td><td>7.344559e-263</td></tr>\n",
       "\t<tr><td>VAR_19_46745734_A_G</td><td>0.7770192</td><td>1.659016e-262</td></tr>\n",
       "\t<tr><td>VAR_19_44899318_G_A</td><td>0.7770589</td><td>7.344290e-263</td></tr>\n",
       "\t<tr><td>VAR_19_45000712_T_C</td><td>0.7771035</td><td>5.379291e-263</td></tr>\n",
       "\t<tr><td>VAR_19_44824679_T_C</td><td>0.7772589</td><td>5.228185e-263</td></tr>\n",
       "\t<tr><td>VAR_19_43944285_C_T</td><td>0.7777029</td><td>1.000168e-262</td></tr>\n",
       "\t<tr><td>VAR_19_45114289_C_G</td><td>0.7779955</td><td>5.502694e-263</td></tr>\n",
       "\t<tr><td>VAR_19_45922323_G_T</td><td>0.7781186</td><td>7.736578e-263</td></tr>\n",
       "\t<tr><td>VAR_19_45912406_G_A</td><td>0.7785805</td><td>7.441813e-263</td></tr>\n",
       "\t<tr><td>VAR_19_44132559_G_C</td><td>0.7785871</td><td>5.230484e-263</td></tr>\n",
       "\t<tr><td>VAR_19_46951805_A_G</td><td>0.7791294</td><td>1.413655e-262</td></tr>\n",
       "\t<tr><td>VAR_19_46725333_T_C</td><td>0.7792503</td><td>1.715311e-262</td></tr>\n",
       "\t<tr><td>VAR_19_46742424_C_T</td><td>0.7793020</td><td>7.221685e-263</td></tr>\n",
       "\t<tr><td>VAR_19_46801465_G_C</td><td>0.7793360</td><td>8.738321e-263</td></tr>\n",
       "\t<tr><td>VAR_19_46350078_G_A</td><td>0.7793981</td><td>1.464935e-262</td></tr>\n",
       "\t<tr><td>VAR_19_44899473_T_C</td><td>0.7795296</td><td>7.335872e-263</td></tr>\n",
       "\t<tr><td>VAR_19_44204384_A_G</td><td>0.7796945</td><td>5.713370e-263</td></tr>\n",
       "\t<tr><td>null               </td><td>       NA</td><td> 0.000000e+00</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 10001 × 3\n",
       "\\begin{tabular}{lll}\n",
       " id & pval & pp\\_snp\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t VAR\\_19\\_45242740\\_A\\_G & 1.057829e-264 &  9.987847e-01\\\\\n",
       "\t VAR\\_19\\_45418790\\_T\\_C & 9.430279e-262 &  1.215255e-03\\\\\n",
       "\t VAR\\_19\\_45416178\\_G\\_T & 6.215120e-246 &  1.916881e-19\\\\\n",
       "\t VAR\\_19\\_45416741\\_C\\_T & 7.463390e-244 &  1.604392e-21\\\\\n",
       "\t VAR\\_19\\_45220896\\_T\\_C & 4.514966e-220 &  3.379888e-45\\\\\n",
       "\t VAR\\_19\\_45384338\\_G\\_A & 4.761959e-219 &  2.395260e-46\\\\\n",
       "\t VAR\\_19\\_45687093\\_C\\_T & 3.200399e-206 &  2.782476e-59\\\\\n",
       "\t VAR\\_19\\_45384105\\_C\\_T & 5.042943e-206 &  2.347705e-59\\\\\n",
       "\t VAR\\_19\\_45372959\\_C\\_T & 2.187813e-202 &  6.078236e-63\\\\\n",
       "\t VAR\\_19\\_45304061\\_G\\_A & 2.825650e-202 &  5.970324e-63\\\\\n",
       "\t VAR\\_19\\_45372184\\_A\\_C & 1.773714e-200 &  7.542997e-65\\\\\n",
       "\t VAR\\_19\\_45395266\\_G\\_A & 1.177904e-196 &  1.053249e-68\\\\\n",
       "\t VAR\\_19\\_45372354\\_T\\_G & 3.842682e-188 &  3.566300e-77\\\\\n",
       "\t VAR\\_19\\_45524593\\_A\\_G & 3.867867e-186 &  1.828176e-79\\\\\n",
       "\t VAR\\_19\\_45372707\\_G\\_A & 6.645271e-184 &  2.087437e-81\\\\\n",
       "\t VAR\\_19\\_45395330\\_A\\_G & 5.710477e-183 &  2.276577e-82\\\\\n",
       "\t VAR\\_19\\_45372129\\_A\\_T & 2.179755e-181 &  6.420226e-84\\\\\n",
       "\t VAR\\_19\\_45186873\\_G\\_T & 6.898697e-180 &  2.789779e-85\\\\\n",
       "\t VAR\\_19\\_45421100\\_G\\_A & 2.281116e-173 &  5.823871e-92\\\\\n",
       "\t VAR\\_19\\_45913250\\_C\\_T & 9.821882e-172 &  1.557319e-93\\\\\n",
       "\t VAR\\_19\\_45410444\\_G\\_A & 1.581927e-171 &  8.507279e-94\\\\\n",
       "\t VAR\\_19\\_45239185\\_A\\_G & 9.988133e-171 &  1.396435e-94\\\\\n",
       "\t VAR\\_19\\_45416291\\_C\\_T & 3.572415e-169 &  3.796783e-96\\\\\n",
       "\t VAR\\_19\\_45160086\\_C\\_A & 7.714231e-168 &  1.341096e-97\\\\\n",
       "\t VAR\\_19\\_45170097\\_A\\_G & 3.514722e-167 &  6.221276e-98\\\\\n",
       "\t VAR\\_19\\_45237812\\_T\\_C & 3.646089e-166 &  3.893403e-99\\\\\n",
       "\t VAR\\_19\\_45380970\\_G\\_T & 3.875917e-166 &  3.479225e-99\\\\\n",
       "\t VAR\\_19\\_45385759\\_G\\_C & 4.248585e-163 & 3.268907e-102\\\\\n",
       "\t VAR\\_19\\_45381917\\_C\\_T & 9.058701e-163 & 1.783299e-102\\\\\n",
       "\t VAR\\_19\\_45402477\\_C\\_T & 8.993050e-162 & 1.550301e-103\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t VAR\\_19\\_45103698\\_T\\_C & 0.7733042 & 5.690196e-263\\\\\n",
       "\t VAR\\_19\\_44850314\\_C\\_T & 0.7734388 & 2.568215e-262\\\\\n",
       "\t VAR\\_19\\_44441694\\_G\\_A & 0.7734785 & 5.551229e-263\\\\\n",
       "\t VAR\\_19\\_46986829\\_G\\_C & 0.7750160 & 7.494146e-263\\\\\n",
       "\t VAR\\_19\\_46950437\\_G\\_C & 0.7751292 & 1.415613e-262\\\\\n",
       "\t VAR\\_19\\_46815373\\_C\\_T & 0.7755157 & 2.113053e-262\\\\\n",
       "\t VAR\\_19\\_44111890\\_C\\_T & 0.7757396 & 1.880876e-262\\\\\n",
       "\t VAR\\_19\\_46615656\\_A\\_G & 0.7759433 & 1.969228e-262\\\\\n",
       "\t VAR\\_19\\_44899111\\_C\\_T & 0.7760955 & 7.347045e-263\\\\\n",
       "\t VAR\\_19\\_46327245\\_G\\_T & 0.7761084 & 1.465946e-262\\\\\n",
       "\t VAR\\_19\\_46950906\\_A\\_G & 0.7764213 & 1.414934e-262\\\\\n",
       "\t VAR\\_19\\_46718040\\_C\\_G & 0.7768282 & 1.718177e-262\\\\\n",
       "\t VAR\\_19\\_44899142\\_G\\_A & 0.7770098 & 7.344559e-263\\\\\n",
       "\t VAR\\_19\\_46745734\\_A\\_G & 0.7770192 & 1.659016e-262\\\\\n",
       "\t VAR\\_19\\_44899318\\_G\\_A & 0.7770589 & 7.344290e-263\\\\\n",
       "\t VAR\\_19\\_45000712\\_T\\_C & 0.7771035 & 5.379291e-263\\\\\n",
       "\t VAR\\_19\\_44824679\\_T\\_C & 0.7772589 & 5.228185e-263\\\\\n",
       "\t VAR\\_19\\_43944285\\_C\\_T & 0.7777029 & 1.000168e-262\\\\\n",
       "\t VAR\\_19\\_45114289\\_C\\_G & 0.7779955 & 5.502694e-263\\\\\n",
       "\t VAR\\_19\\_45922323\\_G\\_T & 0.7781186 & 7.736578e-263\\\\\n",
       "\t VAR\\_19\\_45912406\\_G\\_A & 0.7785805 & 7.441813e-263\\\\\n",
       "\t VAR\\_19\\_44132559\\_G\\_C & 0.7785871 & 5.230484e-263\\\\\n",
       "\t VAR\\_19\\_46951805\\_A\\_G & 0.7791294 & 1.413655e-262\\\\\n",
       "\t VAR\\_19\\_46725333\\_T\\_C & 0.7792503 & 1.715311e-262\\\\\n",
       "\t VAR\\_19\\_46742424\\_C\\_T & 0.7793020 & 7.221685e-263\\\\\n",
       "\t VAR\\_19\\_46801465\\_G\\_C & 0.7793360 & 8.738321e-263\\\\\n",
       "\t VAR\\_19\\_46350078\\_G\\_A & 0.7793981 & 1.464935e-262\\\\\n",
       "\t VAR\\_19\\_44899473\\_T\\_C & 0.7795296 & 7.335872e-263\\\\\n",
       "\t VAR\\_19\\_44204384\\_A\\_G & 0.7796945 & 5.713370e-263\\\\\n",
       "\t null                &        NA &  0.000000e+00\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 10001 × 3\n",
       "\n",
       "| id &lt;chr&gt; | pval &lt;dbl&gt; | pp_snp &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| VAR_19_45242740_A_G | 1.057829e-264 |  9.987847e-01 |\n",
       "| VAR_19_45418790_T_C | 9.430279e-262 |  1.215255e-03 |\n",
       "| VAR_19_45416178_G_T | 6.215120e-246 |  1.916881e-19 |\n",
       "| VAR_19_45416741_C_T | 7.463390e-244 |  1.604392e-21 |\n",
       "| VAR_19_45220896_T_C | 4.514966e-220 |  3.379888e-45 |\n",
       "| VAR_19_45384338_G_A | 4.761959e-219 |  2.395260e-46 |\n",
       "| VAR_19_45687093_C_T | 3.200399e-206 |  2.782476e-59 |\n",
       "| VAR_19_45384105_C_T | 5.042943e-206 |  2.347705e-59 |\n",
       "| VAR_19_45372959_C_T | 2.187813e-202 |  6.078236e-63 |\n",
       "| VAR_19_45304061_G_A | 2.825650e-202 |  5.970324e-63 |\n",
       "| VAR_19_45372184_A_C | 1.773714e-200 |  7.542997e-65 |\n",
       "| VAR_19_45395266_G_A | 1.177904e-196 |  1.053249e-68 |\n",
       "| VAR_19_45372354_T_G | 3.842682e-188 |  3.566300e-77 |\n",
       "| VAR_19_45524593_A_G | 3.867867e-186 |  1.828176e-79 |\n",
       "| VAR_19_45372707_G_A | 6.645271e-184 |  2.087437e-81 |\n",
       "| VAR_19_45395330_A_G | 5.710477e-183 |  2.276577e-82 |\n",
       "| VAR_19_45372129_A_T | 2.179755e-181 |  6.420226e-84 |\n",
       "| VAR_19_45186873_G_T | 6.898697e-180 |  2.789779e-85 |\n",
       "| VAR_19_45421100_G_A | 2.281116e-173 |  5.823871e-92 |\n",
       "| VAR_19_45913250_C_T | 9.821882e-172 |  1.557319e-93 |\n",
       "| VAR_19_45410444_G_A | 1.581927e-171 |  8.507279e-94 |\n",
       "| VAR_19_45239185_A_G | 9.988133e-171 |  1.396435e-94 |\n",
       "| VAR_19_45416291_C_T | 3.572415e-169 |  3.796783e-96 |\n",
       "| VAR_19_45160086_C_A | 7.714231e-168 |  1.341096e-97 |\n",
       "| VAR_19_45170097_A_G | 3.514722e-167 |  6.221276e-98 |\n",
       "| VAR_19_45237812_T_C | 3.646089e-166 |  3.893403e-99 |\n",
       "| VAR_19_45380970_G_T | 3.875917e-166 |  3.479225e-99 |\n",
       "| VAR_19_45385759_G_C | 4.248585e-163 | 3.268907e-102 |\n",
       "| VAR_19_45381917_C_T | 9.058701e-163 | 1.783299e-102 |\n",
       "| VAR_19_45402477_C_T | 8.993050e-162 | 1.550301e-103 |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| VAR_19_45103698_T_C | 0.7733042 | 5.690196e-263 |\n",
       "| VAR_19_44850314_C_T | 0.7734388 | 2.568215e-262 |\n",
       "| VAR_19_44441694_G_A | 0.7734785 | 5.551229e-263 |\n",
       "| VAR_19_46986829_G_C | 0.7750160 | 7.494146e-263 |\n",
       "| VAR_19_46950437_G_C | 0.7751292 | 1.415613e-262 |\n",
       "| VAR_19_46815373_C_T | 0.7755157 | 2.113053e-262 |\n",
       "| VAR_19_44111890_C_T | 0.7757396 | 1.880876e-262 |\n",
       "| VAR_19_46615656_A_G | 0.7759433 | 1.969228e-262 |\n",
       "| VAR_19_44899111_C_T | 0.7760955 | 7.347045e-263 |\n",
       "| VAR_19_46327245_G_T | 0.7761084 | 1.465946e-262 |\n",
       "| VAR_19_46950906_A_G | 0.7764213 | 1.414934e-262 |\n",
       "| VAR_19_46718040_C_G | 0.7768282 | 1.718177e-262 |\n",
       "| VAR_19_44899142_G_A | 0.7770098 | 7.344559e-263 |\n",
       "| VAR_19_46745734_A_G | 0.7770192 | 1.659016e-262 |\n",
       "| VAR_19_44899318_G_A | 0.7770589 | 7.344290e-263 |\n",
       "| VAR_19_45000712_T_C | 0.7771035 | 5.379291e-263 |\n",
       "| VAR_19_44824679_T_C | 0.7772589 | 5.228185e-263 |\n",
       "| VAR_19_43944285_C_T | 0.7777029 | 1.000168e-262 |\n",
       "| VAR_19_45114289_C_G | 0.7779955 | 5.502694e-263 |\n",
       "| VAR_19_45922323_G_T | 0.7781186 | 7.736578e-263 |\n",
       "| VAR_19_45912406_G_A | 0.7785805 | 7.441813e-263 |\n",
       "| VAR_19_44132559_G_C | 0.7785871 | 5.230484e-263 |\n",
       "| VAR_19_46951805_A_G | 0.7791294 | 1.413655e-262 |\n",
       "| VAR_19_46725333_T_C | 0.7792503 | 1.715311e-262 |\n",
       "| VAR_19_46742424_C_T | 0.7793020 | 7.221685e-263 |\n",
       "| VAR_19_46801465_G_C | 0.7793360 | 8.738321e-263 |\n",
       "| VAR_19_46350078_G_A | 0.7793981 | 1.464935e-262 |\n",
       "| VAR_19_44899473_T_C | 0.7795296 | 7.335872e-263 |\n",
       "| VAR_19_44204384_A_G | 0.7796945 | 5.713370e-263 |\n",
       "| null                |        NA |  0.000000e+00 |\n",
       "\n"
      ],
      "text/plain": [
       "      id                  pval          pp_snp       \n",
       "1     VAR_19_45242740_A_G 1.057829e-264  9.987847e-01\n",
       "2     VAR_19_45418790_T_C 9.430279e-262  1.215255e-03\n",
       "3     VAR_19_45416178_G_T 6.215120e-246  1.916881e-19\n",
       "4     VAR_19_45416741_C_T 7.463390e-244  1.604392e-21\n",
       "5     VAR_19_45220896_T_C 4.514966e-220  3.379888e-45\n",
       "6     VAR_19_45384338_G_A 4.761959e-219  2.395260e-46\n",
       "7     VAR_19_45687093_C_T 3.200399e-206  2.782476e-59\n",
       "8     VAR_19_45384105_C_T 5.042943e-206  2.347705e-59\n",
       "9     VAR_19_45372959_C_T 2.187813e-202  6.078236e-63\n",
       "10    VAR_19_45304061_G_A 2.825650e-202  5.970324e-63\n",
       "11    VAR_19_45372184_A_C 1.773714e-200  7.542997e-65\n",
       "12    VAR_19_45395266_G_A 1.177904e-196  1.053249e-68\n",
       "13    VAR_19_45372354_T_G 3.842682e-188  3.566300e-77\n",
       "14    VAR_19_45524593_A_G 3.867867e-186  1.828176e-79\n",
       "15    VAR_19_45372707_G_A 6.645271e-184  2.087437e-81\n",
       "16    VAR_19_45395330_A_G 5.710477e-183  2.276577e-82\n",
       "17    VAR_19_45372129_A_T 2.179755e-181  6.420226e-84\n",
       "18    VAR_19_45186873_G_T 6.898697e-180  2.789779e-85\n",
       "19    VAR_19_45421100_G_A 2.281116e-173  5.823871e-92\n",
       "20    VAR_19_45913250_C_T 9.821882e-172  1.557319e-93\n",
       "21    VAR_19_45410444_G_A 1.581927e-171  8.507279e-94\n",
       "22    VAR_19_45239185_A_G 9.988133e-171  1.396435e-94\n",
       "23    VAR_19_45416291_C_T 3.572415e-169  3.796783e-96\n",
       "24    VAR_19_45160086_C_A 7.714231e-168  1.341096e-97\n",
       "25    VAR_19_45170097_A_G 3.514722e-167  6.221276e-98\n",
       "26    VAR_19_45237812_T_C 3.646089e-166  3.893403e-99\n",
       "27    VAR_19_45380970_G_T 3.875917e-166  3.479225e-99\n",
       "28    VAR_19_45385759_G_C 4.248585e-163 3.268907e-102\n",
       "29    VAR_19_45381917_C_T 9.058701e-163 1.783299e-102\n",
       "30    VAR_19_45402477_C_T 8.993050e-162 1.550301e-103\n",
       "⋮     ⋮                   ⋮             ⋮            \n",
       "9972  VAR_19_45103698_T_C 0.7733042     5.690196e-263\n",
       "9973  VAR_19_44850314_C_T 0.7734388     2.568215e-262\n",
       "9974  VAR_19_44441694_G_A 0.7734785     5.551229e-263\n",
       "9975  VAR_19_46986829_G_C 0.7750160     7.494146e-263\n",
       "9976  VAR_19_46950437_G_C 0.7751292     1.415613e-262\n",
       "9977  VAR_19_46815373_C_T 0.7755157     2.113053e-262\n",
       "9978  VAR_19_44111890_C_T 0.7757396     1.880876e-262\n",
       "9979  VAR_19_46615656_A_G 0.7759433     1.969228e-262\n",
       "9980  VAR_19_44899111_C_T 0.7760955     7.347045e-263\n",
       "9981  VAR_19_46327245_G_T 0.7761084     1.465946e-262\n",
       "9982  VAR_19_46950906_A_G 0.7764213     1.414934e-262\n",
       "9983  VAR_19_46718040_C_G 0.7768282     1.718177e-262\n",
       "9984  VAR_19_44899142_G_A 0.7770098     7.344559e-263\n",
       "9985  VAR_19_46745734_A_G 0.7770192     1.659016e-262\n",
       "9986  VAR_19_44899318_G_A 0.7770589     7.344290e-263\n",
       "9987  VAR_19_45000712_T_C 0.7771035     5.379291e-263\n",
       "9988  VAR_19_44824679_T_C 0.7772589     5.228185e-263\n",
       "9989  VAR_19_43944285_C_T 0.7777029     1.000168e-262\n",
       "9990  VAR_19_45114289_C_G 0.7779955     5.502694e-263\n",
       "9991  VAR_19_45922323_G_T 0.7781186     7.736578e-263\n",
       "9992  VAR_19_45912406_G_A 0.7785805     7.441813e-263\n",
       "9993  VAR_19_44132559_G_C 0.7785871     5.230484e-263\n",
       "9994  VAR_19_46951805_A_G 0.7791294     1.413655e-262\n",
       "9995  VAR_19_46725333_T_C 0.7792503     1.715311e-262\n",
       "9996  VAR_19_46742424_C_T 0.7793020     7.221685e-263\n",
       "9997  VAR_19_46801465_G_C 0.7793360     8.738321e-263\n",
       "9998  VAR_19_46350078_G_A 0.7793981     1.464935e-262\n",
       "9999  VAR_19_44899473_T_C 0.7795296     7.335872e-263\n",
       "10000 VAR_19_44204384_A_G 0.7796945     5.713370e-263\n",
       "10001 null                       NA      0.000000e+00"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "finemap_gwas = function(dataset)\n",
    "{\n",
    "    if(length(dataset$pvalues) > 10000)\n",
    "    {\n",
    "        new_dataset = data.frame(pval = dataset$pvalues, maf = dataset$MAF, snp = dataset$snp)\n",
    "        new_dataset = new_dataset[order(new_dataset$pval),]\n",
    "        new_dataset = new_dataset[1:10000,]\n",
    "        dataset     = list(pvalues = new_dataset$pval, MAF = new_dataset$maf, snp = new_dataset$snp, N = dataset$N, sdY = 1, type = dataset$type)\n",
    "    }\n",
    "    \n",
    "    finemap           = finemap.abf(dataset = dataset)\n",
    "    finemap           = finemap[,c(\"snp\", \"pvalues.\", \"SNP.PP\")]\n",
    "    colnames(finemap) = c(\"id\", \"pval\", \"pp_snp\")\n",
    "    \n",
    "    return(finemap)\n",
    "    \n",
    "    myres             = finemap\n",
    "    myres             = myres[order(myres$pp_snp, decreasing = TRUE), ]\n",
    "    myres$cum         = cumsum(myres$pp_snp)\n",
    "    myres$cs          = FALSE\n",
    "    to_cs             = myres[myres$cum >= 0.99, ][1, \"id\"]\n",
    "    \n",
    "    myres[myres$cum < 0.99 | myres$id == to_cs, \"cs\"] = TRUE\n",
    "\n",
    "    credible_set = myres[ myres$cs == TRUE, \"id\"]\n",
    "    \n",
    "    return(list(pp = myres, credible_set = credible_set))\n",
    "}\n",
    "\n",
    "dataset = x[[1]]\n",
    "trait_type = x[[2]]\n",
    "\n",
    "finemap_gwas(dataset)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finemap = x[[2]]\n",
    "myres   = finemap\n",
    "myres           = myres[order(myres$pp_snp, decreasing = TRUE), ]\n",
    "myres$cum       = cumsum(myres$pp_snp)\n",
    "myres$cs        = FALSE\n",
    "to_cs           = myres[myres$cum >= 0.99, ][1, \"id\"]\n",
    "\n",
    "myres[myres$cum < 0.99 | myres$id == to_cs, \"cs\"] = TRUE\n",
    "\n",
    "credible_set = myres[ myres$cs == TRUE, \"id\"]\n",
    "\n",
    "str(myres)\n",
    "\n",
    "myres"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Integrate coloc with moloc data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_coloc_qtl_by_locus = function(id, moloc_loci2class, coloc_qtls)\n",
    "{\n",
    "    locus       = moloc_loci2class[id, \"locus\"  ]\n",
    "    som         = moloc_loci2class[id, \"class\"  ]\n",
    "    cluster     = moloc_loci2class[id, \"cluster\"]\n",
    "    traits2test = data.frame(trait = colnames(moloc_trait2class), weight = as.numeric(moloc_trait2class[som,]))\n",
    "    traits      = traits2test[traits2test$weight >= 0.5, \"trait\"]\n",
    "    coloc       = coloc_qtls[ coloc_qtls$locus == locus & coloc_qtls$pop == \"meta\" & coloc_qtls$gwas %in% traits,]\n",
    "    \n",
    "    out = list(som            = moloc_loci2class[id,],\n",
    "               traits_cluster = traits,\n",
    "               eqtls          = coloc[ coloc$PP.H4.abf >= 0.8, \"transcript_id\"],\n",
    "               traits         = traits2test,\n",
    "               coloc          = coloc\n",
    "              )\n",
    "    \n",
    "    #return(coloc[ coloc$PP.H4.abf >= 0.8,])\n",
    "    return(out)\n",
    "}\n",
    "\n",
    "id = \"5_130403286_131933599.5\"\n",
    "id = \"5_130403286_131933599.3\"\n",
    "\n",
    "coloc_list        = lapply(moloc_loci2class$id, function(id){get_coloc_qtl_by_locus(id, moloc_loci2class, coloc_qtls)})\n",
    "names(coloc_list) = moloc_loci2class$id\n",
    "\n",
    "saveRDS(coloc_list, \"pipeline/5.4.analyze_coloc_qtl_som_maps/coloc_list.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "locus2coloc = as.data.frame(rbindlist(lapply(coloc_list, function(x)\n",
    "{\n",
    "    out          = x$som\n",
    "    out$traits   = paste(sort(unique(x$traits_cluster)), collapse = \"; \")\n",
    "    out$traits_n = length(unique(x$traits_cluster))\n",
    "    out$eqtls    = paste(sort(unique(x$eqtls)), collapse = \"; \")\n",
    "    out$eqtls_n  = length(unique(x$eqtls))\n",
    "    \n",
    "    return(out)\n",
    "})), stringsAsFactors = FALSE)\n",
    "\n",
    "fwrite(locus2coloc, \"pipeline/5.4.analyze_coloc_qtl_som_maps/locus2coloc.txt\", sep = \"\\t\", col.names = TRUE, row.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "locus2coloc[ locus2coloc$id == \"17_40144007_48070076.2\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "totest = locus2coloc[ locus2coloc$traits_n > 0 & rowSums(locus2coloc[,c(\"traits_n\", \"eqtls_n\")]) > 1, \"id\"]\n",
    "\n",
    "length(totest)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Analyze loci"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "suppressPackageStartupMessages(library(coloc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_gwas_data = function(coord, gwas_file)\n",
    "{\n",
    "    my_head              = colnames(fread(cmd = paste(\"zcat\", gwas_file, \"|\", \"head -n 2\"), sep = \"\\t\", header = TRUE, data.table = FALSE))\n",
    "    gwas_data            = suppressWarnings(tabix.read.table(gwas_file, coord, col.names = TRUE, stringsAsFactors = FALSE))\n",
    "    colnames(gwas_data)  = my_head\n",
    "    gwas_data$id         = paste(\"VAR\", gwas_data$chr, gwas_data$pos, gwas_data$ref, gwas_data$alt, sep = \"_\")\n",
    "    rownames(gwas_data)  = gwas_data$id\n",
    "    \n",
    "    return(gwas_data)\n",
    "}\n",
    "\n",
    "create_dataset = function(study, trait_type, totest, variants, pop, manifest)\n",
    "{\n",
    "    #message(paste(study, trait_type, pop))\n",
    "    if(trait_type %in% c(\"categorical\", \"icd10\", \"phecode\"))\n",
    "    {\n",
    "        populations = unlist(strsplit(manifest[study, \"pops\"], \",\"))\n",
    "        if(pop == \"meta\"){n = sum(manifest[study, paste(\"n_cases\", populations, sep = \"_\")]) + sum(manifest[study, paste(\"n_controls\", populations, sep = \"_\")])}\n",
    "        if(pop != \"meta\"){n =     manifest[study, paste(\"n_cases\", pop        , sep = \"_\")]  +     manifest[study, paste(\"n_controls\", pop        , sep = \"_\")] }\n",
    "        \n",
    "        if(pop == \"meta\"){s = sum(manifest[study, paste(\"n_cases\", populations, sep = \"_\")]) / n}\n",
    "        if(pop != \"meta\"){s =     manifest[study, paste(\"n_cases\", pop        , sep = \"_\")]  / n}\n",
    "        \n",
    "        totest  = totest[is.na(totest[,paste(\"pval\", pop, sep = \"_\")]) == FALSE & is.na(totest[,paste(\"af_controls\", pop, sep = \"_\")]) == FALSE, ]\n",
    "        dataset = list(snp = variants, pvalues = totest[variants, paste(\"pval\", pop, sep = \"_\")], N = n, s = s, MAF = totest[variants, paste(\"af_controls\", pop, sep = \"_\")], type = \"cc\")\n",
    "        \n",
    "    }\n",
    "    if(trait_type %in% c(\"biomarkers\", \"continuous\"))\n",
    "    {\n",
    "        populations = unlist(strsplit(manifest[study, \"pops\"], \",\"))\n",
    "        if(pop == \"meta\"){n = sum(manifest[study, paste(\"n_cases\", populations, sep = \"_\")])}\n",
    "        if(pop != \"meta\"){n =     manifest[study, paste(\"n_cases\", pop        , sep = \"_\")] }\n",
    "        \n",
    "        totest  = totest[is.na(totest[,paste(\"pval\", pop, sep = \"_\")]) == FALSE & is.na(totest[,paste(\"af\", pop, sep = \"_\")]) == FALSE, ]\n",
    "        dataset = list(snp = variants, pvalues = totest[variants, paste(\"pval\", pop, sep = \"_\")], N = n, MAF = totest[variants, paste(\"af\", pop, sep = \"_\")], type = \"quant\")\n",
    "    }\n",
    "    if(trait_type == \"qtl\")\n",
    "    {\n",
    "        rownames(totest) = totest$id\n",
    "        dataset = list(snp     = variants, \n",
    "                       pvalues = totest[variants, \"pval\"], \n",
    "                       N       = 966, \n",
    "                       MAF     = totest[variants, \"af\"], \n",
    "                       type    = \"quant\")\n",
    "    }\n",
    "    \n",
    "    return(dataset)\n",
    "}\n",
    "\n",
    "run_coloc_by_pop = function(locus, study1, study2, pop, totest1, totest2, manifest)\n",
    "{\n",
    "    if (nrow(manifest[ manifest$id == study1,]) > 0){trait_type1 = manifest[study1, \"trait_type\"]}\n",
    "    if (nrow(manifest[ manifest$id == study2,]) > 0){trait_type2 = manifest[study2, \"trait_type\"]}\n",
    "    \n",
    "    if (nrow(manifest[ manifest$id == study1,]) == 0){trait_type1 = \"qtl\"}\n",
    "    if (nrow(manifest[ manifest$id == study2,]) == 0){trait_type2 = \"qtl\"}\n",
    "    \n",
    "    if( trait_type1 %in% c(\"categorical\", \"icd10\", \"phecode\")){variants1  = totest1[is.na(totest1[,paste(\"pval\", pop, sep = \"_\")]) == FALSE & is.na(totest1[,paste(\"af_controls\", pop, sep = \"_\")]) == FALSE, \"id\"]}\n",
    "    if( trait_type1 %in% c(\"biomarkers\", \"continuous\"       )){variants1  = totest1[is.na(totest1[,paste(\"pval\", pop, sep = \"_\")]) == FALSE & is.na(totest1[,paste(\"af\"         , pop, sep = \"_\")]) == FALSE, \"id\"]}\n",
    "    if( trait_type1 %in% c(\"qtl\"                            )){variants1  = totest1$id}\n",
    "    \n",
    "    \n",
    "    if( trait_type2 %in% c(\"categorical\", \"icd10\", \"phecode\")){variants2  = totest2[is.na(totest2[,paste(\"pval\", pop, sep = \"_\")]) == FALSE & is.na(totest2[,paste(\"af_controls\", pop, sep = \"_\")]) == FALSE, \"id\"]}\n",
    "    if( trait_type2 %in% c(\"biomarkers\", \"continuous\"       )){variants2  = totest2[is.na(totest2[,paste(\"pval\", pop, sep = \"_\")]) == FALSE & is.na(totest2[,paste(\"af\"         , pop, sep = \"_\")]) == FALSE, \"id\"]}\n",
    "    if( trait_type2 %in% c(\"qtl\"                            )){variants2  = totest2$id}\n",
    "    \n",
    "    if( trait_type1 %in% c(\"qtl\")){pop1 = \"\"}\n",
    "    if( trait_type2 %in% c(\"qtl\")){pop2 = \"\"}\n",
    "    \n",
    "    if(!trait_type1 %in% c(\"qtl\")){pop1 = pop}\n",
    "    if(!trait_type2 %in% c(\"qtl\")){pop2 = pop}\n",
    "    \n",
    "    variants2   = totest2$id\n",
    "    variants    = intersect(variants1, variants2)\n",
    "    \n",
    "    if(length(variants) > 100)\n",
    "    {\n",
    "        dataset1        = create_dataset(study1, trait_type1, totest1, variants, pop1, manifest)\n",
    "        dataset2        = create_dataset(study2, trait_type2, totest2, variants, pop2, manifest)\n",
    "        coloc_mapped    = coloc.abf(dataset1 = dataset1, dataset2 = dataset2) \n",
    "        probs           = as.data.frame(t(coloc_mapped$summary))\n",
    "        myres           = coloc_mapped$results\n",
    "        myres           = myres[, c(which(colnames(myres) == \"snp\"), ncol(myres))]\n",
    "        colnames(myres) = c(\"id\", \"pp_snp\")\n",
    "        myres           = myres[order(myres$pp_snp, decreasing = TRUE), ]\n",
    "        myres$cum       = cumsum(myres$pp_snp)\n",
    "        myres$cs        = FALSE\n",
    "        to_cs           = myres[myres$cum >= 0.99, ][1, \"id\"]\n",
    "        \n",
    "        myres[myres$cum < 0.99 | myres$id == to_cs, \"cs\"] = TRUE\n",
    "        \n",
    "        out             = cbind(probs, myres[1, ])\n",
    "    }else\n",
    "    {\n",
    "        out = data.frame(nsnps = 0, PP.H0.abf = 1, PP.H1.abf = 0, PP.H2.abf = 0, PP.H3.abf = 0, PP.H4.abf = 0, \n",
    "                         locus = locus, study1 = study1, study2 = study2, id = \"\", pp_snp = 0\n",
    "                        )\n",
    "        myres = data.frame(id = \"\", pp_snp = 0, cum = 1, cs = FALSE)\n",
    "    }\n",
    "\n",
    "    return(list(top = out, pp = myres))\n",
    "}\n",
    "\n",
    "combine_variants = function(totest_comb, coloc_tests, totest_list)\n",
    "{\n",
    "    myvars = sort(unique(unlist(lapply(totest_list, function(x){x$id}))))\n",
    "    out    = matrix(0, nrow = length(myvars), ncol = nrow(totest_comb), dimnames = list(myvars, totest_comb$id))\n",
    "    \n",
    "    if(nrow(totest_comb) > 1)\n",
    "    {\n",
    "        for(id in totest_comb$id)\n",
    "        {\n",
    "            this = coloc_tests[[id]][[\"pp\"]]\n",
    "            \n",
    "            if(nrow(this) > 1){out[ this$id, id] = this$pp_snp}\n",
    "        }\n",
    "\n",
    "        out = out[rownames(out) != \"\",]\n",
    "    }else\n",
    "    {\n",
    "        this = coloc_tests[[totest_comb[1, \"id\"]]][[\"pp\"]]\n",
    "        \n",
    "        if(nrow(this) > 1){out[ this$id, 1] = this$pp_snp}\n",
    "    }\n",
    "    \n",
    "    return(out)\n",
    "}\n",
    "\n",
    "quiet = function(x) \n",
    "{ \n",
    "    sink(tempfile()) \n",
    "    on.exit(sink()) \n",
    "    invisible(force(x)) \n",
    "} \n",
    "\n",
    "analyze_locus = function(id, coloc_list, loci, manifest, to_return = FALSE)\n",
    "{\n",
    "    message(id)\n",
    "    som            = coloc_list[[id]][[\"som\"           ]]\n",
    "    traits_cluster = coloc_list[[id]][[\"traits_cluster\"]]\n",
    "    eqtls          = coloc_list[[id]][[\"eqtls\"         ]]\n",
    "    traits         = coloc_list[[id]][[\"traits\"        ]]\n",
    "    coloc          = coloc_list[[id]][[\"coloc\"         ]]   \n",
    "    \n",
    "    locus         = som[1, \"locus\"]\n",
    "    coloc         = coloc[ coloc$PP.H4.abf >= 0.8,]\n",
    "    coloc         = coloc[order(coloc$PP.H4.abf, decreasing = TRUE),]\n",
    "    coloc$tr2type = paste(coloc$transcript_id, coloc$type)\n",
    "    coord         = loci[locus,]\n",
    "    coord         = paste0(coord$chrom, \":\", coord$from, \"-\", coord$to)\n",
    "    \n",
    "    # get GWAS data\n",
    "    totest_gwas_list        = lapply(traits_cluster, function(gwas){get_gwas_data(coord, manifest[gwas, \"sumstat_file\"])})\n",
    "    names(totest_gwas_list) = traits_cluster\n",
    "    \n",
    "    # get QTL data\n",
    "    qtls          = unique(coloc[,c(\"transcript_id\", \"gene_id\", \"phenotype\", \"type\", \"tr2type\")])\n",
    "    \n",
    "    if(nrow(qtls) > 0)\n",
    "    {\n",
    "        qtls$qtl_file = paste(\"/frazer01/projects/CARDIPS/analysis/cardiac_qtls_combined/input/qtl/processing\", qtls$phenotype, paste(\"qtl\", qtls$transcript_id, \"txt\", sep = \".\"), sep = \"/\")\n",
    "\n",
    "        qtl_list = lapply(1:nrow(qtls), function(ii)\n",
    "        {\n",
    "            indata = fread(qtls[ii, \"qtl_file\"], sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "            type   =       qtls[ii, \"type\"    ]\n",
    "            indata = indata[indata$type == type,]\n",
    "\n",
    "            return(indata)\n",
    "        })\n",
    "\n",
    "        names(qtl_list) = qtls$tr2type\n",
    "        \n",
    "        totest_list = c(totest_gwas_list, qtl_list)\n",
    "    }else\n",
    "    {\n",
    "        totest_list = totest_gwas_list\n",
    "    }\n",
    "    \n",
    "    # Generate matrix for coloc (all vs. all)\n",
    "    \n",
    "    totest_comb           = as.data.frame(t(combn(names(totest_list), 2)))\n",
    "    colnames(totest_comb) = c(\"study1\", \"study2\")\n",
    "    totest_comb$id        = paste(totest_comb$study1, totest_comb$study2, sep = \":\")\n",
    "    rownames(totest_comb) = totest_comb$id\n",
    "    \n",
    "    coloc_tests = lapply(totest_comb$id, function(id)\n",
    "    {\n",
    "        study1 = totest_comb[id, \"study1\"]\n",
    "        study2 = totest_comb[id, \"study2\"]\n",
    "        out    = run_coloc_by_pop(locus, study1, study2, \"meta\", totest_list[[study1]], totest_list[[study2]], manifest)\n",
    "        \n",
    "        return(out)\n",
    "    })\n",
    "    \n",
    "    names(coloc_tests) = totest_comb$id\n",
    "    \n",
    "    pps                = combine_variants(totest_comb, coloc_tests, totest_list)\n",
    "    var2pp             = data.frame(id = rownames(pps), pp_mean = as.numeric(rowMeans(pps)))\n",
    "    var2pp             = var2pp[order(var2pp$pp_mean, decreasing = TRUE),]    \n",
    "    out                = list(comb        = totest_comb, \n",
    "                              totest_list = totest_list, \n",
    "                              coloc_tests = coloc_tests,\n",
    "                              pps         = pps        ,\n",
    "                              var2pp      = var2pp\n",
    "                             )\n",
    "    \n",
    "    saveRDS(out, paste0(\"pipeline/5.4.analyze_coloc_qtl_som_maps/coloc_data/\", id, \".rds\"))\n",
    "    \n",
    "    if(to_return == TRUE){return(out)}\n",
    "}\n",
    "\n",
    "id = \"5_136864391_137972681.2\" # atrial fibrillation + pulse rate\n",
    "#id = \"5_16423249_16671484.1\" \n",
    "\n",
    "x = quiet(analyze_locus(id, coloc_list, loci, manifest, TRUE))\n",
    "\n",
    "#invisible(lapply(totest, function(id){quiet(analyze_locus(id, coloc_list, loci, manifest, FALSE))}))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Summarize colocalization results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "find_lead = function(id)\n",
    "{\n",
    "    x      = readRDS(paste0(\"pipeline/5.4.analyze_coloc_qtl_som_maps/coloc_data/\", id, \".rds\"))\n",
    "    out    = x[[\"var2pp\"]][1,]\n",
    "    out    = cbind(locus2coloc[ locus2coloc$id == id, ], data.frame(variant_id = out$id, pp = out$pp_mean))\n",
    "    \n",
    "    if(out$eqtls_n >  0)\n",
    "    {\n",
    "        genes                = sort(unique(coordinates[ coordinates$transcript_id %in% unlist(strsplit(out$eqtls, \"; \")), \"gene_name\"]))\n",
    "        out$eqtls_gene_names = paste(genes, collapse = \"; \")\n",
    "        out$eqtls_genes_n    = length(genes)\n",
    "    }else\n",
    "    {\n",
    "        out$eqtls_gene_names = \"\"\n",
    "        out$eqtls_genes_n    = 0\n",
    "    }\n",
    "    \n",
    "    out$traits_name = paste(sort(unique(manifest[ unlist(strsplit(out$traits, \"; \")), \"name\"])), collapse = \"; \")\n",
    "    \n",
    "    return(out)\n",
    "}\n",
    "\n",
    "id = \"5_136864391_137972681.2\" # atrial fibrillation + pulse rate\n",
    "#find_lead(id)\n",
    "\n",
    "pps = as.data.frame(rbindlist(lapply(totest, find_lead)), stringsAsFactors = FALSE)\n",
    "pps = pps[order(pps$pp, decreasing = TRUE),]\n",
    "fwrite(pps, \"pipeline/5.4.analyze_coloc_qtl_som_maps/pps.txt\", sep = \"\\t\", col.names = TRUE, row.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "nrow(pps[ pps$traits_n > 0 & pps$eqtls_n > 0 & pps$pp > 0.9, ])\n",
    "nrow(pps[ pps$traits_n > 0 & pps$eqtls_n > 0 & pps$pp > 0.5 & pps$pp <= 0.9, ])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pps[ pps$traits_n > 0 & pps$eqtls_n > 0 & pps$pp > 0.9, ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "coordinates[ coordinates$transcript_id == \"ENSG00000164867.11_5\", ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.0.1",
   "language": "R",
   "name": "ir401"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
