{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/frazer01/projects/CARDIPS/analysis/cardiac_gwas_coloc\")\n",
    "\n",
    "source(\"script/functions.R\"  )\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get LD for figures"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "dir.create(\"pipeline/ld\", showWarnings = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "fm_list   = readRDS(\"pipeline/eqtl.4.finemap_single/fm_list.rds\"  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "coloc        =              fread(\"pipeline/eqtls.0.1.coloc_gwas/coloc.txt\"           , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "enr1         =              fread(\"pipeline/eqtls.0.1.coloc_gwas/enrichment_by_pp.txt\", sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "manifest     = add_rownames(fread(\"pipeline/eqtls.0.1.coloc_gwas/manifest.txt\"        , sep = \"\\t\", header = TRUE, data.table = FALSE))\n",
    "pph4         = coloc[ coloc$PP.H4.abf >= 0.8,]\n",
    "coloc_old    =              fread(\"pipeline/eqtl.2.gwas_coloc/coloc.txt\"           , sep = \"\\t\", header = TRUE, data.table = FALSE)\n",
    "pph4_old     = coloc_old[ coloc_old$PP.H4.abf >= 0.8,]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "toplot1 = merge(pph4    [ pph4$interaction.ipsc_cvpc == 1 & pph4$gwas == \"continuous-PP-both_sexes-combined_medadj_irnt\", c(\"transcript_id\", \"gene_id\", \"type\", \"gwas\")],\n",
    "                pph4_old[ , c(\"transcript_id\", \"gene_id\", \"type\", \"gwas\", \"locus\", \"gene_name\")],\n",
    "                by = c(\"transcript_id\", \"gene_id\", \"type\", \"gwas\")\n",
    "               )\n",
    "toplot2 = merge(pph4    [ pph4$interaction.adult == 1 & pph4$gwas == \"continuous-102-both_sexes-irnt\", c(\"transcript_id\", \"gene_id\", \"type\", \"gwas\")],\n",
    "                pph4_old[ , c(\"transcript_id\", \"gene_id\", \"type\", \"gwas\", \"locus\", \"gene_name\")],\n",
    "                by = c(\"transcript_id\", \"gene_id\", \"type\", \"gwas\")\n",
    "               )\n",
    "\n",
    "toplot       = rbind(toplot1, toplot2)\n",
    "toplot$coloc = paste(toplot$transcript_id, toplot$type, toplot$gwas, toplot$locus, sep = \":\")\n",
    "fm           = fm_list[toplot$coloc]\n",
    "\n",
    "#toplot = toplot[order(toplot$gene_name, toplot$transcript_id),]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "rsid2chrom = data.frame(rsid = c(\"rs28473516\", \"rs17584\", \"rs61876335\", \"rs7589901\", \"rs2549009\", \"rs2738413\", \"rs3493\", \"rs74181299\", \"rs11000060\", \"rs1708618\", \"rs12724121\"), chrom = c(5, 11, 11, 2, 5, 14, 13, 2, 10, 17, 1))\n",
    "rsid2chrom = unique(rbind(rsid2chrom, as.data.frame(rbindlist(lapply(toplot$coloc, function(x)\n",
    "{    \n",
    "    indata            = fm_list[[x]]\n",
    "    gwas              = indata$gwas\n",
    "    gwas              = gwas[is.na(gwas$pval) == FALSE,]\n",
    "    qtl               = indata$qtl\n",
    "    finemap           = indata$finemap\n",
    "    ids               = indata$credible_set\n",
    "    id                = finemap[1,\"id\"]\n",
    "    rsid              = qtl    [id, c(\"rsid\", \"chrom\")]\n",
    "    \n",
    "    return(rsid)\n",
    "})), stringsAsFactors = FALSE))) %>% arrange(chrom, rsid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1\n",
      "/software/plink-2.3/plink2_64 --make-bed --out /frazer01/projects/CARDIPS/analysis/cardiac_gwas_coloc/pipeline/ld/rs12724121 --memory 96000 --threads 12 --pgen /sdsc/cast/ukbb/array_imputed/pfile_converted/chr1.pgen --pvar /sdsc/cast/ukbb/array_imputed/pfile_converted/chr1.pvar --psam /sdsc/cast/ukbb/array_imputed/pfile_converted/chr1.psam --keep /frazer01/projects/CARDIPS/analysis/cardiac_gwas_coloc/pipeline/ld/rs12724121.subjects.txt --maf 0.01 --hwe 1e-06 \n",
      "\n",
      "/software/plink-1.90b3x/plink --out /frazer01/projects/CARDIPS/analysis/cardiac_gwas_coloc/pipeline/ld/rs12724121 --memory 96000 --threads 12 --vcf /frazer01/reference/public/1000Genomes/ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --ld-snp rs12724121 --ld-window 1000 --ld-window-kb 5000 --ld-window-r2 0 --r2 dprime \n",
      "\n",
      "2\n",
      "3\n",
      "4\n",
      "/software/plink-2.3/plink2_64 --make-bed --out /frazer01/projects/CARDIPS/analysis/cardiac_gwas_coloc/pipeline/ld/rs74181299 --memory 96000 --threads 12 --pgen /sdsc/cast/ukbb/array_imputed/pfile_converted/chr2.pgen --pvar /sdsc/cast/ukbb/array_imputed/pfile_converted/chr2.pvar --psam /sdsc/cast/ukbb/array_imputed/pfile_converted/chr2.psam --keep /frazer01/projects/CARDIPS/analysis/cardiac_gwas_coloc/pipeline/ld/rs74181299.subjects.txt --maf 0.01 --hwe 1e-06 \n",
      "\n",
      "/software/plink-1.90b3x/plink --out /frazer01/projects/CARDIPS/analysis/cardiac_gwas_coloc/pipeline/ld/rs74181299 --memory 96000 --threads 12 --vcf /frazer01/reference/public/1000Genomes/ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --ld-snp rs74181299 --ld-window 1000 --ld-window-kb 5000 --ld-window-r2 0 --r2 dprime \n",
      "\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n"
     ]
    }
   ],
   "source": [
    "calculate_ld_by_snp = function(rsid, chrom)\n",
    "{\n",
    "    prefix  = paste(getwd(), \"pipeline/ld\", rsid, sep = \"/\")\n",
    "    vcf     = paste(\"/frazer01/reference/public/1000Genomes/ALL\", paste0(\"chr\", chrom), \"phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\", sep = \".\")\n",
    "    #bfile   = paste(\"/sdsc/cast/ukbb/array_imputed\"   , paste(\"ukb_imp\", paste0(\"chr\", chrom), \"v3\", sep = \"_\"), sep = \"/\")\n",
    "    #bgen    = paste(\"/sdsc/cast/ukbb/array_imputed\"   , paste(\"ukb_imp\", paste0(\"chr\", chrom), \"v3.bgen\", sep = \"_\"), sep = \"/\")\n",
    "    pgen    = paste(\"/sdsc/cast/ukbb/array_imputed/pfile_converted\"   , paste0(\"chr\", chrom), sep = \"/\")\n",
    "    outfile = paste(getwd(), \"pipeline/ld\", paste(rsid, \"ld.txt\", sep = \".\"), sep = \"/\")\n",
    "    \n",
    "    if(file.exists(outfile) == FALSE)\n",
    "    {\n",
    "        subjects_file = paste(prefix, \"subjects.txt\", sep = \".\")\n",
    "        command       = paste(\"head\", \"-n\", 5000, paste(pgen, \"psam\", sep = \".\"), \">\", subjects_file)\n",
    "        system(command) # uncomment to run\n",
    "        \n",
    "        command = paste(plink2,\n",
    "                        \"--make-bed\",\n",
    "                        \"--out\"     , prefix,\n",
    "                        \"--memory\"  , 96000,\n",
    "                        \"--threads\" , 12,\n",
    "                        \"--pgen\"    , paste(pgen, \"pgen\", sep = \".\"),\n",
    "                        \"--pvar\"    , paste(pgen, \"pvar\", sep = \".\"),\n",
    "                        \"--psam\"    , paste(pgen, \"psam\", sep = \".\"),\n",
    "                        \"--keep\"    , subjects_file,\n",
    "                        \"--maf\"     , 0.01,\n",
    "                        \"--hwe\"     , 0.000001,\n",
    "                        \"\"\n",
    "                       )\n",
    "\n",
    "        message(command)\n",
    "        if(file.exists(paste(prefix, \"bed\", sep = \".\")) == FALSE){system(command)} # uncomment to run\n",
    "        \n",
    "        \n",
    "        command = paste(plink,\n",
    "                        \"--out\"           , prefix,\n",
    "                        \"--memory\"        , 96000,\n",
    "                        \"--threads\"       , 12,\n",
    "                        \"--vcf\"           , vcf,\n",
    "                        #\"--bed\"           , paste(prefix, \"bed\", sep = \".\"),\n",
    "                        #\"--bim\"           , paste(prefix, \"bim\", sep = \".\"),\n",
    "                        #\"--fam\"           , paste(prefix, \"fam\", sep = \".\"),\n",
    "                        #\"--bfile\"         , bfile,\n",
    "                        #\"--bgen\"          , bgen,\n",
    "                        #\"--sample\"        , \"/sdsc/cast/ukbb/array_imputed/ukb46122_imp_chr1_v3_s487283.sample\",\n",
    "                        \"--ld-snp\"        , rsid,\n",
    "                        #\"--indep-pairwise\", 1000, 5000, 0.2,\n",
    "                        \"--ld-window\"     , 1000,\n",
    "                        \"--ld-window-kb\"  , 5000,\n",
    "                        \"--ld-window-r2\"  ,    0,\n",
    "                        \"--r2\"            , \"dprime\",\n",
    "                        #\"--maf\"           , 0.01,\n",
    "                        #\"--hwe\"           , 0.000001,\n",
    "                        \"\"\n",
    "                       )\n",
    "\n",
    "        message(command)\n",
    "        if(file.exists(paste(prefix, \"ld\", sep = \".\")) == FALSE){system(command)} # uncomment to run\n",
    "\n",
    "        ld           = read.table(paste(prefix, \"ld\", sep = \".\"), sep = \"\", header = TRUE)[,c(\"SNP_A\", \"SNP_B\", \"R2\", \"DP\")]\n",
    "        colnames(ld) = tolower(colnames(ld))\n",
    "\n",
    "        ld = mutate(ld,\n",
    "               bg = case_when(r2 <= 0.2 ~ \"#00007e\",\n",
    "                              r2 <= 0.4 ~ \"#86cdf9\",\n",
    "                              r2 <= 0.6 ~ \"#01fe02\",\n",
    "                              r2 <= 0.8 ~ \"#fea500\",\n",
    "                              r2 <= 1   ~ \"#fc0000\",\n",
    "                              TRUE ~ NA_character_\n",
    "                             )\n",
    "              )   \n",
    "\n",
    "        fwrite(ld, outfile, sep = \"\\t\", col.names = TRUE, row.names = FALSE)\n",
    "    }\n",
    "}\n",
    "\n",
    "plink    = \"/software/plink-1.90b3x/plink\"\n",
    "plink2   = \"/software/plink-2.3/plink2_64\"\n",
    "ii = 1\n",
    "\n",
    "invisible(lapply(1:nrow(rsid2chrom), function(ii)\n",
    "{\n",
    "    message(ii, appendLF = FALSE)\n",
    "    \n",
    "    try(calculate_ld_by_snp(rsid2chrom[ii, \"rsid\"], rsid2chrom[ii, \"chrom\"]))\n",
    "}))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "List of 11\n",
      " $ transcript_id : chr \"ENSG00000000971.16_4\"\n",
      " $ phenotype     : chr \"rna\"\n",
      " $ type          : int 0\n",
      " $ locus         : chr \"1_196248779_197362184\"\n",
      " $ trait         : chr \"biomarkers-30600-both_sexes-irnt\"\n",
      " $ coord         : chr \"1:196248779-197362184\"\n",
      " $ gwas          :'data.frame':\t2529 obs. of  5 variables:\n",
      "  ..$ pos : int [1:2529] 196248826 196249082 196249394 196250413 196250589 196250988 196251522 196251694 196252118 196252159 ...\n",
      "  ..$ id  : chr [1:2529] \"VAR_1_196248826_G_A\" \"VAR_1_196249082_T_A\" \"VAR_1_196249394_G_C\" \"VAR_1_196250413_C_T\" ...\n",
      "  ..$ beta: num [1:2529] 0.01319 -0.00768 -0.03075 0.00587 -0.01241 ...\n",
      "  ..$ se  : num [1:2529] 0.00276 0.0025 0.06342 0.0033 0.00498 ...\n",
      "  ..$ pval: num [1:2529] 1.69e-06 2.14e-03 6.28e-01 7.58e-02 1.27e-02 ...\n",
      " $ qtl           :'data.frame':\t1293 obs. of  11 variables:\n",
      "  ..$ chrom     : int [1:1293] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  ..$ pos       : int [1:1293] 196121869 196121998 196122305 196122733 196122848 196123070 196123091 196123482 196124222 196124287 ...\n",
      "  ..$ ref       : chr [1:1293] \"C\" \"A\" \"T\" \"A\" ...\n",
      "  ..$ alt       : chr [1:1293] \"T\" \"G\" \"A\" \"G\" ...\n",
      "  ..$ rsid      : chr [1:1293] \"rs61013848\" \"rs16839561\" \"rs61654311\" \"rs72748434\" ...\n",
      "  ..$ id        : chr [1:1293] \"VAR_1_196121869_C_T\" \"VAR_1_196121998_A_G\" \"VAR_1_196122305_T_A\" \"VAR_1_196122733_A_G\" ...\n",
      "  ..$ af        : num [1:1293] 0.074 0.074 0.074 0.074 0.074 0.144 0.074 0.046 0.046 0.074 ...\n",
      "  ..$ beta      : num [1:1293] 0.0692 0.0692 0.0692 0.0692 0.0692 ...\n",
      "  ..$ se        : num [1:1293] 0.0387 0.0387 0.0387 0.0387 0.0387 ...\n",
      "  ..$ pval      : num [1:1293] 0.798 0.798 0.798 0.798 0.798 ...\n",
      "  ..$ bonferroni: num [1:1293] 1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ finemap       :'data.frame':\t994 obs. of  5 variables:\n",
      "  ..$ id : chr [1:994] \"VAR_1_196357746_A_G\" \"VAR_1_196371432_T_C\" \"VAR_1_196400253_C_T\" \"VAR_1_196351150_A_C\" ...\n",
      "  ..$ pp : num [1:994] 0.2782 0.1298 0.0893 0.087 0.0634 ...\n",
      "  ..$ pos: num [1:994] 1.96e+08 1.96e+08 1.96e+08 1.96e+08 1.96e+08 ...\n",
      "  ..$ cum: num [1:994] 0.278 0.408 0.497 0.584 0.648 ...\n",
      "  ..$ cs : logi [1:994] TRUE TRUE TRUE TRUE TRUE TRUE ...\n",
      " $ credible_set  : chr [1:24] \"VAR_1_196357746_A_G\" \"VAR_1_196371432_T_C\" \"VAR_1_196400253_C_T\" \"VAR_1_196351150_A_C\" ...\n",
      " $ credible_set_n: int 24\n"
     ]
    }
   ],
   "source": [
    "str(fm_list[[1]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 23 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>rsid</th><th scope=col>chrom</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>rs1190441 </td><td> 2</td></tr>\n",
       "\t<tr><td>rs1656398 </td><td> 2</td></tr>\n",
       "\t<tr><td>rs7589901 </td><td> 2</td></tr>\n",
       "\t<tr><td>rs7687399 </td><td> 4</td></tr>\n",
       "\t<tr><td>rs2549009 </td><td> 5</td></tr>\n",
       "\t<tr><td>rs28473516</td><td> 5</td></tr>\n",
       "\t<tr><td>rs10247315</td><td> 7</td></tr>\n",
       "\t<tr><td>rs12705096</td><td> 7</td></tr>\n",
       "\t<tr><td>rs478882  </td><td> 9</td></tr>\n",
       "\t<tr><td>rs11000060</td><td>10</td></tr>\n",
       "\t<tr><td>rs10832164</td><td>11</td></tr>\n",
       "\t<tr><td>rs17584   </td><td>11</td></tr>\n",
       "\t<tr><td>rs3923805 </td><td>11</td></tr>\n",
       "\t<tr><td>rs61876335</td><td>11</td></tr>\n",
       "\t<tr><td>rs2738413 </td><td>14</td></tr>\n",
       "\t<tr><td>rs1072911 </td><td>16</td></tr>\n",
       "\t<tr><td>rs2280370 </td><td>16</td></tr>\n",
       "\t<tr><td>rs34938804</td><td>16</td></tr>\n",
       "\t<tr><td>rs35915772</td><td>16</td></tr>\n",
       "\t<tr><td>rs8044091 </td><td>16</td></tr>\n",
       "\t<tr><td>rs1708618 </td><td>17</td></tr>\n",
       "\t<tr><td>rs2109019 </td><td>17</td></tr>\n",
       "\t<tr><td>rs8108864 </td><td>19</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 23 × 2\n",
       "\\begin{tabular}{ll}\n",
       " rsid & chrom\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t rs1190441  &  2\\\\\n",
       "\t rs1656398  &  2\\\\\n",
       "\t rs7589901  &  2\\\\\n",
       "\t rs7687399  &  4\\\\\n",
       "\t rs2549009  &  5\\\\\n",
       "\t rs28473516 &  5\\\\\n",
       "\t rs10247315 &  7\\\\\n",
       "\t rs12705096 &  7\\\\\n",
       "\t rs478882   &  9\\\\\n",
       "\t rs11000060 & 10\\\\\n",
       "\t rs10832164 & 11\\\\\n",
       "\t rs17584    & 11\\\\\n",
       "\t rs3923805  & 11\\\\\n",
       "\t rs61876335 & 11\\\\\n",
       "\t rs2738413  & 14\\\\\n",
       "\t rs1072911  & 16\\\\\n",
       "\t rs2280370  & 16\\\\\n",
       "\t rs34938804 & 16\\\\\n",
       "\t rs35915772 & 16\\\\\n",
       "\t rs8044091  & 16\\\\\n",
       "\t rs1708618  & 17\\\\\n",
       "\t rs2109019  & 17\\\\\n",
       "\t rs8108864  & 19\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 23 × 2\n",
       "\n",
       "| rsid &lt;chr&gt; | chrom &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| rs1190441  |  2 |\n",
       "| rs1656398  |  2 |\n",
       "| rs7589901  |  2 |\n",
       "| rs7687399  |  4 |\n",
       "| rs2549009  |  5 |\n",
       "| rs28473516 |  5 |\n",
       "| rs10247315 |  7 |\n",
       "| rs12705096 |  7 |\n",
       "| rs478882   |  9 |\n",
       "| rs11000060 | 10 |\n",
       "| rs10832164 | 11 |\n",
       "| rs17584    | 11 |\n",
       "| rs3923805  | 11 |\n",
       "| rs61876335 | 11 |\n",
       "| rs2738413  | 14 |\n",
       "| rs1072911  | 16 |\n",
       "| rs2280370  | 16 |\n",
       "| rs34938804 | 16 |\n",
       "| rs35915772 | 16 |\n",
       "| rs8044091  | 16 |\n",
       "| rs1708618  | 17 |\n",
       "| rs2109019  | 17 |\n",
       "| rs8108864  | 19 |\n",
       "\n"
      ],
      "text/plain": [
       "   rsid       chrom\n",
       "1  rs1190441   2   \n",
       "2  rs1656398   2   \n",
       "3  rs7589901   2   \n",
       "4  rs7687399   4   \n",
       "5  rs2549009   5   \n",
       "6  rs28473516  5   \n",
       "7  rs10247315  7   \n",
       "8  rs12705096  7   \n",
       "9  rs478882    9   \n",
       "10 rs11000060 10   \n",
       "11 rs10832164 11   \n",
       "12 rs17584    11   \n",
       "13 rs3923805  11   \n",
       "14 rs61876335 11   \n",
       "15 rs2738413  14   \n",
       "16 rs1072911  16   \n",
       "17 rs2280370  16   \n",
       "18 rs34938804 16   \n",
       "19 rs35915772 16   \n",
       "20 rs8044091  16   \n",
       "21 rs1708618  17   \n",
       "22 rs2109019  17   \n",
       "23 rs8108864  19   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rsid2chrom"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.0.1",
   "language": "R",
   "name": "ir401"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
